{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "knowing-frame",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T13:50:33.650250Z",
     "iopub.status.busy": "2023-01-31T13:50:33.647467Z",
     "iopub.status.idle": "2023-01-31T13:50:42.393329Z",
     "shell.execute_reply": "2023-01-31T13:50:42.391051Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Attaching SeuratObject\n",
      "\n",
      "Attaching sp\n",
      "\n",
      "Loading required package: SingleCellExperiment\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:sp’:\n",
      "\n",
      "    %over%\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘SummarizedExperiment’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SeuratObject’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Seurat’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(Seurat)\n",
    "library(MAST)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "measured-calvin",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T13:50:42.421818Z",
     "iopub.status.busy": "2023-01-31T13:50:42.395779Z",
     "iopub.status.idle": "2023-01-31T13:50:42.458605Z",
     "shell.execute_reply": "2023-01-31T13:50:42.457261Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(dplyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "systematic-consolidation",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:50:42.461515Z",
     "iopub.status.busy": "2023-01-31T13:50:42.460484Z",
     "iopub.status.idle": "2023-01-31T13:50:42.471288Z",
     "shell.execute_reply": "2023-01-31T13:50:42.470005Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "data_root <- '/lustre/scratch126/cellgen/team205/nh3/skin'\n",
    "proj_root <- '~/FCA_Fetal_Skin_priv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "rubber-captain",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:50:42.474172Z",
     "iopub.status.busy": "2023-01-31T13:50:42.473134Z",
     "iopub.status.idle": "2023-01-31T13:52:44.453464Z",
     "shell.execute_reply": "2023-01-31T13:52:44.450497Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "pooled_ad = readRDS(file.path(proj_root, \"data\", \"pooled_fetal_organoid_adult_processed.rds\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "automatic-actor",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:52:44.457611Z",
     "iopub.status.busy": "2023-01-31T13:52:44.456194Z",
     "iopub.status.idle": "2023-01-31T13:52:44.478899Z",
     "shell.execute_reply": "2023-01-31T13:52:44.477429Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sanger_id'</li><li>'chemistry'</li><li>'batch'</li><li>'gender'</li><li>'pcw'</li><li>'sorting'</li><li>'sample'</li><li>'chemistry_sorting'</li><li>'cell_caller'</li><li>'scrublet_score'</li><li>'cluster_scrublet_score'</li><li>'doublet_pval'</li><li>'bh_doublet_pval'</li><li>'nCounts_RNA'</li><li>'nFeaturess_RNA'</li><li>'percent_mito'</li><li>'percent_ribo'</li><li>'percent_hb'</li><li>'independent_annotation_refined'</li><li>'independent_annotation_broad1'</li><li>'independent_annotation_broad2'</li><li>'independent_annotation_broad3'</li><li>'independent_annotation_broad4'</li><li>'independent_annotation_broad5'</li><li>'joint_annotation'</li><li>'is_maternal'</li><li>'fig1b_annotation'</li><li>'independent_annotation_refined_20220201a'</li><li>'independent_annotation_refined_20220201b'</li><li>'independent_annotation_broad1_20220201'</li><li>'independent_annotation_broad2_20220201'</li><li>'independent_annotation_broad3_20220201'</li><li>'joint_annotation_20220201a'</li><li>'joint_annotation_20220201b'</li><li>'fig1b_annotation_20220201'</li><li>'joint_annotation_20220202'</li><li>'independent_annotation_refined_20220202'</li><li>'annot'</li><li>'day'</li><li>'sample_id'</li><li>'version'</li><li>'week'</li><li>'nh3_annot1'</li><li>'log1p_nCounts_RNA'</li><li>'log1p_nFeaturess_RNA'</li><li>'nCounts_RNA_mito'</li><li>'nCounts_RNA_ribo'</li><li>'nCounts_RNA_hb'</li><li>'percent_top50'</li><li>'pass_auto_filter'</li><li>'scrublet_score_z'</li><li>'bh_pval'</li><li>'scrublet_done'</li><li>'doublet'</li><li>'seurat_clusters'</li><li>'fsk_annot1_min30'</li><li>'fsk_annot1_min30_prob'</li><li>'fsk_annot2_min30'</li><li>'fsk_annot2_min30_prob'</li><li>'day8to13'</li><li>'org_annot0'</li><li>'org_annot0_prob'</li><li>'org_annot30'</li><li>'org_annot30_prob'</li><li>'fsk_annot0'</li><li>'fsk_annot0_prob'</li><li>'fsk_annot30'</li><li>'fsk_annot30_prob'</li><li>'mad_prd'</li><li>'Status'</li><li>'Site'</li><li>'Site_old'</li><li>'Tissue'</li><li>'Enrichment'</li><li>'Location'</li><li>'Sex'</li><li>'Age'</li><li>'stage'</li><li>'anno_final'</li><li>'final'</li><li>'dataset'</li><li>'log2pcw'</li><li>'annot_early_late'</li><li>'annot_early_late2'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sanger\\_id'\n",
       "\\item 'chemistry'\n",
       "\\item 'batch'\n",
       "\\item 'gender'\n",
       "\\item 'pcw'\n",
       "\\item 'sorting'\n",
       "\\item 'sample'\n",
       "\\item 'chemistry\\_sorting'\n",
       "\\item 'cell\\_caller'\n",
       "\\item 'scrublet\\_score'\n",
       "\\item 'cluster\\_scrublet\\_score'\n",
       "\\item 'doublet\\_pval'\n",
       "\\item 'bh\\_doublet\\_pval'\n",
       "\\item 'nCounts\\_RNA'\n",
       "\\item 'nFeaturess\\_RNA'\n",
       "\\item 'percent\\_mito'\n",
       "\\item 'percent\\_ribo'\n",
       "\\item 'percent\\_hb'\n",
       "\\item 'independent\\_annotation\\_refined'\n",
       "\\item 'independent\\_annotation\\_broad1'\n",
       "\\item 'independent\\_annotation\\_broad2'\n",
       "\\item 'independent\\_annotation\\_broad3'\n",
       "\\item 'independent\\_annotation\\_broad4'\n",
       "\\item 'independent\\_annotation\\_broad5'\n",
       "\\item 'joint\\_annotation'\n",
       "\\item 'is\\_maternal'\n",
       "\\item 'fig1b\\_annotation'\n",
       "\\item 'independent\\_annotation\\_refined\\_20220201a'\n",
       "\\item 'independent\\_annotation\\_refined\\_20220201b'\n",
       "\\item 'independent\\_annotation\\_broad1\\_20220201'\n",
       "\\item 'independent\\_annotation\\_broad2\\_20220201'\n",
       "\\item 'independent\\_annotation\\_broad3\\_20220201'\n",
       "\\item 'joint\\_annotation\\_20220201a'\n",
       "\\item 'joint\\_annotation\\_20220201b'\n",
       "\\item 'fig1b\\_annotation\\_20220201'\n",
       "\\item 'joint\\_annotation\\_20220202'\n",
       "\\item 'independent\\_annotation\\_refined\\_20220202'\n",
       "\\item 'annot'\n",
       "\\item 'day'\n",
       "\\item 'sample\\_id'\n",
       "\\item 'version'\n",
       "\\item 'week'\n",
       "\\item 'nh3\\_annot1'\n",
       "\\item 'log1p\\_nCounts\\_RNA'\n",
       "\\item 'log1p\\_nFeaturess\\_RNA'\n",
       "\\item 'nCounts\\_RNA\\_mito'\n",
       "\\item 'nCounts\\_RNA\\_ribo'\n",
       "\\item 'nCounts\\_RNA\\_hb'\n",
       "\\item 'percent\\_top50'\n",
       "\\item 'pass\\_auto\\_filter'\n",
       "\\item 'scrublet\\_score\\_z'\n",
       "\\item 'bh\\_pval'\n",
       "\\item 'scrublet\\_done'\n",
       "\\item 'doublet'\n",
       "\\item 'seurat\\_clusters'\n",
       "\\item 'fsk\\_annot1\\_min30'\n",
       "\\item 'fsk\\_annot1\\_min30\\_prob'\n",
       "\\item 'fsk\\_annot2\\_min30'\n",
       "\\item 'fsk\\_annot2\\_min30\\_prob'\n",
       "\\item 'day8to13'\n",
       "\\item 'org\\_annot0'\n",
       "\\item 'org\\_annot0\\_prob'\n",
       "\\item 'org\\_annot30'\n",
       "\\item 'org\\_annot30\\_prob'\n",
       "\\item 'fsk\\_annot0'\n",
       "\\item 'fsk\\_annot0\\_prob'\n",
       "\\item 'fsk\\_annot30'\n",
       "\\item 'fsk\\_annot30\\_prob'\n",
       "\\item 'mad\\_prd'\n",
       "\\item 'Status'\n",
       "\\item 'Site'\n",
       "\\item 'Site\\_old'\n",
       "\\item 'Tissue'\n",
       "\\item 'Enrichment'\n",
       "\\item 'Location'\n",
       "\\item 'Sex'\n",
       "\\item 'Age'\n",
       "\\item 'stage'\n",
       "\\item 'anno\\_final'\n",
       "\\item 'final'\n",
       "\\item 'dataset'\n",
       "\\item 'log2pcw'\n",
       "\\item 'annot\\_early\\_late'\n",
       "\\item 'annot\\_early\\_late2'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sanger_id'\n",
       "2. 'chemistry'\n",
       "3. 'batch'\n",
       "4. 'gender'\n",
       "5. 'pcw'\n",
       "6. 'sorting'\n",
       "7. 'sample'\n",
       "8. 'chemistry_sorting'\n",
       "9. 'cell_caller'\n",
       "10. 'scrublet_score'\n",
       "11. 'cluster_scrublet_score'\n",
       "12. 'doublet_pval'\n",
       "13. 'bh_doublet_pval'\n",
       "14. 'nCounts_RNA'\n",
       "15. 'nFeaturess_RNA'\n",
       "16. 'percent_mito'\n",
       "17. 'percent_ribo'\n",
       "18. 'percent_hb'\n",
       "19. 'independent_annotation_refined'\n",
       "20. 'independent_annotation_broad1'\n",
       "21. 'independent_annotation_broad2'\n",
       "22. 'independent_annotation_broad3'\n",
       "23. 'independent_annotation_broad4'\n",
       "24. 'independent_annotation_broad5'\n",
       "25. 'joint_annotation'\n",
       "26. 'is_maternal'\n",
       "27. 'fig1b_annotation'\n",
       "28. 'independent_annotation_refined_20220201a'\n",
       "29. 'independent_annotation_refined_20220201b'\n",
       "30. 'independent_annotation_broad1_20220201'\n",
       "31. 'independent_annotation_broad2_20220201'\n",
       "32. 'independent_annotation_broad3_20220201'\n",
       "33. 'joint_annotation_20220201a'\n",
       "34. 'joint_annotation_20220201b'\n",
       "35. 'fig1b_annotation_20220201'\n",
       "36. 'joint_annotation_20220202'\n",
       "37. 'independent_annotation_refined_20220202'\n",
       "38. 'annot'\n",
       "39. 'day'\n",
       "40. 'sample_id'\n",
       "41. 'version'\n",
       "42. 'week'\n",
       "43. 'nh3_annot1'\n",
       "44. 'log1p_nCounts_RNA'\n",
       "45. 'log1p_nFeaturess_RNA'\n",
       "46. 'nCounts_RNA_mito'\n",
       "47. 'nCounts_RNA_ribo'\n",
       "48. 'nCounts_RNA_hb'\n",
       "49. 'percent_top50'\n",
       "50. 'pass_auto_filter'\n",
       "51. 'scrublet_score_z'\n",
       "52. 'bh_pval'\n",
       "53. 'scrublet_done'\n",
       "54. 'doublet'\n",
       "55. 'seurat_clusters'\n",
       "56. 'fsk_annot1_min30'\n",
       "57. 'fsk_annot1_min30_prob'\n",
       "58. 'fsk_annot2_min30'\n",
       "59. 'fsk_annot2_min30_prob'\n",
       "60. 'day8to13'\n",
       "61. 'org_annot0'\n",
       "62. 'org_annot0_prob'\n",
       "63. 'org_annot30'\n",
       "64. 'org_annot30_prob'\n",
       "65. 'fsk_annot0'\n",
       "66. 'fsk_annot0_prob'\n",
       "67. 'fsk_annot30'\n",
       "68. 'fsk_annot30_prob'\n",
       "69. 'mad_prd'\n",
       "70. 'Status'\n",
       "71. 'Site'\n",
       "72. 'Site_old'\n",
       "73. 'Tissue'\n",
       "74. 'Enrichment'\n",
       "75. 'Location'\n",
       "76. 'Sex'\n",
       "77. 'Age'\n",
       "78. 'stage'\n",
       "79. 'anno_final'\n",
       "80. 'final'\n",
       "81. 'dataset'\n",
       "82. 'log2pcw'\n",
       "83. 'annot_early_late'\n",
       "84. 'annot_early_late2'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sanger_id\"                               \n",
       " [2] \"chemistry\"                               \n",
       " [3] \"batch\"                                   \n",
       " [4] \"gender\"                                  \n",
       " [5] \"pcw\"                                     \n",
       " [6] \"sorting\"                                 \n",
       " [7] \"sample\"                                  \n",
       " [8] \"chemistry_sorting\"                       \n",
       " [9] \"cell_caller\"                             \n",
       "[10] \"scrublet_score\"                          \n",
       "[11] \"cluster_scrublet_score\"                  \n",
       "[12] \"doublet_pval\"                            \n",
       "[13] \"bh_doublet_pval\"                         \n",
       "[14] \"nCounts_RNA\"                             \n",
       "[15] \"nFeaturess_RNA\"                          \n",
       "[16] \"percent_mito\"                            \n",
       "[17] \"percent_ribo\"                            \n",
       "[18] \"percent_hb\"                              \n",
       "[19] \"independent_annotation_refined\"          \n",
       "[20] \"independent_annotation_broad1\"           \n",
       "[21] \"independent_annotation_broad2\"           \n",
       "[22] \"independent_annotation_broad3\"           \n",
       "[23] \"independent_annotation_broad4\"           \n",
       "[24] \"independent_annotation_broad5\"           \n",
       "[25] \"joint_annotation\"                        \n",
       "[26] \"is_maternal\"                             \n",
       "[27] \"fig1b_annotation\"                        \n",
       "[28] \"independent_annotation_refined_20220201a\"\n",
       "[29] \"independent_annotation_refined_20220201b\"\n",
       "[30] \"independent_annotation_broad1_20220201\"  \n",
       "[31] \"independent_annotation_broad2_20220201\"  \n",
       "[32] \"independent_annotation_broad3_20220201\"  \n",
       "[33] \"joint_annotation_20220201a\"              \n",
       "[34] \"joint_annotation_20220201b\"              \n",
       "[35] \"fig1b_annotation_20220201\"               \n",
       "[36] \"joint_annotation_20220202\"               \n",
       "[37] \"independent_annotation_refined_20220202\" \n",
       "[38] \"annot\"                                   \n",
       "[39] \"day\"                                     \n",
       "[40] \"sample_id\"                               \n",
       "[41] \"version\"                                 \n",
       "[42] \"week\"                                    \n",
       "[43] \"nh3_annot1\"                              \n",
       "[44] \"log1p_nCounts_RNA\"                       \n",
       "[45] \"log1p_nFeaturess_RNA\"                    \n",
       "[46] \"nCounts_RNA_mito\"                        \n",
       "[47] \"nCounts_RNA_ribo\"                        \n",
       "[48] \"nCounts_RNA_hb\"                          \n",
       "[49] \"percent_top50\"                           \n",
       "[50] \"pass_auto_filter\"                        \n",
       "[51] \"scrublet_score_z\"                        \n",
       "[52] \"bh_pval\"                                 \n",
       "[53] \"scrublet_done\"                           \n",
       "[54] \"doublet\"                                 \n",
       "[55] \"seurat_clusters\"                         \n",
       "[56] \"fsk_annot1_min30\"                        \n",
       "[57] \"fsk_annot1_min30_prob\"                   \n",
       "[58] \"fsk_annot2_min30\"                        \n",
       "[59] \"fsk_annot2_min30_prob\"                   \n",
       "[60] \"day8to13\"                                \n",
       "[61] \"org_annot0\"                              \n",
       "[62] \"org_annot0_prob\"                         \n",
       "[63] \"org_annot30\"                             \n",
       "[64] \"org_annot30_prob\"                        \n",
       "[65] \"fsk_annot0\"                              \n",
       "[66] \"fsk_annot0_prob\"                         \n",
       "[67] \"fsk_annot30\"                             \n",
       "[68] \"fsk_annot30_prob\"                        \n",
       "[69] \"mad_prd\"                                 \n",
       "[70] \"Status\"                                  \n",
       "[71] \"Site\"                                    \n",
       "[72] \"Site_old\"                                \n",
       "[73] \"Tissue\"                                  \n",
       "[74] \"Enrichment\"                              \n",
       "[75] \"Location\"                                \n",
       "[76] \"Sex\"                                     \n",
       "[77] \"Age\"                                     \n",
       "[78] \"stage\"                                   \n",
       "[79] \"anno_final\"                              \n",
       "[80] \"final\"                                   \n",
       "[81] \"dataset\"                                 \n",
       "[82] \"log2pcw\"                                 \n",
       "[83] \"annot_early_late\"                        \n",
       "[84] \"annot_early_late2\"                       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(pooled_ad@meta.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "willing-organic",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T09:17:40.490582Z",
     "iopub.status.busy": "2023-01-31T09:17:40.489091Z",
     "iopub.status.idle": "2023-01-31T09:17:40.522505Z",
     "shell.execute_reply": "2023-01-31T09:17:40.520915Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".\n",
       "             adult_Fibroblast                     adult_ILC \n",
       "                        17456                          7038 \n",
       "           adult_Keratinocyte         adult_Langerhans cell \n",
       "                        53844                         12545 \n",
       "  adult_Lymphatic endothelium              adult_Macrophage \n",
       "                         4924                          7205 \n",
       "              adult_Mast cell              adult_Melanocyte \n",
       "                          557                          3682 \n",
       "               adult_Monocyte              adult_Mural cell \n",
       "                         6685                          5006 \n",
       "            adult_Plasma cell            adult_Schwann cell \n",
       "                           71                           269 \n",
       "                 adult_T cell    adult_Vascular endothelium \n",
       "                        34774                         25785 \n",
       "                    adult_cDC               fetal_Adipocyte \n",
       "                        15898                          3203 \n",
       "                 fetal_B cell         fetal_Dermal papillia \n",
       "                         3087                          5637 \n",
       "              fetal_Erythroid              fetal_Fibroblast \n",
       "                         3468                         86785 \n",
       "                    fetal_ILC            fetal_Keratinocyte \n",
       "                         9473                          1469 \n",
       "        fetal_Langerhans cell   fetal_Lymphatic endothelium \n",
       "                          379                           582 \n",
       "             fetal_Macrophage               fetal_Mast cell \n",
       "                        24392                          1886 \n",
       "          fetal_Megakaryocyte              fetal_Melanocyte \n",
       "                          173                           794 \n",
       "               fetal_Monocyte              fetal_Mural cell \n",
       "                         6673                          9377 \n",
       "          fetal_Myofibroblast           fetal_Neuronal cell \n",
       "                         3154                          1704 \n",
       "             fetal_Neutrophil              fetal_Progenitor \n",
       "                          262                           582 \n",
       "           fetal_Schwann cell         fetal_Skeletal muscle \n",
       "                         3939                          5811 \n",
       "                 fetal_T cell    fetal_Vascular endothelium \n",
       "                         8078                          1591 \n",
       "                    fetal_cDC                     fetal_pDC \n",
       "                         3445                           589 \n",
       "           organoid_Adipocyte      organoid_Dermal papillia \n",
       "                         3832                          7318 \n",
       "          organoid_Fibroblast         organoid_Keratinocyte \n",
       "                        73990                         24589 \n",
       "          organoid_Melanocyte           organoid_Mural cell \n",
       "                         1195                           775 \n",
       "       organoid_Neuronal cell         organoid_Schwann cell \n",
       "                         8590                          2872 \n",
       "     organoid_Skeletal muscle organoid_Vascular endothelium \n",
       "                          543                           131 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pooled_ad@meta.data$annot %>% table()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "enclosed-gabriel",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:52:44.482508Z",
     "iopub.status.busy": "2023-01-31T13:52:44.481076Z",
     "iopub.status.idle": "2023-01-31T13:52:45.237657Z",
     "shell.execute_reply": "2023-01-31T13:52:45.235763Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "df = data.frame(pooled_ad@meta.data[grepl(\"fetal|organoid\", rownames(pooled_ad@meta.data)),])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "graduate-atlantic",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:52:45.241214Z",
     "iopub.status.busy": "2023-01-31T13:52:45.239987Z",
     "iopub.status.idle": "2023-01-31T13:52:50.190254Z",
     "shell.execute_reply": "2023-01-31T13:52:50.188361Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'fetal'</li><li>'organoid'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'fetal'\n",
       "\\item 'organoid'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'fetal'\n",
       "2. 'organoid'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"fetal\"    \"organoid\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(sapply(strsplit(rownames(df), split=\"-\"), tail, n=1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "specified-providence",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T13:38:19.290767Z",
     "iopub.status.busy": "2023-01-31T13:38:19.287272Z",
     "iopub.status.idle": "2023-01-31T13:38:19.616044Z",
     "shell.execute_reply": "2023-01-31T13:38:19.614562Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 310368 × 84</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sanger_id</th><th scope=col>chemistry</th><th scope=col>batch</th><th scope=col>gender</th><th scope=col>pcw</th><th scope=col>sorting</th><th scope=col>sample</th><th scope=col>chemistry_sorting</th><th scope=col>cell_caller</th><th scope=col>scrublet_score</th><th scope=col>⋯</th><th scope=col>Location</th><th scope=col>Sex</th><th scope=col>Age</th><th scope=col>stage</th><th scope=col>anno_final</th><th scope=col>final</th><th scope=col>dataset</th><th scope=col>log2pcw</th><th scope=col>annot_early_late</th><th scope=col>annot_early_late2</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.22580645</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAAGATGGTCGATTGT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.14960630</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Monocyte        </td><td>fetal_Monocyte         </td></tr>\n",
       "\t<tr><th scope=row>AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.20197044</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.16455696</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Progenitor      </td><td>fetal_Progenitor       </td></tr>\n",
       "\t<tr><th scope=row>AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.18181818</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.25443787</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.28947368</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AACACGTAGAGAGCTC-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.33333333</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_cDC             </td><td>fetal_cDC              </td></tr>\n",
       "\t<tr><th scope=row>AACCATGAGATCTGCT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.14960630</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_cDC             </td><td>fetal_cDC              </td></tr>\n",
       "\t<tr><th scope=row>AACCATGGTGCCTTGG-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.20197044</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>early-fetal_Macrophage</td><td>fetal_Macrophage, early</td></tr>\n",
       "\t<tr><th scope=row>AACTCAGCAGGACGTA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.10559006</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.18181818</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Monocyte        </td><td>fetal_Monocyte         </td></tr>\n",
       "\t<tr><th scope=row>AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.38983051</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AACTTTCAGTGCCATT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.20197044</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAGACCTCATGAAGTA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.22580645</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Empt</td><td>0.20197044</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.20197044</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.08988764</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.22580645</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AATCGGTAGTACGTAA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.11475410</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Monocyte        </td><td>fetal_Monocyte         </td></tr>\n",
       "\t<tr><th scope=row>AATCGGTAGTGGACGT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Cell</td><td>0.20197044</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.12500000</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>ACACCAATCCTCTAGC-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.14960630</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Monocyte        </td><td>fetal_Monocyte         </td></tr>\n",
       "\t<tr><th scope=row>ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.10559006</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Fibroblast      </td><td>fetal_Fibroblast       </td></tr>\n",
       "\t<tr><th scope=row>ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.25443787</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Macrophage      </td><td>fetal_Macrophage       </td></tr>\n",
       "\t<tr><th scope=row>ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.33333333</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_ILC             </td><td>fetal_ILC              </td></tr>\n",
       "\t<tr><th scope=row>ACATACGAGTGATCGG-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.16455696</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>early-fetal_ILC       </td><td>fetal_ILC, early       </td></tr>\n",
       "\t<tr><th scope=row>ACATCAGGTAAATACG-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.14960630</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_ILC             </td><td>fetal_ILC              </td></tr>\n",
       "\t<tr><th scope=row>ACATGGTCATCGGGTC-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.14960630</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_Monocyte        </td><td>fetal_Monocyte         </td></tr>\n",
       "\t<tr><th scope=row>ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal</th><td>4834STDY7002879</td><td>SC3Pv2</td><td>F16</td><td>male</td><td>8</td><td>CD45P</td><td>F16_male_8+1PCW</td><td>SC3Pv2_CD45P</td><td>Both</td><td>0.25443787</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>fetal</td><td>3</td><td>fetal_B cell          </td><td>fetal_B cell           </td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Skeletal muscle</td><td>organoid_Skeletal muscle</td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Skeletal muscle</td><td>organoid_Skeletal muscle</td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Neuronal cell  </td><td>organoid_Neuronal cell  </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Melanocyte     </td><td>organoid_Melanocyte     </td></tr>\n",
       "\t<tr><th scope=row>day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid</th><td>nan</td><td>nan</td><td>DSP</td><td>nan</td><td>2</td><td>nan</td><td>nan</td><td>nan</td><td>nan</td><td>NaN</td><td>⋯</td><td>nan</td><td>nan</td><td>NaN</td><td>nan</td><td>nan</td><td>nan</td><td>organoid</td><td>1</td><td>organoid_Keratinocyte   </td><td>organoid_Keratinocyte   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 310368 × 84\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & sanger\\_id & chemistry & batch & gender & pcw & sorting & sample & chemistry\\_sorting & cell\\_caller & scrublet\\_score & ⋯ & Location & Sex & Age & stage & anno\\_final & final & dataset & log2pcw & annot\\_early\\_late & annot\\_early\\_late2\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <dbl> & <fct> & <fct> & <fct> & <fct> & <dbl> & ⋯ & <fct> & <fct> & <dbl> & <fct> & <fct> & <fct> & <fct> & <dbl> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tAAACCTGGTCAGTGGA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.22580645 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAAGATGGTCGATTGT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.14960630 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Monocyte         & fetal\\_Monocyte         \\\\\n",
       "\tAAAGCAAAGATGTGGC-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.20197044 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAAGTAGCAGATCGGA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.16455696 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Progenitor       & fetal\\_Progenitor       \\\\\n",
       "\tAAAGTAGTCCGCATCT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.18181818 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAATGCCAGCTGCCCA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.25443787 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAATGCCTCGTGGTCG-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.28947368 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAACACGTAGAGAGCTC-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.33333333 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_cDC              & fetal\\_cDC              \\\\\n",
       "\tAACCATGAGATCTGCT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.14960630 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_cDC              & fetal\\_cDC              \\\\\n",
       "\tAACCATGGTGCCTTGG-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.20197044 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & early-fetal\\_Macrophage & fetal\\_Macrophage, early\\\\\n",
       "\tAACTCAGCAGGACGTA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.10559006 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAACTCAGCAGGTCTCG-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.18181818 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Monocyte         & fetal\\_Monocyte         \\\\\n",
       "\tAACTCAGGTCAGCTAT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.38983051 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAACTTTCAGTGCCATT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.20197044 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAGACCTCATGAAGTA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.22580645 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Empt & 0.20197044 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAGTCTGGTCGACTAT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.20197044 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.08988764 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAATCCAGGTGTTTGTG-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.22580645 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAATCGGTAGTACGTAA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.11475410 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Monocyte         & fetal\\_Monocyte         \\\\\n",
       "\tAATCGGTAGTGGACGT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Cell & 0.20197044 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tAATCGGTTCCTAGAAC-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.12500000 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tACACCAATCCTCTAGC-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.14960630 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Monocyte         & fetal\\_Monocyte         \\\\\n",
       "\tACACCCTGTCGTTGTA-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.10559006 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Fibroblast       & fetal\\_Fibroblast       \\\\\n",
       "\tACACTGAAGCCAGAAC-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.25443787 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Macrophage       & fetal\\_Macrophage       \\\\\n",
       "\tACAGCTAGTAAACGCG-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.33333333 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_ILC              & fetal\\_ILC              \\\\\n",
       "\tACATACGAGTGATCGG-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.16455696 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & early-fetal\\_ILC        & fetal\\_ILC, early       \\\\\n",
       "\tACATCAGGTAAATACG-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.14960630 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_ILC              & fetal\\_ILC              \\\\\n",
       "\tACATGGTCATCGGGTC-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.14960630 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_Monocyte         & fetal\\_Monocyte         \\\\\n",
       "\tACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal & 4834STDY7002879 & SC3Pv2 & F16 & male & 8 & CD45P & F16\\_male\\_8+1PCW & SC3Pv2\\_CD45P & Both & 0.25443787 & ⋯ & nan & nan & NaN & nan & nan & nan & fetal & 3 & fetal\\_B cell           & fetal\\_B cell           \\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGTTTCCAATGCAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCTCTCTCGACx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGTTGTCCAGTTAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTCGATTCATGCTAGx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGTTTCGGTCGGTx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTCCTCTCATAGACCx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCGTTAGGGTGx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGTTGTGAATGATx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Skeletal muscle & organoid\\_Skeletal muscle\\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCGTAGAATGTx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGACTCATGACTCAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGTTTCTCCCAACx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGAGGTGGAAGTCx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Skeletal muscle & organoid\\_Skeletal muscle\\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGTTGAGTCGCCCAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCTCAAACGAAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTCCTCCACTAGGTTx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGAGGTTATTCCTx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCCATCATCTTx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGAGTCGAGCACCx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGAGGTACAGAATx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGTTGGTACAGAATx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTCGATGTCGACGCTx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGTTGTCCGAGCTGx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCAGAGAGTGAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCCAATACCTGx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGGTTAGCGTGAGTx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTCCTCCATACAGAAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTCGATGTTAAAGTGx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCAGTACCCTAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Neuronal cell   & organoid\\_Neuronal cell  \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCTCTGGCCGAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Melanocyte      & organoid\\_Melanocyte     \\\\\n",
       "\tday\\_13\\_IEOdev\\_DSP\\_d13:TTTGATCCAGAACCGAx-organoid & nan & nan & DSP & nan & 2 & nan & nan & nan & nan & NaN & ⋯ & nan & nan & NaN & nan & nan & nan & organoid & 1 & organoid\\_Keratinocyte    & organoid\\_Keratinocyte   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 310368 × 84\n",
       "\n",
       "| <!--/--> | sanger_id &lt;fct&gt; | chemistry &lt;fct&gt; | batch &lt;fct&gt; | gender &lt;fct&gt; | pcw &lt;dbl&gt; | sorting &lt;fct&gt; | sample &lt;fct&gt; | chemistry_sorting &lt;fct&gt; | cell_caller &lt;fct&gt; | scrublet_score &lt;dbl&gt; | ⋯ ⋯ | Location &lt;fct&gt; | Sex &lt;fct&gt; | Age &lt;dbl&gt; | stage &lt;fct&gt; | anno_final &lt;fct&gt; | final &lt;fct&gt; | dataset &lt;fct&gt; | log2pcw &lt;dbl&gt; | annot_early_late &lt;fct&gt; | annot_early_late2 &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.22580645 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAAGATGGTCGATTGT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.14960630 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Monocyte         | fetal_Monocyte          |\n",
       "| AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.20197044 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.16455696 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Progenitor       | fetal_Progenitor        |\n",
       "| AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.18181818 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.25443787 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.28947368 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AACACGTAGAGAGCTC-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.33333333 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_cDC              | fetal_cDC               |\n",
       "| AACCATGAGATCTGCT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.14960630 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_cDC              | fetal_cDC               |\n",
       "| AACCATGGTGCCTTGG-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.20197044 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | early-fetal_Macrophage | fetal_Macrophage, early |\n",
       "| AACTCAGCAGGACGTA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.10559006 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.18181818 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Monocyte         | fetal_Monocyte          |\n",
       "| AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.38983051 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AACTTTCAGTGCCATT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.20197044 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAGACCTCATGAAGTA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.22580645 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Empt | 0.20197044 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.20197044 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.08988764 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.22580645 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AATCGGTAGTACGTAA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.11475410 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Monocyte         | fetal_Monocyte          |\n",
       "| AATCGGTAGTGGACGT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Cell | 0.20197044 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.12500000 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| ACACCAATCCTCTAGC-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.14960630 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Monocyte         | fetal_Monocyte          |\n",
       "| ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.10559006 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Fibroblast       | fetal_Fibroblast        |\n",
       "| ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.25443787 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Macrophage       | fetal_Macrophage        |\n",
       "| ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.33333333 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_ILC              | fetal_ILC               |\n",
       "| ACATACGAGTGATCGG-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.16455696 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | early-fetal_ILC        | fetal_ILC, early        |\n",
       "| ACATCAGGTAAATACG-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.14960630 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_ILC              | fetal_ILC               |\n",
       "| ACATGGTCATCGGGTC-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.14960630 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_Monocyte         | fetal_Monocyte          |\n",
       "| ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal | 4834STDY7002879 | SC3Pv2 | F16 | male | 8 | CD45P | F16_male_8+1PCW | SC3Pv2_CD45P | Both | 0.25443787 | ⋯ | nan | nan | NaN | nan | nan | nan | fetal | 3 | fetal_B cell           | fetal_B cell            |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Skeletal muscle | organoid_Skeletal muscle |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Skeletal muscle | organoid_Skeletal muscle |\n",
       "| day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Neuronal cell   | organoid_Neuronal cell   |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Melanocyte      | organoid_Melanocyte      |\n",
       "| day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid | nan | nan | DSP | nan | 2 | nan | nan | nan | nan | NaN | ⋯ | nan | nan | NaN | nan | nan | nan | organoid | 1 | organoid_Keratinocyte    | organoid_Keratinocyte    |\n",
       "\n"
      ],
      "text/plain": [
       "                                                 sanger_id       chemistry\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         4834STDY7002879 SC3Pv2   \n",
       "⋮                                                ⋮               ⋮        \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid nan             nan      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid nan             nan      \n",
       "                                                 batch gender pcw sorting\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         F16   male   8   CD45P  \n",
       "⋮                                                ⋮     ⋮      ⋮   ⋮      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid DSP   nan    2   nan    \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid DSP   nan    2   nan    \n",
       "                                                 sample         \n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         F16_male_8+1PCW\n",
       "⋮                                                ⋮              \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid nan            \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid nan            \n",
       "                                                 chemistry_sorting cell_caller\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Empt       \n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Cell       \n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         SC3Pv2_CD45P      Both       \n",
       "⋮                                                ⋮                 ⋮          \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid nan               nan        \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid nan               nan        \n",
       "                                                 scrublet_score ⋯ Location Sex\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         0.22580645     ⋯ nan      nan\n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         0.14960630     ⋯ nan      nan\n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         0.20197044     ⋯ nan      nan\n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         0.16455696     ⋯ nan      nan\n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         0.18181818     ⋯ nan      nan\n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         0.25443787     ⋯ nan      nan\n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         0.28947368     ⋯ nan      nan\n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         0.33333333     ⋯ nan      nan\n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         0.14960630     ⋯ nan      nan\n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         0.20197044     ⋯ nan      nan\n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         0.10559006     ⋯ nan      nan\n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         0.18181818     ⋯ nan      nan\n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         0.38983051     ⋯ nan      nan\n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         0.20197044     ⋯ nan      nan\n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         0.22580645     ⋯ nan      nan\n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         0.20197044     ⋯ nan      nan\n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         0.20197044     ⋯ nan      nan\n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         0.08988764     ⋯ nan      nan\n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         0.22580645     ⋯ nan      nan\n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         0.11475410     ⋯ nan      nan\n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         0.20197044     ⋯ nan      nan\n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         0.12500000     ⋯ nan      nan\n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         0.14960630     ⋯ nan      nan\n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         0.10559006     ⋯ nan      nan\n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         0.25443787     ⋯ nan      nan\n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         0.33333333     ⋯ nan      nan\n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         0.16455696     ⋯ nan      nan\n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         0.14960630     ⋯ nan      nan\n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         0.14960630     ⋯ nan      nan\n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         0.25443787     ⋯ nan      nan\n",
       "⋮                                                ⋮              ⋱ ⋮        ⋮  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid NaN            ⋯ nan      nan\n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid NaN            ⋯ nan      nan\n",
       "                                                 Age stage anno_final final\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         NaN nan   nan        nan  \n",
       "⋮                                                ⋮   ⋮     ⋮          ⋮    \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid NaN nan   nan        nan  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid NaN nan   nan        nan  \n",
       "                                                 dataset  log2pcw\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         fetal    3      \n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         fetal    3      \n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         fetal    3      \n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         fetal    3      \n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         fetal    3      \n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         fetal    3      \n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         fetal    3      \n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         fetal    3      \n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         fetal    3      \n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         fetal    3      \n",
       "⋮                                                ⋮        ⋮      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid organoid 1      \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid organoid 1      \n",
       "                                                 annot_early_late        \n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         fetal_Progenitor        \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         fetal_cDC               \n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         fetal_cDC               \n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         early-fetal_Macrophage  \n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         fetal_Fibroblast        \n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         fetal_ILC               \n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         early-fetal_ILC         \n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         fetal_ILC               \n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         fetal_B cell            \n",
       "⋮                                                ⋮                       \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid organoid_Skeletal muscle\n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid organoid_Skeletal muscle\n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid organoid_Melanocyte     \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid organoid_Keratinocyte   \n",
       "                                                 annot_early_late2       \n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal         fetal_Progenitor        \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAATGCCTCGTGGTCG-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AACACGTAGAGAGCTC-1-4834STDY7002879-fetal         fetal_cDC               \n",
       "AACCATGAGATCTGCT-1-4834STDY7002879-fetal         fetal_cDC               \n",
       "AACCATGGTGCCTTGG-1-4834STDY7002879-fetal         fetal_Macrophage, early \n",
       "AACTCAGCAGGACGTA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AACTCAGCAGGTCTCG-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "AACTCAGGTCAGCTAT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AACTTTCAGTGCCATT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGACCTCATGAAGTA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGCCGCAGTGTCTCA-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGTCTGGTCGACTAT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AAGTCTGTCGCCTGTT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AATCCAGGTGTTTGTG-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AATCGGTAGTACGTAA-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "AATCGGTAGTGGACGT-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "AATCGGTTCCTAGAAC-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "ACACCAATCCTCTAGC-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "ACACCCTGTCGTTGTA-1-4834STDY7002879-fetal         fetal_Fibroblast        \n",
       "ACACTGAAGCCAGAAC-1-4834STDY7002879-fetal         fetal_Macrophage        \n",
       "ACAGCTAGTAAACGCG-1-4834STDY7002879-fetal         fetal_ILC               \n",
       "ACATACGAGTGATCGG-1-4834STDY7002879-fetal         fetal_ILC, early        \n",
       "ACATCAGGTAAATACG-1-4834STDY7002879-fetal         fetal_ILC               \n",
       "ACATGGTCATCGGGTC-1-4834STDY7002879-fetal         fetal_Monocyte          \n",
       "ACCAGTAAGGCGCTCT-1-4834STDY7002879-fetal         fetal_B cell            \n",
       "⋮                                                ⋮                       \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCCAATGCAx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTCTCGACx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTCCAGTTAx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTCGATTCATGCTAGx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCGGTCGGTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCTCATAGACCx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTTAGGGTGx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTGTGAATGATx-organoid organoid_Skeletal muscle\n",
       "day_13_IEOdev_DSP_d13:TTTGATCGTAGAATGTx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGACTCATGACTCAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTTCTCCCAACx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTGGAAGTCx-organoid organoid_Skeletal muscle\n",
       "day_13_IEOdev_DSP_d13:TTTGTTGAGTCGCCCAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCAAACGAAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCACTAGGTTx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTTATTCCTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCATCATCTTx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGTCGAGCACCx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGAGGTACAGAATx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGGTACAGAATx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTCGACGCTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGTTGTCCGAGCTGx-organoid organoid_Keratinocyte   \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGAGAGTGAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAATACCTGx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGGTTAGCGTGAGTx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCCTCCATACAGAAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTCGATGTTAAAGTGx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCAGTACCCTAx-organoid organoid_Neuronal cell  \n",
       "day_13_IEOdev_DSP_d13:TTTGATCTCTGGCCGAx-organoid organoid_Melanocyte     \n",
       "day_13_IEOdev_DSP_d13:TTTGATCCAGAACCGAx-organoid organoid_Keratinocyte   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "declared-preview",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:52:50.193554Z",
     "iopub.status.busy": "2023-01-31T13:52:50.192483Z",
     "iopub.status.idle": "2023-01-31T13:53:01.835283Z",
     "shell.execute_reply": "2023-01-31T13:53:01.833693Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "check_column_string = function(df, string) {\n",
    "  sapply(df, function(x) sum(grepl(string, x))) > 0\n",
    "}\n",
    "\n",
    "# get the column names which contain the string\n",
    "cols_with_string = names(df)[check_column_string(df, \"Tip cell\")]\n",
    "\n",
    "# subset the data frame with the column names\n",
    "df = df[cols_with_string]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "noble-prospect",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:53:13.956746Z",
     "iopub.status.busy": "2023-01-31T13:53:13.955424Z",
     "iopub.status.idle": "2023-01-31T13:53:13.969550Z",
     "shell.execute_reply": "2023-01-31T13:53:13.968230Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'independent_annotation_refined'</li><li>'joint_annotation'</li><li>'independent_annotation_refined_20220201a'</li><li>'independent_annotation_refined_20220201b'</li><li>'joint_annotation_20220201a'</li><li>'joint_annotation_20220201b'</li><li>'joint_annotation_20220202'</li><li>'independent_annotation_refined_20220202'</li><li>'fsk_annot1_min30'</li><li>'fsk_annot2_min30'</li><li>'org_annot0'</li><li>'org_annot30'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'independent\\_annotation\\_refined'\n",
       "\\item 'joint\\_annotation'\n",
       "\\item 'independent\\_annotation\\_refined\\_20220201a'\n",
       "\\item 'independent\\_annotation\\_refined\\_20220201b'\n",
       "\\item 'joint\\_annotation\\_20220201a'\n",
       "\\item 'joint\\_annotation\\_20220201b'\n",
       "\\item 'joint\\_annotation\\_20220202'\n",
       "\\item 'independent\\_annotation\\_refined\\_20220202'\n",
       "\\item 'fsk\\_annot1\\_min30'\n",
       "\\item 'fsk\\_annot2\\_min30'\n",
       "\\item 'org\\_annot0'\n",
       "\\item 'org\\_annot30'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'independent_annotation_refined'\n",
       "2. 'joint_annotation'\n",
       "3. 'independent_annotation_refined_20220201a'\n",
       "4. 'independent_annotation_refined_20220201b'\n",
       "5. 'joint_annotation_20220201a'\n",
       "6. 'joint_annotation_20220201b'\n",
       "7. 'joint_annotation_20220202'\n",
       "8. 'independent_annotation_refined_20220202'\n",
       "9. 'fsk_annot1_min30'\n",
       "10. 'fsk_annot2_min30'\n",
       "11. 'org_annot0'\n",
       "12. 'org_annot30'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"independent_annotation_refined\"          \n",
       " [2] \"joint_annotation\"                        \n",
       " [3] \"independent_annotation_refined_20220201a\"\n",
       " [4] \"independent_annotation_refined_20220201b\"\n",
       " [5] \"joint_annotation_20220201a\"              \n",
       " [6] \"joint_annotation_20220201b\"              \n",
       " [7] \"joint_annotation_20220202\"               \n",
       " [8] \"independent_annotation_refined_20220202\" \n",
       " [9] \"fsk_annot1_min30\"                        \n",
       "[10] \"fsk_annot2_min30\"                        \n",
       "[11] \"org_annot0\"                              \n",
       "[12] \"org_annot30\"                             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cols_with_string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "decreased-karma",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:56:28.406723Z",
     "iopub.status.busy": "2023-01-31T13:56:28.405372Z",
     "iopub.status.idle": "2023-01-31T13:56:29.398900Z",
     "shell.execute_reply": "2023-01-31T13:56:29.397449Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"independent_annotation_refined\"\n",
      "[1] Tip cell (arterial)\n",
      "79 Levels: ASDC Activated mac Adipocytes Angiogenic pericyte ... pDC\n",
      "[1] \"joint_annotation\"\n",
      "[1] Tip cell (arterial)\n",
      "85 Levels: ASDC Activated mac Adipocytes Angiogenic pericyte ... pDC\n",
      "[1] \"independent_annotation_refined_20220201a\"\n",
      "[1] Tip cell (arterial)\n",
      "79 Levels: ASDC Activated mac Adipocytes Arterial B cell ... pDC\n",
      "[1] \"independent_annotation_refined_20220201b\"\n",
      "[1] Tip cell (arterial)\n",
      "78 Levels: ASDC Activated mac Adipocytes Arterial B cell ... pDC\n",
      "[1] \"joint_annotation_20220201a\"\n",
      "[1] Tip cell (arterial)\n",
      "85 Levels: ASDC Activated mac Adipocytes Arterial B cell Basal ... pDC\n",
      "[1] \"joint_annotation_20220201b\"\n",
      "[1] Tip cell (arterial)\n",
      "84 Levels: ASDC Activated mac Adipocytes Arterial B cell Basal ... pDC\n",
      "[1] \"joint_annotation_20220202\"\n",
      "[1] Tip cell (arterial)\n",
      "84 Levels: ASDC Adipocytes Arterial B cell Basal CD4 T cell ... pDC\n",
      "[1] \"independent_annotation_refined_20220202\"\n",
      "[1] Tip cell (arterial)\n",
      "78 Levels: ASDC Adipocytes Arterial B cell Basal KC CD4 T cell ... pDC\n",
      "[1] \"fsk_annot1_min30\"\n",
      "[1] fetal_Tip cell (arterial)\n",
      "36 Levels: Uncertain fetal_Adipocytes fetal_Basal ... nan\n",
      "[1] \"fsk_annot2_min30\"\n",
      "[1] Tip cell (arterial)\n",
      "35 Levels: Adipocytes Angiogenic pericyte Basal Basal POSTN+ ... nan\n",
      "[1] \"org_annot0\"\n",
      "[1] organoid_Tip cell (arterial)\n",
      "34 Levels: nan organoid_Adipocytes organoid_Basal ... organoid_Tip cell (arterial)\n",
      "[1] \"org_annot30\"\n",
      "[1] organoid_Tip cell (arterial)\n",
      "35 Levels: Uncertain nan organoid_Adipocytes ... organoid_Tip cell (arterial)\n"
     ]
    }
   ],
   "source": [
    "for (c in colnames(df)){\n",
    "    print(c)\n",
    "    print(unique(df[[c]][grepl(\"Tip cell\", df[[c]])]))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "modular-classic",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T13:59:10.901691Z",
     "iopub.status.busy": "2023-01-31T13:59:10.900471Z",
     "iopub.status.idle": "2023-01-31T13:59:10.991423Z",
     "shell.execute_reply": "2023-01-31T13:59:10.990125Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "130"
      ],
      "text/latex": [
       "130"
      ],
      "text/markdown": [
       "130"
      ],
      "text/plain": [
       "[1] 130"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(df$org_annot30[grepl(\"Tip cell\", df$org_annot30)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "greater-upper",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T14:59:48.431216Z",
     "iopub.status.busy": "2023-01-31T14:59:48.429798Z",
     "iopub.status.idle": "2023-01-31T14:59:48.518752Z",
     "shell.execute_reply": "2023-01-31T14:59:48.517235Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "Tip cell (arterial)\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'Adipocytes'</li><li>'Angiogenic pericyte'</li><li>'Basal'</li><li>'Basal POSTN+'</li><li>'Basal SHH+'</li><li>'Companion layer'</li><li>'Dermal condensate'</li><li>'Dermal papillia'</li><li>'Early fibroblast FRZB+'</li><li>'Early fibroblast HOXC5+'</li><li>'Early myocytes'</li><li>'Fibroblast PEAR1+'</li><li>'Fibroblast WNT2+'</li><li>'Immature basal'</li><li>'Immature suprabasal'</li><li>'MEMP - Early erythroid'</li><li>'Melanoblast'</li><li>'Melanocyte'</li><li>'Myelinating Schwann cells'</li><li>'Myoblasts'</li><li>'Myocytes'</li><li>'Myofibroblasts'</li><li>'Neuroendocrine'</li><li>'Neuron progenitors'</li><li>'Outer root sheath'</li><li>'Pericytes'</li><li>'Periderm'</li><li>'Pre-dermal condensate'</li><li>'SPP1+ proliferating neuron proneitors'</li><li>'Schwann cellls PID1+'</li><li>'Schwann/Schwann precursors'</li><li>'Suprabasal IFE'</li><li>'Tip cell (arterial)'</li><li>'Uncertain'</li><li>'nan'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "Tip cell (arterial)\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Adipocytes'\n",
       "\\item 'Angiogenic pericyte'\n",
       "\\item 'Basal'\n",
       "\\item 'Basal POSTN+'\n",
       "\\item 'Basal SHH+'\n",
       "\\item 'Companion layer'\n",
       "\\item 'Dermal condensate'\n",
       "\\item 'Dermal papillia'\n",
       "\\item 'Early fibroblast FRZB+'\n",
       "\\item 'Early fibroblast HOXC5+'\n",
       "\\item 'Early myocytes'\n",
       "\\item 'Fibroblast PEAR1+'\n",
       "\\item 'Fibroblast WNT2+'\n",
       "\\item 'Immature basal'\n",
       "\\item 'Immature suprabasal'\n",
       "\\item 'MEMP - Early erythroid'\n",
       "\\item 'Melanoblast'\n",
       "\\item 'Melanocyte'\n",
       "\\item 'Myelinating Schwann cells'\n",
       "\\item 'Myoblasts'\n",
       "\\item 'Myocytes'\n",
       "\\item 'Myofibroblasts'\n",
       "\\item 'Neuroendocrine'\n",
       "\\item 'Neuron progenitors'\n",
       "\\item 'Outer root sheath'\n",
       "\\item 'Pericytes'\n",
       "\\item 'Periderm'\n",
       "\\item 'Pre-dermal condensate'\n",
       "\\item 'SPP1+ proliferating neuron proneitors'\n",
       "\\item 'Schwann cellls PID1+'\n",
       "\\item 'Schwann/Schwann precursors'\n",
       "\\item 'Suprabasal IFE'\n",
       "\\item 'Tip cell (arterial)'\n",
       "\\item 'Uncertain'\n",
       "\\item 'nan'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "Tip cell (arterial)\n",
       "**Levels**: 1. 'Adipocytes'\n",
       "2. 'Angiogenic pericyte'\n",
       "3. 'Basal'\n",
       "4. 'Basal POSTN+'\n",
       "5. 'Basal SHH+'\n",
       "6. 'Companion layer'\n",
       "7. 'Dermal condensate'\n",
       "8. 'Dermal papillia'\n",
       "9. 'Early fibroblast FRZB+'\n",
       "10. 'Early fibroblast HOXC5+'\n",
       "11. 'Early myocytes'\n",
       "12. 'Fibroblast PEAR1+'\n",
       "13. 'Fibroblast WNT2+'\n",
       "14. 'Immature basal'\n",
       "15. 'Immature suprabasal'\n",
       "16. 'MEMP - Early erythroid'\n",
       "17. 'Melanoblast'\n",
       "18. 'Melanocyte'\n",
       "19. 'Myelinating Schwann cells'\n",
       "20. 'Myoblasts'\n",
       "21. 'Myocytes'\n",
       "22. 'Myofibroblasts'\n",
       "23. 'Neuroendocrine'\n",
       "24. 'Neuron progenitors'\n",
       "25. 'Outer root sheath'\n",
       "26. 'Pericytes'\n",
       "27. 'Periderm'\n",
       "28. 'Pre-dermal condensate'\n",
       "29. 'SPP1+ proliferating neuron proneitors'\n",
       "30. 'Schwann cellls PID1+'\n",
       "31. 'Schwann/Schwann precursors'\n",
       "32. 'Suprabasal IFE'\n",
       "33. 'Tip cell (arterial)'\n",
       "34. 'Uncertain'\n",
       "35. 'nan'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] Tip cell (arterial)\n",
       "35 Levels: Adipocytes Angiogenic pericyte Basal Basal POSTN+ ... nan"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(df$fsk_annot2_min30[grepl(\"Tip cell\", df$fsk_annot2_min30)])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bigger-veteran",
   "metadata": {},
   "source": [
    "<Font color=\"red\">~130 cells are not enough to produce statistically robust differential expression results</Font>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "bright-trouble",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T20:45:26.655908Z",
     "iopub.status.busy": "2023-01-31T20:45:26.654532Z",
     "iopub.status.idle": "2023-01-31T20:45:26.758396Z",
     "shell.execute_reply": "2023-01-31T20:45:26.756845Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>AGGATCTAGTTGAAAC-1-2-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AGGGCTCCAAGCGCAA-1-2-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTTACAGTCCGCTGTT-1-2-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCGACCTGTACATACC-1-2-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CAATACGAGAAACTGT-1-4-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTTAGACCAGTAGAGC-1-4-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTCGTTCCAATGGGTG-1-5-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTGAGGATCTAGTGTG-1-5-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTTACCCCATCGTGGC-1-5-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TACCGAACAAAGGGTC-1-5-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCAAGACCAACCAATC-1-5-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTGACCCTCCCTTCCC-1-5-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGAGCATCAGCAGTTT-1-6-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTAATGGGTCTCAACA-1-7-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GCGAGAATCTGTTGAG-1-7-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTATCTTCATGTTCCC-1-7-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTCTTAGGTGGGTCAA-1-7-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AAACCCATCTCGACGG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AAGCATCTCCCACAGG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AAGCGTTGTTCTCCCA-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACTTAGGGTTGTTGTG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AGAGCAGTCGAGTGGA-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AGCTACAAGACAACTA-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ATCGTGATCGGTAGAG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ATTACTCCACAGCTTA-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CAACGGCTCTCACTCG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CATGCCTGTTCTAACG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CATGCGGCAATGAAAC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCAATTTTCCGATGTA-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCGAACGAGATTGACA-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGGCAGTAGAGTGACC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGGGACTCACTGGATT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGTAAGTAGTAATCCC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTACCCATCCGTGTCT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTACCTGAGCTGTGCC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTGCCTACAGTGACCC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTGTGGGTCACTGTTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTTCGGTCAGCGACAA-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GAGTCTATCTTGCGCT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGAATCTCATAGCACT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGACGTCGTCTTACAG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGAGAACCAAAGAGTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGAGGATTCGCTATTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGATGTTCATAATCGC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGCTGTGGTTCCATTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGGATCCAGTTCGGTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTCGTTCTCGTAGTGT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTGAGCCGTTCTTGTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTGTTAGCAAGCGAGT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTTATGGTCGAACACT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TAACCAGAGGGCCCTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TAATTCCTCGTGAGAG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TATTGGGAGTACGTCT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCCATCGCAGCGTTGC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCTCACGTCTCAACCC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCTCTGGGTTCTTGTT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGCAGATAGTTCCGGC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGGAACTTCAAGGAGC-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTGGATGCAATCACGT-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTGGGTACAATACCTG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTTGTTGCAGAGCTAG-1-8-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AGCTCAAGTACGACTT-1-9-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCTCACGAGACTTGTC-1-9-organoid</dt><dd>organoid_Early fibroblasts FRZB+</dd><dt>AACCGCGGTACCAGTT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACACCCTGTAGCGTAG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACCAGTACATGTAAGA-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACGATACTCAACTCTT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACGCCAGGTATATGGA-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACTGAACGTGACTCAT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACTGTCCGTATTACCG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AGTTGGTTCAAGGCTT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CAGCTAATCATATCGG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CATCGAAAGGGTATCG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGGAGCTGTATGGTTC-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGTGTAATCTTACCTA-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTAGAGTTCTTAGAGC-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTCCTAGCAAGTCTGT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTCCTAGTCCTCCTAG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTCGAAATCGGCATCG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTTCTCTGTTATCACG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GACGGCTGTGCCTGTG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GCAGTTAAGATGTGGC-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGATGTTGTTATGTGC-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGGCACTAGCGCCTTG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTGAAGGCAAGAAGAG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTGCGGTCACCTCGGA-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TACTTACCACCATGTA-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCTTTCCCAATAGAGT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGGACGCCAAATCCGT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGGCTGGAGTGAATTG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGTGTTTTCCATGAAC-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTAGGACAGTGAACAT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTAGGCACAGGGTATG-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTCTTAGTCCGCGTTT-1-10-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AACGTTGAGGATGTAT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AACTCTTTCCCGGATG-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AAGGTTCTCACATGCA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>ACTGATGGTATCGCAT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AGAGCGATCGCCAGCA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>AGGGTGAGTTCCATGA-1-11-organoid</dt><dd>organoid_Smooth muscle/Pericytes</dd><dt>ATAGACCTCTTAGAGC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CAACTAGCATCGGTTA-1-11-organoid</dt><dd>organoid_Smooth muscle/Pericytes</dd><dt>CATCCACAGGCATGGT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCACGGATCTTGAGGT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCGGTAGCACTGTCGG-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCGGTAGGTTCCACGG-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCGTACTAGCACAGGT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCTTCCCTCCCTTGCA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CCTTCGAAGCGTGTCC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGAGCCATCTCTAGGA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGCTTCACATTCCTGC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGTGTAAAGAGGGCTT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CGTTAGATCACAGGCC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTCTACGCACGTCTCT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTCTGGTGTAGCGTCC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTGATAGAGTTCCACA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>CTTAGGACAAAGTCAA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GACCAATAGAGCTGGT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GCAAACTAGAGTCGGT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GCATGATTCATAACCG-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGGATGACAGACGCTC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GGGTCTGGTGTGAATA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTCACAAAGTATCTCG-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTTAAGCAGCCAGAAC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>GTTACAGAGGACGAAA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCACGAAAGAGTCGGT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCATTACGTGCATCTA-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TCGGGACAGGCTAGAC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGGCCAGCATGGTTGT-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TGGCCAGGTCTAGAGG-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>TTCTACACACTTAAGC-1-11-organoid</dt><dd>organoid_Tip cell (arterial)</dd><dt>day_13_IEOdev_DSP_d13:CGTGAATTCGGTAACTx-organoid</dt><dd>organoid_Tip cell (arterial)</dd></dl>\n",
       "\n",
       "<details>\n",
       "\t<summary style=display:list-item;cursor:pointer>\n",
       "\t\t<strong>Levels</strong>:\n",
       "\t</summary>\n",
       "\t<style>\n",
       "\t.list-inline {list-style: none; margin:0; padding: 0}\n",
       "\t.list-inline>li {display: inline-block}\n",
       "\t.list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "\t</style>\n",
       "\t<ol class=list-inline><li>'Uncertain'</li><li>'nan'</li><li>'organoid_Adipocytes'</li><li>'organoid_Basal'</li><li>'organoid_Basal POSTN+'</li><li>'organoid_Basal SHH+'</li><li>'organoid_Companion layer'</li><li>'organoid_Cuticle/cortex'</li><li>'organoid_Dermal condensate'</li><li>'organoid_Dermal papillia'</li><li>'organoid_Early fibroblasts FRZB+'</li><li>'organoid_Early fibroblasts HOXC5'</li><li>'organoid_Early myocytes'</li><li>'organoid_Fibroblasts POSTN+'</li><li>'organoid_Immature basal'</li><li>'organoid_Immature suprabasal'</li><li>'organoid_Inner root sheath'</li><li>'organoid_Melanoblasts'</li><li>'organoid_Melanocytes'</li><li>'organoid_Merkel cells'</li><li>'organoid_Myelinating Schwann cells'</li><li>'organoid_Myoblasts'</li><li>'organoid_Myocytes'</li><li>'organoid_Neuroepithelial cells'</li><li>'organoid_Neuron progenitors'</li><li>'organoid_Neuron progenitors SPP1+'</li><li>'organoid_Outer root sheath'</li><li>'organoid_Periderm'</li><li>'organoid_Pre-dermal condensate'</li><li>'organoid_Proliferating melanoblasts'</li><li>'organoid_Schwann cells/Schwann precursors'</li><li>'organoid_Secreting neuron'</li><li>'organoid_Smooth muscle/Pericytes'</li><li>'organoid_Suprabasal IFE'</li><li>'organoid_Tip cell (arterial)'</li></ol>\n",
       "</details>"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[AGGATCTAGTTGAAAC-1-2-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AGGGCTCCAAGCGCAA-1-2-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTTACAGTCCGCTGTT-1-2-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCGACCTGTACATACC-1-2-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CAATACGAGAAACTGT-1-4-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTTAGACCAGTAGAGC-1-4-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTCGTTCCAATGGGTG-1-5-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTGAGGATCTAGTGTG-1-5-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTTACCCCATCGTGGC-1-5-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TACCGAACAAAGGGTC-1-5-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCAAGACCAACCAATC-1-5-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTGACCCTCCCTTCCC-1-5-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGAGCATCAGCAGTTT-1-6-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTAATGGGTCTCAACA-1-7-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GCGAGAATCTGTTGAG-1-7-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTATCTTCATGTTCCC-1-7-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTCTTAGGTGGGTCAA-1-7-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AAACCCATCTCGACGG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AAGCATCTCCCACAGG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AAGCGTTGTTCTCCCA-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACTTAGGGTTGTTGTG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AGAGCAGTCGAGTGGA-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AGCTACAAGACAACTA-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ATCGTGATCGGTAGAG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ATTACTCCACAGCTTA-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CAACGGCTCTCACTCG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CATGCCTGTTCTAACG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CATGCGGCAATGAAAC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCAATTTTCCGATGTA-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCGAACGAGATTGACA-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGGCAGTAGAGTGACC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGGGACTCACTGGATT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGTAAGTAGTAATCCC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTACCCATCCGTGTCT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTACCTGAGCTGTGCC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTGCCTACAGTGACCC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTGTGGGTCACTGTTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTTCGGTCAGCGACAA-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GAGTCTATCTTGCGCT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGAATCTCATAGCACT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGACGTCGTCTTACAG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGAGAACCAAAGAGTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGAGGATTCGCTATTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGATGTTCATAATCGC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGCTGTGGTTCCATTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGGATCCAGTTCGGTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTCGTTCTCGTAGTGT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTGAGCCGTTCTTGTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTGTTAGCAAGCGAGT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTTATGGTCGAACACT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TAACCAGAGGGCCCTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TAATTCCTCGTGAGAG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TATTGGGAGTACGTCT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCCATCGCAGCGTTGC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCTCACGTCTCAACCC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCTCTGGGTTCTTGTT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGCAGATAGTTCCGGC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGGAACTTCAAGGAGC-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTGGATGCAATCACGT-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTGGGTACAATACCTG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTTGTTGCAGAGCTAG-1-8-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AGCTCAAGTACGACTT-1-9-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCTCACGAGACTTGTC-1-9-organoid] organoid\\_Early fibroblasts FRZB+\n",
       "\\item[AACCGCGGTACCAGTT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACACCCTGTAGCGTAG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACCAGTACATGTAAGA-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACGATACTCAACTCTT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACGCCAGGTATATGGA-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACTGAACGTGACTCAT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACTGTCCGTATTACCG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AGTTGGTTCAAGGCTT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CAGCTAATCATATCGG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CATCGAAAGGGTATCG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGGAGCTGTATGGTTC-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGTGTAATCTTACCTA-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTAGAGTTCTTAGAGC-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTCCTAGCAAGTCTGT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTCCTAGTCCTCCTAG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTCGAAATCGGCATCG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTTCTCTGTTATCACG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GACGGCTGTGCCTGTG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GCAGTTAAGATGTGGC-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGATGTTGTTATGTGC-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGGCACTAGCGCCTTG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTGAAGGCAAGAAGAG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTGCGGTCACCTCGGA-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TACTTACCACCATGTA-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCTTTCCCAATAGAGT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGGACGCCAAATCCGT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGGCTGGAGTGAATTG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGTGTTTTCCATGAAC-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTAGGACAGTGAACAT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTAGGCACAGGGTATG-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTCTTAGTCCGCGTTT-1-10-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AACGTTGAGGATGTAT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AACTCTTTCCCGGATG-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AAGGTTCTCACATGCA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[ACTGATGGTATCGCAT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AGAGCGATCGCCAGCA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[AGGGTGAGTTCCATGA-1-11-organoid] organoid\\_Smooth muscle/Pericytes\n",
       "\\item[ATAGACCTCTTAGAGC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CAACTAGCATCGGTTA-1-11-organoid] organoid\\_Smooth muscle/Pericytes\n",
       "\\item[CATCCACAGGCATGGT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCACGGATCTTGAGGT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCGGTAGCACTGTCGG-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCGGTAGGTTCCACGG-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCGTACTAGCACAGGT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCTTCCCTCCCTTGCA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CCTTCGAAGCGTGTCC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGAGCCATCTCTAGGA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGCTTCACATTCCTGC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGTGTAAAGAGGGCTT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CGTTAGATCACAGGCC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTCTACGCACGTCTCT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTCTGGTGTAGCGTCC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTGATAGAGTTCCACA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[CTTAGGACAAAGTCAA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GACCAATAGAGCTGGT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GCAAACTAGAGTCGGT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GCATGATTCATAACCG-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGGATGACAGACGCTC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GGGTCTGGTGTGAATA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTCACAAAGTATCTCG-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTTAAGCAGCCAGAAC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[GTTACAGAGGACGAAA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCACGAAAGAGTCGGT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCATTACGTGCATCTA-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TCGGGACAGGCTAGAC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGGCCAGCATGGTTGT-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TGGCCAGGTCTAGAGG-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[TTCTACACACTTAAGC-1-11-organoid] organoid\\_Tip cell (arterial)\n",
       "\\item[day\\textbackslash{}\\_13\\textbackslash{}\\_IEOdev\\textbackslash{}\\_DSP\\textbackslash{}\\_d13:CGTGAATTCGGTAACTx-organoid] organoid\\_Tip cell (arterial)\n",
       "\\end{description*}\n",
       "\n",
       "\\emph{Levels}: \\begin{enumerate*}\n",
       "\\item 'Uncertain'\n",
       "\\item 'nan'\n",
       "\\item 'organoid\\_Adipocytes'\n",
       "\\item 'organoid\\_Basal'\n",
       "\\item 'organoid\\_Basal POSTN+'\n",
       "\\item 'organoid\\_Basal SHH+'\n",
       "\\item 'organoid\\_Companion layer'\n",
       "\\item 'organoid\\_Cuticle/cortex'\n",
       "\\item 'organoid\\_Dermal condensate'\n",
       "\\item 'organoid\\_Dermal papillia'\n",
       "\\item 'organoid\\_Early fibroblasts FRZB+'\n",
       "\\item 'organoid\\_Early fibroblasts HOXC5'\n",
       "\\item 'organoid\\_Early myocytes'\n",
       "\\item 'organoid\\_Fibroblasts POSTN+'\n",
       "\\item 'organoid\\_Immature basal'\n",
       "\\item 'organoid\\_Immature suprabasal'\n",
       "\\item 'organoid\\_Inner root sheath'\n",
       "\\item 'organoid\\_Melanoblasts'\n",
       "\\item 'organoid\\_Melanocytes'\n",
       "\\item 'organoid\\_Merkel cells'\n",
       "\\item 'organoid\\_Myelinating Schwann cells'\n",
       "\\item 'organoid\\_Myoblasts'\n",
       "\\item 'organoid\\_Myocytes'\n",
       "\\item 'organoid\\_Neuroepithelial cells'\n",
       "\\item 'organoid\\_Neuron progenitors'\n",
       "\\item 'organoid\\_Neuron progenitors SPP1+'\n",
       "\\item 'organoid\\_Outer root sheath'\n",
       "\\item 'organoid\\_Periderm'\n",
       "\\item 'organoid\\_Pre-dermal condensate'\n",
       "\\item 'organoid\\_Proliferating melanoblasts'\n",
       "\\item 'organoid\\_Schwann cells/Schwann precursors'\n",
       "\\item 'organoid\\_Secreting neuron'\n",
       "\\item 'organoid\\_Smooth muscle/Pericytes'\n",
       "\\item 'organoid\\_Suprabasal IFE'\n",
       "\\item 'organoid\\_Tip cell (arterial)'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "AGGATCTAGTTGAAAC-1-2-organoid\n",
       ":   organoid_Tip cell (arterial)AGGGCTCCAAGCGCAA-1-2-organoid\n",
       ":   organoid_Tip cell (arterial)GTTACAGTCCGCTGTT-1-2-organoid\n",
       ":   organoid_Tip cell (arterial)TCGACCTGTACATACC-1-2-organoid\n",
       ":   organoid_Tip cell (arterial)CAATACGAGAAACTGT-1-4-organoid\n",
       ":   organoid_Tip cell (arterial)GTTAGACCAGTAGAGC-1-4-organoid\n",
       ":   organoid_Tip cell (arterial)GTCGTTCCAATGGGTG-1-5-organoid\n",
       ":   organoid_Tip cell (arterial)GTGAGGATCTAGTGTG-1-5-organoid\n",
       ":   organoid_Tip cell (arterial)GTTACCCCATCGTGGC-1-5-organoid\n",
       ":   organoid_Tip cell (arterial)TACCGAACAAAGGGTC-1-5-organoid\n",
       ":   organoid_Tip cell (arterial)TCAAGACCAACCAATC-1-5-organoid\n",
       ":   organoid_Tip cell (arterial)TTGACCCTCCCTTCCC-1-5-organoid\n",
       ":   organoid_Tip cell (arterial)TGAGCATCAGCAGTTT-1-6-organoid\n",
       ":   organoid_Tip cell (arterial)CTAATGGGTCTCAACA-1-7-organoid\n",
       ":   organoid_Tip cell (arterial)GCGAGAATCTGTTGAG-1-7-organoid\n",
       ":   organoid_Tip cell (arterial)GTATCTTCATGTTCCC-1-7-organoid\n",
       ":   organoid_Tip cell (arterial)TTCTTAGGTGGGTCAA-1-7-organoid\n",
       ":   organoid_Tip cell (arterial)AAACCCATCTCGACGG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)AAGCATCTCCCACAGG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)AAGCGTTGTTCTCCCA-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)ACTTAGGGTTGTTGTG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)AGAGCAGTCGAGTGGA-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)AGCTACAAGACAACTA-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)ATCGTGATCGGTAGAG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)ATTACTCCACAGCTTA-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CAACGGCTCTCACTCG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CATGCCTGTTCTAACG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CATGCGGCAATGAAAC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CCAATTTTCCGATGTA-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CCGAACGAGATTGACA-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CGGCAGTAGAGTGACC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CGGGACTCACTGGATT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CGTAAGTAGTAATCCC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CTACCCATCCGTGTCT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CTACCTGAGCTGTGCC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CTGCCTACAGTGACCC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CTGTGGGTCACTGTTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)CTTCGGTCAGCGACAA-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GAGTCTATCTTGCGCT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GGAATCTCATAGCACT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GGACGTCGTCTTACAG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GGAGAACCAAAGAGTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GGAGGATTCGCTATTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GGATGTTCATAATCGC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GGCTGTGGTTCCATTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GGGATCCAGTTCGGTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GTCGTTCTCGTAGTGT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GTGAGCCGTTCTTGTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GTGTTAGCAAGCGAGT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)GTTATGGTCGAACACT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TAACCAGAGGGCCCTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TAATTCCTCGTGAGAG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TATTGGGAGTACGTCT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TCCATCGCAGCGTTGC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TCTCACGTCTCAACCC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TCTCTGGGTTCTTGTT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TGCAGATAGTTCCGGC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TGGAACTTCAAGGAGC-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TTGGATGCAATCACGT-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TTGGGTACAATACCTG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)TTTGTTGCAGAGCTAG-1-8-organoid\n",
       ":   organoid_Tip cell (arterial)AGCTCAAGTACGACTT-1-9-organoid\n",
       ":   organoid_Tip cell (arterial)TCTCACGAGACTTGTC-1-9-organoid\n",
       ":   organoid_Early fibroblasts FRZB+AACCGCGGTACCAGTT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)ACACCCTGTAGCGTAG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)ACCAGTACATGTAAGA-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)ACGATACTCAACTCTT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)ACGCCAGGTATATGGA-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)ACTGAACGTGACTCAT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)ACTGTCCGTATTACCG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)AGTTGGTTCAAGGCTT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CAGCTAATCATATCGG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CATCGAAAGGGTATCG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CGGAGCTGTATGGTTC-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CGTGTAATCTTACCTA-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CTAGAGTTCTTAGAGC-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CTCCTAGCAAGTCTGT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CTCCTAGTCCTCCTAG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CTCGAAATCGGCATCG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)CTTCTCTGTTATCACG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)GACGGCTGTGCCTGTG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)GCAGTTAAGATGTGGC-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)GGATGTTGTTATGTGC-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)GGGCACTAGCGCCTTG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)GTGAAGGCAAGAAGAG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)GTGCGGTCACCTCGGA-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TACTTACCACCATGTA-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TCTTTCCCAATAGAGT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TGGACGCCAAATCCGT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TGGCTGGAGTGAATTG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TGTGTTTTCCATGAAC-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TTAGGACAGTGAACAT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TTAGGCACAGGGTATG-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)TTCTTAGTCCGCGTTT-1-10-organoid\n",
       ":   organoid_Tip cell (arterial)AACGTTGAGGATGTAT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)AACTCTTTCCCGGATG-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)AAGGTTCTCACATGCA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)ACTGATGGTATCGCAT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)AGAGCGATCGCCAGCA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)AGGGTGAGTTCCATGA-1-11-organoid\n",
       ":   organoid_Smooth muscle/PericytesATAGACCTCTTAGAGC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CAACTAGCATCGGTTA-1-11-organoid\n",
       ":   organoid_Smooth muscle/PericytesCATCCACAGGCATGGT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CCACGGATCTTGAGGT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CCGGTAGCACTGTCGG-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CCGGTAGGTTCCACGG-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CCGTACTAGCACAGGT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CCTTCCCTCCCTTGCA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CCTTCGAAGCGTGTCC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CGAGCCATCTCTAGGA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CGCTTCACATTCCTGC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CGTGTAAAGAGGGCTT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CGTTAGATCACAGGCC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CTCTACGCACGTCTCT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CTCTGGTGTAGCGTCC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CTGATAGAGTTCCACA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)CTTAGGACAAAGTCAA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GACCAATAGAGCTGGT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GCAAACTAGAGTCGGT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GCATGATTCATAACCG-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GGGATGACAGACGCTC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GGGTCTGGTGTGAATA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GTCACAAAGTATCTCG-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GTTAAGCAGCCAGAAC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)GTTACAGAGGACGAAA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)TCACGAAAGAGTCGGT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)TCATTACGTGCATCTA-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)TCGGGACAGGCTAGAC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)TGGCCAGCATGGTTGT-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)TGGCCAGGTCTAGAGG-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)TTCTACACACTTAAGC-1-11-organoid\n",
       ":   organoid_Tip cell (arterial)day_13_IEOdev_DSP_d13:CGTGAATTCGGTAACTx-organoid\n",
       ":   organoid_Tip cell (arterial)\n",
       "\n",
       "\n",
       "**Levels**: 1. 'Uncertain'\n",
       "2. 'nan'\n",
       "3. 'organoid_Adipocytes'\n",
       "4. 'organoid_Basal'\n",
       "5. 'organoid_Basal POSTN+'\n",
       "6. 'organoid_Basal SHH+'\n",
       "7. 'organoid_Companion layer'\n",
       "8. 'organoid_Cuticle/cortex'\n",
       "9. 'organoid_Dermal condensate'\n",
       "10. 'organoid_Dermal papillia'\n",
       "11. 'organoid_Early fibroblasts FRZB+'\n",
       "12. 'organoid_Early fibroblasts HOXC5'\n",
       "13. 'organoid_Early myocytes'\n",
       "14. 'organoid_Fibroblasts POSTN+'\n",
       "15. 'organoid_Immature basal'\n",
       "16. 'organoid_Immature suprabasal'\n",
       "17. 'organoid_Inner root sheath'\n",
       "18. 'organoid_Melanoblasts'\n",
       "19. 'organoid_Melanocytes'\n",
       "20. 'organoid_Merkel cells'\n",
       "21. 'organoid_Myelinating Schwann cells'\n",
       "22. 'organoid_Myoblasts'\n",
       "23. 'organoid_Myocytes'\n",
       "24. 'organoid_Neuroepithelial cells'\n",
       "25. 'organoid_Neuron progenitors'\n",
       "26. 'organoid_Neuron progenitors SPP1+'\n",
       "27. 'organoid_Outer root sheath'\n",
       "28. 'organoid_Periderm'\n",
       "29. 'organoid_Pre-dermal condensate'\n",
       "30. 'organoid_Proliferating melanoblasts'\n",
       "31. 'organoid_Schwann cells/Schwann precursors'\n",
       "32. 'organoid_Secreting neuron'\n",
       "33. 'organoid_Smooth muscle/Pericytes'\n",
       "34. 'organoid_Suprabasal IFE'\n",
       "35. 'organoid_Tip cell (arterial)'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "                   AGGATCTAGTTGAAAC-1-2-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   AGGGCTCCAAGCGCAA-1-2-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTTACAGTCCGCTGTT-1-2-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TCGACCTGTACATACC-1-2-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CAATACGAGAAACTGT-1-4-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTTAGACCAGTAGAGC-1-4-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTCGTTCCAATGGGTG-1-5-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTGAGGATCTAGTGTG-1-5-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTTACCCCATCGTGGC-1-5-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TACCGAACAAAGGGTC-1-5-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TCAAGACCAACCAATC-1-5-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TTGACCCTCCCTTCCC-1-5-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TGAGCATCAGCAGTTT-1-6-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CTAATGGGTCTCAACA-1-7-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GCGAGAATCTGTTGAG-1-7-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTATCTTCATGTTCCC-1-7-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TTCTTAGGTGGGTCAA-1-7-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   AAACCCATCTCGACGG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   AAGCATCTCCCACAGG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   AAGCGTTGTTCTCCCA-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   ACTTAGGGTTGTTGTG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   AGAGCAGTCGAGTGGA-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   AGCTACAAGACAACTA-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   ATCGTGATCGGTAGAG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   ATTACTCCACAGCTTA-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CAACGGCTCTCACTCG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CATGCCTGTTCTAACG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CATGCGGCAATGAAAC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CCAATTTTCCGATGTA-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CCGAACGAGATTGACA-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CGGCAGTAGAGTGACC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CGGGACTCACTGGATT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CGTAAGTAGTAATCCC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CTACCCATCCGTGTCT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CTACCTGAGCTGTGCC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CTGCCTACAGTGACCC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CTGTGGGTCACTGTTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   CTTCGGTCAGCGACAA-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GAGTCTATCTTGCGCT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GGAATCTCATAGCACT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GGACGTCGTCTTACAG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GGAGAACCAAAGAGTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GGAGGATTCGCTATTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GGATGTTCATAATCGC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GGCTGTGGTTCCATTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GGGATCCAGTTCGGTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTCGTTCTCGTAGTGT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTGAGCCGTTCTTGTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTGTTAGCAAGCGAGT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   GTTATGGTCGAACACT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TAACCAGAGGGCCCTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TAATTCCTCGTGAGAG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TATTGGGAGTACGTCT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TCCATCGCAGCGTTGC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TCTCACGTCTCAACCC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TCTCTGGGTTCTTGTT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TGCAGATAGTTCCGGC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TGGAACTTCAAGGAGC-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TTGGATGCAATCACGT-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TTGGGTACAATACCTG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TTTGTTGCAGAGCTAG-1-8-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   AGCTCAAGTACGACTT-1-9-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                   TCTCACGAGACTTGTC-1-9-organoid \n",
       "                organoid_Early fibroblasts FRZB+ \n",
       "                  AACCGCGGTACCAGTT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  ACACCCTGTAGCGTAG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  ACCAGTACATGTAAGA-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  ACGATACTCAACTCTT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  ACGCCAGGTATATGGA-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  ACTGAACGTGACTCAT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  ACTGTCCGTATTACCG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  AGTTGGTTCAAGGCTT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CAGCTAATCATATCGG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CATCGAAAGGGTATCG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CGGAGCTGTATGGTTC-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CGTGTAATCTTACCTA-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTAGAGTTCTTAGAGC-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTCCTAGCAAGTCTGT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTCCTAGTCCTCCTAG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTCGAAATCGGCATCG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTTCTCTGTTATCACG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GACGGCTGTGCCTGTG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GCAGTTAAGATGTGGC-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GGATGTTGTTATGTGC-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GGGCACTAGCGCCTTG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GTGAAGGCAAGAAGAG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GTGCGGTCACCTCGGA-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TACTTACCACCATGTA-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TCTTTCCCAATAGAGT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TGGACGCCAAATCCGT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TGGCTGGAGTGAATTG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TGTGTTTTCCATGAAC-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TTAGGACAGTGAACAT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TTAGGCACAGGGTATG-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TTCTTAGTCCGCGTTT-1-10-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  AACGTTGAGGATGTAT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  AACTCTTTCCCGGATG-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  AAGGTTCTCACATGCA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  ACTGATGGTATCGCAT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  AGAGCGATCGCCAGCA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  AGGGTGAGTTCCATGA-1-11-organoid \n",
       "                organoid_Smooth muscle/Pericytes \n",
       "                  ATAGACCTCTTAGAGC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CAACTAGCATCGGTTA-1-11-organoid \n",
       "                organoid_Smooth muscle/Pericytes \n",
       "                  CATCCACAGGCATGGT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CCACGGATCTTGAGGT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CCGGTAGCACTGTCGG-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CCGGTAGGTTCCACGG-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CCGTACTAGCACAGGT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CCTTCCCTCCCTTGCA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CCTTCGAAGCGTGTCC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CGAGCCATCTCTAGGA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CGCTTCACATTCCTGC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CGTGTAAAGAGGGCTT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CGTTAGATCACAGGCC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTCTACGCACGTCTCT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTCTGGTGTAGCGTCC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTGATAGAGTTCCACA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  CTTAGGACAAAGTCAA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GACCAATAGAGCTGGT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GCAAACTAGAGTCGGT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GCATGATTCATAACCG-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GGGATGACAGACGCTC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GGGTCTGGTGTGAATA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GTCACAAAGTATCTCG-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GTTAAGCAGCCAGAAC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  GTTACAGAGGACGAAA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TCACGAAAGAGTCGGT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TCATTACGTGCATCTA-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TCGGGACAGGCTAGAC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TGGCCAGCATGGTTGT-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TGGCCAGGTCTAGAGG-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "                  TTCTACACACTTAAGC-1-11-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "day_13_IEOdev_DSP_d13:CGTGAATTCGGTAACTx-organoid \n",
       "                    organoid_Tip cell (arterial) \n",
       "35 Levels: Uncertain nan organoid_Adipocytes ... organoid_Tip cell (arterial)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pooled_ad$org_annot30[pooled_ad$fsk_annot2_min30 == \"Tip cell (arterial)\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "concerned-indiana",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T15:02:46.235977Z",
     "iopub.status.busy": "2023-01-31T15:02:46.234427Z",
     "iopub.status.idle": "2023-01-31T15:02:46.310599Z",
     "shell.execute_reply": "2023-01-31T15:02:46.308732Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in `[[<-`(`*tmp*`, pooled_ad$fsk_annot2_min30 == \"Tip cell (arterial)\", : attempt to select more than one element in vectorIndex\n",
     "output_type": "error",
     "traceback": [
      "Error in `[[<-`(`*tmp*`, pooled_ad$fsk_annot2_min30 == \"Tip cell (arterial)\", : attempt to select more than one element in vectorIndex\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "pooled_ad$capillary_arterioles = NA\n",
    "pooled_ad$capillary_arterioles[pooled_ad$fsk_annot2_min30 == \"Tip cell (arterial)\"] = \"fetal_capillary_arterioles\"\n",
    "pooled_ad$capillary_arterioles[pooled_ad$org_annot30 == \"organoid_Tip cell (arterial)\"] = \"organoid_capillary_arterioles\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "decimal-slovakia",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T14:58:39.735800Z",
     "iopub.status.busy": "2023-01-31T14:58:39.734355Z",
     "iopub.status.idle": "2023-01-31T14:58:39.804793Z",
     "shell.execute_reply": "2023-01-31T14:58:39.803280Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>AGGATCTAGTTGAAAC-1-2-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AGGGCTCCAAGCGCAA-1-2-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTTACAGTCCGCTGTT-1-2-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCGACCTGTACATACC-1-2-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CAATACGAGAAACTGT-1-4-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTTAGACCAGTAGAGC-1-4-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTCGTTCCAATGGGTG-1-5-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTGAGGATCTAGTGTG-1-5-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTTACCCCATCGTGGC-1-5-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TACCGAACAAAGGGTC-1-5-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCAAGACCAACCAATC-1-5-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTGACCCTCCCTTCCC-1-5-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGAGCATCAGCAGTTT-1-6-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTAATGGGTCTCAACA-1-7-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GCGAGAATCTGTTGAG-1-7-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTATCTTCATGTTCCC-1-7-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGTTCCGAGAGGACGG-1-7-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTCTTAGGTGGGTCAA-1-7-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AAACCCATCTCGACGG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AAGCATCTCCCACAGG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AAGCGTTGTTCTCCCA-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACTTAGGGTTGTTGTG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AGAGCAGTCGAGTGGA-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AGCTACAAGACAACTA-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ATCGTGATCGGTAGAG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ATTACTCCACAGCTTA-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CAACGGCTCTCACTCG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CATGCCTGTTCTAACG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CATGCGGCAATGAAAC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCAATTTTCCGATGTA-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCGAACGAGATTGACA-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGGCAGTAGAGTGACC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGGGACTCACTGGATT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGTAAGTAGTAATCCC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTACCCATCCGTGTCT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTACCTGAGCTGTGCC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTGCCTACAGTGACCC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTGTGGGTCACTGTTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTTCGGTCAGCGACAA-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GAGTCTATCTTGCGCT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGAATCTCATAGCACT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGACGTCGTCTTACAG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGAGAACCAAAGAGTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGAGGATTCGCTATTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGATGTTCATAATCGC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGCTGTGGTTCCATTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGGATCCAGTTCGGTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTCGTTCTCGTAGTGT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTGAGCCGTTCTTGTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTGTTAGCAAGCGAGT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTTATGGTCGAACACT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TAACCAGAGGGCCCTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TAATTCCTCGTGAGAG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TATTGGGAGTACGTCT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCCATCGCAGCGTTGC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCTCACGTCTCAACCC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCTCTGGGTTCTTGTT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGCAGATAGTTCCGGC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGGAACTTCAAGGAGC-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTGGATGCAATCACGT-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTGGGTACAATACCTG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTTGTTGCAGAGCTAG-1-8-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AGCTCAAGTACGACTT-1-9-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCTCACGAGACTTGTC-1-9-organoid</dt><dd>'fetal_capillary_arterioles'</dd><dt>AACCGCGGTACCAGTT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACACCCTGTAGCGTAG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACCAGTACATGTAAGA-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACGATACTCAACTCTT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACGCCAGGTATATGGA-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACTGAACGTGACTCAT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACTGTCCGTATTACCG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AGTTGGTTCAAGGCTT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CAGCTAATCATATCGG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CATCGAAAGGGTATCG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGGAGCTGTATGGTTC-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGTGTAATCTTACCTA-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTAGAGTTCTTAGAGC-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTCCTAGCAAGTCTGT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTCCTAGTCCTCCTAG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTCGAAATCGGCATCG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTTCTCTGTTATCACG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GACGGCTGTGCCTGTG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GCAGTTAAGATGTGGC-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGATGTTGTTATGTGC-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGGCACTAGCGCCTTG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTGAAGGCAAGAAGAG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTGCGGTCACCTCGGA-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TACTTACCACCATGTA-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCTTTCCCAATAGAGT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGGACGCCAAATCCGT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGGCTGGAGTGAATTG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGTGTTTTCCATGAAC-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTAGGACAGTGAACAT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTAGGCACAGGGTATG-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTCTTAGTCCGCGTTT-1-10-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AACGTTGAGGATGTAT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AACTCTTTCCCGGATG-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AAGGTTCTCACATGCA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>ACTGATGGTATCGCAT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AGAGCGATCGCCAGCA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>AGGGTGAGTTCCATGA-1-11-organoid</dt><dd>'fetal_capillary_arterioles'</dd><dt>ATAGACCTCTTAGAGC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CAACTAGCATCGGTTA-1-11-organoid</dt><dd>'fetal_capillary_arterioles'</dd><dt>CATCCACAGGCATGGT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCACGGATCTTGAGGT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCGGTAGCACTGTCGG-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCGGTAGGTTCCACGG-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCGTACTAGCACAGGT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCTTCCCTCCCTTGCA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CCTTCGAAGCGTGTCC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGAGCCATCTCTAGGA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGCTTCACATTCCTGC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGTGTAAAGAGGGCTT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CGTTAGATCACAGGCC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTCTACGCACGTCTCT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTCTGGTGTAGCGTCC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTGATAGAGTTCCACA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>CTTAGGACAAAGTCAA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GACCAATAGAGCTGGT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GCAAACTAGAGTCGGT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GCATGATTCATAACCG-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGGATGACAGACGCTC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GGGTCTGGTGTGAATA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTCACAAAGTATCTCG-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTTAAGCAGCCAGAAC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>GTTACAGAGGACGAAA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCACGAAAGAGTCGGT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCATTACGTGCATCTA-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TCGGGACAGGCTAGAC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGGCCAGCATGGTTGT-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TGGCCAGGTCTAGAGG-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>TTCTACACACTTAAGC-1-11-organoid</dt><dd>'organoid_capillary_arterioles'</dd><dt>day_13_IEOdev_DSP_d13:CGTGAATTCGGTAACTx-organoid</dt><dd>'organoid_capillary_arterioles'</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[AGGATCTAGTTGAAAC-1-2-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AGGGCTCCAAGCGCAA-1-2-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTTACAGTCCGCTGTT-1-2-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCGACCTGTACATACC-1-2-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CAATACGAGAAACTGT-1-4-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTTAGACCAGTAGAGC-1-4-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTCGTTCCAATGGGTG-1-5-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTGAGGATCTAGTGTG-1-5-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTTACCCCATCGTGGC-1-5-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TACCGAACAAAGGGTC-1-5-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCAAGACCAACCAATC-1-5-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTGACCCTCCCTTCCC-1-5-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGAGCATCAGCAGTTT-1-6-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTAATGGGTCTCAACA-1-7-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GCGAGAATCTGTTGAG-1-7-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTATCTTCATGTTCCC-1-7-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGTTCCGAGAGGACGG-1-7-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTCTTAGGTGGGTCAA-1-7-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AAACCCATCTCGACGG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AAGCATCTCCCACAGG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AAGCGTTGTTCTCCCA-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACTTAGGGTTGTTGTG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AGAGCAGTCGAGTGGA-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AGCTACAAGACAACTA-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ATCGTGATCGGTAGAG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ATTACTCCACAGCTTA-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CAACGGCTCTCACTCG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CATGCCTGTTCTAACG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CATGCGGCAATGAAAC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCAATTTTCCGATGTA-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCGAACGAGATTGACA-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGGCAGTAGAGTGACC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGGGACTCACTGGATT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGTAAGTAGTAATCCC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTACCCATCCGTGTCT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTACCTGAGCTGTGCC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTGCCTACAGTGACCC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTGTGGGTCACTGTTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTTCGGTCAGCGACAA-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GAGTCTATCTTGCGCT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGAATCTCATAGCACT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGACGTCGTCTTACAG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGAGAACCAAAGAGTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGAGGATTCGCTATTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGATGTTCATAATCGC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGCTGTGGTTCCATTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGGATCCAGTTCGGTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTCGTTCTCGTAGTGT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTGAGCCGTTCTTGTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTGTTAGCAAGCGAGT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTTATGGTCGAACACT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TAACCAGAGGGCCCTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TAATTCCTCGTGAGAG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TATTGGGAGTACGTCT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCCATCGCAGCGTTGC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCTCACGTCTCAACCC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCTCTGGGTTCTTGTT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGCAGATAGTTCCGGC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGGAACTTCAAGGAGC-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTGGATGCAATCACGT-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTGGGTACAATACCTG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTTGTTGCAGAGCTAG-1-8-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AGCTCAAGTACGACTT-1-9-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCTCACGAGACTTGTC-1-9-organoid] 'fetal\\_capillary\\_arterioles'\n",
       "\\item[AACCGCGGTACCAGTT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACACCCTGTAGCGTAG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACCAGTACATGTAAGA-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACGATACTCAACTCTT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACGCCAGGTATATGGA-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACTGAACGTGACTCAT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACTGTCCGTATTACCG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AGTTGGTTCAAGGCTT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CAGCTAATCATATCGG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CATCGAAAGGGTATCG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGGAGCTGTATGGTTC-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGTGTAATCTTACCTA-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTAGAGTTCTTAGAGC-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTCCTAGCAAGTCTGT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTCCTAGTCCTCCTAG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTCGAAATCGGCATCG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTTCTCTGTTATCACG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GACGGCTGTGCCTGTG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GCAGTTAAGATGTGGC-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGATGTTGTTATGTGC-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGGCACTAGCGCCTTG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTGAAGGCAAGAAGAG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTGCGGTCACCTCGGA-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TACTTACCACCATGTA-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCTTTCCCAATAGAGT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGGACGCCAAATCCGT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGGCTGGAGTGAATTG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGTGTTTTCCATGAAC-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTAGGACAGTGAACAT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTAGGCACAGGGTATG-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTCTTAGTCCGCGTTT-1-10-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AACGTTGAGGATGTAT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AACTCTTTCCCGGATG-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AAGGTTCTCACATGCA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[ACTGATGGTATCGCAT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AGAGCGATCGCCAGCA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[AGGGTGAGTTCCATGA-1-11-organoid] 'fetal\\_capillary\\_arterioles'\n",
       "\\item[ATAGACCTCTTAGAGC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CAACTAGCATCGGTTA-1-11-organoid] 'fetal\\_capillary\\_arterioles'\n",
       "\\item[CATCCACAGGCATGGT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCACGGATCTTGAGGT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCGGTAGCACTGTCGG-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCGGTAGGTTCCACGG-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCGTACTAGCACAGGT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCTTCCCTCCCTTGCA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CCTTCGAAGCGTGTCC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGAGCCATCTCTAGGA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGCTTCACATTCCTGC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGTGTAAAGAGGGCTT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CGTTAGATCACAGGCC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTCTACGCACGTCTCT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTCTGGTGTAGCGTCC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTGATAGAGTTCCACA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[CTTAGGACAAAGTCAA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GACCAATAGAGCTGGT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GCAAACTAGAGTCGGT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GCATGATTCATAACCG-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGGATGACAGACGCTC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GGGTCTGGTGTGAATA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTCACAAAGTATCTCG-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTTAAGCAGCCAGAAC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[GTTACAGAGGACGAAA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCACGAAAGAGTCGGT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCATTACGTGCATCTA-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TCGGGACAGGCTAGAC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGGCCAGCATGGTTGT-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TGGCCAGGTCTAGAGG-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[TTCTACACACTTAAGC-1-11-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\item[day\\textbackslash{}\\_13\\textbackslash{}\\_IEOdev\\textbackslash{}\\_DSP\\textbackslash{}\\_d13:CGTGAATTCGGTAACTx-organoid] 'organoid\\_capillary\\_arterioles'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "AGGATCTAGTTGAAAC-1-2-organoid\n",
       ":   'organoid_capillary_arterioles'AGGGCTCCAAGCGCAA-1-2-organoid\n",
       ":   'organoid_capillary_arterioles'GTTACAGTCCGCTGTT-1-2-organoid\n",
       ":   'organoid_capillary_arterioles'TCGACCTGTACATACC-1-2-organoid\n",
       ":   'organoid_capillary_arterioles'CAATACGAGAAACTGT-1-4-organoid\n",
       ":   'organoid_capillary_arterioles'GTTAGACCAGTAGAGC-1-4-organoid\n",
       ":   'organoid_capillary_arterioles'GTCGTTCCAATGGGTG-1-5-organoid\n",
       ":   'organoid_capillary_arterioles'GTGAGGATCTAGTGTG-1-5-organoid\n",
       ":   'organoid_capillary_arterioles'GTTACCCCATCGTGGC-1-5-organoid\n",
       ":   'organoid_capillary_arterioles'TACCGAACAAAGGGTC-1-5-organoid\n",
       ":   'organoid_capillary_arterioles'TCAAGACCAACCAATC-1-5-organoid\n",
       ":   'organoid_capillary_arterioles'TTGACCCTCCCTTCCC-1-5-organoid\n",
       ":   'organoid_capillary_arterioles'TGAGCATCAGCAGTTT-1-6-organoid\n",
       ":   'organoid_capillary_arterioles'CTAATGGGTCTCAACA-1-7-organoid\n",
       ":   'organoid_capillary_arterioles'GCGAGAATCTGTTGAG-1-7-organoid\n",
       ":   'organoid_capillary_arterioles'GTATCTTCATGTTCCC-1-7-organoid\n",
       ":   'organoid_capillary_arterioles'TGTTCCGAGAGGACGG-1-7-organoid\n",
       ":   'organoid_capillary_arterioles'TTCTTAGGTGGGTCAA-1-7-organoid\n",
       ":   'organoid_capillary_arterioles'AAACCCATCTCGACGG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'AAGCATCTCCCACAGG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'AAGCGTTGTTCTCCCA-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'ACTTAGGGTTGTTGTG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'AGAGCAGTCGAGTGGA-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'AGCTACAAGACAACTA-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'ATCGTGATCGGTAGAG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'ATTACTCCACAGCTTA-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CAACGGCTCTCACTCG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CATGCCTGTTCTAACG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CATGCGGCAATGAAAC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CCAATTTTCCGATGTA-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CCGAACGAGATTGACA-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CGGCAGTAGAGTGACC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CGGGACTCACTGGATT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CGTAAGTAGTAATCCC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CTACCCATCCGTGTCT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CTACCTGAGCTGTGCC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CTGCCTACAGTGACCC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CTGTGGGTCACTGTTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'CTTCGGTCAGCGACAA-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GAGTCTATCTTGCGCT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GGAATCTCATAGCACT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GGACGTCGTCTTACAG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GGAGAACCAAAGAGTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GGAGGATTCGCTATTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GGATGTTCATAATCGC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GGCTGTGGTTCCATTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GGGATCCAGTTCGGTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GTCGTTCTCGTAGTGT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GTGAGCCGTTCTTGTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GTGTTAGCAAGCGAGT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'GTTATGGTCGAACACT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TAACCAGAGGGCCCTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TAATTCCTCGTGAGAG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TATTGGGAGTACGTCT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TCCATCGCAGCGTTGC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TCTCACGTCTCAACCC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TCTCTGGGTTCTTGTT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TGCAGATAGTTCCGGC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TGGAACTTCAAGGAGC-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TTGGATGCAATCACGT-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TTGGGTACAATACCTG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'TTTGTTGCAGAGCTAG-1-8-organoid\n",
       ":   'organoid_capillary_arterioles'AGCTCAAGTACGACTT-1-9-organoid\n",
       ":   'organoid_capillary_arterioles'TCTCACGAGACTTGTC-1-9-organoid\n",
       ":   'fetal_capillary_arterioles'AACCGCGGTACCAGTT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'ACACCCTGTAGCGTAG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'ACCAGTACATGTAAGA-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'ACGATACTCAACTCTT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'ACGCCAGGTATATGGA-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'ACTGAACGTGACTCAT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'ACTGTCCGTATTACCG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'AGTTGGTTCAAGGCTT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CAGCTAATCATATCGG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CATCGAAAGGGTATCG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CGGAGCTGTATGGTTC-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CGTGTAATCTTACCTA-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CTAGAGTTCTTAGAGC-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CTCCTAGCAAGTCTGT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CTCCTAGTCCTCCTAG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CTCGAAATCGGCATCG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'CTTCTCTGTTATCACG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'GACGGCTGTGCCTGTG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'GCAGTTAAGATGTGGC-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'GGATGTTGTTATGTGC-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'GGGCACTAGCGCCTTG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'GTGAAGGCAAGAAGAG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'GTGCGGTCACCTCGGA-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TACTTACCACCATGTA-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TCTTTCCCAATAGAGT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TGGACGCCAAATCCGT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TGGCTGGAGTGAATTG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TGTGTTTTCCATGAAC-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TTAGGACAGTGAACAT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TTAGGCACAGGGTATG-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'TTCTTAGTCCGCGTTT-1-10-organoid\n",
       ":   'organoid_capillary_arterioles'AACGTTGAGGATGTAT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'AACTCTTTCCCGGATG-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'AAGGTTCTCACATGCA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'ACTGATGGTATCGCAT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'AGAGCGATCGCCAGCA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'AGGGTGAGTTCCATGA-1-11-organoid\n",
       ":   'fetal_capillary_arterioles'ATAGACCTCTTAGAGC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CAACTAGCATCGGTTA-1-11-organoid\n",
       ":   'fetal_capillary_arterioles'CATCCACAGGCATGGT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CCACGGATCTTGAGGT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CCGGTAGCACTGTCGG-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CCGGTAGGTTCCACGG-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CCGTACTAGCACAGGT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CCTTCCCTCCCTTGCA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CCTTCGAAGCGTGTCC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CGAGCCATCTCTAGGA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CGCTTCACATTCCTGC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CGTGTAAAGAGGGCTT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CGTTAGATCACAGGCC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CTCTACGCACGTCTCT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CTCTGGTGTAGCGTCC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CTGATAGAGTTCCACA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'CTTAGGACAAAGTCAA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GACCAATAGAGCTGGT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GCAAACTAGAGTCGGT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GCATGATTCATAACCG-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GGGATGACAGACGCTC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GGGTCTGGTGTGAATA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GTCACAAAGTATCTCG-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GTTAAGCAGCCAGAAC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'GTTACAGAGGACGAAA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'TCACGAAAGAGTCGGT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'TCATTACGTGCATCTA-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'TCGGGACAGGCTAGAC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'TGGCCAGCATGGTTGT-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'TGGCCAGGTCTAGAGG-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'TTCTACACACTTAAGC-1-11-organoid\n",
       ":   'organoid_capillary_arterioles'day_13_IEOdev_DSP_d13:CGTGAATTCGGTAACTx-organoid\n",
       ":   'organoid_capillary_arterioles'\n",
       "\n"
      ],
      "text/plain": [
       "                   AGGATCTAGTTGAAAC-1-2-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   AGGGCTCCAAGCGCAA-1-2-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTTACAGTCCGCTGTT-1-2-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TCGACCTGTACATACC-1-2-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CAATACGAGAAACTGT-1-4-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTTAGACCAGTAGAGC-1-4-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTCGTTCCAATGGGTG-1-5-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTGAGGATCTAGTGTG-1-5-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTTACCCCATCGTGGC-1-5-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TACCGAACAAAGGGTC-1-5-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TCAAGACCAACCAATC-1-5-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TTGACCCTCCCTTCCC-1-5-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TGAGCATCAGCAGTTT-1-6-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CTAATGGGTCTCAACA-1-7-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GCGAGAATCTGTTGAG-1-7-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTATCTTCATGTTCCC-1-7-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TGTTCCGAGAGGACGG-1-7-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TTCTTAGGTGGGTCAA-1-7-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   AAACCCATCTCGACGG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   AAGCATCTCCCACAGG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   AAGCGTTGTTCTCCCA-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   ACTTAGGGTTGTTGTG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   AGAGCAGTCGAGTGGA-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   AGCTACAAGACAACTA-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   ATCGTGATCGGTAGAG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   ATTACTCCACAGCTTA-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CAACGGCTCTCACTCG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CATGCCTGTTCTAACG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CATGCGGCAATGAAAC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CCAATTTTCCGATGTA-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CCGAACGAGATTGACA-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CGGCAGTAGAGTGACC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CGGGACTCACTGGATT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CGTAAGTAGTAATCCC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CTACCCATCCGTGTCT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CTACCTGAGCTGTGCC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CTGCCTACAGTGACCC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CTGTGGGTCACTGTTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   CTTCGGTCAGCGACAA-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GAGTCTATCTTGCGCT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GGAATCTCATAGCACT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GGACGTCGTCTTACAG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GGAGAACCAAAGAGTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GGAGGATTCGCTATTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GGATGTTCATAATCGC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GGCTGTGGTTCCATTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GGGATCCAGTTCGGTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTCGTTCTCGTAGTGT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTGAGCCGTTCTTGTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTGTTAGCAAGCGAGT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   GTTATGGTCGAACACT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TAACCAGAGGGCCCTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TAATTCCTCGTGAGAG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TATTGGGAGTACGTCT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TCCATCGCAGCGTTGC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TCTCACGTCTCAACCC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TCTCTGGGTTCTTGTT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TGCAGATAGTTCCGGC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TGGAACTTCAAGGAGC-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TTGGATGCAATCACGT-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TTGGGTACAATACCTG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TTTGTTGCAGAGCTAG-1-8-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   AGCTCAAGTACGACTT-1-9-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                   TCTCACGAGACTTGTC-1-9-organoid \n",
       "                    \"fetal_capillary_arterioles\" \n",
       "                  AACCGCGGTACCAGTT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  ACACCCTGTAGCGTAG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  ACCAGTACATGTAAGA-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  ACGATACTCAACTCTT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  ACGCCAGGTATATGGA-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  ACTGAACGTGACTCAT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  ACTGTCCGTATTACCG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  AGTTGGTTCAAGGCTT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CAGCTAATCATATCGG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CATCGAAAGGGTATCG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CGGAGCTGTATGGTTC-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CGTGTAATCTTACCTA-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTAGAGTTCTTAGAGC-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTCCTAGCAAGTCTGT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTCCTAGTCCTCCTAG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTCGAAATCGGCATCG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTTCTCTGTTATCACG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GACGGCTGTGCCTGTG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GCAGTTAAGATGTGGC-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GGATGTTGTTATGTGC-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GGGCACTAGCGCCTTG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GTGAAGGCAAGAAGAG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GTGCGGTCACCTCGGA-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TACTTACCACCATGTA-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TCTTTCCCAATAGAGT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TGGACGCCAAATCCGT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TGGCTGGAGTGAATTG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TGTGTTTTCCATGAAC-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TTAGGACAGTGAACAT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TTAGGCACAGGGTATG-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TTCTTAGTCCGCGTTT-1-10-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  AACGTTGAGGATGTAT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  AACTCTTTCCCGGATG-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  AAGGTTCTCACATGCA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  ACTGATGGTATCGCAT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  AGAGCGATCGCCAGCA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  AGGGTGAGTTCCATGA-1-11-organoid \n",
       "                    \"fetal_capillary_arterioles\" \n",
       "                  ATAGACCTCTTAGAGC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CAACTAGCATCGGTTA-1-11-organoid \n",
       "                    \"fetal_capillary_arterioles\" \n",
       "                  CATCCACAGGCATGGT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CCACGGATCTTGAGGT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CCGGTAGCACTGTCGG-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CCGGTAGGTTCCACGG-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CCGTACTAGCACAGGT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CCTTCCCTCCCTTGCA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CCTTCGAAGCGTGTCC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CGAGCCATCTCTAGGA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CGCTTCACATTCCTGC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CGTGTAAAGAGGGCTT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CGTTAGATCACAGGCC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTCTACGCACGTCTCT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTCTGGTGTAGCGTCC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTGATAGAGTTCCACA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  CTTAGGACAAAGTCAA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GACCAATAGAGCTGGT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GCAAACTAGAGTCGGT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GCATGATTCATAACCG-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GGGATGACAGACGCTC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GGGTCTGGTGTGAATA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GTCACAAAGTATCTCG-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GTTAAGCAGCCAGAAC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  GTTACAGAGGACGAAA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TCACGAAAGAGTCGGT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TCATTACGTGCATCTA-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TCGGGACAGGCTAGAC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TGGCCAGCATGGTTGT-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TGGCCAGGTCTAGAGG-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "                  TTCTACACACTTAAGC-1-11-organoid \n",
       "                 \"organoid_capillary_arterioles\" \n",
       "day_13_IEOdev_DSP_d13:CGTGAATTCGGTAACTx-organoid \n",
       "                 \"organoid_capillary_arterioles\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pooled_ad$capillary_arterioles[!is.na(pooled_ad$capillary_arterioles)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "friendly-midwest",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T14:54:28.766237Z",
     "iopub.status.busy": "2023-01-31T14:54:28.764905Z",
     "iopub.status.idle": "2023-01-31T14:54:37.316210Z",
     "shell.execute_reply": "2023-01-31T14:54:37.313866Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Subset on the `$capillary_arterioles` value in the object meta data\n",
    "# cells_1 = subset(pooled_ad, subset = pooled_ad$capillary_arterioles == \"fetal_capillary_arterioles\")\n",
    "cells_1 = colnames(pooled_ad[, pooled_ad$capillary_arterioles == \"fetal_capillary_arterioles\"])\n",
    "# cells_2 = subset(pooled_ad, subset = pooled_ad$capillary_arterioles == \"organoid_capillary_arterioles\")\n",
    "cells_2 = colnames(pooled_ad[, pooled_ad$capillary_arterioles == \"organoid_capillary_arterioles\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "pending-minute",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T14:54:37.319933Z",
     "iopub.status.busy": "2023-01-31T14:54:37.318677Z",
     "iopub.status.idle": "2023-01-31T14:54:37.334008Z",
     "shell.execute_reply": "2023-01-31T14:54:37.332573Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'TCTCACGAGACTTGTC-1-9-organoid'</li><li>'AGGGTGAGTTCCATGA-1-11-organoid'</li><li>'CAACTAGCATCGGTTA-1-11-organoid'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'TCTCACGAGACTTGTC-1-9-organoid'\n",
       "\\item 'AGGGTGAGTTCCATGA-1-11-organoid'\n",
       "\\item 'CAACTAGCATCGGTTA-1-11-organoid'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'TCTCACGAGACTTGTC-1-9-organoid'\n",
       "2. 'AGGGTGAGTTCCATGA-1-11-organoid'\n",
       "3. 'CAACTAGCATCGGTTA-1-11-organoid'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"TCTCACGAGACTTGTC-1-9-organoid\"  \"AGGGTGAGTTCCATGA-1-11-organoid\"\n",
       "[3] \"CAACTAGCATCGGTTA-1-11-organoid\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cells_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "hungry-cedar",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T14:29:37.432394Z",
     "iopub.status.busy": "2023-01-31T14:29:37.430959Z",
     "iopub.status.idle": "2023-01-31T14:29:37.445590Z",
     "shell.execute_reply": "2023-01-31T14:29:37.443945Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error: 'MASTDETest' is not an exported object from 'namespace:MAST'\n",
     "output_type": "error",
     "traceback": [
      "Error: 'MASTDETest' is not an exported object from 'namespace:MAST'\nTraceback:\n",
      "1. MAST::MASTDETest",
      "2. getExportedValue(pkg, name)",
      "3. stop(gettextf(\"'%s' is not an exported object from 'namespace:%s'\", \n .     name, getNamespaceName(ns)), call. = FALSE, domain = NA)"
     ]
    }
   ],
   "source": [
    "diff_exp_genes = MAST::MASTDETest(pooled_ad, cells_1, cells_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "encouraging-moderator",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T18:03:12.148170Z",
     "iopub.status.busy": "2023-01-31T18:03:12.146639Z",
     "iopub.status.idle": "2023-01-31T18:03:12.517018Z",
     "shell.execute_reply": "2023-01-31T18:03:12.514032Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for FindMarkers {Seurat}\"><tr><td>FindMarkers {Seurat}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Gene expression markers of identity classes</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>Finds markers (differentially expressed genes) for identity classes\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "FindMarkers(object, ...)\n",
       "\n",
       "## Default S3 method:\n",
       "FindMarkers(\n",
       "  object,\n",
       "  slot = \"data\",\n",
       "  counts = numeric(),\n",
       "  cells.1 = NULL,\n",
       "  cells.2 = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  fc.results = NULL,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Assay'\n",
       "FindMarkers(\n",
       "  object,\n",
       "  slot = \"data\",\n",
       "  cells.1 = NULL,\n",
       "  cells.2 = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  mean.fxn = NULL,\n",
       "  fc.name = NULL,\n",
       "  base = 2,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'DimReduc'\n",
       "FindMarkers(\n",
       "  object,\n",
       "  cells.1 = NULL,\n",
       "  cells.2 = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  mean.fxn = rowMeans,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "FindMarkers(\n",
       "  object,\n",
       "  ident.1 = NULL,\n",
       "  ident.2 = NULL,\n",
       "  group.by = NULL,\n",
       "  subset.ident = NULL,\n",
       "  assay = NULL,\n",
       "  slot = \"data\",\n",
       "  reduction = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  mean.fxn = NULL,\n",
       "  fc.name = NULL,\n",
       "  base = 2,\n",
       "  ...\n",
       ")\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>object</code></td>\n",
       "<td>\n",
       "<p>An object</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>...</code></td>\n",
       "<td>\n",
       "<p>Arguments passed to other methods and to specific DE methods</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>slot</code></td>\n",
       "<td>\n",
       "<p>Slot to pull data from; note that if <code>test.use</code> is &quot;negbinom&quot;, &quot;poisson&quot;, or &quot;DESeq2&quot;,\n",
       "<code>slot</code> will be set to &quot;counts&quot;</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>counts</code></td>\n",
       "<td>\n",
       "<p>Count matrix if using scale.data for DE tests. This is used for\n",
       "computing pct.1 and pct.2 and for filtering features based on fraction\n",
       "expressing</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>cells.1</code></td>\n",
       "<td>\n",
       "<p>Vector of cell names belonging to group 1</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>cells.2</code></td>\n",
       "<td>\n",
       "<p>Vector of cell names belonging to group 2</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>features</code></td>\n",
       "<td>\n",
       "<p>Genes to test. Default is to use all genes</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>logfc.threshold</code></td>\n",
       "<td>\n",
       "<p>Limit testing to genes which show, on average, at least\n",
       "X-fold difference (log-scale) between the two groups of cells. Default is 0.25\n",
       "Increasing logfc.threshold speeds up the function, but can miss weaker signals.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>test.use</code></td>\n",
       "<td>\n",
       "<p>Denotes which test to use. Available options are:\n",
       "</p>\n",
       "\n",
       "<ul>\n",
       "<li><p>&quot;wilcox&quot; : Identifies differentially expressed genes between two\n",
       "groups of cells using a Wilcoxon Rank Sum test (default)\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;bimod&quot; : Likelihood-ratio test for single cell gene expression,\n",
       "(McDavid et al., Bioinformatics, 2013)\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;roc&quot; : Identifies 'markers' of gene expression using ROC analysis.\n",
       "For each gene, evaluates (using AUC) a classifier built on that gene alone,\n",
       "to classify between two groups of cells. An AUC value of 1 means that\n",
       "expression values for this gene alone can perfectly classify the two\n",
       "groupings (i.e. Each of the cells in cells.1 exhibit a higher level than\n",
       "each of the cells in cells.2). An AUC value of 0 also means there is perfect\n",
       "classification, but in the other direction. A value of 0.5 implies that\n",
       "the gene has no predictive power to classify the two groups. Returns a\n",
       "'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially\n",
       "expressed genes.\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;t&quot; : Identify differentially expressed genes between two groups of\n",
       "cells using the Student's t-test.\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;negbinom&quot; : Identifies differentially expressed genes between two\n",
       "groups of cells using a negative binomial generalized linear model.\n",
       "Use only for UMI-based datasets\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;poisson&quot; : Identifies differentially expressed genes between two\n",
       "groups of cells using a poisson generalized linear model.\n",
       "Use only for UMI-based datasets\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;LR&quot; : Uses a logistic regression framework to determine differentially\n",
       "expressed genes. Constructs a logistic regression model predicting group\n",
       "membership based on each feature individually and compares this to a null\n",
       "model with a likelihood ratio test.\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;MAST&quot; : Identifies differentially expressed genes between two groups\n",
       "of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST\n",
       "package to run the DE testing.\n",
       "</p>\n",
       "</li>\n",
       "<li><p>&quot;DESeq2&quot; : Identifies differentially expressed genes between two groups\n",
       "of cells based on a model using DESeq2 which uses a negative binomial\n",
       "distribution (Love et al, Genome Biology, 2014).This test does not support\n",
       "pre-filtering of genes based on average difference (or percent detection rate)\n",
       "between cell groups. However, genes may be pre-filtered based on their\n",
       "minimum detection rate (min.pct) across both cell groups. To use this method,\n",
       "please install DESeq2, using the instructions at\n",
       "https://bioconductor.org/packages/release/bioc/html/DESeq2.html\n",
       "</p>\n",
       "</li></ul>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>min.pct</code></td>\n",
       "<td>\n",
       "<p>only test genes that are detected in a minimum fraction of\n",
       "min.pct cells in either of the two populations. Meant to speed up the function\n",
       "by not testing genes that are very infrequently expressed. Default is 0.1</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>min.diff.pct</code></td>\n",
       "<td>\n",
       "<p>only test genes that show a minimum difference in the\n",
       "fraction of detection between the two groups. Set to -Inf by default</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>verbose</code></td>\n",
       "<td>\n",
       "<p>Print a progress bar once expression testing begins</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>only.pos</code></td>\n",
       "<td>\n",
       "<p>Only return positive markers (FALSE by default)</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>max.cells.per.ident</code></td>\n",
       "<td>\n",
       "<p>Down sample each identity class to a max number.\n",
       "Default is no downsampling. Not activated by default (set to Inf)</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>random.seed</code></td>\n",
       "<td>\n",
       "<p>Random seed for downsampling</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>latent.vars</code></td>\n",
       "<td>\n",
       "<p>Variables to test, used only when <code>test.use</code> is one of\n",
       "'LR', 'negbinom', 'poisson', or 'MAST'</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>min.cells.feature</code></td>\n",
       "<td>\n",
       "<p>Minimum number of cells expressing the feature in at least one\n",
       "of the two groups, currently only used for poisson and negative binomial tests</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>min.cells.group</code></td>\n",
       "<td>\n",
       "<p>Minimum number of cells in one of the groups</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>pseudocount.use</code></td>\n",
       "<td>\n",
       "<p>Pseudocount to add to averaged expression values when\n",
       "calculating logFC. 1 by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fc.results</code></td>\n",
       "<td>\n",
       "<p>data.frame from FoldChange</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>mean.fxn</code></td>\n",
       "<td>\n",
       "<p>Function to use for fold change or average difference calculation.\n",
       "If NULL, the appropriate function will be chose according to the slot used</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fc.name</code></td>\n",
       "<td>\n",
       "<p>Name of the fold change, average difference, or custom function column\n",
       "in the output data.frame. If NULL, the fold change column will be named\n",
       "according to the logarithm base (eg, &quot;avg_log2FC&quot;), or if using the scale.data\n",
       "slot &quot;avg_diff&quot;.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>base</code></td>\n",
       "<td>\n",
       "<p>The base with respect to which logarithms are computed.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>ident.1</code></td>\n",
       "<td>\n",
       "<p>Identity class to define markers for; pass an object of class\n",
       "<code>phylo</code> or 'clustertree' to find markers for a node in a cluster tree;\n",
       "passing 'clustertree' requires <code>BuildClusterTree</code> to have been run</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>ident.2</code></td>\n",
       "<td>\n",
       "<p>A second identity class for comparison; if <code>NULL</code>,\n",
       "use all other cells for comparison; if an object of class <code>phylo</code> or\n",
       "'clustertree' is passed to <code>ident.1</code>, must pass a node to find markers for</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>group.by</code></td>\n",
       "<td>\n",
       "<p>Regroup cells into a different identity class prior to performing differential expression (see example)</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset.ident</code></td>\n",
       "<td>\n",
       "<p>Subset a particular identity class prior to regrouping. Only relevant if group.by is set (see example)</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>assay</code></td>\n",
       "<td>\n",
       "<p>Assay to use in differential expression testing</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>reduction</code></td>\n",
       "<td>\n",
       "<p>Reduction to use in differential expression testing - will test for DE on cell embeddings</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p>p-value adjustment is performed using bonferroni correction based on\n",
       "the total number of genes in the dataset. Other correction methods are not\n",
       "recommended, as Seurat pre-filters genes using the arguments above, reducing\n",
       "the number of tests performed. Lastly, as Aaron Lun has pointed out, p-values\n",
       "should be interpreted cautiously, as the genes used for clustering are the\n",
       "same genes tested for differential expression.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<p>data.frame with a ranked list of putative markers as rows, and associated\n",
       "statistics as columns (p-values, ROC score, etc., depending on the test used (<code>test.use</code>)). The following columns are always present:\n",
       "</p>\n",
       "\n",
       "<ul>\n",
       "<li> <p><code>avg_logFC</code>: log fold-chage of the average expression between the two groups. Positive values indicate that the gene is more highly expressed in the first group\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>pct.1</code>: The percentage of cells where the gene is detected in the first group\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>pct.2</code>: The percentage of cells where the gene is detected in the second group\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>p_val_adj</code>: Adjusted p-value, based on bonferroni correction using all genes in the dataset\n",
       "</p>\n",
       "</li></ul>\n",
       "\n",
       "\n",
       "\n",
       "<h3>References</h3>\n",
       "\n",
       "<p>McDavid A, Finak G, Chattopadyay PK, et al. Data exploration,\n",
       "quality control and testing in single-cell qPCR-based gene expression experiments.\n",
       "Bioinformatics. 2013;29(4):461-467. doi:10.1093/bioinformatics/bts714\n",
       "</p>\n",
       "<p>Trapnell C, et al. The dynamics and regulators of cell fate\n",
       "decisions are revealed by pseudotemporal ordering of single cells. Nature\n",
       "Biotechnology volume 32, pages 381-386 (2014)\n",
       "</p>\n",
       "<p>Andrew McDavid, Greg Finak and Masanao Yajima (2017). MAST: Model-based\n",
       "Analysis of Single Cell Transcriptomics. R package version 1.2.1.\n",
       "https://github.com/RGLab/MAST/\n",
       "</p>\n",
       "<p>Love MI, Huber W and Anders S (2014). &quot;Moderated estimation of\n",
       "fold change and dispersion for RNA-seq data with DESeq2.&quot; Genome Biology.\n",
       "https://bioconductor.org/packages/release/bioc/html/DESeq2.html\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>See Also</h3>\n",
       "\n",
       "<p><code>FoldChange</code>\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "data(\"pbmc_small\")\n",
       "# Find markers for cluster 2\n",
       "markers &lt;- FindMarkers(object = pbmc_small, ident.1 = 2)\n",
       "head(x = markers)\n",
       "\n",
       "# Take all cells in cluster 2, and find markers that separate cells in the 'g1' group (metadata\n",
       "# variable 'group')\n",
       "markers &lt;- FindMarkers(pbmc_small, ident.1 = \"g1\", group.by = 'groups', subset.ident = \"2\")\n",
       "head(x = markers)\n",
       "\n",
       "# Pass 'clustertree' or an object of class phylo to ident.1 and\n",
       "# a node to ident.2 as a replacement for FindMarkersNode\n",
       "pbmc_small &lt;- BuildClusterTree(object = pbmc_small)\n",
       "markers &lt;- FindMarkers(object = pbmc_small, ident.1 = 'clustertree', ident.2 = 5)\n",
       "head(x = markers)\n",
       "\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>Seurat</em> version 4.0.0 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{FindMarkers}{Gene expression markers of identity classes}{FindMarkers}\n",
       "\\methaliasA{FindMarkers.Assay}{FindMarkers}{FindMarkers.Assay}\n",
       "\\methaliasA{FindMarkers.default}{FindMarkers}{FindMarkers.default}\n",
       "\\methaliasA{FindMarkers.DimReduc}{FindMarkers}{FindMarkers.DimReduc}\n",
       "\\methaliasA{FindMarkers.Seurat}{FindMarkers}{FindMarkers.Seurat}\n",
       "\\aliasA{FindMarkersNode}{FindMarkers}{FindMarkersNode}\n",
       "\\keyword{differential\\_expression}{FindMarkers}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "Finds markers (differentially expressed genes) for identity classes\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "FindMarkers(object, ...)\n",
       "\n",
       "## Default S3 method:\n",
       "FindMarkers(\n",
       "  object,\n",
       "  slot = \"data\",\n",
       "  counts = numeric(),\n",
       "  cells.1 = NULL,\n",
       "  cells.2 = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  fc.results = NULL,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Assay'\n",
       "FindMarkers(\n",
       "  object,\n",
       "  slot = \"data\",\n",
       "  cells.1 = NULL,\n",
       "  cells.2 = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  mean.fxn = NULL,\n",
       "  fc.name = NULL,\n",
       "  base = 2,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'DimReduc'\n",
       "FindMarkers(\n",
       "  object,\n",
       "  cells.1 = NULL,\n",
       "  cells.2 = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  mean.fxn = rowMeans,\n",
       "  ...\n",
       ")\n",
       "\n",
       "## S3 method for class 'Seurat'\n",
       "FindMarkers(\n",
       "  object,\n",
       "  ident.1 = NULL,\n",
       "  ident.2 = NULL,\n",
       "  group.by = NULL,\n",
       "  subset.ident = NULL,\n",
       "  assay = NULL,\n",
       "  slot = \"data\",\n",
       "  reduction = NULL,\n",
       "  features = NULL,\n",
       "  logfc.threshold = 0.25,\n",
       "  test.use = \"wilcox\",\n",
       "  min.pct = 0.1,\n",
       "  min.diff.pct = -Inf,\n",
       "  verbose = TRUE,\n",
       "  only.pos = FALSE,\n",
       "  max.cells.per.ident = Inf,\n",
       "  random.seed = 1,\n",
       "  latent.vars = NULL,\n",
       "  min.cells.feature = 3,\n",
       "  min.cells.group = 3,\n",
       "  pseudocount.use = 1,\n",
       "  mean.fxn = NULL,\n",
       "  fc.name = NULL,\n",
       "  base = 2,\n",
       "  ...\n",
       ")\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{object}] An object\n",
       "\n",
       "\\item[\\code{...}] Arguments passed to other methods and to specific DE methods\n",
       "\n",
       "\\item[\\code{slot}] Slot to pull data from; note that if \\code{test.use} is \"negbinom\", \"poisson\", or \"DESeq2\",\n",
       "\\code{slot} will be set to \"counts\"\n",
       "\n",
       "\\item[\\code{counts}] Count matrix if using scale.data for DE tests. This is used for\n",
       "computing pct.1 and pct.2 and for filtering features based on fraction\n",
       "expressing\n",
       "\n",
       "\\item[\\code{cells.1}] Vector of cell names belonging to group 1\n",
       "\n",
       "\\item[\\code{cells.2}] Vector of cell names belonging to group 2\n",
       "\n",
       "\\item[\\code{features}] Genes to test. Default is to use all genes\n",
       "\n",
       "\\item[\\code{logfc.threshold}] Limit testing to genes which show, on average, at least\n",
       "X-fold difference (log-scale) between the two groups of cells. Default is 0.25\n",
       "Increasing logfc.threshold speeds up the function, but can miss weaker signals.\n",
       "\n",
       "\\item[\\code{test.use}] Denotes which test to use. Available options are:\n",
       "\\begin{itemize}\n",
       "\n",
       "\\item{} \"wilcox\" : Identifies differentially expressed genes between two\n",
       "groups of cells using a Wilcoxon Rank Sum test (default)\n",
       "\\item{} \"bimod\" : Likelihood-ratio test for single cell gene expression,\n",
       "(McDavid et al., Bioinformatics, 2013)\n",
       "\\item{} \"roc\" : Identifies 'markers' of gene expression using ROC analysis.\n",
       "For each gene, evaluates (using AUC) a classifier built on that gene alone,\n",
       "to classify between two groups of cells. An AUC value of 1 means that\n",
       "expression values for this gene alone can perfectly classify the two\n",
       "groupings (i.e. Each of the cells in cells.1 exhibit a higher level than\n",
       "each of the cells in cells.2). An AUC value of 0 also means there is perfect\n",
       "classification, but in the other direction. A value of 0.5 implies that\n",
       "the gene has no predictive power to classify the two groups. Returns a\n",
       "'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially\n",
       "expressed genes.\n",
       "\\item{} \"t\" : Identify differentially expressed genes between two groups of\n",
       "cells using the Student's t-test.\n",
       "\\item{} \"negbinom\" : Identifies differentially expressed genes between two\n",
       "groups of cells using a negative binomial generalized linear model.\n",
       "Use only for UMI-based datasets\n",
       "\\item{} \"poisson\" : Identifies differentially expressed genes between two\n",
       "groups of cells using a poisson generalized linear model.\n",
       "Use only for UMI-based datasets\n",
       "\\item{} \"LR\" : Uses a logistic regression framework to determine differentially\n",
       "expressed genes. Constructs a logistic regression model predicting group\n",
       "membership based on each feature individually and compares this to a null\n",
       "model with a likelihood ratio test.\n",
       "\\item{} \"MAST\" : Identifies differentially expressed genes between two groups\n",
       "of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST\n",
       "package to run the DE testing.\n",
       "\\item{} \"DESeq2\" : Identifies differentially expressed genes between two groups\n",
       "of cells based on a model using DESeq2 which uses a negative binomial\n",
       "distribution (Love et al, Genome Biology, 2014).This test does not support\n",
       "pre-filtering of genes based on average difference (or percent detection rate)\n",
       "between cell groups. However, genes may be pre-filtered based on their\n",
       "minimum detection rate (min.pct) across both cell groups. To use this method,\n",
       "please install DESeq2, using the instructions at\n",
       "https://bioconductor.org/packages/release/bioc/html/DESeq2.html\n",
       "\n",
       "\\end{itemize}\n",
       "\n",
       "\n",
       "\\item[\\code{min.pct}] only test genes that are detected in a minimum fraction of\n",
       "min.pct cells in either of the two populations. Meant to speed up the function\n",
       "by not testing genes that are very infrequently expressed. Default is 0.1\n",
       "\n",
       "\\item[\\code{min.diff.pct}] only test genes that show a minimum difference in the\n",
       "fraction of detection between the two groups. Set to -Inf by default\n",
       "\n",
       "\\item[\\code{verbose}] Print a progress bar once expression testing begins\n",
       "\n",
       "\\item[\\code{only.pos}] Only return positive markers (FALSE by default)\n",
       "\n",
       "\\item[\\code{max.cells.per.ident}] Down sample each identity class to a max number.\n",
       "Default is no downsampling. Not activated by default (set to Inf)\n",
       "\n",
       "\\item[\\code{random.seed}] Random seed for downsampling\n",
       "\n",
       "\\item[\\code{latent.vars}] Variables to test, used only when \\code{test.use} is one of\n",
       "'LR', 'negbinom', 'poisson', or 'MAST'\n",
       "\n",
       "\\item[\\code{min.cells.feature}] Minimum number of cells expressing the feature in at least one\n",
       "of the two groups, currently only used for poisson and negative binomial tests\n",
       "\n",
       "\\item[\\code{min.cells.group}] Minimum number of cells in one of the groups\n",
       "\n",
       "\\item[\\code{pseudocount.use}] Pseudocount to add to averaged expression values when\n",
       "calculating logFC. 1 by default.\n",
       "\n",
       "\\item[\\code{fc.results}] data.frame from FoldChange\n",
       "\n",
       "\\item[\\code{mean.fxn}] Function to use for fold change or average difference calculation.\n",
       "If NULL, the appropriate function will be chose according to the slot used\n",
       "\n",
       "\\item[\\code{fc.name}] Name of the fold change, average difference, or custom function column\n",
       "in the output data.frame. If NULL, the fold change column will be named\n",
       "according to the logarithm base (eg, \"avg\\_log2FC\"), or if using the scale.data\n",
       "slot \"avg\\_diff\".\n",
       "\n",
       "\\item[\\code{base}] The base with respect to which logarithms are computed.\n",
       "\n",
       "\\item[\\code{ident.1}] Identity class to define markers for; pass an object of class\n",
       "\\code{phylo} or 'clustertree' to find markers for a node in a cluster tree;\n",
       "passing 'clustertree' requires \\code{\\LinkA{BuildClusterTree}{BuildClusterTree}} to have been run\n",
       "\n",
       "\\item[\\code{ident.2}] A second identity class for comparison; if \\code{NULL},\n",
       "use all other cells for comparison; if an object of class \\code{phylo} or\n",
       "'clustertree' is passed to \\code{ident.1}, must pass a node to find markers for\n",
       "\n",
       "\\item[\\code{group.by}] Regroup cells into a different identity class prior to performing differential expression (see example)\n",
       "\n",
       "\\item[\\code{subset.ident}] Subset a particular identity class prior to regrouping. Only relevant if group.by is set (see example)\n",
       "\n",
       "\\item[\\code{assay}] Assay to use in differential expression testing\n",
       "\n",
       "\\item[\\code{reduction}] Reduction to use in differential expression testing - will test for DE on cell embeddings\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "p-value adjustment is performed using bonferroni correction based on\n",
       "the total number of genes in the dataset. Other correction methods are not\n",
       "recommended, as Seurat pre-filters genes using the arguments above, reducing\n",
       "the number of tests performed. Lastly, as Aaron Lun has pointed out, p-values\n",
       "should be interpreted cautiously, as the genes used for clustering are the\n",
       "same genes tested for differential expression.\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Value}\n",
       "data.frame with a ranked list of putative markers as rows, and associated\n",
       "statistics as columns (p-values, ROC score, etc., depending on the test used (\\code{test.use})). The following columns are always present:\n",
       "\\begin{itemize}\n",
       "\n",
       "\\item{} \\code{avg\\_logFC}: log fold-chage of the average expression between the two groups. Positive values indicate that the gene is more highly expressed in the first group\n",
       "\\item{} \\code{pct.1}: The percentage of cells where the gene is detected in the first group\n",
       "\\item{} \\code{pct.2}: The percentage of cells where the gene is detected in the second group\n",
       "\\item{} \\code{p\\_val\\_adj}: Adjusted p-value, based on bonferroni correction using all genes in the dataset\n",
       "\n",
       "\\end{itemize}\n",
       "\n",
       "\\end{Value}\n",
       "%\n",
       "\\begin{References}\\relax\n",
       "McDavid A, Finak G, Chattopadyay PK, et al. Data exploration,\n",
       "quality control and testing in single-cell qPCR-based gene expression experiments.\n",
       "Bioinformatics. 2013;29(4):461-467. doi:10.1093/bioinformatics/bts714\n",
       "\n",
       "Trapnell C, et al. The dynamics and regulators of cell fate\n",
       "decisions are revealed by pseudotemporal ordering of single cells. Nature\n",
       "Biotechnology volume 32, pages 381-386 (2014)\n",
       "\n",
       "Andrew McDavid, Greg Finak and Masanao Yajima (2017). MAST: Model-based\n",
       "Analysis of Single Cell Transcriptomics. R package version 1.2.1.\n",
       "https://github.com/RGLab/MAST/\n",
       "\n",
       "Love MI, Huber W and Anders S (2014). \"Moderated estimation of\n",
       "fold change and dispersion for RNA-seq data with DESeq2.\" Genome Biology.\n",
       "https://bioconductor.org/packages/release/bioc/html/DESeq2.html\n",
       "\\end{References}\n",
       "%\n",
       "\\begin{SeeAlso}\\relax\n",
       "\\code{FoldChange}\n",
       "\\end{SeeAlso}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "data(\"pbmc_small\")\n",
       "# Find markers for cluster 2\n",
       "markers <- FindMarkers(object = pbmc_small, ident.1 = 2)\n",
       "head(x = markers)\n",
       "\n",
       "# Take all cells in cluster 2, and find markers that separate cells in the 'g1' group (metadata\n",
       "# variable 'group')\n",
       "markers <- FindMarkers(pbmc_small, ident.1 = \"g1\", group.by = 'groups', subset.ident = \"2\")\n",
       "head(x = markers)\n",
       "\n",
       "# Pass 'clustertree' or an object of class phylo to ident.1 and\n",
       "# a node to ident.2 as a replacement for FindMarkersNode\n",
       "pbmc_small <- BuildClusterTree(object = pbmc_small)\n",
       "markers <- FindMarkers(object = pbmc_small, ident.1 = 'clustertree', ident.2 = 5)\n",
       "head(x = markers)\n",
       "\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "FindMarkers               package:Seurat               R Documentation\n",
       "\n",
       "_\bG_\be_\bn_\be _\be_\bx_\bp_\br_\be_\bs_\bs_\bi_\bo_\bn _\bm_\ba_\br_\bk_\be_\br_\bs _\bo_\bf _\bi_\bd_\be_\bn_\bt_\bi_\bt_\by _\bc_\bl_\ba_\bs_\bs_\be_\bs\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     Finds markers (differentially expressed genes) for identity\n",
       "     classes\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     FindMarkers(object, ...)\n",
       "     \n",
       "     ## Default S3 method:\n",
       "     FindMarkers(\n",
       "       object,\n",
       "       slot = \"data\",\n",
       "       counts = numeric(),\n",
       "       cells.1 = NULL,\n",
       "       cells.2 = NULL,\n",
       "       features = NULL,\n",
       "       logfc.threshold = 0.25,\n",
       "       test.use = \"wilcox\",\n",
       "       min.pct = 0.1,\n",
       "       min.diff.pct = -Inf,\n",
       "       verbose = TRUE,\n",
       "       only.pos = FALSE,\n",
       "       max.cells.per.ident = Inf,\n",
       "       random.seed = 1,\n",
       "       latent.vars = NULL,\n",
       "       min.cells.feature = 3,\n",
       "       min.cells.group = 3,\n",
       "       pseudocount.use = 1,\n",
       "       fc.results = NULL,\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "     ## S3 method for class 'Assay'\n",
       "     FindMarkers(\n",
       "       object,\n",
       "       slot = \"data\",\n",
       "       cells.1 = NULL,\n",
       "       cells.2 = NULL,\n",
       "       features = NULL,\n",
       "       logfc.threshold = 0.25,\n",
       "       test.use = \"wilcox\",\n",
       "       min.pct = 0.1,\n",
       "       min.diff.pct = -Inf,\n",
       "       verbose = TRUE,\n",
       "       only.pos = FALSE,\n",
       "       max.cells.per.ident = Inf,\n",
       "       random.seed = 1,\n",
       "       latent.vars = NULL,\n",
       "       min.cells.feature = 3,\n",
       "       min.cells.group = 3,\n",
       "       pseudocount.use = 1,\n",
       "       mean.fxn = NULL,\n",
       "       fc.name = NULL,\n",
       "       base = 2,\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "     ## S3 method for class 'DimReduc'\n",
       "     FindMarkers(\n",
       "       object,\n",
       "       cells.1 = NULL,\n",
       "       cells.2 = NULL,\n",
       "       features = NULL,\n",
       "       logfc.threshold = 0.25,\n",
       "       test.use = \"wilcox\",\n",
       "       min.pct = 0.1,\n",
       "       min.diff.pct = -Inf,\n",
       "       verbose = TRUE,\n",
       "       only.pos = FALSE,\n",
       "       max.cells.per.ident = Inf,\n",
       "       random.seed = 1,\n",
       "       latent.vars = NULL,\n",
       "       min.cells.feature = 3,\n",
       "       min.cells.group = 3,\n",
       "       pseudocount.use = 1,\n",
       "       mean.fxn = rowMeans,\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "     ## S3 method for class 'Seurat'\n",
       "     FindMarkers(\n",
       "       object,\n",
       "       ident.1 = NULL,\n",
       "       ident.2 = NULL,\n",
       "       group.by = NULL,\n",
       "       subset.ident = NULL,\n",
       "       assay = NULL,\n",
       "       slot = \"data\",\n",
       "       reduction = NULL,\n",
       "       features = NULL,\n",
       "       logfc.threshold = 0.25,\n",
       "       test.use = \"wilcox\",\n",
       "       min.pct = 0.1,\n",
       "       min.diff.pct = -Inf,\n",
       "       verbose = TRUE,\n",
       "       only.pos = FALSE,\n",
       "       max.cells.per.ident = Inf,\n",
       "       random.seed = 1,\n",
       "       latent.vars = NULL,\n",
       "       min.cells.feature = 3,\n",
       "       min.cells.group = 3,\n",
       "       pseudocount.use = 1,\n",
       "       mean.fxn = NULL,\n",
       "       fc.name = NULL,\n",
       "       base = 2,\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "  object: An object\n",
       "\n",
       "     ...: Arguments passed to other methods and to specific DE methods\n",
       "\n",
       "    slot: Slot to pull data from; note that if ‘test.use’ is\n",
       "          \"negbinom\", \"poisson\", or \"DESeq2\", ‘slot’ will be set to\n",
       "          \"counts\"\n",
       "\n",
       "  counts: Count matrix if using scale.data for DE tests. This is used\n",
       "          for computing pct.1 and pct.2 and for filtering features\n",
       "          based on fraction expressing\n",
       "\n",
       " cells.1: Vector of cell names belonging to group 1\n",
       "\n",
       " cells.2: Vector of cell names belonging to group 2\n",
       "\n",
       "features: Genes to test. Default is to use all genes\n",
       "\n",
       "logfc.threshold: Limit testing to genes which show, on average, at\n",
       "          least X-fold difference (log-scale) between the two groups of\n",
       "          cells. Default is 0.25 Increasing logfc.threshold speeds up\n",
       "          the function, but can miss weaker signals.\n",
       "\n",
       "test.use: Denotes which test to use. Available options are:\n",
       "\n",
       "            • \"wilcox\" : Identifies differentially expressed genes\n",
       "              between two groups of cells using a Wilcoxon Rank Sum\n",
       "              test (default)\n",
       "\n",
       "            • \"bimod\" : Likelihood-ratio test for single cell gene\n",
       "              expression, (McDavid et al., Bioinformatics, 2013)\n",
       "\n",
       "            • \"roc\" : Identifies 'markers' of gene expression using ROC\n",
       "              analysis.  For each gene, evaluates (using AUC) a\n",
       "              classifier built on that gene alone, to classify between\n",
       "              two groups of cells. An AUC value of 1 means that\n",
       "              expression values for this gene alone can perfectly\n",
       "              classify the two groupings (i.e. Each of the cells in\n",
       "              cells.1 exhibit a higher level than each of the cells in\n",
       "              cells.2). An AUC value of 0 also means there is perfect\n",
       "              classification, but in the other direction. A value of\n",
       "              0.5 implies that the gene has no predictive power to\n",
       "              classify the two groups. Returns a 'predictive power'\n",
       "              (abs(AUC-0.5) * 2) ranked matrix of putative\n",
       "              differentially expressed genes.\n",
       "\n",
       "            • \"t\" : Identify differentially expressed genes between two\n",
       "              groups of cells using the Student's t-test.\n",
       "\n",
       "            • \"negbinom\" : Identifies differentially expressed genes\n",
       "              between two groups of cells using a negative binomial\n",
       "              generalized linear model.  Use only for UMI-based\n",
       "              datasets\n",
       "\n",
       "            • \"poisson\" : Identifies differentially expressed genes\n",
       "              between two groups of cells using a poisson generalized\n",
       "              linear model.  Use only for UMI-based datasets\n",
       "\n",
       "            • \"LR\" : Uses a logistic regression framework to determine\n",
       "              differentially expressed genes. Constructs a logistic\n",
       "              regression model predicting group membership based on\n",
       "              each feature individually and compares this to a null\n",
       "              model with a likelihood ratio test.\n",
       "\n",
       "            • \"MAST\" : Identifies differentially expressed genes\n",
       "              between two groups of cells using a hurdle model tailored\n",
       "              to scRNA-seq data. Utilizes the MAST package to run the\n",
       "              DE testing.\n",
       "\n",
       "            • \"DESeq2\" : Identifies differentially expressed genes\n",
       "              between two groups of cells based on a model using DESeq2\n",
       "              which uses a negative binomial distribution (Love et al,\n",
       "              Genome Biology, 2014).This test does not support\n",
       "              pre-filtering of genes based on average difference (or\n",
       "              percent detection rate) between cell groups. However,\n",
       "              genes may be pre-filtered based on their minimum\n",
       "              detection rate (min.pct) across both cell groups. To use\n",
       "              this method, please install DESeq2, using the\n",
       "              instructions at\n",
       "              https://bioconductor.org/packages/release/bioc/html/DESeq2.html\n",
       "\n",
       " min.pct: only test genes that are detected in a minimum fraction of\n",
       "          min.pct cells in either of the two populations. Meant to\n",
       "          speed up the function by not testing genes that are very\n",
       "          infrequently expressed. Default is 0.1\n",
       "\n",
       "min.diff.pct: only test genes that show a minimum difference in the\n",
       "          fraction of detection between the two groups. Set to -Inf by\n",
       "          default\n",
       "\n",
       " verbose: Print a progress bar once expression testing begins\n",
       "\n",
       "only.pos: Only return positive markers (FALSE by default)\n",
       "\n",
       "max.cells.per.ident: Down sample each identity class to a max number.\n",
       "          Default is no downsampling. Not activated by default (set to\n",
       "          Inf)\n",
       "\n",
       "random.seed: Random seed for downsampling\n",
       "\n",
       "latent.vars: Variables to test, used only when ‘test.use’ is one of\n",
       "          'LR', 'negbinom', 'poisson', or 'MAST'\n",
       "\n",
       "min.cells.feature: Minimum number of cells expressing the feature in at\n",
       "          least one of the two groups, currently only used for poisson\n",
       "          and negative binomial tests\n",
       "\n",
       "min.cells.group: Minimum number of cells in one of the groups\n",
       "\n",
       "pseudocount.use: Pseudocount to add to averaged expression values when\n",
       "          calculating logFC. 1 by default.\n",
       "\n",
       "fc.results: data.frame from FoldChange\n",
       "\n",
       "mean.fxn: Function to use for fold change or average difference\n",
       "          calculation. If NULL, the appropriate function will be chose\n",
       "          according to the slot used\n",
       "\n",
       " fc.name: Name of the fold change, average difference, or custom\n",
       "          function column in the output data.frame. If NULL, the fold\n",
       "          change column will be named according to the logarithm base\n",
       "          (eg, \"avg_log2FC\"), or if using the scale.data slot\n",
       "          \"avg_diff\".\n",
       "\n",
       "    base: The base with respect to which logarithms are computed.\n",
       "\n",
       " ident.1: Identity class to define markers for; pass an object of class\n",
       "          ‘phylo’ or 'clustertree' to find markers for a node in a\n",
       "          cluster tree; passing 'clustertree' requires\n",
       "          ‘BuildClusterTree’ to have been run\n",
       "\n",
       " ident.2: A second identity class for comparison; if ‘NULL’, use all\n",
       "          other cells for comparison; if an object of class ‘phylo’ or\n",
       "          'clustertree' is passed to ‘ident.1’, must pass a node to\n",
       "          find markers for\n",
       "\n",
       "group.by: Regroup cells into a different identity class prior to\n",
       "          performing differential expression (see example)\n",
       "\n",
       "subset.ident: Subset a particular identity class prior to regrouping.\n",
       "          Only relevant if group.by is set (see example)\n",
       "\n",
       "   assay: Assay to use in differential expression testing\n",
       "\n",
       "reduction: Reduction to use in differential expression testing - will\n",
       "          test for DE on cell embeddings\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     p-value adjustment is performed using bonferroni correction based\n",
       "     on the total number of genes in the dataset. Other correction\n",
       "     methods are not recommended, as Seurat pre-filters genes using the\n",
       "     arguments above, reducing the number of tests performed. Lastly,\n",
       "     as Aaron Lun has pointed out, p-values should be interpreted\n",
       "     cautiously, as the genes used for clustering are the same genes\n",
       "     tested for differential expression.\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     data.frame with a ranked list of putative markers as rows, and\n",
       "     associated statistics as columns (p-values, ROC score, etc.,\n",
       "     depending on the test used (‘test.use’)). The following columns\n",
       "     are always present:\n",
       "\n",
       "        • ‘avg_logFC’: log fold-chage of the average expression between\n",
       "          the two groups. Positive values indicate that the gene is\n",
       "          more highly expressed in the first group\n",
       "\n",
       "        • ‘pct.1’: The percentage of cells where the gene is detected\n",
       "          in the first group\n",
       "\n",
       "        • ‘pct.2’: The percentage of cells where the gene is detected\n",
       "          in the second group\n",
       "\n",
       "        • ‘p_val_adj’: Adjusted p-value, based on bonferroni correction\n",
       "          using all genes in the dataset\n",
       "\n",
       "_\bR_\be_\bf_\be_\br_\be_\bn_\bc_\be_\bs:\n",
       "\n",
       "     McDavid A, Finak G, Chattopadyay PK, et al. Data exploration,\n",
       "     quality control and testing in single-cell qPCR-based gene\n",
       "     expression experiments. Bioinformatics. 2013;29(4):461-467.\n",
       "     doi:10.1093/bioinformatics/bts714\n",
       "\n",
       "     Trapnell C, et al. The dynamics and regulators of cell fate\n",
       "     decisions are revealed by pseudotemporal ordering of single cells.\n",
       "     Nature Biotechnology volume 32, pages 381-386 (2014)\n",
       "\n",
       "     Andrew McDavid, Greg Finak and Masanao Yajima (2017). MAST:\n",
       "     Model-based Analysis of Single Cell Transcriptomics. R package\n",
       "     version 1.2.1. https://github.com/RGLab/MAST/\n",
       "\n",
       "     Love MI, Huber W and Anders S (2014). \"Moderated estimation of\n",
       "     fold change and dispersion for RNA-seq data with DESeq2.\" Genome\n",
       "     Biology.\n",
       "     https://bioconductor.org/packages/release/bioc/html/DESeq2.html\n",
       "\n",
       "_\bS_\be_\be _\bA_\bl_\bs_\bo:\n",
       "\n",
       "     ‘FoldChange’\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     data(\"pbmc_small\")\n",
       "     # Find markers for cluster 2\n",
       "     markers <- FindMarkers(object = pbmc_small, ident.1 = 2)\n",
       "     head(x = markers)\n",
       "     \n",
       "     # Take all cells in cluster 2, and find markers that separate cells in the 'g1' group (metadata\n",
       "     # variable 'group')\n",
       "     markers <- FindMarkers(pbmc_small, ident.1 = \"g1\", group.by = 'groups', subset.ident = \"2\")\n",
       "     head(x = markers)\n",
       "     \n",
       "     # Pass 'clustertree' or an object of class phylo to ident.1 and\n",
       "     # a node to ident.2 as a replacement for FindMarkersNode\n",
       "     pbmc_small <- BuildClusterTree(object = pbmc_small)\n",
       "     markers <- FindMarkers(object = pbmc_small, ident.1 = 'clustertree', ident.2 = 5)\n",
       "     head(x = markers)\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?FindMarkers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "blank-thompson",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T18:06:58.778162Z",
     "iopub.status.busy": "2023-01-31T18:06:58.776814Z",
     "iopub.status.idle": "2023-01-31T18:07:52.313243Z",
     "shell.execute_reply": "2023-01-31T18:07:52.311551Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Done!\n",
      "\n",
      "Combining coefficients and standard errors\n",
      "\n",
      "Calculating log-fold changes\n",
      "\n",
      "Calculating likelihood ratio tests\n",
      "\n",
      "Refitting on reduced model...\n",
      "\n",
      "\n",
      "Done!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Perform differential expression analysis\n",
    "de_results <- FindMarkers(pooled_ad, cells_1, cells_2, test.use = \"MAST\", logfc.threshold = 0.25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "cathedral-proposition",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T18:29:39.853587Z",
     "iopub.status.busy": "2023-01-31T18:29:39.852232Z",
     "iopub.status.idle": "2023-01-31T18:29:39.997177Z",
     "shell.execute_reply": "2023-01-31T18:29:39.995722Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4886 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>p_val</th><th scope=col>avg_log2FC</th><th scope=col>pct.1</th><th scope=col>pct.2</th><th scope=col>p_val_adj</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>PTN</th><td>0.0135631530</td><td> 3.481791</td><td>0.333</td><td>0.169</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>UBE2C</th><td>0.0037808086</td><td> 3.038616</td><td>0.667</td><td>0.108</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>COL3A1</th><td>0.0002788539</td><td> 2.802564</td><td>0.333</td><td>0.431</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>PTTG1</th><td>0.0157257432</td><td> 2.527251</td><td>0.667</td><td>0.138</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ARL6IP1</th><td>0.0008037031</td><td> 2.104016</td><td>0.667</td><td>0.654</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SFRP2</th><td>0.0825681795</td><td> 2.083583</td><td>0.333</td><td>0.092</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>RPS4Y1</th><td>0.0341172639</td><td>-2.015701</td><td>0.000</td><td>0.592</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MGP</th><td>0.1816760752</td><td> 2.002652</td><td>0.333</td><td>0.115</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CDK1</th><td>0.0265915497</td><td> 1.967298</td><td>0.667</td><td>0.123</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>HMGB2</th><td>0.0073991748</td><td> 1.916832</td><td>1.000</td><td>0.400</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>LDHA</th><td>0.0558685935</td><td>-1.875169</td><td>0.333</td><td>0.900</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>UBE2S</th><td>0.0000481710</td><td> 1.874852</td><td>0.667</td><td>0.400</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>EIF4EBP1</th><td>0.0205379982</td><td>-1.773034</td><td>0.333</td><td>0.923</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>BCAP31</th><td>0.0027296316</td><td>-1.759390</td><td>0.000</td><td>0.823</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ACKR3</th><td>0.0072208498</td><td>-1.758801</td><td>0.000</td><td>0.754</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>PCLAF</th><td>0.0132228222</td><td> 1.744362</td><td>1.000</td><td>0.215</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>HMGN2</th><td>0.0084806258</td><td> 1.742515</td><td>1.000</td><td>0.808</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MKI67</th><td>0.0052457153</td><td> 1.730225</td><td>0.667</td><td>0.108</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>PRRX1</th><td>0.0819901800</td><td> 1.725540</td><td>0.333</td><td>0.077</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>PRSS2</th><td>0.1368676593</td><td>-1.720786</td><td>1.000</td><td>0.746</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SMAD1</th><td>0.0024005704</td><td>-1.701940</td><td>0.000</td><td>0.831</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CDC20</th><td>0.0135808127</td><td> 1.701473</td><td>0.667</td><td>0.038</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CKS2</th><td>0.0054080547</td><td> 1.697888</td><td>0.667</td><td>0.438</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>P4HB</th><td>0.0210497524</td><td>-1.686653</td><td>0.333</td><td>0.915</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CENPN</th><td>0.0031935513</td><td> 1.686444</td><td>0.667</td><td>0.154</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CCNA2</th><td>0.0216060250</td><td> 1.665068</td><td>0.667</td><td>0.069</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CRABP1</th><td>0.0029080452</td><td> 1.655757</td><td>0.333</td><td>0.292</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ESM1</th><td>0.2845798816</td><td>-1.628091</td><td>0.333</td><td>0.738</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CENPW</th><td>0.0078197942</td><td> 1.628023</td><td>0.667</td><td>0.277</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>C2orf69</th><td>0.0048087666</td><td> 1.612651</td><td>0.667</td><td>0.146</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>SARAF</th><td>0.8394885</td><td>-0.2511795</td><td>0.667</td><td>0.815</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>IDS</th><td>0.7640309</td><td> 0.2511068</td><td>0.667</td><td>0.477</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ARRB2</th><td>0.5680651</td><td> 0.2511033</td><td>0.333</td><td>0.100</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>RIOK1</th><td>0.3045630</td><td>-0.2510549</td><td>0.000</td><td>0.223</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SRD5A3</th><td>0.9271491</td><td>-0.2510266</td><td>0.333</td><td>0.415</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ELMOD2</th><td>0.2932709</td><td>-0.2510231</td><td>0.000</td><td>0.231</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ARMC7</th><td>0.6083595</td><td> 0.2510068</td><td>0.333</td><td>0.115</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>HAUS2</th><td>0.3283637</td><td>-0.2509677</td><td>0.000</td><td>0.208</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>FAM136A</th><td>0.8431908</td><td>-0.2509642</td><td>0.333</td><td>0.500</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>BAK1</th><td>0.2155462</td><td>-0.2509196</td><td>0.000</td><td>0.292</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>E2F1</th><td>0.3338776</td><td> 0.2508956</td><td>0.333</td><td>0.054</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CCDC85C</th><td>0.2422634</td><td>-0.2508947</td><td>0.000</td><td>0.269</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>PSAT1</th><td>0.5969442</td><td> 0.2508419</td><td>0.333</td><td>0.108</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>KDELC2</th><td>0.6733051</td><td> 0.2508333</td><td>0.333</td><td>0.169</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>KAT6B</th><td>0.3045630</td><td>-0.2507993</td><td>0.000</td><td>0.223</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>VKORC1L1</th><td>0.2155462</td><td>-0.2507013</td><td>0.000</td><td>0.292</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CD58</th><td>0.6240221</td><td> 0.2506750</td><td>0.333</td><td>0.123</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ME3</th><td>0.3409089</td><td>-0.2506518</td><td>0.000</td><td>0.200</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>RANBP6</th><td>0.2616326</td><td>-0.2506493</td><td>0.000</td><td>0.254</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>TAPBPL</th><td>0.6733858</td><td> 0.2506337</td><td>0.333</td><td>0.154</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MSMP</th><td>0.3813766</td><td>-0.2505910</td><td>0.000</td><td>0.177</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SLC15A4</th><td>0.6375639</td><td> 0.2505787</td><td>0.333</td><td>0.131</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>NUS1</th><td>0.2517845</td><td>-0.2505175</td><td>0.000</td><td>0.262</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>TERF2</th><td>0.2330574</td><td>-0.2504391</td><td>0.000</td><td>0.277</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ARL15</th><td>0.3045630</td><td>-0.2503660</td><td>0.000</td><td>0.223</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>IPO7</th><td>0.8205137</td><td>-0.2503627</td><td>0.333</td><td>0.515</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>WDR6</th><td>0.5869650</td><td> 0.2503068</td><td>0.333</td><td>0.346</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MUL1</th><td>0.2330574</td><td>-0.2502621</td><td>0.000</td><td>0.277</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>OPA3</th><td>0.2932709</td><td>-0.2501218</td><td>0.000</td><td>0.231</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MTMR2</th><td>0.4871587</td><td> 0.2501110</td><td>0.333</td><td>0.385</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4886 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & p\\_val & avg\\_log2FC & pct.1 & pct.2 & p\\_val\\_adj\\\\\n",
       "  & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tPTN & 0.0135631530 &  3.481791 & 0.333 & 0.169 & 1\\\\\n",
       "\tUBE2C & 0.0037808086 &  3.038616 & 0.667 & 0.108 & 1\\\\\n",
       "\tCOL3A1 & 0.0002788539 &  2.802564 & 0.333 & 0.431 & 1\\\\\n",
       "\tPTTG1 & 0.0157257432 &  2.527251 & 0.667 & 0.138 & 1\\\\\n",
       "\tARL6IP1 & 0.0008037031 &  2.104016 & 0.667 & 0.654 & 1\\\\\n",
       "\tSFRP2 & 0.0825681795 &  2.083583 & 0.333 & 0.092 & 1\\\\\n",
       "\tRPS4Y1 & 0.0341172639 & -2.015701 & 0.000 & 0.592 & 1\\\\\n",
       "\tMGP & 0.1816760752 &  2.002652 & 0.333 & 0.115 & 1\\\\\n",
       "\tCDK1 & 0.0265915497 &  1.967298 & 0.667 & 0.123 & 1\\\\\n",
       "\tHMGB2 & 0.0073991748 &  1.916832 & 1.000 & 0.400 & 1\\\\\n",
       "\tLDHA & 0.0558685935 & -1.875169 & 0.333 & 0.900 & 1\\\\\n",
       "\tUBE2S & 0.0000481710 &  1.874852 & 0.667 & 0.400 & 1\\\\\n",
       "\tEIF4EBP1 & 0.0205379982 & -1.773034 & 0.333 & 0.923 & 1\\\\\n",
       "\tBCAP31 & 0.0027296316 & -1.759390 & 0.000 & 0.823 & 1\\\\\n",
       "\tACKR3 & 0.0072208498 & -1.758801 & 0.000 & 0.754 & 1\\\\\n",
       "\tPCLAF & 0.0132228222 &  1.744362 & 1.000 & 0.215 & 1\\\\\n",
       "\tHMGN2 & 0.0084806258 &  1.742515 & 1.000 & 0.808 & 1\\\\\n",
       "\tMKI67 & 0.0052457153 &  1.730225 & 0.667 & 0.108 & 1\\\\\n",
       "\tPRRX1 & 0.0819901800 &  1.725540 & 0.333 & 0.077 & 1\\\\\n",
       "\tPRSS2 & 0.1368676593 & -1.720786 & 1.000 & 0.746 & 1\\\\\n",
       "\tSMAD1 & 0.0024005704 & -1.701940 & 0.000 & 0.831 & 1\\\\\n",
       "\tCDC20 & 0.0135808127 &  1.701473 & 0.667 & 0.038 & 1\\\\\n",
       "\tCKS2 & 0.0054080547 &  1.697888 & 0.667 & 0.438 & 1\\\\\n",
       "\tP4HB & 0.0210497524 & -1.686653 & 0.333 & 0.915 & 1\\\\\n",
       "\tCENPN & 0.0031935513 &  1.686444 & 0.667 & 0.154 & 1\\\\\n",
       "\tCCNA2 & 0.0216060250 &  1.665068 & 0.667 & 0.069 & 1\\\\\n",
       "\tCRABP1 & 0.0029080452 &  1.655757 & 0.333 & 0.292 & 1\\\\\n",
       "\tESM1 & 0.2845798816 & -1.628091 & 0.333 & 0.738 & 1\\\\\n",
       "\tCENPW & 0.0078197942 &  1.628023 & 0.667 & 0.277 & 1\\\\\n",
       "\tC2orf69 & 0.0048087666 &  1.612651 & 0.667 & 0.146 & 1\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tSARAF & 0.8394885 & -0.2511795 & 0.667 & 0.815 & 1\\\\\n",
       "\tIDS & 0.7640309 &  0.2511068 & 0.667 & 0.477 & 1\\\\\n",
       "\tARRB2 & 0.5680651 &  0.2511033 & 0.333 & 0.100 & 1\\\\\n",
       "\tRIOK1 & 0.3045630 & -0.2510549 & 0.000 & 0.223 & 1\\\\\n",
       "\tSRD5A3 & 0.9271491 & -0.2510266 & 0.333 & 0.415 & 1\\\\\n",
       "\tELMOD2 & 0.2932709 & -0.2510231 & 0.000 & 0.231 & 1\\\\\n",
       "\tARMC7 & 0.6083595 &  0.2510068 & 0.333 & 0.115 & 1\\\\\n",
       "\tHAUS2 & 0.3283637 & -0.2509677 & 0.000 & 0.208 & 1\\\\\n",
       "\tFAM136A & 0.8431908 & -0.2509642 & 0.333 & 0.500 & 1\\\\\n",
       "\tBAK1 & 0.2155462 & -0.2509196 & 0.000 & 0.292 & 1\\\\\n",
       "\tE2F1 & 0.3338776 &  0.2508956 & 0.333 & 0.054 & 1\\\\\n",
       "\tCCDC85C & 0.2422634 & -0.2508947 & 0.000 & 0.269 & 1\\\\\n",
       "\tPSAT1 & 0.5969442 &  0.2508419 & 0.333 & 0.108 & 1\\\\\n",
       "\tKDELC2 & 0.6733051 &  0.2508333 & 0.333 & 0.169 & 1\\\\\n",
       "\tKAT6B & 0.3045630 & -0.2507993 & 0.000 & 0.223 & 1\\\\\n",
       "\tVKORC1L1 & 0.2155462 & -0.2507013 & 0.000 & 0.292 & 1\\\\\n",
       "\tCD58 & 0.6240221 &  0.2506750 & 0.333 & 0.123 & 1\\\\\n",
       "\tME3 & 0.3409089 & -0.2506518 & 0.000 & 0.200 & 1\\\\\n",
       "\tRANBP6 & 0.2616326 & -0.2506493 & 0.000 & 0.254 & 1\\\\\n",
       "\tTAPBPL & 0.6733858 &  0.2506337 & 0.333 & 0.154 & 1\\\\\n",
       "\tMSMP & 0.3813766 & -0.2505910 & 0.000 & 0.177 & 1\\\\\n",
       "\tSLC15A4 & 0.6375639 &  0.2505787 & 0.333 & 0.131 & 1\\\\\n",
       "\tNUS1 & 0.2517845 & -0.2505175 & 0.000 & 0.262 & 1\\\\\n",
       "\tTERF2 & 0.2330574 & -0.2504391 & 0.000 & 0.277 & 1\\\\\n",
       "\tARL15 & 0.3045630 & -0.2503660 & 0.000 & 0.223 & 1\\\\\n",
       "\tIPO7 & 0.8205137 & -0.2503627 & 0.333 & 0.515 & 1\\\\\n",
       "\tWDR6 & 0.5869650 &  0.2503068 & 0.333 & 0.346 & 1\\\\\n",
       "\tMUL1 & 0.2330574 & -0.2502621 & 0.000 & 0.277 & 1\\\\\n",
       "\tOPA3 & 0.2932709 & -0.2501218 & 0.000 & 0.231 & 1\\\\\n",
       "\tMTMR2 & 0.4871587 &  0.2501110 & 0.333 & 0.385 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4886 × 5\n",
       "\n",
       "| <!--/--> | p_val &lt;dbl&gt; | avg_log2FC &lt;dbl&gt; | pct.1 &lt;dbl&gt; | pct.2 &lt;dbl&gt; | p_val_adj &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| PTN | 0.0135631530 |  3.481791 | 0.333 | 0.169 | 1 |\n",
       "| UBE2C | 0.0037808086 |  3.038616 | 0.667 | 0.108 | 1 |\n",
       "| COL3A1 | 0.0002788539 |  2.802564 | 0.333 | 0.431 | 1 |\n",
       "| PTTG1 | 0.0157257432 |  2.527251 | 0.667 | 0.138 | 1 |\n",
       "| ARL6IP1 | 0.0008037031 |  2.104016 | 0.667 | 0.654 | 1 |\n",
       "| SFRP2 | 0.0825681795 |  2.083583 | 0.333 | 0.092 | 1 |\n",
       "| RPS4Y1 | 0.0341172639 | -2.015701 | 0.000 | 0.592 | 1 |\n",
       "| MGP | 0.1816760752 |  2.002652 | 0.333 | 0.115 | 1 |\n",
       "| CDK1 | 0.0265915497 |  1.967298 | 0.667 | 0.123 | 1 |\n",
       "| HMGB2 | 0.0073991748 |  1.916832 | 1.000 | 0.400 | 1 |\n",
       "| LDHA | 0.0558685935 | -1.875169 | 0.333 | 0.900 | 1 |\n",
       "| UBE2S | 0.0000481710 |  1.874852 | 0.667 | 0.400 | 1 |\n",
       "| EIF4EBP1 | 0.0205379982 | -1.773034 | 0.333 | 0.923 | 1 |\n",
       "| BCAP31 | 0.0027296316 | -1.759390 | 0.000 | 0.823 | 1 |\n",
       "| ACKR3 | 0.0072208498 | -1.758801 | 0.000 | 0.754 | 1 |\n",
       "| PCLAF | 0.0132228222 |  1.744362 | 1.000 | 0.215 | 1 |\n",
       "| HMGN2 | 0.0084806258 |  1.742515 | 1.000 | 0.808 | 1 |\n",
       "| MKI67 | 0.0052457153 |  1.730225 | 0.667 | 0.108 | 1 |\n",
       "| PRRX1 | 0.0819901800 |  1.725540 | 0.333 | 0.077 | 1 |\n",
       "| PRSS2 | 0.1368676593 | -1.720786 | 1.000 | 0.746 | 1 |\n",
       "| SMAD1 | 0.0024005704 | -1.701940 | 0.000 | 0.831 | 1 |\n",
       "| CDC20 | 0.0135808127 |  1.701473 | 0.667 | 0.038 | 1 |\n",
       "| CKS2 | 0.0054080547 |  1.697888 | 0.667 | 0.438 | 1 |\n",
       "| P4HB | 0.0210497524 | -1.686653 | 0.333 | 0.915 | 1 |\n",
       "| CENPN | 0.0031935513 |  1.686444 | 0.667 | 0.154 | 1 |\n",
       "| CCNA2 | 0.0216060250 |  1.665068 | 0.667 | 0.069 | 1 |\n",
       "| CRABP1 | 0.0029080452 |  1.655757 | 0.333 | 0.292 | 1 |\n",
       "| ESM1 | 0.2845798816 | -1.628091 | 0.333 | 0.738 | 1 |\n",
       "| CENPW | 0.0078197942 |  1.628023 | 0.667 | 0.277 | 1 |\n",
       "| C2orf69 | 0.0048087666 |  1.612651 | 0.667 | 0.146 | 1 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| SARAF | 0.8394885 | -0.2511795 | 0.667 | 0.815 | 1 |\n",
       "| IDS | 0.7640309 |  0.2511068 | 0.667 | 0.477 | 1 |\n",
       "| ARRB2 | 0.5680651 |  0.2511033 | 0.333 | 0.100 | 1 |\n",
       "| RIOK1 | 0.3045630 | -0.2510549 | 0.000 | 0.223 | 1 |\n",
       "| SRD5A3 | 0.9271491 | -0.2510266 | 0.333 | 0.415 | 1 |\n",
       "| ELMOD2 | 0.2932709 | -0.2510231 | 0.000 | 0.231 | 1 |\n",
       "| ARMC7 | 0.6083595 |  0.2510068 | 0.333 | 0.115 | 1 |\n",
       "| HAUS2 | 0.3283637 | -0.2509677 | 0.000 | 0.208 | 1 |\n",
       "| FAM136A | 0.8431908 | -0.2509642 | 0.333 | 0.500 | 1 |\n",
       "| BAK1 | 0.2155462 | -0.2509196 | 0.000 | 0.292 | 1 |\n",
       "| E2F1 | 0.3338776 |  0.2508956 | 0.333 | 0.054 | 1 |\n",
       "| CCDC85C | 0.2422634 | -0.2508947 | 0.000 | 0.269 | 1 |\n",
       "| PSAT1 | 0.5969442 |  0.2508419 | 0.333 | 0.108 | 1 |\n",
       "| KDELC2 | 0.6733051 |  0.2508333 | 0.333 | 0.169 | 1 |\n",
       "| KAT6B | 0.3045630 | -0.2507993 | 0.000 | 0.223 | 1 |\n",
       "| VKORC1L1 | 0.2155462 | -0.2507013 | 0.000 | 0.292 | 1 |\n",
       "| CD58 | 0.6240221 |  0.2506750 | 0.333 | 0.123 | 1 |\n",
       "| ME3 | 0.3409089 | -0.2506518 | 0.000 | 0.200 | 1 |\n",
       "| RANBP6 | 0.2616326 | -0.2506493 | 0.000 | 0.254 | 1 |\n",
       "| TAPBPL | 0.6733858 |  0.2506337 | 0.333 | 0.154 | 1 |\n",
       "| MSMP | 0.3813766 | -0.2505910 | 0.000 | 0.177 | 1 |\n",
       "| SLC15A4 | 0.6375639 |  0.2505787 | 0.333 | 0.131 | 1 |\n",
       "| NUS1 | 0.2517845 | -0.2505175 | 0.000 | 0.262 | 1 |\n",
       "| TERF2 | 0.2330574 | -0.2504391 | 0.000 | 0.277 | 1 |\n",
       "| ARL15 | 0.3045630 | -0.2503660 | 0.000 | 0.223 | 1 |\n",
       "| IPO7 | 0.8205137 | -0.2503627 | 0.333 | 0.515 | 1 |\n",
       "| WDR6 | 0.5869650 |  0.2503068 | 0.333 | 0.346 | 1 |\n",
       "| MUL1 | 0.2330574 | -0.2502621 | 0.000 | 0.277 | 1 |\n",
       "| OPA3 | 0.2932709 | -0.2501218 | 0.000 | 0.231 | 1 |\n",
       "| MTMR2 | 0.4871587 |  0.2501110 | 0.333 | 0.385 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "         p_val        avg_log2FC pct.1 pct.2 p_val_adj\n",
       "PTN      0.0135631530  3.481791  0.333 0.169 1        \n",
       "UBE2C    0.0037808086  3.038616  0.667 0.108 1        \n",
       "COL3A1   0.0002788539  2.802564  0.333 0.431 1        \n",
       "PTTG1    0.0157257432  2.527251  0.667 0.138 1        \n",
       "ARL6IP1  0.0008037031  2.104016  0.667 0.654 1        \n",
       "SFRP2    0.0825681795  2.083583  0.333 0.092 1        \n",
       "RPS4Y1   0.0341172639 -2.015701  0.000 0.592 1        \n",
       "MGP      0.1816760752  2.002652  0.333 0.115 1        \n",
       "CDK1     0.0265915497  1.967298  0.667 0.123 1        \n",
       "HMGB2    0.0073991748  1.916832  1.000 0.400 1        \n",
       "LDHA     0.0558685935 -1.875169  0.333 0.900 1        \n",
       "UBE2S    0.0000481710  1.874852  0.667 0.400 1        \n",
       "EIF4EBP1 0.0205379982 -1.773034  0.333 0.923 1        \n",
       "BCAP31   0.0027296316 -1.759390  0.000 0.823 1        \n",
       "ACKR3    0.0072208498 -1.758801  0.000 0.754 1        \n",
       "PCLAF    0.0132228222  1.744362  1.000 0.215 1        \n",
       "HMGN2    0.0084806258  1.742515  1.000 0.808 1        \n",
       "MKI67    0.0052457153  1.730225  0.667 0.108 1        \n",
       "PRRX1    0.0819901800  1.725540  0.333 0.077 1        \n",
       "PRSS2    0.1368676593 -1.720786  1.000 0.746 1        \n",
       "SMAD1    0.0024005704 -1.701940  0.000 0.831 1        \n",
       "CDC20    0.0135808127  1.701473  0.667 0.038 1        \n",
       "CKS2     0.0054080547  1.697888  0.667 0.438 1        \n",
       "P4HB     0.0210497524 -1.686653  0.333 0.915 1        \n",
       "CENPN    0.0031935513  1.686444  0.667 0.154 1        \n",
       "CCNA2    0.0216060250  1.665068  0.667 0.069 1        \n",
       "CRABP1   0.0029080452  1.655757  0.333 0.292 1        \n",
       "ESM1     0.2845798816 -1.628091  0.333 0.738 1        \n",
       "CENPW    0.0078197942  1.628023  0.667 0.277 1        \n",
       "C2orf69  0.0048087666  1.612651  0.667 0.146 1        \n",
       "⋮        ⋮            ⋮          ⋮     ⋮     ⋮        \n",
       "SARAF    0.8394885    -0.2511795 0.667 0.815 1        \n",
       "IDS      0.7640309     0.2511068 0.667 0.477 1        \n",
       "ARRB2    0.5680651     0.2511033 0.333 0.100 1        \n",
       "RIOK1    0.3045630    -0.2510549 0.000 0.223 1        \n",
       "SRD5A3   0.9271491    -0.2510266 0.333 0.415 1        \n",
       "ELMOD2   0.2932709    -0.2510231 0.000 0.231 1        \n",
       "ARMC7    0.6083595     0.2510068 0.333 0.115 1        \n",
       "HAUS2    0.3283637    -0.2509677 0.000 0.208 1        \n",
       "FAM136A  0.8431908    -0.2509642 0.333 0.500 1        \n",
       "BAK1     0.2155462    -0.2509196 0.000 0.292 1        \n",
       "E2F1     0.3338776     0.2508956 0.333 0.054 1        \n",
       "CCDC85C  0.2422634    -0.2508947 0.000 0.269 1        \n",
       "PSAT1    0.5969442     0.2508419 0.333 0.108 1        \n",
       "KDELC2   0.6733051     0.2508333 0.333 0.169 1        \n",
       "KAT6B    0.3045630    -0.2507993 0.000 0.223 1        \n",
       "VKORC1L1 0.2155462    -0.2507013 0.000 0.292 1        \n",
       "CD58     0.6240221     0.2506750 0.333 0.123 1        \n",
       "ME3      0.3409089    -0.2506518 0.000 0.200 1        \n",
       "RANBP6   0.2616326    -0.2506493 0.000 0.254 1        \n",
       "TAPBPL   0.6733858     0.2506337 0.333 0.154 1        \n",
       "MSMP     0.3813766    -0.2505910 0.000 0.177 1        \n",
       "SLC15A4  0.6375639     0.2505787 0.333 0.131 1        \n",
       "NUS1     0.2517845    -0.2505175 0.000 0.262 1        \n",
       "TERF2    0.2330574    -0.2504391 0.000 0.277 1        \n",
       "ARL15    0.3045630    -0.2503660 0.000 0.223 1        \n",
       "IPO7     0.8205137    -0.2503627 0.333 0.515 1        \n",
       "WDR6     0.5869650     0.2503068 0.333 0.346 1        \n",
       "MUL1     0.2330574    -0.2502621 0.000 0.277 1        \n",
       "OPA3     0.2932709    -0.2501218 0.000 0.231 1        \n",
       "MTMR2    0.4871587     0.2501110 0.333 0.385 1        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "de_results[order(abs(de_results[, \"avg_log2FC\"]), decreasing = TRUE), ]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "flying-drain",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T18:08:36.522881Z",
     "iopub.status.busy": "2023-01-31T18:08:36.521494Z",
     "iopub.status.idle": "2023-01-31T18:08:36.548651Z",
     "shell.execute_reply": "2023-01-31T18:08:36.547233Z"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Extract the differentially expressed genes\n",
    "sig_genes <- rownames(de_results[de_results$p_val < 0.05, ])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "considered-angle",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T18:23:42.615386Z",
     "iopub.status.busy": "2023-01-31T18:23:42.613988Z",
     "iopub.status.idle": "2023-01-31T18:23:42.633342Z",
     "shell.execute_reply": "2023-01-31T18:23:42.631993Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'UBE2S'</li><li>'KIF18B'</li><li>'COL3A1'</li><li>'SCUBE2'</li><li>'UBE2H'</li><li>'ARL6IP1'</li><li>'ZNF706'</li><li>'STX3'</li><li>'CHID1'</li><li>'DTYMK'</li><li>'PSMB6'</li><li>'SMAD1'</li><li>'GABPB1'</li><li>'PPP4R2'</li><li>'BCAP31'</li><li>'CRABP1'</li><li>'COL1A2'</li><li>'CENPN'</li><li>'WRB'</li><li>'UBE2C'</li><li>'RALB'</li><li>'CDCA3'</li><li>'C2orf69'</li><li>'TEN1'</li><li>'MKI67'</li><li>'CKS2'</li><li>'ERGIC3'</li><li>'HIGD2A'</li><li>'SERPINB6'</li><li>'ACKR3'</li><li>'PPP1CB'</li><li>'KIF2C'</li><li>'HMGB2'</li><li>'ZSCAN21'</li><li>'IL17RC'</li><li>'GLCCI1'</li><li>'VPS37D'</li><li>'YIF1B'</li><li>'CENPW'</li><li>'PFDN4'</li><li>'DBF4'</li><li>'CUL3'</li><li>'CDCA8'</li><li>'HMGN2'</li><li>'PXMP4'</li><li>'NONO'</li><li>'PNRC1'</li><li>'STAT1'</li><li>'TROAP'</li><li>'BTF3'</li><li>'GTSE1'</li><li>'NDC80'</li><li>'MED29'</li><li>'MIGA1'</li><li>'OSBPL9'</li><li>'ZNF558'</li><li>'DDX3X'</li><li>'SBDS'</li><li>'SLC4A7'</li><li>'SYNCRIP'</li><li>'C11orf74'</li><li>'PRDX1'</li><li>'CD163L1'</li><li>'KIF23'</li><li>'SPPL3'</li><li>'CCDC167'</li><li>'MAP4'</li><li>'RAPGEF5'</li><li>'PPP2R2D'</li><li>'TYMS'</li><li>'KNL1'</li><li>'RASGRP3'</li><li>'PCLAF'</li><li>'PTN'</li><li>'CDC20'</li><li>'TAZ'</li><li>'AC010240.3'</li><li>'AC019171.1'</li><li>'AC133550.2'</li><li>'AC146944.4'</li><li>'AC245123.1'</li><li>'AF111167.2'</li><li>'AL096865.1'</li><li>'AL121821.1'</li><li>'ANKRD29'</li><li>'AP000866.1'</li><li>'C1QTNF7'</li><li>'CAPN6'</li><li>'CCDC136'</li><li>'CCDC177'</li><li>'CELF4'</li><li>'CXorf67'</li><li>'DTX3'</li><li>'DYTN'</li><li>'ENPP1'</li><li>'EYA4'</li><li>'HMGCLL1'</li><li>'LINC00626'</li><li>'LINC01173'</li><li>'LTC4S'</li><li>'MRGPRF'</li><li>'NPHS1'</li><li>'PAPPA2'</li><li>'PLSCR4'</li><li>'PNMA6A'</li><li>'PRR19'</li><li>'SRCIN1'</li><li>'STAM-AS1'</li><li>'SUSD4'</li><li>'SYT14'</li><li>'TMEM59L'</li><li>'VCAM1'</li><li>'ZFP3'</li><li>'ZIC4'</li><li>'AC008496.2'</li><li>'AC068790.9'</li><li>'AC245052.4'</li><li>'AL121718.1'</li><li>'AL359715.2'</li><li>'AP001363.2'</li><li>'HIST1H3B'</li><li>'TRPM1'</li><li>'Z99774.1'</li><li>'AC090627.1'</li><li>'GALNT9'</li><li>'IL10RB-DT'</li><li>'KCNE5'</li><li>'MIAT'</li><li>'NDUFA13'</li><li>'FOXM1'</li><li>'TNRC6B'</li><li>'PFDN1'</li><li>'RASSF2'</li><li>'RBM33'</li><li>'KCNQ1OT1'</li><li>'PTTG1'</li><li>'SKA3'</li><li>'PRC1'</li><li>'ASXL1'</li><li>'AKAP13'</li><li>'LIPA'</li><li>'TECR'</li><li>'NCAPG'</li><li>'TFPT'</li><li>'AMD1'</li><li>'QPCT'</li><li>'DIAPH3'</li><li>'EIF4A2'</li><li>'ISG15'</li><li>'NAMPT'</li><li>'CCDC25'</li><li>'NAT14'</li><li>'RPA1'</li><li>'CNTROB'</li><li>'DLEU2'</li><li>'KDM4A'</li><li>'OSER1'</li><li>'ANLN'</li><li>'BGN'</li><li>'ARHGEF28'</li><li>'EIF4EBP1'</li><li>'SAE1'</li><li>'SNRPE'</li><li>'P4HB'</li><li>'POMT2'</li><li>'APLP2'</li><li>'CCNA2'</li><li>'TMEM60'</li><li>'ADIPOR1'</li><li>'LSM8'</li><li>'PSMB9'</li><li>'TNFRSF1A'</li><li>'HNRNPR'</li><li>'ILK'</li><li>'SPHK1'</li><li>'CMPK1'</li><li>'PFKL'</li><li>'RNPS1'</li><li>'CCNB1'</li><li>'PRNP'</li><li>'EIF2S2'</li><li>'PPM1F'</li><li>'KIAA1143'</li><li>'KIF14'</li><li>'SOGA1'</li><li>'MAPKAPK5'</li><li>'DDX19B'</li><li>'LAMB2'</li><li>'TUBB'</li><li>'CENPF'</li><li>'RHEB'</li><li>'C12orf45'</li><li>'RABGAP1'</li><li>'CDK1'</li><li>'FNBP1L'</li><li>'LYRM4'</li><li>'U2SURP'</li><li>'PLK1'</li><li>'TOP2A'</li><li>'FBRS'</li><li>⋯</li><li>'GPR161'</li><li>'AURKB'</li><li>'FREM1'</li><li>'FAM50A'</li><li>'STARD4'</li><li>'AP001505.1'</li><li>'ARHGEF4'</li><li>'B3GALT1'</li><li>'BNC2'</li><li>'CD74'</li><li>'CPED1'</li><li>'GRID2'</li><li>'RND2'</li><li>'THRB'</li><li>'TNMD'</li><li>'AIF1'</li><li>'AL163051.1'</li><li>'AL354836.1'</li><li>'AL590560.2'</li><li>'ASXL3'</li><li>'BARX1'</li><li>'BMP5'</li><li>'C19orf38'</li><li>'CDON'</li><li>'EPHA3'</li><li>'EPS8'</li><li>'FBXO2'</li><li>'FGF13'</li><li>'FSTL5'</li><li>'GAS7'</li><li>'LRRC4B'</li><li>'NHS'</li><li>'NKD1'</li><li>'PALM2'</li><li>'PHACTR3'</li><li>'PLPPR3'</li><li>'SAMD11'</li><li>'SIX1'</li><li>'SULF1'</li><li>'TCEAL5'</li><li>'ZNF700'</li><li>'AC018695.2'</li><li>'AC022306.2'</li><li>'BDNF-AS'</li><li>'BICC1'</li><li>'CDH23'</li><li>'COL13A1'</li><li>'GDF10'</li><li>'GXYLT2'</li><li>'JAKMIP2'</li><li>'KIF18A'</li><li>'NRK'</li><li>'STXBP2'</li><li>'Z97832.2'</li><li>'HCG17'</li><li>'AC110285.2'</li><li>'AL731661.1'</li><li>'ARHGEF39'</li><li>'C17orf53'</li><li>'C3orf67'</li><li>'C8orf88'</li><li>'CD274'</li><li>'E2F8'</li><li>'ERC2'</li><li>'HLA-DMA'</li><li>'HSD17B2'</li><li>'MIRLET7BHG'</li><li>'PODXL2'</li><li>'ZNF501'</li><li>'ZNF726'</li><li>'ZSWIM3'</li><li>'ZSWIM6'</li><li>'ATF6B'</li><li>'GMDS'</li><li>'LOX'</li><li>'PJA2'</li><li>'SRSF4'</li><li>'THRAP3'</li><li>'UBE2A'</li><li>'CELF1'</li><li>'CGNL1'</li><li>'S100A13'</li><li>'MELK'</li><li>'C6orf89'</li><li>'STAG3'</li><li>'WSB2'</li><li>'RC3H2'</li><li>'C12orf10'</li><li>'JUNB'</li><li>'PDHB'</li><li>'RPS4Y1'</li><li>'ARL4C'</li><li>'F2R'</li><li>'PRPF38B'</li><li>'DNM2'</li><li>'MTRNR2L8'</li><li>'SRBD1'</li><li>'PDZD11'</li><li>'MTRNR2L10'</li><li>'TPST2'</li><li>'PARG'</li><li>'FANCI'</li><li>'EIF2AK2'</li><li>'ERLEC1'</li><li>'PPFIBP1'</li><li>'AC010642.2'</li><li>'CUEDC2'</li><li>'HIGD1A'</li><li>'MAF'</li><li>'RNF187'</li><li>'COL1A1'</li><li>'GGNBP2'</li><li>'CCNB2'</li><li>'VAPA'</li><li>'CXCL12'</li><li>'LIMD2'</li><li>'UGGT2'</li><li>'EIF2S3'</li><li>'KANK3'</li><li>'PCSK6'</li><li>'PSMD6'</li><li>'RAPGEF2'</li><li>'SH3GL1'</li><li>'VMP1'</li><li>'ZFYVE21'</li><li>'RPS26'</li><li>'ENO1'</li><li>'LINC01704'</li><li>'TSTD2'</li><li>'SNRPC'</li><li>'AGAP1'</li><li>'ARCN1'</li><li>'CDK17'</li><li>'GIPC1'</li><li>'MRPS6'</li><li>'SMIM10L1'</li><li>'DDRGK1'</li><li>'HEXA'</li><li>'TIAL1'</li><li>'H2AFY'</li><li>'BOD1'</li><li>'PDXK'</li><li>'STX7'</li><li>'TMEM14A'</li><li>'YIPF5'</li><li>'ZCCHC17'</li><li>'KRT8'</li><li>'AMZ2'</li><li>'IFITM2'</li><li>'SKP2'</li><li>'CKAP2'</li><li>'CCDC34'</li><li>'SLC25A38'</li><li>'NUP153'</li><li>'DDX46'</li><li>'SSBP3'</li><li>'WDFY1'</li><li>'FAU'</li><li>'ADSS'</li><li>'BCHE'</li><li>'EFNB2'</li><li>'FKBP10'</li><li>'GRPEL1'</li><li>'PPP1R7'</li><li>'TTC37'</li><li>'GNA13'</li><li>'LBR'</li><li>'PCID2'</li><li>'ADAR'</li><li>'CAPRIN1'</li><li>'SGO2'</li><li>'TOP2B'</li><li>'MED22'</li><li>'EIF4H'</li><li>'SPCS1'</li><li>'BUB3'</li><li>'PRRC1'</li><li>'RRM2'</li><li>'CD2BP2'</li><li>'IGFBP7'</li><li>'PLCG1'</li><li>'ABCD4'</li><li>'CASP4'</li><li>'CDC42BPA'</li><li>'DNAJB14'</li><li>'HNRNPH2'</li><li>'MRPS12'</li><li>'MRPS5'</li><li>'NANS'</li><li>'NEMF'</li><li>'P4HA1'</li><li>'MGAT4B'</li><li>'SCD'</li><li>'CNTRL'</li><li>'MGAT2'</li><li>'TSPAN14'</li><li>'HUWE1'</li><li>'HYI'</li><li>'HMGB3'</li><li>'CKLF'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'UBE2S'\n",
       "\\item 'KIF18B'\n",
       "\\item 'COL3A1'\n",
       "\\item 'SCUBE2'\n",
       "\\item 'UBE2H'\n",
       "\\item 'ARL6IP1'\n",
       "\\item 'ZNF706'\n",
       "\\item 'STX3'\n",
       "\\item 'CHID1'\n",
       "\\item 'DTYMK'\n",
       "\\item 'PSMB6'\n",
       "\\item 'SMAD1'\n",
       "\\item 'GABPB1'\n",
       "\\item 'PPP4R2'\n",
       "\\item 'BCAP31'\n",
       "\\item 'CRABP1'\n",
       "\\item 'COL1A2'\n",
       "\\item 'CENPN'\n",
       "\\item 'WRB'\n",
       "\\item 'UBE2C'\n",
       "\\item 'RALB'\n",
       "\\item 'CDCA3'\n",
       "\\item 'C2orf69'\n",
       "\\item 'TEN1'\n",
       "\\item 'MKI67'\n",
       "\\item 'CKS2'\n",
       "\\item 'ERGIC3'\n",
       "\\item 'HIGD2A'\n",
       "\\item 'SERPINB6'\n",
       "\\item 'ACKR3'\n",
       "\\item 'PPP1CB'\n",
       "\\item 'KIF2C'\n",
       "\\item 'HMGB2'\n",
       "\\item 'ZSCAN21'\n",
       "\\item 'IL17RC'\n",
       "\\item 'GLCCI1'\n",
       "\\item 'VPS37D'\n",
       "\\item 'YIF1B'\n",
       "\\item 'CENPW'\n",
       "\\item 'PFDN4'\n",
       "\\item 'DBF4'\n",
       "\\item 'CUL3'\n",
       "\\item 'CDCA8'\n",
       "\\item 'HMGN2'\n",
       "\\item 'PXMP4'\n",
       "\\item 'NONO'\n",
       "\\item 'PNRC1'\n",
       "\\item 'STAT1'\n",
       "\\item 'TROAP'\n",
       "\\item 'BTF3'\n",
       "\\item 'GTSE1'\n",
       "\\item 'NDC80'\n",
       "\\item 'MED29'\n",
       "\\item 'MIGA1'\n",
       "\\item 'OSBPL9'\n",
       "\\item 'ZNF558'\n",
       "\\item 'DDX3X'\n",
       "\\item 'SBDS'\n",
       "\\item 'SLC4A7'\n",
       "\\item 'SYNCRIP'\n",
       "\\item 'C11orf74'\n",
       "\\item 'PRDX1'\n",
       "\\item 'CD163L1'\n",
       "\\item 'KIF23'\n",
       "\\item 'SPPL3'\n",
       "\\item 'CCDC167'\n",
       "\\item 'MAP4'\n",
       "\\item 'RAPGEF5'\n",
       "\\item 'PPP2R2D'\n",
       "\\item 'TYMS'\n",
       "\\item 'KNL1'\n",
       "\\item 'RASGRP3'\n",
       "\\item 'PCLAF'\n",
       "\\item 'PTN'\n",
       "\\item 'CDC20'\n",
       "\\item 'TAZ'\n",
       "\\item 'AC010240.3'\n",
       "\\item 'AC019171.1'\n",
       "\\item 'AC133550.2'\n",
       "\\item 'AC146944.4'\n",
       "\\item 'AC245123.1'\n",
       "\\item 'AF111167.2'\n",
       "\\item 'AL096865.1'\n",
       "\\item 'AL121821.1'\n",
       "\\item 'ANKRD29'\n",
       "\\item 'AP000866.1'\n",
       "\\item 'C1QTNF7'\n",
       "\\item 'CAPN6'\n",
       "\\item 'CCDC136'\n",
       "\\item 'CCDC177'\n",
       "\\item 'CELF4'\n",
       "\\item 'CXorf67'\n",
       "\\item 'DTX3'\n",
       "\\item 'DYTN'\n",
       "\\item 'ENPP1'\n",
       "\\item 'EYA4'\n",
       "\\item 'HMGCLL1'\n",
       "\\item 'LINC00626'\n",
       "\\item 'LINC01173'\n",
       "\\item 'LTC4S'\n",
       "\\item 'MRGPRF'\n",
       "\\item 'NPHS1'\n",
       "\\item 'PAPPA2'\n",
       "\\item 'PLSCR4'\n",
       "\\item 'PNMA6A'\n",
       "\\item 'PRR19'\n",
       "\\item 'SRCIN1'\n",
       "\\item 'STAM-AS1'\n",
       "\\item 'SUSD4'\n",
       "\\item 'SYT14'\n",
       "\\item 'TMEM59L'\n",
       "\\item 'VCAM1'\n",
       "\\item 'ZFP3'\n",
       "\\item 'ZIC4'\n",
       "\\item 'AC008496.2'\n",
       "\\item 'AC068790.9'\n",
       "\\item 'AC245052.4'\n",
       "\\item 'AL121718.1'\n",
       "\\item 'AL359715.2'\n",
       "\\item 'AP001363.2'\n",
       "\\item 'HIST1H3B'\n",
       "\\item 'TRPM1'\n",
       "\\item 'Z99774.1'\n",
       "\\item 'AC090627.1'\n",
       "\\item 'GALNT9'\n",
       "\\item 'IL10RB-DT'\n",
       "\\item 'KCNE5'\n",
       "\\item 'MIAT'\n",
       "\\item 'NDUFA13'\n",
       "\\item 'FOXM1'\n",
       "\\item 'TNRC6B'\n",
       "\\item 'PFDN1'\n",
       "\\item 'RASSF2'\n",
       "\\item 'RBM33'\n",
       "\\item 'KCNQ1OT1'\n",
       "\\item 'PTTG1'\n",
       "\\item 'SKA3'\n",
       "\\item 'PRC1'\n",
       "\\item 'ASXL1'\n",
       "\\item 'AKAP13'\n",
       "\\item 'LIPA'\n",
       "\\item 'TECR'\n",
       "\\item 'NCAPG'\n",
       "\\item 'TFPT'\n",
       "\\item 'AMD1'\n",
       "\\item 'QPCT'\n",
       "\\item 'DIAPH3'\n",
       "\\item 'EIF4A2'\n",
       "\\item 'ISG15'\n",
       "\\item 'NAMPT'\n",
       "\\item 'CCDC25'\n",
       "\\item 'NAT14'\n",
       "\\item 'RPA1'\n",
       "\\item 'CNTROB'\n",
       "\\item 'DLEU2'\n",
       "\\item 'KDM4A'\n",
       "\\item 'OSER1'\n",
       "\\item 'ANLN'\n",
       "\\item 'BGN'\n",
       "\\item 'ARHGEF28'\n",
       "\\item 'EIF4EBP1'\n",
       "\\item 'SAE1'\n",
       "\\item 'SNRPE'\n",
       "\\item 'P4HB'\n",
       "\\item 'POMT2'\n",
       "\\item 'APLP2'\n",
       "\\item 'CCNA2'\n",
       "\\item 'TMEM60'\n",
       "\\item 'ADIPOR1'\n",
       "\\item 'LSM8'\n",
       "\\item 'PSMB9'\n",
       "\\item 'TNFRSF1A'\n",
       "\\item 'HNRNPR'\n",
       "\\item 'ILK'\n",
       "\\item 'SPHK1'\n",
       "\\item 'CMPK1'\n",
       "\\item 'PFKL'\n",
       "\\item 'RNPS1'\n",
       "\\item 'CCNB1'\n",
       "\\item 'PRNP'\n",
       "\\item 'EIF2S2'\n",
       "\\item 'PPM1F'\n",
       "\\item 'KIAA1143'\n",
       "\\item 'KIF14'\n",
       "\\item 'SOGA1'\n",
       "\\item 'MAPKAPK5'\n",
       "\\item 'DDX19B'\n",
       "\\item 'LAMB2'\n",
       "\\item 'TUBB'\n",
       "\\item 'CENPF'\n",
       "\\item 'RHEB'\n",
       "\\item 'C12orf45'\n",
       "\\item 'RABGAP1'\n",
       "\\item 'CDK1'\n",
       "\\item 'FNBP1L'\n",
       "\\item 'LYRM4'\n",
       "\\item 'U2SURP'\n",
       "\\item 'PLK1'\n",
       "\\item 'TOP2A'\n",
       "\\item 'FBRS'\n",
       "\\item ⋯\n",
       "\\item 'GPR161'\n",
       "\\item 'AURKB'\n",
       "\\item 'FREM1'\n",
       "\\item 'FAM50A'\n",
       "\\item 'STARD4'\n",
       "\\item 'AP001505.1'\n",
       "\\item 'ARHGEF4'\n",
       "\\item 'B3GALT1'\n",
       "\\item 'BNC2'\n",
       "\\item 'CD74'\n",
       "\\item 'CPED1'\n",
       "\\item 'GRID2'\n",
       "\\item 'RND2'\n",
       "\\item 'THRB'\n",
       "\\item 'TNMD'\n",
       "\\item 'AIF1'\n",
       "\\item 'AL163051.1'\n",
       "\\item 'AL354836.1'\n",
       "\\item 'AL590560.2'\n",
       "\\item 'ASXL3'\n",
       "\\item 'BARX1'\n",
       "\\item 'BMP5'\n",
       "\\item 'C19orf38'\n",
       "\\item 'CDON'\n",
       "\\item 'EPHA3'\n",
       "\\item 'EPS8'\n",
       "\\item 'FBXO2'\n",
       "\\item 'FGF13'\n",
       "\\item 'FSTL5'\n",
       "\\item 'GAS7'\n",
       "\\item 'LRRC4B'\n",
       "\\item 'NHS'\n",
       "\\item 'NKD1'\n",
       "\\item 'PALM2'\n",
       "\\item 'PHACTR3'\n",
       "\\item 'PLPPR3'\n",
       "\\item 'SAMD11'\n",
       "\\item 'SIX1'\n",
       "\\item 'SULF1'\n",
       "\\item 'TCEAL5'\n",
       "\\item 'ZNF700'\n",
       "\\item 'AC018695.2'\n",
       "\\item 'AC022306.2'\n",
       "\\item 'BDNF-AS'\n",
       "\\item 'BICC1'\n",
       "\\item 'CDH23'\n",
       "\\item 'COL13A1'\n",
       "\\item 'GDF10'\n",
       "\\item 'GXYLT2'\n",
       "\\item 'JAKMIP2'\n",
       "\\item 'KIF18A'\n",
       "\\item 'NRK'\n",
       "\\item 'STXBP2'\n",
       "\\item 'Z97832.2'\n",
       "\\item 'HCG17'\n",
       "\\item 'AC110285.2'\n",
       "\\item 'AL731661.1'\n",
       "\\item 'ARHGEF39'\n",
       "\\item 'C17orf53'\n",
       "\\item 'C3orf67'\n",
       "\\item 'C8orf88'\n",
       "\\item 'CD274'\n",
       "\\item 'E2F8'\n",
       "\\item 'ERC2'\n",
       "\\item 'HLA-DMA'\n",
       "\\item 'HSD17B2'\n",
       "\\item 'MIRLET7BHG'\n",
       "\\item 'PODXL2'\n",
       "\\item 'ZNF501'\n",
       "\\item 'ZNF726'\n",
       "\\item 'ZSWIM3'\n",
       "\\item 'ZSWIM6'\n",
       "\\item 'ATF6B'\n",
       "\\item 'GMDS'\n",
       "\\item 'LOX'\n",
       "\\item 'PJA2'\n",
       "\\item 'SRSF4'\n",
       "\\item 'THRAP3'\n",
       "\\item 'UBE2A'\n",
       "\\item 'CELF1'\n",
       "\\item 'CGNL1'\n",
       "\\item 'S100A13'\n",
       "\\item 'MELK'\n",
       "\\item 'C6orf89'\n",
       "\\item 'STAG3'\n",
       "\\item 'WSB2'\n",
       "\\item 'RC3H2'\n",
       "\\item 'C12orf10'\n",
       "\\item 'JUNB'\n",
       "\\item 'PDHB'\n",
       "\\item 'RPS4Y1'\n",
       "\\item 'ARL4C'\n",
       "\\item 'F2R'\n",
       "\\item 'PRPF38B'\n",
       "\\item 'DNM2'\n",
       "\\item 'MTRNR2L8'\n",
       "\\item 'SRBD1'\n",
       "\\item 'PDZD11'\n",
       "\\item 'MTRNR2L10'\n",
       "\\item 'TPST2'\n",
       "\\item 'PARG'\n",
       "\\item 'FANCI'\n",
       "\\item 'EIF2AK2'\n",
       "\\item 'ERLEC1'\n",
       "\\item 'PPFIBP1'\n",
       "\\item 'AC010642.2'\n",
       "\\item 'CUEDC2'\n",
       "\\item 'HIGD1A'\n",
       "\\item 'MAF'\n",
       "\\item 'RNF187'\n",
       "\\item 'COL1A1'\n",
       "\\item 'GGNBP2'\n",
       "\\item 'CCNB2'\n",
       "\\item 'VAPA'\n",
       "\\item 'CXCL12'\n",
       "\\item 'LIMD2'\n",
       "\\item 'UGGT2'\n",
       "\\item 'EIF2S3'\n",
       "\\item 'KANK3'\n",
       "\\item 'PCSK6'\n",
       "\\item 'PSMD6'\n",
       "\\item 'RAPGEF2'\n",
       "\\item 'SH3GL1'\n",
       "\\item 'VMP1'\n",
       "\\item 'ZFYVE21'\n",
       "\\item 'RPS26'\n",
       "\\item 'ENO1'\n",
       "\\item 'LINC01704'\n",
       "\\item 'TSTD2'\n",
       "\\item 'SNRPC'\n",
       "\\item 'AGAP1'\n",
       "\\item 'ARCN1'\n",
       "\\item 'CDK17'\n",
       "\\item 'GIPC1'\n",
       "\\item 'MRPS6'\n",
       "\\item 'SMIM10L1'\n",
       "\\item 'DDRGK1'\n",
       "\\item 'HEXA'\n",
       "\\item 'TIAL1'\n",
       "\\item 'H2AFY'\n",
       "\\item 'BOD1'\n",
       "\\item 'PDXK'\n",
       "\\item 'STX7'\n",
       "\\item 'TMEM14A'\n",
       "\\item 'YIPF5'\n",
       "\\item 'ZCCHC17'\n",
       "\\item 'KRT8'\n",
       "\\item 'AMZ2'\n",
       "\\item 'IFITM2'\n",
       "\\item 'SKP2'\n",
       "\\item 'CKAP2'\n",
       "\\item 'CCDC34'\n",
       "\\item 'SLC25A38'\n",
       "\\item 'NUP153'\n",
       "\\item 'DDX46'\n",
       "\\item 'SSBP3'\n",
       "\\item 'WDFY1'\n",
       "\\item 'FAU'\n",
       "\\item 'ADSS'\n",
       "\\item 'BCHE'\n",
       "\\item 'EFNB2'\n",
       "\\item 'FKBP10'\n",
       "\\item 'GRPEL1'\n",
       "\\item 'PPP1R7'\n",
       "\\item 'TTC37'\n",
       "\\item 'GNA13'\n",
       "\\item 'LBR'\n",
       "\\item 'PCID2'\n",
       "\\item 'ADAR'\n",
       "\\item 'CAPRIN1'\n",
       "\\item 'SGO2'\n",
       "\\item 'TOP2B'\n",
       "\\item 'MED22'\n",
       "\\item 'EIF4H'\n",
       "\\item 'SPCS1'\n",
       "\\item 'BUB3'\n",
       "\\item 'PRRC1'\n",
       "\\item 'RRM2'\n",
       "\\item 'CD2BP2'\n",
       "\\item 'IGFBP7'\n",
       "\\item 'PLCG1'\n",
       "\\item 'ABCD4'\n",
       "\\item 'CASP4'\n",
       "\\item 'CDC42BPA'\n",
       "\\item 'DNAJB14'\n",
       "\\item 'HNRNPH2'\n",
       "\\item 'MRPS12'\n",
       "\\item 'MRPS5'\n",
       "\\item 'NANS'\n",
       "\\item 'NEMF'\n",
       "\\item 'P4HA1'\n",
       "\\item 'MGAT4B'\n",
       "\\item 'SCD'\n",
       "\\item 'CNTRL'\n",
       "\\item 'MGAT2'\n",
       "\\item 'TSPAN14'\n",
       "\\item 'HUWE1'\n",
       "\\item 'HYI'\n",
       "\\item 'HMGB3'\n",
       "\\item 'CKLF'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'UBE2S'\n",
       "2. 'KIF18B'\n",
       "3. 'COL3A1'\n",
       "4. 'SCUBE2'\n",
       "5. 'UBE2H'\n",
       "6. 'ARL6IP1'\n",
       "7. 'ZNF706'\n",
       "8. 'STX3'\n",
       "9. 'CHID1'\n",
       "10. 'DTYMK'\n",
       "11. 'PSMB6'\n",
       "12. 'SMAD1'\n",
       "13. 'GABPB1'\n",
       "14. 'PPP4R2'\n",
       "15. 'BCAP31'\n",
       "16. 'CRABP1'\n",
       "17. 'COL1A2'\n",
       "18. 'CENPN'\n",
       "19. 'WRB'\n",
       "20. 'UBE2C'\n",
       "21. 'RALB'\n",
       "22. 'CDCA3'\n",
       "23. 'C2orf69'\n",
       "24. 'TEN1'\n",
       "25. 'MKI67'\n",
       "26. 'CKS2'\n",
       "27. 'ERGIC3'\n",
       "28. 'HIGD2A'\n",
       "29. 'SERPINB6'\n",
       "30. 'ACKR3'\n",
       "31. 'PPP1CB'\n",
       "32. 'KIF2C'\n",
       "33. 'HMGB2'\n",
       "34. 'ZSCAN21'\n",
       "35. 'IL17RC'\n",
       "36. 'GLCCI1'\n",
       "37. 'VPS37D'\n",
       "38. 'YIF1B'\n",
       "39. 'CENPW'\n",
       "40. 'PFDN4'\n",
       "41. 'DBF4'\n",
       "42. 'CUL3'\n",
       "43. 'CDCA8'\n",
       "44. 'HMGN2'\n",
       "45. 'PXMP4'\n",
       "46. 'NONO'\n",
       "47. 'PNRC1'\n",
       "48. 'STAT1'\n",
       "49. 'TROAP'\n",
       "50. 'BTF3'\n",
       "51. 'GTSE1'\n",
       "52. 'NDC80'\n",
       "53. 'MED29'\n",
       "54. 'MIGA1'\n",
       "55. 'OSBPL9'\n",
       "56. 'ZNF558'\n",
       "57. 'DDX3X'\n",
       "58. 'SBDS'\n",
       "59. 'SLC4A7'\n",
       "60. 'SYNCRIP'\n",
       "61. 'C11orf74'\n",
       "62. 'PRDX1'\n",
       "63. 'CD163L1'\n",
       "64. 'KIF23'\n",
       "65. 'SPPL3'\n",
       "66. 'CCDC167'\n",
       "67. 'MAP4'\n",
       "68. 'RAPGEF5'\n",
       "69. 'PPP2R2D'\n",
       "70. 'TYMS'\n",
       "71. 'KNL1'\n",
       "72. 'RASGRP3'\n",
       "73. 'PCLAF'\n",
       "74. 'PTN'\n",
       "75. 'CDC20'\n",
       "76. 'TAZ'\n",
       "77. 'AC010240.3'\n",
       "78. 'AC019171.1'\n",
       "79. 'AC133550.2'\n",
       "80. 'AC146944.4'\n",
       "81. 'AC245123.1'\n",
       "82. 'AF111167.2'\n",
       "83. 'AL096865.1'\n",
       "84. 'AL121821.1'\n",
       "85. 'ANKRD29'\n",
       "86. 'AP000866.1'\n",
       "87. 'C1QTNF7'\n",
       "88. 'CAPN6'\n",
       "89. 'CCDC136'\n",
       "90. 'CCDC177'\n",
       "91. 'CELF4'\n",
       "92. 'CXorf67'\n",
       "93. 'DTX3'\n",
       "94. 'DYTN'\n",
       "95. 'ENPP1'\n",
       "96. 'EYA4'\n",
       "97. 'HMGCLL1'\n",
       "98. 'LINC00626'\n",
       "99. 'LINC01173'\n",
       "100. 'LTC4S'\n",
       "101. 'MRGPRF'\n",
       "102. 'NPHS1'\n",
       "103. 'PAPPA2'\n",
       "104. 'PLSCR4'\n",
       "105. 'PNMA6A'\n",
       "106. 'PRR19'\n",
       "107. 'SRCIN1'\n",
       "108. 'STAM-AS1'\n",
       "109. 'SUSD4'\n",
       "110. 'SYT14'\n",
       "111. 'TMEM59L'\n",
       "112. 'VCAM1'\n",
       "113. 'ZFP3'\n",
       "114. 'ZIC4'\n",
       "115. 'AC008496.2'\n",
       "116. 'AC068790.9'\n",
       "117. 'AC245052.4'\n",
       "118. 'AL121718.1'\n",
       "119. 'AL359715.2'\n",
       "120. 'AP001363.2'\n",
       "121. 'HIST1H3B'\n",
       "122. 'TRPM1'\n",
       "123. 'Z99774.1'\n",
       "124. 'AC090627.1'\n",
       "125. 'GALNT9'\n",
       "126. 'IL10RB-DT'\n",
       "127. 'KCNE5'\n",
       "128. 'MIAT'\n",
       "129. 'NDUFA13'\n",
       "130. 'FOXM1'\n",
       "131. 'TNRC6B'\n",
       "132. 'PFDN1'\n",
       "133. 'RASSF2'\n",
       "134. 'RBM33'\n",
       "135. 'KCNQ1OT1'\n",
       "136. 'PTTG1'\n",
       "137. 'SKA3'\n",
       "138. 'PRC1'\n",
       "139. 'ASXL1'\n",
       "140. 'AKAP13'\n",
       "141. 'LIPA'\n",
       "142. 'TECR'\n",
       "143. 'NCAPG'\n",
       "144. 'TFPT'\n",
       "145. 'AMD1'\n",
       "146. 'QPCT'\n",
       "147. 'DIAPH3'\n",
       "148. 'EIF4A2'\n",
       "149. 'ISG15'\n",
       "150. 'NAMPT'\n",
       "151. 'CCDC25'\n",
       "152. 'NAT14'\n",
       "153. 'RPA1'\n",
       "154. 'CNTROB'\n",
       "155. 'DLEU2'\n",
       "156. 'KDM4A'\n",
       "157. 'OSER1'\n",
       "158. 'ANLN'\n",
       "159. 'BGN'\n",
       "160. 'ARHGEF28'\n",
       "161. 'EIF4EBP1'\n",
       "162. 'SAE1'\n",
       "163. 'SNRPE'\n",
       "164. 'P4HB'\n",
       "165. 'POMT2'\n",
       "166. 'APLP2'\n",
       "167. 'CCNA2'\n",
       "168. 'TMEM60'\n",
       "169. 'ADIPOR1'\n",
       "170. 'LSM8'\n",
       "171. 'PSMB9'\n",
       "172. 'TNFRSF1A'\n",
       "173. 'HNRNPR'\n",
       "174. 'ILK'\n",
       "175. 'SPHK1'\n",
       "176. 'CMPK1'\n",
       "177. 'PFKL'\n",
       "178. 'RNPS1'\n",
       "179. 'CCNB1'\n",
       "180. 'PRNP'\n",
       "181. 'EIF2S2'\n",
       "182. 'PPM1F'\n",
       "183. 'KIAA1143'\n",
       "184. 'KIF14'\n",
       "185. 'SOGA1'\n",
       "186. 'MAPKAPK5'\n",
       "187. 'DDX19B'\n",
       "188. 'LAMB2'\n",
       "189. 'TUBB'\n",
       "190. 'CENPF'\n",
       "191. 'RHEB'\n",
       "192. 'C12orf45'\n",
       "193. 'RABGAP1'\n",
       "194. 'CDK1'\n",
       "195. 'FNBP1L'\n",
       "196. 'LYRM4'\n",
       "197. 'U2SURP'\n",
       "198. 'PLK1'\n",
       "199. 'TOP2A'\n",
       "200. 'FBRS'\n",
       "201. ⋯\n",
       "202. 'GPR161'\n",
       "203. 'AURKB'\n",
       "204. 'FREM1'\n",
       "205. 'FAM50A'\n",
       "206. 'STARD4'\n",
       "207. 'AP001505.1'\n",
       "208. 'ARHGEF4'\n",
       "209. 'B3GALT1'\n",
       "210. 'BNC2'\n",
       "211. 'CD74'\n",
       "212. 'CPED1'\n",
       "213. 'GRID2'\n",
       "214. 'RND2'\n",
       "215. 'THRB'\n",
       "216. 'TNMD'\n",
       "217. 'AIF1'\n",
       "218. 'AL163051.1'\n",
       "219. 'AL354836.1'\n",
       "220. 'AL590560.2'\n",
       "221. 'ASXL3'\n",
       "222. 'BARX1'\n",
       "223. 'BMP5'\n",
       "224. 'C19orf38'\n",
       "225. 'CDON'\n",
       "226. 'EPHA3'\n",
       "227. 'EPS8'\n",
       "228. 'FBXO2'\n",
       "229. 'FGF13'\n",
       "230. 'FSTL5'\n",
       "231. 'GAS7'\n",
       "232. 'LRRC4B'\n",
       "233. 'NHS'\n",
       "234. 'NKD1'\n",
       "235. 'PALM2'\n",
       "236. 'PHACTR3'\n",
       "237. 'PLPPR3'\n",
       "238. 'SAMD11'\n",
       "239. 'SIX1'\n",
       "240. 'SULF1'\n",
       "241. 'TCEAL5'\n",
       "242. 'ZNF700'\n",
       "243. 'AC018695.2'\n",
       "244. 'AC022306.2'\n",
       "245. 'BDNF-AS'\n",
       "246. 'BICC1'\n",
       "247. 'CDH23'\n",
       "248. 'COL13A1'\n",
       "249. 'GDF10'\n",
       "250. 'GXYLT2'\n",
       "251. 'JAKMIP2'\n",
       "252. 'KIF18A'\n",
       "253. 'NRK'\n",
       "254. 'STXBP2'\n",
       "255. 'Z97832.2'\n",
       "256. 'HCG17'\n",
       "257. 'AC110285.2'\n",
       "258. 'AL731661.1'\n",
       "259. 'ARHGEF39'\n",
       "260. 'C17orf53'\n",
       "261. 'C3orf67'\n",
       "262. 'C8orf88'\n",
       "263. 'CD274'\n",
       "264. 'E2F8'\n",
       "265. 'ERC2'\n",
       "266. 'HLA-DMA'\n",
       "267. 'HSD17B2'\n",
       "268. 'MIRLET7BHG'\n",
       "269. 'PODXL2'\n",
       "270. 'ZNF501'\n",
       "271. 'ZNF726'\n",
       "272. 'ZSWIM3'\n",
       "273. 'ZSWIM6'\n",
       "274. 'ATF6B'\n",
       "275. 'GMDS'\n",
       "276. 'LOX'\n",
       "277. 'PJA2'\n",
       "278. 'SRSF4'\n",
       "279. 'THRAP3'\n",
       "280. 'UBE2A'\n",
       "281. 'CELF1'\n",
       "282. 'CGNL1'\n",
       "283. 'S100A13'\n",
       "284. 'MELK'\n",
       "285. 'C6orf89'\n",
       "286. 'STAG3'\n",
       "287. 'WSB2'\n",
       "288. 'RC3H2'\n",
       "289. 'C12orf10'\n",
       "290. 'JUNB'\n",
       "291. 'PDHB'\n",
       "292. 'RPS4Y1'\n",
       "293. 'ARL4C'\n",
       "294. 'F2R'\n",
       "295. 'PRPF38B'\n",
       "296. 'DNM2'\n",
       "297. 'MTRNR2L8'\n",
       "298. 'SRBD1'\n",
       "299. 'PDZD11'\n",
       "300. 'MTRNR2L10'\n",
       "301. 'TPST2'\n",
       "302. 'PARG'\n",
       "303. 'FANCI'\n",
       "304. 'EIF2AK2'\n",
       "305. 'ERLEC1'\n",
       "306. 'PPFIBP1'\n",
       "307. 'AC010642.2'\n",
       "308. 'CUEDC2'\n",
       "309. 'HIGD1A'\n",
       "310. 'MAF'\n",
       "311. 'RNF187'\n",
       "312. 'COL1A1'\n",
       "313. 'GGNBP2'\n",
       "314. 'CCNB2'\n",
       "315. 'VAPA'\n",
       "316. 'CXCL12'\n",
       "317. 'LIMD2'\n",
       "318. 'UGGT2'\n",
       "319. 'EIF2S3'\n",
       "320. 'KANK3'\n",
       "321. 'PCSK6'\n",
       "322. 'PSMD6'\n",
       "323. 'RAPGEF2'\n",
       "324. 'SH3GL1'\n",
       "325. 'VMP1'\n",
       "326. 'ZFYVE21'\n",
       "327. 'RPS26'\n",
       "328. 'ENO1'\n",
       "329. 'LINC01704'\n",
       "330. 'TSTD2'\n",
       "331. 'SNRPC'\n",
       "332. 'AGAP1'\n",
       "333. 'ARCN1'\n",
       "334. 'CDK17'\n",
       "335. 'GIPC1'\n",
       "336. 'MRPS6'\n",
       "337. 'SMIM10L1'\n",
       "338. 'DDRGK1'\n",
       "339. 'HEXA'\n",
       "340. 'TIAL1'\n",
       "341. 'H2AFY'\n",
       "342. 'BOD1'\n",
       "343. 'PDXK'\n",
       "344. 'STX7'\n",
       "345. 'TMEM14A'\n",
       "346. 'YIPF5'\n",
       "347. 'ZCCHC17'\n",
       "348. 'KRT8'\n",
       "349. 'AMZ2'\n",
       "350. 'IFITM2'\n",
       "351. 'SKP2'\n",
       "352. 'CKAP2'\n",
       "353. 'CCDC34'\n",
       "354. 'SLC25A38'\n",
       "355. 'NUP153'\n",
       "356. 'DDX46'\n",
       "357. 'SSBP3'\n",
       "358. 'WDFY1'\n",
       "359. 'FAU'\n",
       "360. 'ADSS'\n",
       "361. 'BCHE'\n",
       "362. 'EFNB2'\n",
       "363. 'FKBP10'\n",
       "364. 'GRPEL1'\n",
       "365. 'PPP1R7'\n",
       "366. 'TTC37'\n",
       "367. 'GNA13'\n",
       "368. 'LBR'\n",
       "369. 'PCID2'\n",
       "370. 'ADAR'\n",
       "371. 'CAPRIN1'\n",
       "372. 'SGO2'\n",
       "373. 'TOP2B'\n",
       "374. 'MED22'\n",
       "375. 'EIF4H'\n",
       "376. 'SPCS1'\n",
       "377. 'BUB3'\n",
       "378. 'PRRC1'\n",
       "379. 'RRM2'\n",
       "380. 'CD2BP2'\n",
       "381. 'IGFBP7'\n",
       "382. 'PLCG1'\n",
       "383. 'ABCD4'\n",
       "384. 'CASP4'\n",
       "385. 'CDC42BPA'\n",
       "386. 'DNAJB14'\n",
       "387. 'HNRNPH2'\n",
       "388. 'MRPS12'\n",
       "389. 'MRPS5'\n",
       "390. 'NANS'\n",
       "391. 'NEMF'\n",
       "392. 'P4HA1'\n",
       "393. 'MGAT4B'\n",
       "394. 'SCD'\n",
       "395. 'CNTRL'\n",
       "396. 'MGAT2'\n",
       "397. 'TSPAN14'\n",
       "398. 'HUWE1'\n",
       "399. 'HYI'\n",
       "400. 'HMGB3'\n",
       "401. 'CKLF'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"UBE2S\"      \"KIF18B\"     \"COL3A1\"     \"SCUBE2\"     \"UBE2H\"     \n",
       "  [6] \"ARL6IP1\"    \"ZNF706\"     \"STX3\"       \"CHID1\"      \"DTYMK\"     \n",
       " [11] \"PSMB6\"      \"SMAD1\"      \"GABPB1\"     \"PPP4R2\"     \"BCAP31\"    \n",
       " [16] \"CRABP1\"     \"COL1A2\"     \"CENPN\"      \"WRB\"        \"UBE2C\"     \n",
       " [21] \"RALB\"       \"CDCA3\"      \"C2orf69\"    \"TEN1\"       \"MKI67\"     \n",
       " [26] \"CKS2\"       \"ERGIC3\"     \"HIGD2A\"     \"SERPINB6\"   \"ACKR3\"     \n",
       " [31] \"PPP1CB\"     \"KIF2C\"      \"HMGB2\"      \"ZSCAN21\"    \"IL17RC\"    \n",
       " [36] \"GLCCI1\"     \"VPS37D\"     \"YIF1B\"      \"CENPW\"      \"PFDN4\"     \n",
       " [41] \"DBF4\"       \"CUL3\"       \"CDCA8\"      \"HMGN2\"      \"PXMP4\"     \n",
       " [46] \"NONO\"       \"PNRC1\"      \"STAT1\"      \"TROAP\"      \"BTF3\"      \n",
       " [51] \"GTSE1\"      \"NDC80\"      \"MED29\"      \"MIGA1\"      \"OSBPL9\"    \n",
       " [56] \"ZNF558\"     \"DDX3X\"      \"SBDS\"       \"SLC4A7\"     \"SYNCRIP\"   \n",
       " [61] \"C11orf74\"   \"PRDX1\"      \"CD163L1\"    \"KIF23\"      \"SPPL3\"     \n",
       " [66] \"CCDC167\"    \"MAP4\"       \"RAPGEF5\"    \"PPP2R2D\"    \"TYMS\"      \n",
       " [71] \"KNL1\"       \"RASGRP3\"    \"PCLAF\"      \"PTN\"        \"CDC20\"     \n",
       " [76] \"TAZ\"        \"AC010240.3\" \"AC019171.1\" \"AC133550.2\" \"AC146944.4\"\n",
       " [81] \"AC245123.1\" \"AF111167.2\" \"AL096865.1\" \"AL121821.1\" \"ANKRD29\"   \n",
       " [86] \"AP000866.1\" \"C1QTNF7\"    \"CAPN6\"      \"CCDC136\"    \"CCDC177\"   \n",
       " [91] \"CELF4\"      \"CXorf67\"    \"DTX3\"       \"DYTN\"       \"ENPP1\"     \n",
       " [96] \"EYA4\"       \"HMGCLL1\"    \"LINC00626\"  \"LINC01173\"  \"LTC4S\"     \n",
       "[101] \"MRGPRF\"     \"NPHS1\"      \"PAPPA2\"     \"PLSCR4\"     \"PNMA6A\"    \n",
       "[106] \"PRR19\"      \"SRCIN1\"     \"STAM-AS1\"   \"SUSD4\"      \"SYT14\"     \n",
       "[111] \"TMEM59L\"    \"VCAM1\"      \"ZFP3\"       \"ZIC4\"       \"AC008496.2\"\n",
       "[116] \"AC068790.9\" \"AC245052.4\" \"AL121718.1\" \"AL359715.2\" \"AP001363.2\"\n",
       "[121] \"HIST1H3B\"   \"TRPM1\"      \"Z99774.1\"   \"AC090627.1\" \"GALNT9\"    \n",
       "[126] \"IL10RB-DT\"  \"KCNE5\"      \"MIAT\"       \"NDUFA13\"    \"FOXM1\"     \n",
       "[131] \"TNRC6B\"     \"PFDN1\"      \"RASSF2\"     \"RBM33\"      \"KCNQ1OT1\"  \n",
       "[136] \"PTTG1\"      \"SKA3\"       \"PRC1\"       \"ASXL1\"      \"AKAP13\"    \n",
       "[141] \"LIPA\"       \"TECR\"       \"NCAPG\"      \"TFPT\"       \"AMD1\"      \n",
       "[146] \"QPCT\"       \"DIAPH3\"     \"EIF4A2\"     \"ISG15\"      \"NAMPT\"     \n",
       "[151] \"CCDC25\"     \"NAT14\"      \"RPA1\"       \"CNTROB\"     \"DLEU2\"     \n",
       "[156] \"KDM4A\"      \"OSER1\"      \"ANLN\"       \"BGN\"        \"ARHGEF28\"  \n",
       "[161] \"EIF4EBP1\"   \"SAE1\"       \"SNRPE\"      \"P4HB\"       \"POMT2\"     \n",
       "[166] \"APLP2\"      \"CCNA2\"      \"TMEM60\"     \"ADIPOR1\"    \"LSM8\"      \n",
       "[171] \"PSMB9\"      \"TNFRSF1A\"   \"HNRNPR\"     \"ILK\"        \"SPHK1\"     \n",
       "[176] \"CMPK1\"      \"PFKL\"       \"RNPS1\"      \"CCNB1\"      \"PRNP\"      \n",
       "[181] \"EIF2S2\"     \"PPM1F\"      \"KIAA1143\"   \"KIF14\"      \"SOGA1\"     \n",
       "[186] \"MAPKAPK5\"   \"DDX19B\"     \"LAMB2\"      \"TUBB\"       \"CENPF\"     \n",
       "[191] \"RHEB\"       \"C12orf45\"   \"RABGAP1\"    \"CDK1\"       \"FNBP1L\"    \n",
       "[196] \"LYRM4\"      \"U2SURP\"     \"PLK1\"       \"TOP2A\"      \"FBRS\"      \n",
       "[201] \"STAU2\"      \"SMARCB1\"    \"RAP1A\"      \"BAALC\"      \"DLL4\"      \n",
       "[206] \"MTSS1\"      \"RIT1\"       \"ROCK2\"      \"ANAPC5\"     \"MXRA7\"     \n",
       "[211] \"TMEM106C\"   \"CHMP3\"      \"G3BP2\"      \"GOLIM4\"     \"SEC23A\"    \n",
       "[216] \"POLR2L\"     \"CPE\"        \"GPR161\"     \"AURKB\"      \"FREM1\"     \n",
       "[221] \"FAM50A\"     \"STARD4\"     \"AP001505.1\" \"ARHGEF4\"    \"B3GALT1\"   \n",
       "[226] \"BNC2\"       \"CD74\"       \"CPED1\"      \"GRID2\"      \"RND2\"      \n",
       "[231] \"THRB\"       \"TNMD\"       \"AIF1\"       \"AL163051.1\" \"AL354836.1\"\n",
       "[236] \"AL590560.2\" \"ASXL3\"      \"BARX1\"      \"BMP5\"       \"C19orf38\"  \n",
       "[241] \"CDON\"       \"EPHA3\"      \"EPS8\"       \"FBXO2\"      \"FGF13\"     \n",
       "[246] \"FSTL5\"      \"GAS7\"       \"LRRC4B\"     \"NHS\"        \"NKD1\"      \n",
       "[251] \"PALM2\"      \"PHACTR3\"    \"PLPPR3\"     \"SAMD11\"     \"SIX1\"      \n",
       "[256] \"SULF1\"      \"TCEAL5\"     \"ZNF700\"     \"AC018695.2\" \"AC022306.2\"\n",
       "[261] \"BDNF-AS\"    \"BICC1\"      \"CDH23\"      \"COL13A1\"    \"GDF10\"     \n",
       "[266] \"GXYLT2\"     \"JAKMIP2\"    \"KIF18A\"     \"NRK\"        \"STXBP2\"    \n",
       "[271] \"Z97832.2\"   \"HCG17\"      \"AC110285.2\" \"AL731661.1\" \"ARHGEF39\"  \n",
       "[276] \"C17orf53\"   \"C3orf67\"    \"C8orf88\"    \"CD274\"      \"E2F8\"      \n",
       "[281] \"ERC2\"       \"HLA-DMA\"    \"HSD17B2\"    \"MIRLET7BHG\" \"PODXL2\"    \n",
       "[286] \"ZNF501\"     \"ZNF726\"     \"ZSWIM3\"     \"ZSWIM6\"     \"ATF6B\"     \n",
       "[291] \"GMDS\"       \"LOX\"        \"PJA2\"       \"SRSF4\"      \"THRAP3\"    \n",
       "[296] \"UBE2A\"      \"CELF1\"      \"CGNL1\"      \"S100A13\"    \"MELK\"      \n",
       "[301] \"C6orf89\"    \"STAG3\"      \"WSB2\"       \"RC3H2\"      \"C12orf10\"  \n",
       "[306] \"JUNB\"       \"PDHB\"       \"RPS4Y1\"     \"ARL4C\"      \"F2R\"       \n",
       "[311] \"PRPF38B\"    \"DNM2\"       \"MTRNR2L8\"   \"SRBD1\"      \"PDZD11\"    \n",
       "[316] \"MTRNR2L10\"  \"TPST2\"      \"PARG\"       \"FANCI\"      \"EIF2AK2\"   \n",
       "[321] \"ERLEC1\"     \"PPFIBP1\"    \"AC010642.2\" \"CUEDC2\"     \"HIGD1A\"    \n",
       "[326] \"MAF\"        \"RNF187\"     \"COL1A1\"     \"GGNBP2\"     \"CCNB2\"     \n",
       "[331] \"VAPA\"       \"CXCL12\"     \"LIMD2\"      \"UGGT2\"      \"EIF2S3\"    \n",
       "[336] \"KANK3\"      \"PCSK6\"      \"PSMD6\"      \"RAPGEF2\"    \"SH3GL1\"    \n",
       "[341] \"VMP1\"       \"ZFYVE21\"    \"RPS26\"      \"ENO1\"       \"LINC01704\" \n",
       "[346] \"TSTD2\"      \"SNRPC\"      \"AGAP1\"      \"ARCN1\"      \"CDK17\"     \n",
       "[351] \"GIPC1\"      \"MRPS6\"      \"SMIM10L1\"   \"DDRGK1\"     \"HEXA\"      \n",
       "[356] \"TIAL1\"      \"H2AFY\"      \"BOD1\"       \"PDXK\"       \"STX7\"      \n",
       "[361] \"TMEM14A\"    \"YIPF5\"      \"ZCCHC17\"    \"KRT8\"       \"AMZ2\"      \n",
       "[366] \"IFITM2\"     \"SKP2\"       \"CKAP2\"      \"CCDC34\"     \"SLC25A38\"  \n",
       "[371] \"NUP153\"     \"DDX46\"      \"SSBP3\"      \"WDFY1\"      \"FAU\"       \n",
       "[376] \"ADSS\"       \"BCHE\"       \"EFNB2\"      \"FKBP10\"     \"GRPEL1\"    \n",
       "[381] \"PPP1R7\"     \"TTC37\"      \"GNA13\"      \"LBR\"        \"PCID2\"     \n",
       "[386] \"ADAR\"       \"CAPRIN1\"    \"SGO2\"       \"TOP2B\"      \"MED22\"     \n",
       "[391] \"EIF4H\"      \"SPCS1\"      \"BUB3\"       \"PRRC1\"      \"RRM2\"      \n",
       "[396] \"CD2BP2\"     \"IGFBP7\"     \"PLCG1\"      \"ABCD4\"      \"CASP4\"     \n",
       "[401] \"CDC42BPA\"   \"DNAJB14\"    \"HNRNPH2\"    \"MRPS12\"     \"MRPS5\"     \n",
       "[406] \"NANS\"       \"NEMF\"       \"P4HA1\"      \"MGAT4B\"     \"SCD\"       \n",
       "[411] \"CNTRL\"      \"MGAT2\"      \"TSPAN14\"    \"HUWE1\"      \"HYI\"       \n",
       "[416] \"HMGB3\"      \"CKLF\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sig_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "forced-organization",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T18:04:39.292188Z",
     "iopub.status.busy": "2023-01-31T18:04:39.290853Z",
     "iopub.status.idle": "2023-01-31T18:04:39.562753Z",
     "shell.execute_reply": "2023-01-31T18:04:39.559337Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for SummarizedExperiment-class {SummarizedExperiment}\"><tr><td>SummarizedExperiment-class {SummarizedExperiment}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>SummarizedExperiment objects</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>The SummarizedExperiment class is a matrix-like container where rows\n",
       "represent features of interest (e.g. genes, transcripts, exons, etc...)\n",
       "and columns represent samples (with sample data summarized as a\n",
       "DataFrame). A SummarizedExperiment object contains one or more\n",
       "assays, each represented by a matrix-like object of numeric or other mode.\n",
       "</p>\n",
       "<p>Note that SummarizedExperiment is the parent of the\n",
       "RangedSummarizedExperiment class which means that all the methods\n",
       "documented below also work on a RangedSummarizedExperiment object.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "\n",
       "## Constructor\n",
       "\n",
       "# See ?RangedSummarizedExperiment for the constructor function.\n",
       "\n",
       "## Accessors\n",
       "\n",
       "assayNames(x, ...)\n",
       "assayNames(x, ...) &lt;- value\n",
       "assays(x, withDimnames=TRUE, ...)\n",
       "assays(x, withDimnames=TRUE, ...) &lt;- value\n",
       "assay(x, i, withDimnames=TRUE, ...)\n",
       "assay(x, i, withDimnames=TRUE, ...) &lt;- value\n",
       "rowData(x, use.names=TRUE, ...)\n",
       "rowData(x, ...) &lt;- value\n",
       "colData(x, ...)\n",
       "colData(x, ...) &lt;- value\n",
       "#dim(x)\n",
       "#dimnames(x)\n",
       "#dimnames(x) &lt;- value\n",
       "\n",
       "## Quick colData access\n",
       "\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "x$name\n",
       "## S4 replacement method for signature 'SummarizedExperiment'\n",
       "x$name &lt;- value\n",
       "## S4 method for signature 'SummarizedExperiment,ANY,missing'\n",
       "x[[i, j, ...]]\n",
       "## S4 replacement method for signature 'SummarizedExperiment,ANY,missing'\n",
       "x[[i, j, ...]] &lt;- value\n",
       "\n",
       "## Subsetting\n",
       "\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "x[i, j, ..., drop=TRUE]\n",
       "## S4 replacement method for signature 'SummarizedExperiment,ANY,ANY,SummarizedExperiment'\n",
       "x[i, j] &lt;- value\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "subset(x, subset, select, ...)\n",
       "\n",
       "## Combining\n",
       "\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "cbind(..., deparse.level=1)\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "rbind(..., deparse.level=1)\n",
       "\n",
       "## On-disk realization\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "realize(x, BACKEND=getAutoRealizationBackend())\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>x</code></td>\n",
       "<td>\n",
       "<p>A SummarizedExperiment object.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>...</code></td>\n",
       "<td>\n",
       "\n",
       "<p>For <code>assay</code>, arguments in <code>...</code> are forwarded to <code>assays</code>.\n",
       "</p>\n",
       "<p>For <code>cbind</code>, <code>rbind</code>, <code>...</code> contains SummarizedExperiment\n",
       "objects to be combined.\n",
       "</p>\n",
       "<p>For other accessors, ignored.\n",
       "</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>value</code></td>\n",
       "<td>\n",
       "<p>An object of a class specified in the S4 method\n",
       "signature or as outlined in &lsquo;Details&rsquo;.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>i, j</code></td>\n",
       "<td>\n",
       "\n",
       "<p>For <code>assay</code>, <code>assay&lt;-</code>, <code>i</code> is an integer or\n",
       "numeric scalar; see &lsquo;Details&rsquo; for additional constraints.\n",
       "</p>\n",
       "<p>For <code>[,SummarizedExperiment</code>,\n",
       "<code>[,SummarizedExperiment&lt;-</code>, <code>i</code>, <code>j</code> are subscripts\n",
       "that can act to subset the rows and columns of <code>x</code>, that is the\n",
       "<code>matrix</code> elements of <code>assays</code>.\n",
       "</p>\n",
       "<p>For <code>[[,SummarizedExperiment</code>,\n",
       "<code>[[&lt;-,SummarizedExperiment</code>, <code>i</code> is a scalar index (e.g.,\n",
       "<code>character(1)</code> or <code>integer(1)</code>) into a column of\n",
       "<code>colData</code>.\n",
       "</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>name</code></td>\n",
       "<td>\n",
       "<p>A symbol representing the name of a column of\n",
       "<code>colData</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>withDimnames</code></td>\n",
       "<td>\n",
       "<p>A <code>logical(1)</code>, indicating whether the\n",
       "dimnames of the SummarizedExperiment object should be applied (i.e.\n",
       "copied) to the extracted assays. More precisely, setting\n",
       "<code>withDimnames=FALSE</code> in the <em>getter</em> returns the\n",
       "assays <em>as-is</em> whereas setting <code>withDimnames=FALSE</code>\n",
       "return them with possibly modified dimnames.\n",
       "</p>\n",
       "<p>Setting <code>withDimnames=FALSE</code> in the <em>setter</em> (<code>assays&lt;-</code>)\n",
       "is required when the dimnames on the supplied assays are not identical\n",
       "to the dimnames on the SummarizedExperiment object; it does not\n",
       "influence actual assignment of dimnames to assays (they're always\n",
       "stored as-is).\n",
       "</p>\n",
       "<p>Note that </p>\n",
       "<pre>  assays(x, withDimnames=FALSE) &lt;- assays(x, withDimnames=FALSE)</pre>\n",
       "<p>is guaranteed to always work and be a no-op. This is not the case\n",
       "if <code>withDimnames=TRUE</code> is used or if <code>withDimnames</code> is\n",
       "not specified.\n",
       "</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>use.names</code></td>\n",
       "<td>\n",
       "<p>Like <code>mcols(x)</code>, by default\n",
       "<code>rowData(x)</code> propagates the rownames of <code>x</code> to the returned\n",
       "DataFrame object (note that for a SummarizedExperiment\n",
       "object, the rownames are also the names i.e. <code>rownames(x)</code> is\n",
       "always the same as <code>names(x)</code>). Setting <code>use.names=FALSE</code>\n",
       "suppresses this propagation i.e. it returns a DataFrame\n",
       "object with no rownames. Use this when <code>rowData(x)</code> fails,\n",
       "which can happen when the rownames contain NAs (because the rownames\n",
       "of a SummarizedExperiment object can contain NAs, but the rownames of\n",
       "a DataFrame object cannot).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>drop</code></td>\n",
       "<td>\n",
       "<p>A <code>logical(1)</code>, ignored by these methods.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>deparse.level</code></td>\n",
       "<td>\n",
       "<p>See <code>?base::cbind</code> for a description of\n",
       "this argument.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>subset</code></td>\n",
       "<td>\n",
       "<p>An expression which, when evaluated in the\n",
       "context of <code>rowData(x)</code>, is a logical vector indicating\n",
       "elements or rows to keep: missing values are taken as false.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>select</code></td>\n",
       "<td>\n",
       "<p>An expression which, when evaluated in the\n",
       "context of <code>colData(x)</code>, is a logical vector indicating\n",
       "elements or rows to keep: missing values are taken as false.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>BACKEND</code></td>\n",
       "<td>\n",
       "<p><code>NULL</code> (the default), or a single string specifying the\n",
       "name of the backend. When the backend is set to <code>NULL</code>,\n",
       "each element of <code>assays(x)</code> is realized in memory as an ordinary array\n",
       "by just calling <code>as.array</code> on it.</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p>The SummarizedExperiment class is meant for numeric and other\n",
       "data types derived from a sequencing experiment. The structure is\n",
       "rectangular like a <code>matrix</code>, but with additional annotations on\n",
       "the rows and columns, and with the possibility to manage several\n",
       "assays simultaneously.\n",
       "</p>\n",
       "<p>The rows of a SummarizedExperiment object represent features\n",
       "of interest. Information about these features is stored in a\n",
       "DataFrame object, accessible using the function\n",
       "<code>rowData</code>. The DataFrame must have as many rows\n",
       "as there are rows in the SummarizedExperiment object, with each row\n",
       "of the DataFrame providing information on the feature in the\n",
       "corresponding row of the SummarizedExperiment object. Columns of the\n",
       "DataFrame represent different attributes of the features\n",
       "of interest, e.g., gene or transcript IDs, etc.\n",
       "</p>\n",
       "<p>Each column of a SummarizedExperiment object represents a sample.\n",
       "Information about the samples are stored in a DataFrame,\n",
       "accessible using the function <code>colData</code>, described below.\n",
       "The DataFrame must have as many rows as there are\n",
       "columns in the SummarizedExperiment object, with each row of the\n",
       "DataFrame providing information on the sample in the\n",
       "corresponding column of the SummarizedExperiment object.\n",
       "Columns of the DataFrame represent different sample\n",
       "attributes, e.g., tissue of origin, etc. Columns of the\n",
       "DataFrame can themselves be annotated (via the\n",
       "<code>mcols</code> function). Column names typically\n",
       "provide a short identifier unique to each sample.\n",
       "</p>\n",
       "<p>A SummarizedExperiment object can also contain information about\n",
       "the overall experiment, for instance the lab in which it was conducted,\n",
       "the publications with which it is associated, etc. This information is\n",
       "stored as a <code>list</code> object, accessible using the <code>metadata</code>\n",
       "function. The form of the data associated with the experiment is left to\n",
       "the discretion of the user.\n",
       "</p>\n",
       "<p>The SummarizedExperiment container is appropriate for matrix-like\n",
       "data. The data are accessed using the <code>assays</code> function,\n",
       "described below. This returns a SimpleList object. Each\n",
       "element of the list must itself be a matrix (of any mode) and must\n",
       "have dimensions that are the same as the dimensions of the\n",
       "SummarizedExperiment in which they are stored. Row and column\n",
       "names of each matrix must either be <code>NULL</code> or match those of the\n",
       "SummarizedExperiment during construction. It is convenient for\n",
       "the elements of SimpleList of assays to be named.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Constructor</h3>\n",
       "\n",
       "<p>SummarizedExperiment instances are constructed using the\n",
       "<code>SummarizedExperiment</code> function documented in\n",
       "<code>?RangedSummarizedExperiment</code>.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Accessors</h3>\n",
       "\n",
       "<p>In the following code snippets, <code>x</code> is a SummarizedExperiment\n",
       "object.\n",
       "</p>\n",
       "\n",
       "<dl>\n",
       "<dt><code>assays(x)</code>, <code>assays(x) &lt;- value</code>:</dt><dd><p>Get or set the\n",
       "assays. <code>value</code> is a <code>list</code> or <code>SimpleList</code>, each\n",
       "element of which is a matrix with the same dimensions as\n",
       "<code>x</code>.</p>\n",
       "</dd>\n",
       "<dt><code>assay(x, i)</code>, <code>assay(x, i) &lt;- value</code>:</dt><dd><p>A convenient\n",
       "alternative (to <code>assays(x)[[i]]</code>, <code>assays(x)[[i]] &lt;-\n",
       "      value</code>) to get or set the <code>i</code>th (default first) assay\n",
       "element. <code>value</code> must be a matrix of the same dimension as\n",
       "<code>x</code>, and with dimension names <code>NULL</code> or consistent with\n",
       "those of <code>x</code>.</p>\n",
       "</dd>\n",
       "<dt><code>assayNames(x)</code>, <code>assayNames(x) &lt;- value</code>:</dt><dd><p>Get or\n",
       "set the names of <code>assay()</code> elements.</p>\n",
       "</dd>\n",
       "<dt><code>rowData(x, use.names=TRUE)</code>, <code>rowData(x) &lt;- value</code>:</dt><dd>\n",
       "<p>Get or set the row data. <code>value</code> is a DataFrame object.</p>\n",
       "</dd>\n",
       "<dt><code>colData(x)</code>, <code>colData(x) &lt;- value</code>:</dt><dd><p>Get or set the\n",
       "column data. <code>value</code> is a DataFrame object. Row\n",
       "names of <code>value</code> must be NULL or consistent with the existing\n",
       "column names of <code>x</code>.</p>\n",
       "</dd>\n",
       "<dt><code>metadata(x)</code>, <code>metadata(x) &lt;- value</code>:</dt><dd><p>Get or set\n",
       "the experiment data. <code>value</code> is a <code>list</code> with arbitrary\n",
       "content.</p>\n",
       "</dd>\n",
       "<dt><code>dim(x)</code>:</dt><dd><p>Get the dimensions (features of interest x samples)\n",
       "of the SummarizedExperiment.</p>\n",
       "</dd>\n",
       "<dt><code>dimnames(x)</code>, <code>dimnames(x) &lt;- value</code>:</dt><dd><p>Get or set\n",
       "the dimension names. <code>value</code> is usually a list of length 2,\n",
       "containing elements that are either <code>NULL</code> or vectors of\n",
       "appropriate length for the corresponding dimension. <code>value</code>\n",
       "can be <code>NULL</code>, which removes dimension names. This method\n",
       "implies that <code>rownames</code>, <code>rownames&lt;-</code>, <code>colnames</code>,\n",
       "and <code>colnames&lt;-</code> are all available.</p>\n",
       "</dd>\n",
       "</dl>\n",
       "\n",
       "\n",
       "\n",
       "<h3>Subsetting</h3>\n",
       "\n",
       "<p>In the code snippets below, <code>x</code> is a SummarizedExperiment object.\n",
       "</p>\n",
       "\n",
       "<dl>\n",
       "<dt><code>x[i,j]</code>, <code>x[i,j] &lt;- value</code>:</dt><dd><p>Create or replace a\n",
       "subset of <code>x</code>. <code>i</code>, <code>j</code> can be <code>numeric</code>,\n",
       "<code>logical</code>, <code>character</code>, or <code>missing</code>. <code>value</code>\n",
       "must be a SummarizedExperiment object with dimensions,\n",
       "dimension names, and assay elements consistent with the subset\n",
       "<code>x[i,j]</code> being replaced.</p>\n",
       "</dd>\n",
       "<dt><code>subset(x, subset, select)</code>:</dt><dd><p>Create a subset of <code>x</code>\n",
       "using an expression <code>subset</code> referring to columns of\n",
       "<code>rowData(x)</code> and / or <code>select</code> referring to column names\n",
       "of <code>colData(x)</code>.</p>\n",
       "</dd>\n",
       "</dl>\n",
       "\n",
       "<p>Additional subsetting accessors provide convenient access to\n",
       "<code>colData</code> columns\n",
       "</p>\n",
       "\n",
       "<dl>\n",
       "<dt><code>x$name</code>, <code>x$name &lt;- value</code></dt><dd><p>Access or replace\n",
       "column <code>name</code> in <code>x</code>.</p>\n",
       "</dd>\n",
       "<dt><code>x[[i, ...]]</code>, <code>x[[i, ...]] &lt;- value</code></dt><dd><p>Access or\n",
       "replace column <code>i</code> in <code>x</code>.</p>\n",
       "</dd>\n",
       "</dl>\n",
       "\n",
       "\n",
       "\n",
       "<h3>Combining</h3>\n",
       "\n",
       "<p>In the code snippets below, <code>...</code> are SummarizedExperiment objects\n",
       "to be combined.\n",
       "</p>\n",
       "\n",
       "<dl>\n",
       "<dt><code>cbind(...)</code>:</dt><dd>\n",
       "<p><code>cbind</code> combines objects with the same features of interest\n",
       "but different samples (columns in <code>assays</code>).\n",
       "The colnames in <code>colData(SummarizedExperiment)</code> must match or\n",
       "an error is thrown.\n",
       "Duplicate columns of <code>rowData(SummarizedExperiment)</code> must\n",
       "contain the same data.\n",
       "</p>\n",
       "<p>Data in <code>assays</code> are combined by name matching; if all assay\n",
       "names are NULL matching is by position. A mixture of names and NULL\n",
       "throws an error.\n",
       "</p>\n",
       "<p><code>metadata</code> from all objects are combined into a <code>list</code>\n",
       "with no name checking.\n",
       "</p>\n",
       "</dd>\n",
       "<dt><code>rbind(...)</code>:</dt><dd>\n",
       "<p><code>rbind</code> combines objects with the same samples\n",
       "but different features of interest (rows in <code>assays</code>).\n",
       "The colnames in <code>rowData(SummarizedExperiment)</code> must match or\n",
       "an error is thrown.\n",
       "Duplicate columns of <code>colData(SummarizedExperiment)</code> must\n",
       "contain the same data.\n",
       "</p>\n",
       "<p>Data in <code>assays</code> are combined by name matching; if all assay\n",
       "names are NULL matching is by position. A mixture of names and NULL\n",
       "throws an error.\n",
       "</p>\n",
       "<p><code>metadata</code> from all objects are combined into a <code>list</code>\n",
       "with no name checking.\n",
       "</p>\n",
       "</dd>\n",
       "</dl>\n",
       "\n",
       "\n",
       "\n",
       "<h3>Implementation and Extension</h3>\n",
       "\n",
       "<p>This section contains advanced material meant for package developers.\n",
       "</p>\n",
       "<p>SummarizedExperiment is implemented as an S4 class, and can be extended in\n",
       "the usual way, using <code>contains=\"SummarizedExperiment\"</code> in the new\n",
       "class definition.\n",
       "</p>\n",
       "<p>In addition, the representation of the <code>assays</code> slot of\n",
       "SummarizedExperiment is as a virtual class Assays. This\n",
       "allows derived classes (<code>contains=\"Assays\"</code>) to implement\n",
       "alternative requirements for the assays, e.g., backed by file-based\n",
       "storage like NetCDF or the <code>ff</code> package, while re-using the existing\n",
       "SummarizedExperiment class without modification.\n",
       "See Assays for more information.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Author(s)</h3>\n",
       "\n",
       "<p>Martin Morgan, <a href=\"mtmorgan@fhcrc.org\">mtmorgan@fhcrc.org</a></p>\n",
       "\n",
       "\n",
       "<h3>See Also</h3>\n",
       "\n",
       "\n",
       "<ul>\n",
       "<li> <p>RangedSummarizedExperiment objects.\n",
       "</p>\n",
       "</li>\n",
       "<li> <p>DataFrame, SimpleList, and\n",
       "Annotated objects in the <span class=\"pkg\">S4Vectors</span> package.\n",
       "</p>\n",
       "</li>\n",
       "<li><p> The <code>metadata</code> and\n",
       "<code>mcols</code> accessors in the <span class=\"pkg\">S4Vectors</span>\n",
       "package.\n",
       "</p>\n",
       "</li>\n",
       "<li> <p><code>saveHDF5SummarizedExperiment</code> and\n",
       "<code>loadHDF5SummarizedExperiment</code> in the\n",
       "<span class=\"pkg\">HDF5Array</span> package for saving/loading an HDF5-based\n",
       "SummarizedExperiment object to/from disk.\n",
       "</p>\n",
       "</li>\n",
       "<li><p> The <code>realize</code> generic function in the\n",
       "<span class=\"pkg\">DelayedArray</span> package for more information about on-disk\n",
       "realization of objects carrying delayed operations.\n",
       "</p>\n",
       "</li></ul>\n",
       "\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "nrows &lt;- 200; ncols &lt;- 6\n",
       "counts &lt;- matrix(runif(nrows * ncols, 1, 1e4), nrows)\n",
       "colData &lt;- DataFrame(Treatment=rep(c(\"ChIP\", \"Input\"), 3),\n",
       "                     row.names=LETTERS[1:6])\n",
       "se0 &lt;- SummarizedExperiment(assays=SimpleList(counts=counts),\n",
       "                            colData=colData)\n",
       "se0\n",
       "dim(se0)\n",
       "dimnames(se0)\n",
       "assayNames(se0)\n",
       "head(assay(se0))\n",
       "assays(se0) &lt;- endoapply(assays(se0), asinh)\n",
       "head(assay(se0))\n",
       "\n",
       "rowData(se0)\n",
       "colData(se0)\n",
       "\n",
       "se0[, se0$Treatment == \"ChIP\"]\n",
       "subset(se0, select = Treatment == \"ChIP\")\n",
       "\n",
       "## cbind() combines objects with the same features of interest\n",
       "## but different samples:\n",
       "se1 &lt;- se0\n",
       "se2 &lt;- se1[,1:3]\n",
       "colnames(se2) &lt;- letters[seq_len(ncol(se2))]\n",
       "cmb1 &lt;- cbind(se1, se2)\n",
       "dim(cmb1)\n",
       "dimnames(cmb1)\n",
       "\n",
       "## rbind() combines objects with the same samples but different\n",
       "## features of interest:\n",
       "se1 &lt;- se0\n",
       "se2 &lt;- se1[1:50,]\n",
       "rownames(se2) &lt;- letters[seq_len(nrow(se2))]\n",
       "cmb2 &lt;- rbind(se1, se2)\n",
       "dim(cmb2)\n",
       "dimnames(cmb2)\n",
       "\n",
       "## ---------------------------------------------------------------------\n",
       "## ON-DISK REALIZATION\n",
       "## ---------------------------------------------------------------------\n",
       "library(DelayedArray)\n",
       "setAutoRealizationBackend(\"HDF5Array\")\n",
       "cmb3 &lt;- realize(cmb2)\n",
       "assay(cmb3, withDimnames=FALSE)  # an HDF5Matrix object\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>SummarizedExperiment</em> version 1.20.0 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{SummarizedExperiment-class}{SummarizedExperiment objects}{SummarizedExperiment.Rdash.class}\n",
       "\\aliasA{[,SummarizedExperiment-method}{SummarizedExperiment-class}{[,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{[,SummarizedExperiment,ANY-method}{SummarizedExperiment-class}{[,SummarizedExperiment,ANY.Rdash.method}\n",
       "\\aliasA{[,SummarizedExperiment,ANY,ANY,ANY-method}{SummarizedExperiment-class}{[,SummarizedExperiment,ANY,ANY,ANY.Rdash.method}\n",
       "\\aliasA{[[,SummarizedExperiment,ANY,missing-method}{SummarizedExperiment-class}{[[,SummarizedExperiment,ANY,missing.Rdash.method}\n",
       "\\aliasA{[[<\\Rdash{},SummarizedExperiment,ANY,missing-method}{SummarizedExperiment-class}{[[<.Rdash.,SummarizedExperiment,ANY,missing.Rdash.method}\n",
       "\\aliasA{[<\\Rdash{},SummarizedExperiment,ANY,ANY,SummarizedExperiment-method}{SummarizedExperiment-class}{[<.Rdash.,SummarizedExperiment,ANY,ANY,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{\\$,SummarizedExperiment-method}{SummarizedExperiment-class}{.Rdol.,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{\\$<\\Rdash{},SummarizedExperiment-method}{SummarizedExperiment-class}{.Rdol.<.Rdash.,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{assay}{SummarizedExperiment-class}{assay}\n",
       "\\aliasA{assay,SummarizedExperiment,character-method}{SummarizedExperiment-class}{assay,SummarizedExperiment,character.Rdash.method}\n",
       "\\aliasA{assay,SummarizedExperiment,missing-method}{SummarizedExperiment-class}{assay,SummarizedExperiment,missing.Rdash.method}\n",
       "\\aliasA{assay,SummarizedExperiment,numeric-method}{SummarizedExperiment-class}{assay,SummarizedExperiment,numeric.Rdash.method}\n",
       "\\aliasA{assay<\\Rdash{}}{SummarizedExperiment-class}{assay<.Rdash.}\n",
       "\\aliasA{assay<\\Rdash{},SummarizedExperiment,character-method}{SummarizedExperiment-class}{assay<.Rdash.,SummarizedExperiment,character.Rdash.method}\n",
       "\\aliasA{assay<\\Rdash{},SummarizedExperiment,missing-method}{SummarizedExperiment-class}{assay<.Rdash.,SummarizedExperiment,missing.Rdash.method}\n",
       "\\aliasA{assay<\\Rdash{},SummarizedExperiment,numeric-method}{SummarizedExperiment-class}{assay<.Rdash.,SummarizedExperiment,numeric.Rdash.method}\n",
       "\\aliasA{assayNames}{SummarizedExperiment-class}{assayNames}\n",
       "\\aliasA{assayNames,SummarizedExperiment-method}{SummarizedExperiment-class}{assayNames,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{assayNames<\\Rdash{}}{SummarizedExperiment-class}{assayNames<.Rdash.}\n",
       "\\aliasA{assayNames<\\Rdash{},SummarizedExperiment,character-method}{SummarizedExperiment-class}{assayNames<.Rdash.,SummarizedExperiment,character.Rdash.method}\n",
       "\\aliasA{assays}{SummarizedExperiment-class}{assays}\n",
       "\\aliasA{Assays\\_OR\\_NULL}{SummarizedExperiment-class}{Assays.Rul.OR.Rul.NULL}\n",
       "\\aliasA{Assays\\_OR\\_NULL-class}{SummarizedExperiment-class}{Assays.Rul.OR.Rul.NULL.Rdash.class}\n",
       "\\aliasA{assays,SummarizedExperiment-method}{SummarizedExperiment-class}{assays,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{assays<\\Rdash{}}{SummarizedExperiment-class}{assays<.Rdash.}\n",
       "\\aliasA{assays<\\Rdash{},SummarizedExperiment,list-method}{SummarizedExperiment-class}{assays<.Rdash.,SummarizedExperiment,list.Rdash.method}\n",
       "\\aliasA{assays<\\Rdash{},SummarizedExperiment,SimpleList-method}{SummarizedExperiment-class}{assays<.Rdash.,SummarizedExperiment,SimpleList.Rdash.method}\n",
       "\\aliasA{cbind,SummarizedExperiment-method}{SummarizedExperiment-class}{cbind,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{class:Assays\\_OR\\_NULL}{SummarizedExperiment-class}{class:Assays.Rul.OR.Rul.NULL}\n",
       "\\aliasA{class:SummarizedExperiment}{SummarizedExperiment-class}{class:SummarizedExperiment}\n",
       "\\aliasA{colData}{SummarizedExperiment-class}{colData}\n",
       "\\aliasA{colData,SummarizedExperiment-method}{SummarizedExperiment-class}{colData,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{colData<\\Rdash{}}{SummarizedExperiment-class}{colData<.Rdash.}\n",
       "\\aliasA{colData<\\Rdash{},SummarizedExperiment,DataFrame-method}{SummarizedExperiment-class}{colData<.Rdash.,SummarizedExperiment,DataFrame.Rdash.method}\n",
       "\\aliasA{colData<\\Rdash{},SummarizedExperiment,NULL-method}{SummarizedExperiment-class}{colData<.Rdash.,SummarizedExperiment,NULL.Rdash.method}\n",
       "\\aliasA{dim,SummarizedExperiment-method}{SummarizedExperiment-class}{dim,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{dimnames,SummarizedExperiment-method}{SummarizedExperiment-class}{dimnames,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{dimnames<\\Rdash{},SummarizedExperiment,list-method}{SummarizedExperiment-class}{dimnames<.Rdash.,SummarizedExperiment,list.Rdash.method}\n",
       "\\aliasA{dimnames<\\Rdash{},SummarizedExperiment,NULL-method}{SummarizedExperiment-class}{dimnames<.Rdash.,SummarizedExperiment,NULL.Rdash.method}\n",
       "\\aliasA{horizontal\\_slot\\_names,SummarizedExperiment-method}{SummarizedExperiment-class}{horizontal.Rul.slot.Rul.names,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{length,SummarizedExperiment-method}{SummarizedExperiment-class}{length,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{names,SummarizedExperiment-method}{SummarizedExperiment-class}{names,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{names<\\Rdash{},SummarizedExperiment-method}{SummarizedExperiment-class}{names<.Rdash.,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{parallel\\_slot\\_names,SummarizedExperiment-method}{SummarizedExperiment-class}{parallel.Rul.slot.Rul.names,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{rbind,SummarizedExperiment-method}{SummarizedExperiment-class}{rbind,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{realize,SummarizedExperiment-method}{SummarizedExperiment-class}{realize,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{rowData}{SummarizedExperiment-class}{rowData}\n",
       "\\aliasA{rowData,SummarizedExperiment-method}{SummarizedExperiment-class}{rowData,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{rowData<\\Rdash{}}{SummarizedExperiment-class}{rowData<.Rdash.}\n",
       "\\aliasA{rowData<\\Rdash{},SummarizedExperiment-method}{SummarizedExperiment-class}{rowData<.Rdash.,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{show,SummarizedExperiment-method}{SummarizedExperiment-class}{show,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{subset,SummarizedExperiment-method}{SummarizedExperiment-class}{subset,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{updateObject,SummarizedExperiment-method}{SummarizedExperiment-class}{updateObject,SummarizedExperiment.Rdash.method}\n",
       "\\aliasA{vertical\\_slot\\_names,SummarizedExperiment-method}{SummarizedExperiment-class}{vertical.Rul.slot.Rul.names,SummarizedExperiment.Rdash.method}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "\n",
       "The SummarizedExperiment class is a matrix-like container where rows\n",
       "represent features of interest (e.g. genes, transcripts, exons, etc...)\n",
       "and columns represent samples (with sample data summarized as a\n",
       "\\LinkA{DataFrame}{DataFrame}). A SummarizedExperiment object contains one or more\n",
       "assays, each represented by a matrix-like object of numeric or other mode.\n",
       "\n",
       "Note that SummarizedExperiment is the parent of the\n",
       "\\LinkA{RangedSummarizedExperiment}{RangedSummarizedExperiment} class which means that all the methods\n",
       "documented below also work on a \\LinkA{RangedSummarizedExperiment}{RangedSummarizedExperiment} object.\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "\n",
       "## Constructor\n",
       "\n",
       "# See ?RangedSummarizedExperiment for the constructor function.\n",
       "\n",
       "## Accessors\n",
       "\n",
       "assayNames(x, ...)\n",
       "assayNames(x, ...) <- value\n",
       "assays(x, withDimnames=TRUE, ...)\n",
       "assays(x, withDimnames=TRUE, ...) <- value\n",
       "assay(x, i, withDimnames=TRUE, ...)\n",
       "assay(x, i, withDimnames=TRUE, ...) <- value\n",
       "rowData(x, use.names=TRUE, ...)\n",
       "rowData(x, ...) <- value\n",
       "colData(x, ...)\n",
       "colData(x, ...) <- value\n",
       "#dim(x)\n",
       "#dimnames(x)\n",
       "#dimnames(x) <- value\n",
       "\n",
       "## Quick colData access\n",
       "\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "x$name\n",
       "## S4 replacement method for signature 'SummarizedExperiment'\n",
       "x$name <- value\n",
       "## S4 method for signature 'SummarizedExperiment,ANY,missing'\n",
       "x[[i, j, ...]]\n",
       "## S4 replacement method for signature 'SummarizedExperiment,ANY,missing'\n",
       "x[[i, j, ...]] <- value\n",
       "\n",
       "## Subsetting\n",
       "\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "x[i, j, ..., drop=TRUE]\n",
       "## S4 replacement method for signature 'SummarizedExperiment,ANY,ANY,SummarizedExperiment'\n",
       "x[i, j] <- value\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "subset(x, subset, select, ...)\n",
       "\n",
       "## Combining\n",
       "\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "cbind(..., deparse.level=1)\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "rbind(..., deparse.level=1)\n",
       "\n",
       "## On-disk realization\n",
       "## S4 method for signature 'SummarizedExperiment'\n",
       "realize(x, BACKEND=getAutoRealizationBackend())\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\n",
       "\\item[\\code{x}] A SummarizedExperiment object.\n",
       "\n",
       "\\item[\\code{...}] \n",
       "For \\code{assay}, arguments in \\code{...} are forwarded to \\code{assays}.\n",
       "\n",
       "For \\code{cbind}, \\code{rbind}, \\code{...} contains SummarizedExperiment\n",
       "objects to be combined.\n",
       "\n",
       "For other accessors, ignored.\n",
       "\n",
       "\n",
       "\\item[\\code{value}] An object of a class specified in the S4 method\n",
       "signature or as outlined in `Details'.\n",
       "\n",
       "\\item[\\code{i, j}] \n",
       "For \\code{assay}, \\code{assay<-}, \\code{i} is an integer or\n",
       "numeric scalar; see `Details' for additional constraints.\n",
       "\n",
       "For \\code{[,SummarizedExperiment},\n",
       "\\code{[,SummarizedExperiment<-}, \\code{i}, \\code{j} are subscripts\n",
       "that can act to subset the rows and columns of \\code{x}, that is the\n",
       "\\code{matrix} elements of \\code{assays}.\n",
       "\n",
       "For \\code{[[,SummarizedExperiment},\n",
       "\\code{[[<-,SummarizedExperiment}, \\code{i} is a scalar index (e.g.,\n",
       "\\code{character(1)} or \\code{integer(1)}) into a column of\n",
       "\\code{colData}.\n",
       "\n",
       "\n",
       "\\item[\\code{name}] A symbol representing the name of a column of\n",
       "\\code{colData}.\n",
       "\n",
       "\\item[\\code{withDimnames}] A \\code{logical(1)}, indicating whether the\n",
       "dimnames of the SummarizedExperiment object should be applied (i.e.\n",
       "copied) to the extracted assays. More precisely, setting\n",
       "\\code{withDimnames=FALSE} in the \\emph{getter} returns the\n",
       "assays \\emph{as-is} whereas setting \\code{withDimnames=FALSE}\n",
       "return them with possibly modified dimnames.\n",
       "\n",
       "Setting \\code{withDimnames=FALSE} in the \\emph{setter} (\\code{assays<-})\n",
       "is required when the dimnames on the supplied assays are not identical\n",
       "to the dimnames on the SummarizedExperiment object; it does not\n",
       "influence actual assignment of dimnames to assays (they're always\n",
       "stored as-is).\n",
       "\n",
       "Note that \\begin{alltt}  assays(x, withDimnames=FALSE) <- assays(x, withDimnames=FALSE)\\end{alltt}\n",
       "\n",
       "is guaranteed to always work and be a no-op. This is not the case\n",
       "if \\code{withDimnames=TRUE} is used or if \\code{withDimnames} is\n",
       "not specified.\n",
       "\n",
       "\n",
       "\\item[\\code{use.names}] Like \\code{\\LinkA{mcols}{mcols}(x)}, by default\n",
       "\\code{rowData(x)} propagates the rownames of \\code{x} to the returned\n",
       "\\LinkA{DataFrame}{DataFrame} object (note that for a SummarizedExperiment\n",
       "object, the rownames are also the names i.e. \\code{rownames(x)} is\n",
       "always the same as \\code{names(x)}). Setting \\code{use.names=FALSE}\n",
       "suppresses this propagation i.e. it returns a \\LinkA{DataFrame}{DataFrame}\n",
       "object with no rownames. Use this when \\code{rowData(x)} fails,\n",
       "which can happen when the rownames contain NAs (because the rownames\n",
       "of a SummarizedExperiment object can contain NAs, but the rownames of\n",
       "a \\LinkA{DataFrame}{DataFrame} object cannot).\n",
       "\n",
       "\\item[\\code{drop}] A \\code{logical(1)}, ignored by these methods.\n",
       "\n",
       "\\item[\\code{deparse.level}] See \\code{?base::\\LinkA{cbind}{cbind}} for a description of\n",
       "this argument.\n",
       "\n",
       "\\item[\\code{subset}] An expression which, when evaluated in the\n",
       "context of \\code{rowData(x)}, is a logical vector indicating\n",
       "elements or rows to keep: missing values are taken as false.\n",
       "\n",
       "\\item[\\code{select}] An expression which, when evaluated in the\n",
       "context of \\code{colData(x)}, is a logical vector indicating\n",
       "elements or rows to keep: missing values are taken as false.\n",
       "\n",
       "\\item[\\code{BACKEND}] \\code{NULL} (the default), or a single string specifying the\n",
       "name of the backend. When the backend is set to \\code{NULL},\n",
       "each element of \\code{assays(x)} is realized in memory as an ordinary array\n",
       "by just calling \\code{as.array} on it.\n",
       "\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "\n",
       "The SummarizedExperiment class is meant for numeric and other\n",
       "data types derived from a sequencing experiment. The structure is\n",
       "rectangular like a \\code{matrix}, but with additional annotations on\n",
       "the rows and columns, and with the possibility to manage several\n",
       "assays simultaneously.\n",
       "\n",
       "The rows of a SummarizedExperiment object represent features\n",
       "of interest. Information about these features is stored in a\n",
       "\\LinkA{DataFrame}{DataFrame} object, accessible using the function\n",
       "\\code{rowData}. The \\LinkA{DataFrame}{DataFrame} must have as many rows\n",
       "as there are rows in the SummarizedExperiment object, with each row\n",
       "of the \\LinkA{DataFrame}{DataFrame} providing information on the feature in the\n",
       "corresponding row of the SummarizedExperiment object. Columns of the\n",
       "\\LinkA{DataFrame}{DataFrame} represent different attributes of the features\n",
       "of interest, e.g., gene or transcript IDs, etc.\n",
       "\n",
       "Each column of a SummarizedExperiment object represents a sample.\n",
       "Information about the samples are stored in a \\LinkA{DataFrame}{DataFrame},\n",
       "accessible using the function \\code{colData}, described below.\n",
       "The \\LinkA{DataFrame}{DataFrame} must have as many rows as there are\n",
       "columns in the SummarizedExperiment object, with each row of the\n",
       "\\LinkA{DataFrame}{DataFrame} providing information on the sample in the\n",
       "corresponding column of the SummarizedExperiment object.\n",
       "Columns of the \\LinkA{DataFrame}{DataFrame} represent different sample\n",
       "attributes, e.g., tissue of origin, etc. Columns of the\n",
       "\\LinkA{DataFrame}{DataFrame} can themselves be annotated (via the\n",
       "\\code{\\LinkA{mcols}{mcols}} function). Column names typically\n",
       "provide a short identifier unique to each sample.\n",
       "\n",
       "A SummarizedExperiment object can also contain information about\n",
       "the overall experiment, for instance the lab in which it was conducted,\n",
       "the publications with which it is associated, etc. This information is\n",
       "stored as a \\code{list} object, accessible using the \\code{metadata}\n",
       "function. The form of the data associated with the experiment is left to\n",
       "the discretion of the user.\n",
       "\n",
       "The SummarizedExperiment container is appropriate for matrix-like\n",
       "data. The data are accessed using the \\code{assays} function,\n",
       "described below. This returns a \\LinkA{SimpleList}{SimpleList} object. Each\n",
       "element of the list must itself be a matrix (of any mode) and must\n",
       "have dimensions that are the same as the dimensions of the\n",
       "SummarizedExperiment in which they are stored. Row and column\n",
       "names of each matrix must either be \\code{NULL} or match those of the\n",
       "SummarizedExperiment during construction. It is convenient for\n",
       "the elements of \\LinkA{SimpleList}{SimpleList} of assays to be named.\n",
       "\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Section}{Constructor}\n",
       "\n",
       "SummarizedExperiment instances are constructed using the\n",
       "\\code{SummarizedExperiment} function documented in\n",
       "\\code{?\\LinkA{RangedSummarizedExperiment}{RangedSummarizedExperiment}}.\n",
       "\n",
       "\\end{Section}\n",
       "%\n",
       "\\begin{Section}{Accessors}\n",
       "\n",
       "In the following code snippets, \\code{x} is a SummarizedExperiment\n",
       "object.\n",
       "\n",
       "\\begin{description}\n",
       "\n",
       "\n",
       "\\item[\\code{assays(x)}, \\code{assays(x) <- value}:] Get or set the\n",
       "assays. \\code{value} is a \\code{list} or \\code{SimpleList}, each\n",
       "element of which is a matrix with the same dimensions as\n",
       "\\code{x}.\n",
       "\n",
       "\\item[\\code{assay(x, i)}, \\code{assay(x, i) <- value}:] A convenient\n",
       "alternative (to \\code{assays(x)[[i]]}, \\code{assays(x)[[i]] <-\n",
       "      value}) to get or set the \\code{i}th (default first) assay\n",
       "element. \\code{value} must be a matrix of the same dimension as\n",
       "\\code{x}, and with dimension names \\code{NULL} or consistent with\n",
       "those of \\code{x}.\n",
       "\n",
       "\\item[\\code{assayNames(x)}, \\code{assayNames(x) <- value}:] Get or\n",
       "set the names of \\code{assay()} elements.\n",
       "\n",
       "\\item[\\code{rowData(x, use.names=TRUE)}, \\code{rowData(x) <- value}:] \n",
       "Get or set the row data. \\code{value} is a \\LinkA{DataFrame}{DataFrame} object.\n",
       "\n",
       "\\item[\\code{colData(x)}, \\code{colData(x) <- value}:] Get or set the\n",
       "column data. \\code{value} is a \\LinkA{DataFrame}{DataFrame} object. Row\n",
       "names of \\code{value} must be NULL or consistent with the existing\n",
       "column names of \\code{x}.\n",
       "\n",
       "\\item[\\code{metadata(x)}, \\code{metadata(x) <- value}:] Get or set\n",
       "the experiment data. \\code{value} is a \\code{list} with arbitrary\n",
       "content.\n",
       "\n",
       "\\item[\\code{dim(x)}:] Get the dimensions (features of interest x samples)\n",
       "of the SummarizedExperiment.\n",
       "\n",
       "\\item[\\code{dimnames(x)}, \\code{dimnames(x) <- value}:] Get or set\n",
       "the dimension names. \\code{value} is usually a list of length 2,\n",
       "containing elements that are either \\code{NULL} or vectors of\n",
       "appropriate length for the corresponding dimension. \\code{value}\n",
       "can be \\code{NULL}, which removes dimension names. This method\n",
       "implies that \\code{rownames}, \\code{rownames<-}, \\code{colnames},\n",
       "and \\code{colnames<-} are all available.\n",
       "\n",
       "\n",
       "\\end{description}\n",
       "\n",
       "\\end{Section}\n",
       "%\n",
       "\\begin{Section}{Subsetting}\n",
       "\n",
       "In the code snippets below, \\code{x} is a SummarizedExperiment object.\n",
       "\n",
       "\\begin{description}\n",
       "\n",
       "\n",
       "\\item[\\code{x[i,j]}, \\code{x[i,j] <- value}:] Create or replace a\n",
       "subset of \\code{x}. \\code{i}, \\code{j} can be \\code{numeric},\n",
       "\\code{logical}, \\code{character}, or \\code{missing}. \\code{value}\n",
       "must be a SummarizedExperiment object with dimensions,\n",
       "dimension names, and assay elements consistent with the subset\n",
       "\\code{x[i,j]} being replaced.\n",
       "\n",
       "\\item[\\code{subset(x, subset, select)}:] Create a subset of \\code{x}\n",
       "using an expression \\code{subset} referring to columns of\n",
       "\\code{rowData(x)} and / or \\code{select} referring to column names\n",
       "of \\code{colData(x)}.\n",
       "\n",
       "\n",
       "\\end{description}\n",
       "\n",
       "\n",
       "Additional subsetting accessors provide convenient access to\n",
       "\\code{colData} columns\n",
       "\n",
       "\\begin{description}\n",
       "\n",
       "\n",
       "\\item[\\code{x\\$name}, \\code{x\\$name <- value}] Access or replace\n",
       "column \\code{name} in \\code{x}.\n",
       "\n",
       "\\item[\\code{x[[i, ...]]}, \\code{x[[i, ...]] <- value}] Access or\n",
       "replace column \\code{i} in \\code{x}.\n",
       "\n",
       "\n",
       "\\end{description}\n",
       "\n",
       "\n",
       "\\end{Section}\n",
       "%\n",
       "\\begin{Section}{Combining}\n",
       "\n",
       "In the code snippets below, \\code{...} are SummarizedExperiment objects\n",
       "to be combined.\n",
       "\n",
       "\\begin{description}\n",
       "\n",
       "\n",
       "\\item[\\code{cbind(...)}:] \n",
       "\\code{cbind} combines objects with the same features of interest\n",
       "but different samples (columns in \\code{assays}).\n",
       "The colnames in \\code{colData(SummarizedExperiment)} must match or\n",
       "an error is thrown.\n",
       "Duplicate columns of \\code{rowData(SummarizedExperiment)} must\n",
       "contain the same data.\n",
       "\n",
       "Data in \\code{assays} are combined by name matching; if all assay\n",
       "names are NULL matching is by position. A mixture of names and NULL\n",
       "throws an error.\n",
       "\n",
       "\\code{metadata} from all objects are combined into a \\code{list}\n",
       "with no name checking.\n",
       "\n",
       "\n",
       "\\item[\\code{rbind(...)}:] \n",
       "\\code{rbind} combines objects with the same samples\n",
       "but different features of interest (rows in \\code{assays}).\n",
       "The colnames in \\code{rowData(SummarizedExperiment)} must match or\n",
       "an error is thrown.\n",
       "Duplicate columns of \\code{colData(SummarizedExperiment)} must\n",
       "contain the same data.\n",
       "\n",
       "Data in \\code{assays} are combined by name matching; if all assay\n",
       "names are NULL matching is by position. A mixture of names and NULL\n",
       "throws an error.\n",
       "\n",
       "\\code{metadata} from all objects are combined into a \\code{list}\n",
       "with no name checking.\n",
       "\n",
       "\n",
       "\n",
       "\\end{description}\n",
       "\n",
       "\n",
       "\\end{Section}\n",
       "%\n",
       "\\begin{Section}{Implementation and Extension}\n",
       "\n",
       "This section contains advanced material meant for package developers.\n",
       "\n",
       "SummarizedExperiment is implemented as an S4 class, and can be extended in\n",
       "the usual way, using \\code{contains=\"SummarizedExperiment\"} in the new\n",
       "class definition.\n",
       "\n",
       "In addition, the representation of the \\code{assays} slot of\n",
       "SummarizedExperiment is as a virtual class Assays. This\n",
       "allows derived classes (\\code{contains=\"Assays\"}) to implement\n",
       "alternative requirements for the assays, e.g., backed by file-based\n",
       "storage like NetCDF or the \\code{ff} package, while re-using the existing\n",
       "SummarizedExperiment class without modification.\n",
       "See \\LinkA{Assays}{Assays} for more information.\n",
       "\n",
       "\\end{Section}\n",
       "%\n",
       "\\begin{Author}\\relax\n",
       "Martin Morgan, \\url{mtmorgan@fhcrc.org}\n",
       "\\end{Author}\n",
       "%\n",
       "\\begin{SeeAlso}\\relax\n",
       "\\begin{itemize}\n",
       "\n",
       "\\item{} \\LinkA{RangedSummarizedExperiment}{RangedSummarizedExperiment} objects.\n",
       "\n",
       "\\item{} \\LinkA{DataFrame}{DataFrame}, \\LinkA{SimpleList}{SimpleList}, and\n",
       "\\LinkA{Annotated}{Annotated} objects in the \\pkg{S4Vectors} package.\n",
       "\n",
       "\\item{} The \\code{\\LinkA{metadata}{metadata}} and\n",
       "\\code{\\LinkA{mcols}{mcols}} accessors in the \\pkg{S4Vectors}\n",
       "package.\n",
       "\n",
       "\\item{} \\code{\\LinkA{saveHDF5SummarizedExperiment}{saveHDF5SummarizedExperiment}} and\n",
       "\\code{\\LinkA{loadHDF5SummarizedExperiment}{loadHDF5SummarizedExperiment}} in the\n",
       "\\pkg{HDF5Array} package for saving/loading an HDF5-based\n",
       "SummarizedExperiment object to/from disk.\n",
       "\n",
       "\\item{} The \\code{\\LinkA{realize}{realize}} generic function in the\n",
       "\\pkg{DelayedArray} package for more information about on-disk\n",
       "realization of objects carrying delayed operations.\n",
       "\n",
       "\\end{itemize}\n",
       "\n",
       "\\end{SeeAlso}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "nrows <- 200; ncols <- 6\n",
       "counts <- matrix(runif(nrows * ncols, 1, 1e4), nrows)\n",
       "colData <- DataFrame(Treatment=rep(c(\"ChIP\", \"Input\"), 3),\n",
       "                     row.names=LETTERS[1:6])\n",
       "se0 <- SummarizedExperiment(assays=SimpleList(counts=counts),\n",
       "                            colData=colData)\n",
       "se0\n",
       "dim(se0)\n",
       "dimnames(se0)\n",
       "assayNames(se0)\n",
       "head(assay(se0))\n",
       "assays(se0) <- endoapply(assays(se0), asinh)\n",
       "head(assay(se0))\n",
       "\n",
       "rowData(se0)\n",
       "colData(se0)\n",
       "\n",
       "se0[, se0$Treatment == \"ChIP\"]\n",
       "subset(se0, select = Treatment == \"ChIP\")\n",
       "\n",
       "## cbind() combines objects with the same features of interest\n",
       "## but different samples:\n",
       "se1 <- se0\n",
       "se2 <- se1[,1:3]\n",
       "colnames(se2) <- letters[seq_len(ncol(se2))]\n",
       "cmb1 <- cbind(se1, se2)\n",
       "dim(cmb1)\n",
       "dimnames(cmb1)\n",
       "\n",
       "## rbind() combines objects with the same samples but different\n",
       "## features of interest:\n",
       "se1 <- se0\n",
       "se2 <- se1[1:50,]\n",
       "rownames(se2) <- letters[seq_len(nrow(se2))]\n",
       "cmb2 <- rbind(se1, se2)\n",
       "dim(cmb2)\n",
       "dimnames(cmb2)\n",
       "\n",
       "## ---------------------------------------------------------------------\n",
       "## ON-DISK REALIZATION\n",
       "## ---------------------------------------------------------------------\n",
       "library(DelayedArray)\n",
       "setAutoRealizationBackend(\"HDF5Array\")\n",
       "cmb3 <- realize(cmb2)\n",
       "assay(cmb3, withDimnames=FALSE)  # an HDF5Matrix object\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "SummarizedExperiment-classpackage:SummarizedExperiment R Documentation\n",
       "\n",
       "_\bS_\bu_\bm_\bm_\ba_\br_\bi_\bz_\be_\bd_\bE_\bx_\bp_\be_\br_\bi_\bm_\be_\bn_\bt _\bo_\bb_\bj_\be_\bc_\bt_\bs\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     The SummarizedExperiment class is a matrix-like container where\n",
       "     rows represent features of interest (e.g. genes, transcripts,\n",
       "     exons, etc...)  and columns represent samples (with sample data\n",
       "     summarized as a DataFrame). A SummarizedExperiment object contains\n",
       "     one or more assays, each represented by a matrix-like object of\n",
       "     numeric or other mode.\n",
       "\n",
       "     Note that SummarizedExperiment is the parent of the\n",
       "     RangedSummarizedExperiment class which means that all the methods\n",
       "     documented below also work on a RangedSummarizedExperiment object.\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     ## Constructor\n",
       "     \n",
       "     # See ?RangedSummarizedExperiment for the constructor function.\n",
       "     \n",
       "     ## Accessors\n",
       "     \n",
       "     assayNames(x, ...)\n",
       "     assayNames(x, ...) <- value\n",
       "     assays(x, withDimnames=TRUE, ...)\n",
       "     assays(x, withDimnames=TRUE, ...) <- value\n",
       "     assay(x, i, withDimnames=TRUE, ...)\n",
       "     assay(x, i, withDimnames=TRUE, ...) <- value\n",
       "     rowData(x, use.names=TRUE, ...)\n",
       "     rowData(x, ...) <- value\n",
       "     colData(x, ...)\n",
       "     colData(x, ...) <- value\n",
       "     #dim(x)\n",
       "     #dimnames(x)\n",
       "     #dimnames(x) <- value\n",
       "     \n",
       "     ## Quick colData access\n",
       "     \n",
       "     ## S4 method for signature 'SummarizedExperiment'\n",
       "     x$name\n",
       "     ## S4 replacement method for signature 'SummarizedExperiment'\n",
       "     x$name <- value\n",
       "     ## S4 method for signature 'SummarizedExperiment,ANY,missing'\n",
       "     x[[i, j, ...]]\n",
       "     ## S4 replacement method for signature 'SummarizedExperiment,ANY,missing'\n",
       "     x[[i, j, ...]] <- value\n",
       "     \n",
       "     ## Subsetting\n",
       "     \n",
       "     ## S4 method for signature 'SummarizedExperiment'\n",
       "     x[i, j, ..., drop=TRUE]\n",
       "     ## S4 replacement method for signature 'SummarizedExperiment,ANY,ANY,SummarizedExperiment'\n",
       "     x[i, j] <- value\n",
       "     ## S4 method for signature 'SummarizedExperiment'\n",
       "     subset(x, subset, select, ...)\n",
       "     \n",
       "     ## Combining\n",
       "     \n",
       "     ## S4 method for signature 'SummarizedExperiment'\n",
       "     cbind(..., deparse.level=1)\n",
       "     ## S4 method for signature 'SummarizedExperiment'\n",
       "     rbind(..., deparse.level=1)\n",
       "     \n",
       "     ## On-disk realization\n",
       "     ## S4 method for signature 'SummarizedExperiment'\n",
       "     realize(x, BACKEND=getAutoRealizationBackend())\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "       x: A SummarizedExperiment object.\n",
       "\n",
       "     ...: For ‘assay’, arguments in ‘...’ are forwarded to ‘assays’.\n",
       "\n",
       "          For ‘cbind’, ‘rbind’, ‘...’ contains SummarizedExperiment\n",
       "          objects to be combined.\n",
       "\n",
       "          For other accessors, ignored.\n",
       "\n",
       "   value: An object of a class specified in the S4 method signature or\n",
       "          as outlined in ‘Details’.\n",
       "\n",
       "    i, j: For ‘assay’, ‘assay<-’, ‘i’ is an integer or numeric scalar;\n",
       "          see ‘Details’ for additional constraints.\n",
       "\n",
       "          For ‘[,SummarizedExperiment’, ‘[,SummarizedExperiment<-’,\n",
       "          ‘i’, ‘j’ are subscripts that can act to subset the rows and\n",
       "          columns of ‘x’, that is the ‘matrix’ elements of ‘assays’.\n",
       "\n",
       "          For ‘[[,SummarizedExperiment’, ‘[[<-,SummarizedExperiment’,\n",
       "          ‘i’ is a scalar index (e.g., ‘character(1)’ or ‘integer(1)’)\n",
       "          into a column of ‘colData’.\n",
       "\n",
       "    name: A symbol representing the name of a column of ‘colData’.\n",
       "\n",
       "withDimnames: A ‘logical(1)’, indicating whether the dimnames of the\n",
       "          SummarizedExperiment object should be applied (i.e.  copied)\n",
       "          to the extracted assays. More precisely, setting\n",
       "          ‘withDimnames=FALSE’ in the _getter_ returns the assays\n",
       "          _as-is_ whereas setting ‘withDimnames=FALSE’ return them with\n",
       "          possibly modified dimnames.\n",
       "\n",
       "          Setting ‘withDimnames=FALSE’ in the _setter_ (‘assays<-’) is\n",
       "          required when the dimnames on the supplied assays are not\n",
       "          identical to the dimnames on the SummarizedExperiment object;\n",
       "          it does not influence actual assignment of dimnames to assays\n",
       "          (they're always stored as-is).\n",
       "\n",
       "          Note that\n",
       "\n",
       "            assays(x, withDimnames=FALSE) <- assays(x, withDimnames=FALSE)\n",
       "          \n",
       "          is guaranteed to always work and be a no-op. This is not the\n",
       "          case if ‘withDimnames=TRUE’ is used or if ‘withDimnames’ is\n",
       "          not specified.\n",
       "\n",
       "use.names: Like ‘mcols(x)’, by default ‘rowData(x)’ propagates the\n",
       "          rownames of ‘x’ to the returned DataFrame object (note that\n",
       "          for a SummarizedExperiment object, the rownames are also the\n",
       "          names i.e. ‘rownames(x)’ is always the same as ‘names(x)’).\n",
       "          Setting ‘use.names=FALSE’ suppresses this propagation i.e. it\n",
       "          returns a DataFrame object with no rownames. Use this when\n",
       "          ‘rowData(x)’ fails, which can happen when the rownames\n",
       "          contain NAs (because the rownames of a SummarizedExperiment\n",
       "          object can contain NAs, but the rownames of a DataFrame\n",
       "          object cannot).\n",
       "\n",
       "    drop: A ‘logical(1)’, ignored by these methods.\n",
       "\n",
       "deparse.level: See ‘?base::cbind’ for a description of this argument.\n",
       "\n",
       "  subset: An expression which, when evaluated in the context of\n",
       "          ‘rowData(x)’, is a logical vector indicating elements or rows\n",
       "          to keep: missing values are taken as false.\n",
       "\n",
       "  select: An expression which, when evaluated in the context of\n",
       "          ‘colData(x)’, is a logical vector indicating elements or rows\n",
       "          to keep: missing values are taken as false.\n",
       "\n",
       " BACKEND: ‘NULL’ (the default), or a single string specifying the name\n",
       "          of the backend. When the backend is set to ‘NULL’, each\n",
       "          element of ‘assays(x)’ is realized in memory as an ordinary\n",
       "          array by just calling ‘as.array’ on it.\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     The SummarizedExperiment class is meant for numeric and other data\n",
       "     types derived from a sequencing experiment. The structure is\n",
       "     rectangular like a ‘matrix’, but with additional annotations on\n",
       "     the rows and columns, and with the possibility to manage several\n",
       "     assays simultaneously.\n",
       "\n",
       "     The rows of a SummarizedExperiment object represent features of\n",
       "     interest. Information about these features is stored in a\n",
       "     DataFrame object, accessible using the function ‘rowData’. The\n",
       "     DataFrame must have as many rows as there are rows in the\n",
       "     SummarizedExperiment object, with each row of the DataFrame\n",
       "     providing information on the feature in the corresponding row of\n",
       "     the SummarizedExperiment object. Columns of the DataFrame\n",
       "     represent different attributes of the features of interest, e.g.,\n",
       "     gene or transcript IDs, etc.\n",
       "\n",
       "     Each column of a SummarizedExperiment object represents a sample.\n",
       "     Information about the samples are stored in a DataFrame,\n",
       "     accessible using the function ‘colData’, described below.  The\n",
       "     DataFrame must have as many rows as there are columns in the\n",
       "     SummarizedExperiment object, with each row of the DataFrame\n",
       "     providing information on the sample in the corresponding column of\n",
       "     the SummarizedExperiment object.  Columns of the DataFrame\n",
       "     represent different sample attributes, e.g., tissue of origin,\n",
       "     etc. Columns of the DataFrame can themselves be annotated (via the\n",
       "     ‘mcols’ function). Column names typically provide a short\n",
       "     identifier unique to each sample.\n",
       "\n",
       "     A SummarizedExperiment object can also contain information about\n",
       "     the overall experiment, for instance the lab in which it was\n",
       "     conducted, the publications with which it is associated, etc. This\n",
       "     information is stored as a ‘list’ object, accessible using the\n",
       "     ‘metadata’ function. The form of the data associated with the\n",
       "     experiment is left to the discretion of the user.\n",
       "\n",
       "     The SummarizedExperiment container is appropriate for matrix-like\n",
       "     data. The data are accessed using the ‘assays’ function, described\n",
       "     below. This returns a SimpleList object. Each element of the list\n",
       "     must itself be a matrix (of any mode) and must have dimensions\n",
       "     that are the same as the dimensions of the SummarizedExperiment in\n",
       "     which they are stored. Row and column names of each matrix must\n",
       "     either be ‘NULL’ or match those of the SummarizedExperiment during\n",
       "     construction. It is convenient for the elements of SimpleList of\n",
       "     assays to be named.\n",
       "\n",
       "_\bC_\bo_\bn_\bs_\bt_\br_\bu_\bc_\bt_\bo_\br:\n",
       "\n",
       "     SummarizedExperiment instances are constructed using the\n",
       "     ‘SummarizedExperiment’ function documented in\n",
       "     ‘?RangedSummarizedExperiment’.\n",
       "\n",
       "_\bA_\bc_\bc_\be_\bs_\bs_\bo_\br_\bs:\n",
       "\n",
       "     In the following code snippets, ‘x’ is a SummarizedExperiment\n",
       "     object.\n",
       "\n",
       "     ‘assays(x)’, ‘assays(x) <- value’: Get or set the assays. ‘value’\n",
       "          is a ‘list’ or ‘SimpleList’, each element of which is a\n",
       "          matrix with the same dimensions as ‘x’.\n",
       "\n",
       "     ‘assay(x, i)’, ‘assay(x, i) <- value’: A convenient alternative\n",
       "          (to ‘assays(x)[[i]]’, ‘assays(x)[[i]] <- value’) to get or\n",
       "          set the ‘i’th (default first) assay element. ‘value’ must be\n",
       "          a matrix of the same dimension as ‘x’, and with dimension\n",
       "          names ‘NULL’ or consistent with those of ‘x’.\n",
       "\n",
       "     ‘assayNames(x)’, ‘assayNames(x) <- value’: Get or set the names of\n",
       "          ‘assay()’ elements.\n",
       "\n",
       "     ‘rowData(x, use.names=TRUE)’, ‘rowData(x) <- value’: Get or set\n",
       "          the row data. ‘value’ is a DataFrame object.\n",
       "\n",
       "     ‘colData(x)’, ‘colData(x) <- value’: Get or set the column data.\n",
       "          ‘value’ is a DataFrame object. Row names of ‘value’ must be\n",
       "          NULL or consistent with the existing column names of ‘x’.\n",
       "\n",
       "     ‘metadata(x)’, ‘metadata(x) <- value’: Get or set the experiment\n",
       "          data. ‘value’ is a ‘list’ with arbitrary content.\n",
       "\n",
       "     ‘dim(x)’: Get the dimensions (features of interest x samples) of\n",
       "          the SummarizedExperiment.\n",
       "\n",
       "     ‘dimnames(x)’, ‘dimnames(x) <- value’: Get or set the dimension\n",
       "          names. ‘value’ is usually a list of length 2, containing\n",
       "          elements that are either ‘NULL’ or vectors of appropriate\n",
       "          length for the corresponding dimension. ‘value’ can be\n",
       "          ‘NULL’, which removes dimension names. This method implies\n",
       "          that ‘rownames’, ‘rownames<-’, ‘colnames’, and ‘colnames<-’\n",
       "          are all available.\n",
       "\n",
       "_\bS_\bu_\bb_\bs_\be_\bt_\bt_\bi_\bn_\bg:\n",
       "\n",
       "     In the code snippets below, ‘x’ is a SummarizedExperiment object.\n",
       "\n",
       "     ‘x[i,j]’, ‘x[i,j] <- value’: Create or replace a subset of ‘x’.\n",
       "          ‘i’, ‘j’ can be ‘numeric’, ‘logical’, ‘character’, or\n",
       "          ‘missing’. ‘value’ must be a SummarizedExperiment object with\n",
       "          dimensions, dimension names, and assay elements consistent\n",
       "          with the subset ‘x[i,j]’ being replaced.\n",
       "\n",
       "     ‘subset(x, subset, select)’: Create a subset of ‘x’ using an\n",
       "          expression ‘subset’ referring to columns of ‘rowData(x)’ and\n",
       "          / or ‘select’ referring to column names of ‘colData(x)’.\n",
       "\n",
       "     Additional subsetting accessors provide convenient access to\n",
       "     ‘colData’ columns\n",
       "\n",
       "     ‘x$name’, ‘x$name <- value’ Access or replace column ‘name’ in\n",
       "          ‘x’.\n",
       "\n",
       "     ‘x[[i, ...]]’, ‘x[[i, ...]] <- value’ Access or replace column ‘i’\n",
       "          in ‘x’.\n",
       "\n",
       "_\bC_\bo_\bm_\bb_\bi_\bn_\bi_\bn_\bg:\n",
       "\n",
       "     In the code snippets below, ‘...’ are SummarizedExperiment objects\n",
       "     to be combined.\n",
       "\n",
       "     ‘cbind(...)’: ‘cbind’ combines objects with the same features of\n",
       "          interest but different samples (columns in ‘assays’).  The\n",
       "          colnames in ‘colData(SummarizedExperiment)’ must match or an\n",
       "          error is thrown.  Duplicate columns of\n",
       "          ‘rowData(SummarizedExperiment)’ must contain the same data.\n",
       "\n",
       "          Data in ‘assays’ are combined by name matching; if all assay\n",
       "          names are NULL matching is by position. A mixture of names\n",
       "          and NULL throws an error.\n",
       "\n",
       "          ‘metadata’ from all objects are combined into a ‘list’ with\n",
       "          no name checking.\n",
       "\n",
       "     ‘rbind(...)’: ‘rbind’ combines objects with the same samples but\n",
       "          different features of interest (rows in ‘assays’).  The\n",
       "          colnames in ‘rowData(SummarizedExperiment)’ must match or an\n",
       "          error is thrown.  Duplicate columns of\n",
       "          ‘colData(SummarizedExperiment)’ must contain the same data.\n",
       "\n",
       "          Data in ‘assays’ are combined by name matching; if all assay\n",
       "          names are NULL matching is by position. A mixture of names\n",
       "          and NULL throws an error.\n",
       "\n",
       "          ‘metadata’ from all objects are combined into a ‘list’ with\n",
       "          no name checking.\n",
       "\n",
       "_\bI_\bm_\bp_\bl_\be_\bm_\be_\bn_\bt_\ba_\bt_\bi_\bo_\bn _\ba_\bn_\bd _\bE_\bx_\bt_\be_\bn_\bs_\bi_\bo_\bn:\n",
       "\n",
       "     This section contains advanced material meant for package\n",
       "     developers.\n",
       "\n",
       "     SummarizedExperiment is implemented as an S4 class, and can be\n",
       "     extended in the usual way, using ‘contains=\"SummarizedExperiment\"’\n",
       "     in the new class definition.\n",
       "\n",
       "     In addition, the representation of the ‘assays’ slot of\n",
       "     SummarizedExperiment is as a virtual class Assays. This allows\n",
       "     derived classes (‘contains=\"Assays\"’) to implement alternative\n",
       "     requirements for the assays, e.g., backed by file-based storage\n",
       "     like NetCDF or the ‘ff’ package, while re-using the existing\n",
       "     SummarizedExperiment class without modification.  See Assays for\n",
       "     more information.\n",
       "\n",
       "_\bA_\bu_\bt_\bh_\bo_\br(_\bs):\n",
       "\n",
       "     Martin Morgan, <URL: mtmorgan@fhcrc.org>\n",
       "\n",
       "_\bS_\be_\be _\bA_\bl_\bs_\bo:\n",
       "\n",
       "        • RangedSummarizedExperiment objects.\n",
       "\n",
       "        • DataFrame, SimpleList, and Annotated objects in the\n",
       "          ‘S4Vectors’ package.\n",
       "\n",
       "        • The ‘metadata’ and ‘mcols’ accessors in the ‘S4Vectors’\n",
       "          package.\n",
       "\n",
       "        • ‘saveHDF5SummarizedExperiment’ and\n",
       "          ‘loadHDF5SummarizedExperiment’ in the ‘HDF5Array’ package for\n",
       "          saving/loading an HDF5-based SummarizedExperiment object\n",
       "          to/from disk.\n",
       "\n",
       "        • The ‘realize’ generic function in the ‘DelayedArray’ package\n",
       "          for more information about on-disk realization of objects\n",
       "          carrying delayed operations.\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     nrows <- 200; ncols <- 6\n",
       "     counts <- matrix(runif(nrows * ncols, 1, 1e4), nrows)\n",
       "     colData <- DataFrame(Treatment=rep(c(\"ChIP\", \"Input\"), 3),\n",
       "                          row.names=LETTERS[1:6])\n",
       "     se0 <- SummarizedExperiment(assays=SimpleList(counts=counts),\n",
       "                                 colData=colData)\n",
       "     se0\n",
       "     dim(se0)\n",
       "     dimnames(se0)\n",
       "     assayNames(se0)\n",
       "     head(assay(se0))\n",
       "     assays(se0) <- endoapply(assays(se0), asinh)\n",
       "     head(assay(se0))\n",
       "     \n",
       "     rowData(se0)\n",
       "     colData(se0)\n",
       "     \n",
       "     se0[, se0$Treatment == \"ChIP\"]\n",
       "     subset(se0, select = Treatment == \"ChIP\")\n",
       "     \n",
       "     ## cbind() combines objects with the same features of interest\n",
       "     ## but different samples:\n",
       "     se1 <- se0\n",
       "     se2 <- se1[,1:3]\n",
       "     colnames(se2) <- letters[seq_len(ncol(se2))]\n",
       "     cmb1 <- cbind(se1, se2)\n",
       "     dim(cmb1)\n",
       "     dimnames(cmb1)\n",
       "     \n",
       "     ## rbind() combines objects with the same samples but different\n",
       "     ## features of interest:\n",
       "     se1 <- se0\n",
       "     se2 <- se1[1:50,]\n",
       "     rownames(se2) <- letters[seq_len(nrow(se2))]\n",
       "     cmb2 <- rbind(se1, se2)\n",
       "     dim(cmb2)\n",
       "     dimnames(cmb2)\n",
       "     \n",
       "     ## ---------------------------------------------------------------------\n",
       "     ## ON-DISK REALIZATION\n",
       "     ## ---------------------------------------------------------------------\n",
       "     library(DelayedArray)\n",
       "     setAutoRealizationBackend(\"HDF5Array\")\n",
       "     cmb3 <- realize(cmb2)\n",
       "     assay(cmb3, withDimnames=FALSE)  # an HDF5Matrix object\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?assay"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "alive-rwanda",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T18:34:18.021504Z",
     "iopub.status.busy": "2023-01-31T18:34:18.020152Z",
     "iopub.status.idle": "2023-01-31T18:34:18.033745Z",
     "shell.execute_reply": "2023-01-31T18:34:18.032498Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "$RNA\n",
       "Assay data with 46794 features for 506107 cells\n",
       "First 10 features:\n",
       " A1BG, A1BG-AS1, A1CF, A2M, A2M-AS1, A2ML1, A2ML1-AS1, A2ML1-AS2,\n",
       "A3GALT2, A4GALT \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pooled_ad@assays"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "essential-parcel",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T18:43:27.560872Z",
     "iopub.status.busy": "2023-01-31T18:43:27.559562Z",
     "iopub.status.idle": "2023-01-31T18:43:27.572919Z",
     "shell.execute_reply": "2023-01-31T18:43:27.571595Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'TCTCACGAGACTTGTC-1-9-organoid'</li><li>'AGGGTGAGTTCCATGA-1-11-organoid'</li><li>'CAACTAGCATCGGTTA-1-11-organoid'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'TCTCACGAGACTTGTC-1-9-organoid'\n",
       "\\item 'AGGGTGAGTTCCATGA-1-11-organoid'\n",
       "\\item 'CAACTAGCATCGGTTA-1-11-organoid'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'TCTCACGAGACTTGTC-1-9-organoid'\n",
       "2. 'AGGGTGAGTTCCATGA-1-11-organoid'\n",
       "3. 'CAACTAGCATCGGTTA-1-11-organoid'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"TCTCACGAGACTTGTC-1-9-organoid\"  \"AGGGTGAGTTCCATGA-1-11-organoid\"\n",
       "[3] \"CAACTAGCATCGGTTA-1-11-organoid\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cells_1 %>% head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "fossil-updating",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-31T18:41:37.145812Z",
     "iopub.status.busy": "2023-01-31T18:41:37.144453Z",
     "iopub.status.idle": "2023-01-31T18:41:37.159085Z",
     "shell.execute_reply": "2023-01-31T18:41:37.157557Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in pooled_ad@assays[rownames(pooled_ad@assays) %in% c(cells_1, cells_2), : incorrect number of dimensions\n",
     "output_type": "error",
     "traceback": [
      "Error in pooled_ad@assays[rownames(pooled_ad@assays) %in% c(cells_1, cells_2), : incorrect number of dimensions\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "# Convert the Seurat object to a matrix for input to MAST\n",
    "mat <- pooled_ad@assays[rownames(pooled_ad@assays) %in% c(cells_1, cells_2), ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "collect-currency",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-31T18:31:56.344745Z",
     "iopub.status.busy": "2023-01-31T18:31:56.343311Z",
     "iopub.status.idle": "2023-01-31T18:31:56.361059Z",
     "shell.execute_reply": "2023-01-31T18:31:56.359421Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in (function (classes, fdef, mtable) : unable to find an inherited method for function ‘colData’ for signature ‘\"Seurat\"’\n",
     "output_type": "error",
     "traceback": [
      "Error in (function (classes, fdef, mtable) : unable to find an inherited method for function ‘colData’ for signature ‘\"Seurat\"’\nTraceback:\n",
      "1. model.matrix(~0 + condition, data = colData(pooled_ad))",
      "2. model.matrix.default(~0 + condition, data = colData(pooled_ad))",
      "3. terms(object, data = data)",
      "4. terms.formula(object, data = data)",
      "5. colData(pooled_ad)",
      "6. (function (classes, fdef, mtable) \n . {\n .     methods <- .findInheritedMethods(classes, fdef, mtable)\n .     if (length(methods) == 1L) \n .         return(methods[[1L]])\n .     else if (length(methods) == 0L) {\n .         cnames <- paste0(\"\\\"\", vapply(classes, as.character, \n .             \"\"), \"\\\"\", collapse = \", \")\n .         stop(gettextf(\"unable to find an inherited method for function %s for signature %s\", \n .             sQuote(fdef@generic), sQuote(cnames)), domain = NA)\n .     }\n .     else stop(\"Internal error in finding inherited methods; didn't return a unique method\", \n .         domain = NA)\n . })(list(structure(\"Seurat\", package = \"SeuratObject\")), new(\"standardGeneric\", \n .     .Data = function (x, ...) \n .     standardGeneric(\"colData\"), generic = structure(\"colData\", package = \"SummarizedExperiment\"), \n .     package = \"SummarizedExperiment\", group = list(), valueClass = character(0), \n .     signature = \"x\", default = NULL, skeleton = (function (x, \n .         ...) \n .     stop(\"invalid call in method dispatch to 'colData' (no default method)\", \n .         domain = NA))(x, ...)), <environment>)",
      "7. stop(gettextf(\"unable to find an inherited method for function %s for signature %s\", \n .     sQuote(fdef@generic), sQuote(cnames)), domain = NA)"
     ]
    }
   ],
   "source": [
    "# Specify the model for MAST\n",
    "mod <- model.matrix(~0 + condition, data = colData(pooled_ad@assays))\n",
    "colnames(mod) <- levels(colData(pooled_ad)$condition)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "handmade-france",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run MAST\n",
    "fit <- mast(mat, mod)\n",
    "\n",
    "# Results interpretation: Extract the differentially expressed genes\n",
    "res <- tidy(fit)\n",
    "res_sig <- res[res$qval < 0.05, ]\n",
    "\n",
    "# Print the top 10 differentially expressed genes\n",
    "head(res_sig[order(res_sig$logFC), ], 10)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mast",
   "language": "R",
   "name": "mast"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
