{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Perform NicheNet analysis starting from a Seurat object: step-by-step analysis \n",
    "____________________________________\n",
    "Robin Browaeys \n",
    "2019-11-12 \n",
    " \n",
    "<!-- github markdown built using  \n",
    "rmarkdown::render(\"vignettes/seurat_steps.Rmd\", output_format = \"github_document\") \n",
    "--> \n",
    " \n",
    "In this vignette, you can learn how to perform a basic NicheNet analysis \n",
    "on a Seurat v3 object. Such a NicheNet analysis can help you to generate \n",
    "hypotheses about an intercellular communication process of interest for \n",
    "which you have single-cell gene expression data as a Seurat object. \n",
    "Specifically, NicheNet can predict 1) which ligands from one or more \n",
    "cell population(s) (“sender/niche”) are most likely to affect target \n",
    "gene expression in an interacting cell population (“receiver/target”) \n",
    "and 2) which specific target genes are affected by which of these \n",
    "predicted ligands. \n",
    " \n",
    "Because NicheNet studies how ligands affect gene expression in \n",
    "putatively neighboring/interacting cells, you need to have data about \n",
    "this effect in gene expression you want to study. So, there need to be \n",
    "‘some kind of’ differential expression in a receiver cell population, \n",
    "caused by ligands from one of more interacting sender cell populations. \n",
    " \n",
    "In this vignette, we demonstrate the use of NicheNet on a Seurat Object. \n",
    "The steps of the analysis we show here are also discussed in detail in \n",
    "the main, basis, NicheNet vignette [NicheNet’s ligand activity analysis \n",
    "on a gene set of interest: predict active ligands and their target \n",
    "genes](ligand_activity_geneset.md):`vignette(\"ligand_activity_geneset\", package=\"nichenetr\")`. \n",
    "Make sure you understand the different steps in a NicheNet analysis that \n",
    "are described in that vignette before proceeding with this vignette and \n",
    "performing a real NicheNet analysis on your data. This vignette \n",
    "describes the different steps behind the wrapper functions that are \n",
    "shown in [Perform NicheNet analysis starting from a Seurat \n",
    "object](seurat_wrapper.md):`vignette(\"seurat_wrapper\", package=\"nichenetr\")`. \n",
    "Following this vignette has the advantage that it allows users to adapt \n",
    "specific steps of the pipeline to make them more appropriate for their \n",
    "data. \n",
    " \n",
    "As example expression data of interacting cells, we will use mouse \n",
    "NICHE-seq data from Medaglia et al. to explore intercellular \n",
    "communication in the T cell area in the inguinal lymph node before and \n",
    "72 hours after lymphocytic choriomeningitis virus (LCMV) infection \n",
    "(Medaglia et al. 2017). We will NicheNet to explore immune cell \n",
    "crosstalk in response to this LCMV infection. \n",
    " \n",
    "In this dataset, differential expression is observed between CD8 T cells \n",
    "in steady-state and CD8 T cells after LCMV infection. NicheNet can be \n",
    "applied to look at how several immune cell populations in the lymph node \n",
    "(i.e., monocytes, dendritic cells, NK cells, B cells, CD4 T cells) can \n",
    "regulate and induce these observed gene expression changes. NicheNet \n",
    "will specifically prioritize ligands from these immune cells and their \n",
    "target genes that change in expression upon LCMV infection. \n",
    " \n",
    "The used NicheNet networks, ligand-target matrix and example expression \n",
    "data of interacting cells can be downloaded from Zenodo. The NicheNet \n",
    "networks and ligand-target matrix at \n",
    "[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.3260758.svg)](https://doi.org/10.5281/zenodo.3260758) \n",
    "and the Seurat object of the processed NICHE-seq single-cell data at \n",
    "[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.3531889.svg)](https://doi.org/10.5281/zenodo.3531889). \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare NicheNet analysis \n",
    " \n",
    "## Load required packages, read in the Seurat object with processed expression data of interacting cells and NicheNet’s ligand-target prior model, ligand-receptor network and weighted integrated networks. \n",
    " \n",
    "The NicheNet ligand-receptor network and weighted networks are necessary \n",
    "to define and show possible ligand-receptor interactions between two \n",
    "cell populations. The ligand-target matrix denotes the prior potential \n",
    "that particular ligands might regulate the expression of particular \n",
    "target genes. This matrix is necessary to prioritize possible \n",
    "ligand-receptor interactions based on observed gene expression effects \n",
    "(i.e. NicheNet’s ligand activity analysis) and infer affected target \n",
    "genes of these prioritized ligands. \n",
    " \n",
    "### Load Packages and set data paths: \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-19T17:43:00.281937Z",
     "iopub.status.busy": "2023-01-19T17:43:00.278378Z",
     "iopub.status.idle": "2023-01-19T17:43:08.642212Z",
     "shell.execute_reply": "2023-01-19T17:43:08.640535Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 method overwritten by 'SeuratDisk':\n",
      "  method            from  \n",
      "  as.sparse.H5Group Seurat\n",
      "\n",
      "Warning message in system(\"timedatectl\", intern = TRUE):\n",
      "“running command 'timedatectl' had status 1”\n",
      "Attaching SeuratObject\n",
      "\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.2 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.4.0      \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 1.0.1 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.8      \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.10\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.2.1      \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.5.0 \n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.3      \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.2 \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n"
     ]
    }
   ],
   "source": [
    "library(SeuratDisk)\n",
    "library(nichenetr)\n",
    "library(Seurat) # please update to Seurat V4 \n",
    "library(tidyverse) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-19T17:47:17.912688Z",
     "iopub.status.busy": "2023-01-19T17:47:17.911025Z",
     "iopub.status.idle": "2023-01-19T17:47:17.926956Z",
     "shell.execute_reply": "2023-01-19T17:47:17.925190Z"
    }
   },
   "outputs": [],
   "source": [
    "data_root <- '/lustre/scratch126/cellgen/team205/nh3/skin'\n",
    "proj_root <- '~/FCA_Fetal_Skin_priv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "If you would use and load other packages, we recommend to load these 3 \n",
    "packages after the others. \n",
    " \n",
    "### Read in the expression data of interacting cells: \n",
    " \n",
    "The dataset used here is publicly available single-cell data from immune \n",
    "cells in the T cell area of the inguinal lymph node. The data was \n",
    "processed and aggregated by applying the Seurat alignment pipeline. The \n",
    "Seurat object contains this aggregated data. Note that this should be a \n",
    "Seurat v3 object and that gene should be named by their official \n",
    "mouse/human gene symbol. \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-19T17:47:22.728408Z",
     "iopub.status.busy": "2023-01-19T17:47:22.726539Z",
     "iopub.status.idle": "2023-01-19T17:56:01.832247Z",
     "shell.execute_reply": "2023-01-19T17:56:01.830551Z"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Validating h5Seurat file\n",
      "\n",
      "Warning message:\n",
      "“Feature names cannot have underscores ('_'), replacing with dashes ('-')”\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding feature-level metadata for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>dataset</th><th scope=col>donor</th><th scope=col>pcw</th><th scope=col>day</th><th scope=col>sample_id</th><th scope=col>fetal_annot</th><th scope=col>organoid_annot</th><th scope=col>pooled_annot</th><th scope=col>chemistry</th><th scope=col>gender</th><th scope=col>sorting</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal</th><td>fetal</td><td>F16</td><td>8</td><td>NaN</td><td>4834STDY7002879</td><td>LYVE1++ macrophage                  </td><td>nan</td><td>fetal_LYVE1++ macrophage                  </td><td>SC3Pv2</td><td>male</td><td>CD45P</td></tr>\n",
       "\t<tr><th scope=row>AAAGATGGTCGATTGT-1-4834STDY7002879-fetal</th><td>fetal</td><td>F16</td><td>8</td><td>NaN</td><td>4834STDY7002879</td><td>Monocyte (activated/differentiating)</td><td>nan</td><td>fetal_Monocyte (activated/differentiating)</td><td>SC3Pv2</td><td>male</td><td>CD45P</td></tr>\n",
       "\t<tr><th scope=row>AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal</th><td>fetal</td><td>F16</td><td>8</td><td>NaN</td><td>4834STDY7002879</td><td>LYVE1++ macrophage                  </td><td>nan</td><td>fetal_LYVE1++ macrophage                  </td><td>SC3Pv2</td><td>male</td><td>CD45P</td></tr>\n",
       "\t<tr><th scope=row>AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal</th><td>fetal</td><td>F16</td><td>8</td><td>NaN</td><td>4834STDY7002879</td><td>Eo/baso/mast cell progenitor        </td><td>nan</td><td>fetal_Eo/baso/mast cell progenitor        </td><td>SC3Pv2</td><td>male</td><td>CD45P</td></tr>\n",
       "\t<tr><th scope=row>AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal</th><td>fetal</td><td>F16</td><td>8</td><td>NaN</td><td>4834STDY7002879</td><td>LYVE1++ macrophage                  </td><td>nan</td><td>fetal_LYVE1++ macrophage                  </td><td>SC3Pv2</td><td>male</td><td>CD45P</td></tr>\n",
       "\t<tr><th scope=row>AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal</th><td>fetal</td><td>F16</td><td>8</td><td>NaN</td><td>4834STDY7002879</td><td>LYVE1++ macrophage                  </td><td>nan</td><td>fetal_LYVE1++ macrophage                  </td><td>SC3Pv2</td><td>male</td><td>CD45P</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & dataset & donor & pcw & day & sample\\_id & fetal\\_annot & organoid\\_annot & pooled\\_annot & chemistry & gender & sorting\\\\\n",
       "  & <fct> & <fct> & <int> & <dbl> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tAAACCTGGTCAGTGGA-1-4834STDY7002879-fetal & fetal & F16 & 8 & NaN & 4834STDY7002879 & LYVE1++ macrophage                   & nan & fetal\\_LYVE1++ macrophage                   & SC3Pv2 & male & CD45P\\\\\n",
       "\tAAAGATGGTCGATTGT-1-4834STDY7002879-fetal & fetal & F16 & 8 & NaN & 4834STDY7002879 & Monocyte (activated/differentiating) & nan & fetal\\_Monocyte (activated/differentiating) & SC3Pv2 & male & CD45P\\\\\n",
       "\tAAAGCAAAGATGTGGC-1-4834STDY7002879-fetal & fetal & F16 & 8 & NaN & 4834STDY7002879 & LYVE1++ macrophage                   & nan & fetal\\_LYVE1++ macrophage                   & SC3Pv2 & male & CD45P\\\\\n",
       "\tAAAGTAGCAGATCGGA-1-4834STDY7002879-fetal & fetal & F16 & 8 & NaN & 4834STDY7002879 & Eo/baso/mast cell progenitor         & nan & fetal\\_Eo/baso/mast cell progenitor         & SC3Pv2 & male & CD45P\\\\\n",
       "\tAAAGTAGTCCGCATCT-1-4834STDY7002879-fetal & fetal & F16 & 8 & NaN & 4834STDY7002879 & LYVE1++ macrophage                   & nan & fetal\\_LYVE1++ macrophage                   & SC3Pv2 & male & CD45P\\\\\n",
       "\tAAATGCCAGCTGCCCA-1-4834STDY7002879-fetal & fetal & F16 & 8 & NaN & 4834STDY7002879 & LYVE1++ macrophage                   & nan & fetal\\_LYVE1++ macrophage                   & SC3Pv2 & male & CD45P\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | dataset &lt;fct&gt; | donor &lt;fct&gt; | pcw &lt;int&gt; | day &lt;dbl&gt; | sample_id &lt;fct&gt; | fetal_annot &lt;fct&gt; | organoid_annot &lt;fct&gt; | pooled_annot &lt;fct&gt; | chemistry &lt;fct&gt; | gender &lt;fct&gt; | sorting &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal | fetal | F16 | 8 | NaN | 4834STDY7002879 | LYVE1++ macrophage                   | nan | fetal_LYVE1++ macrophage                   | SC3Pv2 | male | CD45P |\n",
       "| AAAGATGGTCGATTGT-1-4834STDY7002879-fetal | fetal | F16 | 8 | NaN | 4834STDY7002879 | Monocyte (activated/differentiating) | nan | fetal_Monocyte (activated/differentiating) | SC3Pv2 | male | CD45P |\n",
       "| AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal | fetal | F16 | 8 | NaN | 4834STDY7002879 | LYVE1++ macrophage                   | nan | fetal_LYVE1++ macrophage                   | SC3Pv2 | male | CD45P |\n",
       "| AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal | fetal | F16 | 8 | NaN | 4834STDY7002879 | Eo/baso/mast cell progenitor         | nan | fetal_Eo/baso/mast cell progenitor         | SC3Pv2 | male | CD45P |\n",
       "| AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal | fetal | F16 | 8 | NaN | 4834STDY7002879 | LYVE1++ macrophage                   | nan | fetal_LYVE1++ macrophage                   | SC3Pv2 | male | CD45P |\n",
       "| AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal | fetal | F16 | 8 | NaN | 4834STDY7002879 | LYVE1++ macrophage                   | nan | fetal_LYVE1++ macrophage                   | SC3Pv2 | male | CD45P |\n",
       "\n"
      ],
      "text/plain": [
       "                                         dataset donor pcw day sample_id      \n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal fetal   F16   8   NaN 4834STDY7002879\n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal fetal   F16   8   NaN 4834STDY7002879\n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal fetal   F16   8   NaN 4834STDY7002879\n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal fetal   F16   8   NaN 4834STDY7002879\n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal fetal   F16   8   NaN 4834STDY7002879\n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal fetal   F16   8   NaN 4834STDY7002879\n",
       "                                         fetal_annot                         \n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal LYVE1++ macrophage                  \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal Monocyte (activated/differentiating)\n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal LYVE1++ macrophage                  \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal Eo/baso/mast cell progenitor        \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal LYVE1++ macrophage                  \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal LYVE1++ macrophage                  \n",
       "                                         organoid_annot\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal nan           \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal nan           \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal nan           \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal nan           \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal nan           \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal nan           \n",
       "                                         pooled_annot                              \n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal fetal_LYVE1++ macrophage                  \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal fetal_Monocyte (activated/differentiating)\n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal fetal_LYVE1++ macrophage                  \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal fetal_Eo/baso/mast cell progenitor        \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal fetal_LYVE1++ macrophage                  \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal fetal_LYVE1++ macrophage                  \n",
       "                                         chemistry gender sorting\n",
       "AAACCTGGTCAGTGGA-1-4834STDY7002879-fetal SC3Pv2    male   CD45P  \n",
       "AAAGATGGTCGATTGT-1-4834STDY7002879-fetal SC3Pv2    male   CD45P  \n",
       "AAAGCAAAGATGTGGC-1-4834STDY7002879-fetal SC3Pv2    male   CD45P  \n",
       "AAAGTAGCAGATCGGA-1-4834STDY7002879-fetal SC3Pv2    male   CD45P  \n",
       "AAAGTAGTCCGCATCT-1-4834STDY7002879-fetal SC3Pv2    male   CD45P  \n",
       "AAATGCCAGCTGCCCA-1-4834STDY7002879-fetal SC3Pv2    male   CD45P  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "seuratObj = LoadH5Seurat(file.path(proj_root, \"data\", \"pooled_fetal_organoid.count_with_annot.h5seurat\"))\n",
    "seuratObj@meta.data %>% head() \n",
    "##         nGene nUMI orig.ident aggregate res.0.6 celltype nCount_RNA nFeature_RNA \n",
    "## W380370   880 1611      LN_SS        SS       1    CD8 T       1607          876 \n",
    "## W380372   541  891      LN_SS        SS       0    CD4 T        885          536 \n",
    "## W380374   742 1229      LN_SS        SS       0    CD4 T       1223          737 \n",
    "## W380378   847 1546      LN_SS        SS       1    CD8 T       1537          838 \n",
    "## W380379   839 1606      LN_SS        SS       0    CD4 T       1603          836 \n",
    "## W380381   517  844      LN_SS        SS       0    CD4 T        840          513 \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Subsetting to fetal data only."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-19T17:56:01.835832Z",
     "iopub.status.busy": "2023-01-19T17:56:01.834559Z",
     "iopub.status.idle": "2023-01-19T17:56:27.474473Z",
     "shell.execute_reply": "2023-01-19T17:56:27.472850Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "An object of class Seurat \n",
       "46794 features across 186533 samples within 1 assay \n",
       "Active assay: RNA (46794 features, 0 variable features)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "seuratObj = seuratObj[, seuratObj$dataset==\"fetal\"]\n",
    "seuratObj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressWarnings(suppressMessages(seuratObj <- FindVariableFeatures(seuratObj, selection.method = \"vst\",\n",
    "                                                                  nfeatures = 2000, verbose = FALSE, assay = \"RNA\")))\n",
    "top20 <- head(VariableFeatures(seuratObj), 20)\n",
    "\n",
    "LabelPoints(plot = VariableFeaturePlot(alldata), points = top20, repel = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "Visualize which cell populations are present: CD4 T cells (including \n",
    "regulatory T cells), CD8 T cells, B cells, NK cells, dendritic cells \n",
    "(DCs) and inflammatory monocytes \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-01-19T18:00:26.429051Z",
     "iopub.status.busy": "2023-01-19T18:00:26.427344Z",
     "iopub.status.idle": "2023-01-19T18:00:26.447775Z",
     "shell.execute_reply": "2023-01-19T18:00:26.446142Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<pre class=language-r><code>function (object, dims = c(1, 2), cells = NULL, cols = NULL, \n",
       "<span style=white-space:pre-wrap>    pt.size = NULL, reduction = NULL, group.by = NULL, split.by = NULL, </span>\n",
       "<span style=white-space:pre-wrap>    shape.by = NULL, order = NULL, shuffle = FALSE, seed = 1, </span>\n",
       "<span style=white-space:pre-wrap>    label = FALSE, label.size = 4, label.color = \"black\", label.box = FALSE, </span>\n",
       "<span style=white-space:pre-wrap>    repel = FALSE, cells.highlight = NULL, cols.highlight = \"#DE2D26\", </span>\n",
       "<span style=white-space:pre-wrap>    sizes.highlight = 1, na.value = \"grey50\", ncol = NULL, combine = TRUE, </span>\n",
       "<span style=white-space:pre-wrap>    raster = NULL, raster.dpi = c(512, 512)) </span>\n",
       "{\n",
       "<span style=white-space:pre-wrap>    if (length(x = dims) != 2) {</span>\n",
       "<span style=white-space:pre-wrap>        stop(\"'dims' must be a two-length vector\")</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    reduction &lt;- reduction %||% DefaultDimReduc(object = object)</span>\n",
       "<span style=white-space:pre-wrap>    cells &lt;- cells %||% colnames(x = object)</span>\n",
       "<span style=white-space:pre-wrap>    data &lt;- Embeddings(object = object[[reduction]])[cells, dims]</span>\n",
       "<span style=white-space:pre-wrap>    data &lt;- as.data.frame(x = data)</span>\n",
       "<span style=white-space:pre-wrap>    dims &lt;- paste0(Key(object = object[[reduction]]), dims)</span>\n",
       "<span style=white-space:pre-wrap>    object[[\"ident\"]] &lt;- Idents(object = object)</span>\n",
       "<span style=white-space:pre-wrap>    orig.groups &lt;- group.by</span>\n",
       "<span style=white-space:pre-wrap>    group.by &lt;- group.by %||% \"ident\"</span>\n",
       "<span style=white-space:pre-wrap>    data &lt;- cbind(data, object[[group.by]][cells, , drop = FALSE])</span>\n",
       "<span style=white-space:pre-wrap>    group.by &lt;- colnames(x = data)[3:ncol(x = data)]</span>\n",
       "<span style=white-space:pre-wrap>    for (group in group.by) {</span>\n",
       "<span style=white-space:pre-wrap>        if (!is.factor(x = data[, group])) {</span>\n",
       "<span style=white-space:pre-wrap>            data[, group] &lt;- factor(x = data[, group])</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (!is.null(x = shape.by)) {</span>\n",
       "<span style=white-space:pre-wrap>        data[, shape.by] &lt;- object[[shape.by, drop = TRUE]]</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (!is.null(x = split.by)) {</span>\n",
       "<span style=white-space:pre-wrap>        data[, split.by] &lt;- object[[split.by, drop = TRUE]]</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (isTRUE(x = shuffle)) {</span>\n",
       "<span style=white-space:pre-wrap>        set.seed(seed = seed)</span>\n",
       "<span style=white-space:pre-wrap>        data &lt;- data[sample(x = 1:nrow(x = data)), ]</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    plots &lt;- lapply(X = group.by, FUN = function(x) {</span>\n",
       "<span style=white-space:pre-wrap>        plot &lt;- SingleDimPlot(data = data[, c(dims, x, split.by, </span>\n",
       "<span style=white-space:pre-wrap>            shape.by)], dims = dims, col.by = x, cols = cols, </span>\n",
       "<span style=white-space:pre-wrap>            pt.size = pt.size, shape.by = shape.by, order = order, </span>\n",
       "<span style=white-space:pre-wrap>            label = FALSE, cells.highlight = cells.highlight, </span>\n",
       "<span style=white-space:pre-wrap>            cols.highlight = cols.highlight, sizes.highlight = sizes.highlight, </span>\n",
       "<span style=white-space:pre-wrap>            na.value = na.value, raster = raster, raster.dpi = raster.dpi)</span>\n",
       "<span style=white-space:pre-wrap>        if (label) {</span>\n",
       "<span style=white-space:pre-wrap>            plot &lt;- LabelClusters(plot = plot, id = x, repel = repel, </span>\n",
       "<span style=white-space:pre-wrap>                size = label.size, split.by = split.by, box = label.box, </span>\n",
       "<span style=white-space:pre-wrap>                color = label.color)</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        if (!is.null(x = split.by)) {</span>\n",
       "<span style=white-space:pre-wrap>            plot &lt;- plot + FacetTheme() + facet_wrap(facets = vars(!!sym(x = split.by)), </span>\n",
       "<span style=white-space:pre-wrap>                ncol = if (length(x = group.by) &gt; 1 || is.null(x = ncol)) {</span>\n",
       "<span style=white-space:pre-wrap>                  length(x = unique(x = data[, split.by]))</span>\n",
       "<span style=white-space:pre-wrap>                }</span>\n",
       "<span style=white-space:pre-wrap>                else {</span>\n",
       "<span style=white-space:pre-wrap>                  ncol</span>\n",
       "<span style=white-space:pre-wrap>                })</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        plot &lt;- if (is.null(x = orig.groups)) {</span>\n",
       "<span style=white-space:pre-wrap>            plot + labs(title = NULL)</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>        else {</span>\n",
       "<span style=white-space:pre-wrap>            plot + CenterTitle()</span>\n",
       "<span style=white-space:pre-wrap>        }</span>\n",
       "<span style=white-space:pre-wrap>    })</span>\n",
       "<span style=white-space:pre-wrap>    if (!is.null(x = split.by)) {</span>\n",
       "<span style=white-space:pre-wrap>        ncol &lt;- 1</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    if (combine) {</span>\n",
       "<span style=white-space:pre-wrap>        plots &lt;- wrap_plots(plots, ncol = orig.groups %iff% ncol)</span>\n",
       "<span style=white-space:pre-wrap>    }</span>\n",
       "<span style=white-space:pre-wrap>    return(plots)</span>\n",
       "}</code></pre>"
      ],
      "text/latex": [
       "\\begin{minted}{r}\n",
       "function (object, dims = c(1, 2), cells = NULL, cols = NULL, \n",
       "    pt.size = NULL, reduction = NULL, group.by = NULL, split.by = NULL, \n",
       "    shape.by = NULL, order = NULL, shuffle = FALSE, seed = 1, \n",
       "    label = FALSE, label.size = 4, label.color = \"black\", label.box = FALSE, \n",
       "    repel = FALSE, cells.highlight = NULL, cols.highlight = \"\\#DE2D26\", \n",
       "    sizes.highlight = 1, na.value = \"grey50\", ncol = NULL, combine = TRUE, \n",
       "    raster = NULL, raster.dpi = c(512, 512)) \n",
       "\\{\n",
       "    if (length(x = dims) != 2) \\{\n",
       "        stop(\"'dims' must be a two-length vector\")\n",
       "    \\}\n",
       "    reduction <- reduction \\%\\textbar{}\\textbar{}\\% DefaultDimReduc(object = object)\n",
       "    cells <- cells \\%\\textbar{}\\textbar{}\\% colnames(x = object)\n",
       "    data <- Embeddings(object = object{[}{[}reduction{]}{]}){[}cells, dims{]}\n",
       "    data <- as.data.frame(x = data)\n",
       "    dims <- paste0(Key(object = object{[}{[}reduction{]}{]}), dims)\n",
       "    object{[}{[}\"ident\"{]}{]} <- Idents(object = object)\n",
       "    orig.groups <- group.by\n",
       "    group.by <- group.by \\%\\textbar{}\\textbar{}\\% \"ident\"\n",
       "    data <- cbind(data, object{[}{[}group.by{]}{]}{[}cells, , drop = FALSE{]})\n",
       "    group.by <- colnames(x = data){[}3:ncol(x = data){]}\n",
       "    for (group in group.by) \\{\n",
       "        if (!is.factor(x = data{[}, group{]})) \\{\n",
       "            data{[}, group{]} <- factor(x = data{[}, group{]})\n",
       "        \\}\n",
       "    \\}\n",
       "    if (!is.null(x = shape.by)) \\{\n",
       "        data{[}, shape.by{]} <- object{[}{[}shape.by, drop = TRUE{]}{]}\n",
       "    \\}\n",
       "    if (!is.null(x = split.by)) \\{\n",
       "        data{[}, split.by{]} <- object{[}{[}split.by, drop = TRUE{]}{]}\n",
       "    \\}\n",
       "    if (isTRUE(x = shuffle)) \\{\n",
       "        set.seed(seed = seed)\n",
       "        data <- data{[}sample(x = 1:nrow(x = data)), {]}\n",
       "    \\}\n",
       "    plots <- lapply(X = group.by, FUN = function(x) \\{\n",
       "        plot <- SingleDimPlot(data = data{[}, c(dims, x, split.by, \n",
       "            shape.by){]}, dims = dims, col.by = x, cols = cols, \n",
       "            pt.size = pt.size, shape.by = shape.by, order = order, \n",
       "            label = FALSE, cells.highlight = cells.highlight, \n",
       "            cols.highlight = cols.highlight, sizes.highlight = sizes.highlight, \n",
       "            na.value = na.value, raster = raster, raster.dpi = raster.dpi)\n",
       "        if (label) \\{\n",
       "            plot <- LabelClusters(plot = plot, id = x, repel = repel, \n",
       "                size = label.size, split.by = split.by, box = label.box, \n",
       "                color = label.color)\n",
       "        \\}\n",
       "        if (!is.null(x = split.by)) \\{\n",
       "            plot <- plot + FacetTheme() + facet\\_wrap(facets = vars(!!sym(x = split.by)), \n",
       "                ncol = if (length(x = group.by) > 1 \\textbar{}\\textbar{} is.null(x = ncol)) \\{\n",
       "                  length(x = unique(x = data{[}, split.by{]}))\n",
       "                \\}\n",
       "                else \\{\n",
       "                  ncol\n",
       "                \\})\n",
       "        \\}\n",
       "        plot <- if (is.null(x = orig.groups)) \\{\n",
       "            plot + labs(title = NULL)\n",
       "        \\}\n",
       "        else \\{\n",
       "            plot + CenterTitle()\n",
       "        \\}\n",
       "    \\})\n",
       "    if (!is.null(x = split.by)) \\{\n",
       "        ncol <- 1\n",
       "    \\}\n",
       "    if (combine) \\{\n",
       "        plots <- wrap\\_plots(plots, ncol = orig.groups \\%iff\\% ncol)\n",
       "    \\}\n",
       "    return(plots)\n",
       "\\}\n",
       "\\end{minted}"
      ],
      "text/markdown": [
       "```r\n",
       "function (object, dims = c(1, 2), cells = NULL, cols = NULL, \n",
       "    pt.size = NULL, reduction = NULL, group.by = NULL, split.by = NULL, \n",
       "    shape.by = NULL, order = NULL, shuffle = FALSE, seed = 1, \n",
       "    label = FALSE, label.size = 4, label.color = \"black\", label.box = FALSE, \n",
       "    repel = FALSE, cells.highlight = NULL, cols.highlight = \"#DE2D26\", \n",
       "    sizes.highlight = 1, na.value = \"grey50\", ncol = NULL, combine = TRUE, \n",
       "    raster = NULL, raster.dpi = c(512, 512)) \n",
       "{\n",
       "    if (length(x = dims) != 2) {\n",
       "        stop(\"'dims' must be a two-length vector\")\n",
       "    }\n",
       "    reduction <- reduction %||% DefaultDimReduc(object = object)\n",
       "    cells <- cells %||% colnames(x = object)\n",
       "    data <- Embeddings(object = object[[reduction]])[cells, dims]\n",
       "    data <- as.data.frame(x = data)\n",
       "    dims <- paste0(Key(object = object[[reduction]]), dims)\n",
       "    object[[\"ident\"]] <- Idents(object = object)\n",
       "    orig.groups <- group.by\n",
       "    group.by <- group.by %||% \"ident\"\n",
       "    data <- cbind(data, object[[group.by]][cells, , drop = FALSE])\n",
       "    group.by <- colnames(x = data)[3:ncol(x = data)]\n",
       "    for (group in group.by) {\n",
       "        if (!is.factor(x = data[, group])) {\n",
       "            data[, group] <- factor(x = data[, group])\n",
       "        }\n",
       "    }\n",
       "    if (!is.null(x = shape.by)) {\n",
       "        data[, shape.by] <- object[[shape.by, drop = TRUE]]\n",
       "    }\n",
       "    if (!is.null(x = split.by)) {\n",
       "        data[, split.by] <- object[[split.by, drop = TRUE]]\n",
       "    }\n",
       "    if (isTRUE(x = shuffle)) {\n",
       "        set.seed(seed = seed)\n",
       "        data <- data[sample(x = 1:nrow(x = data)), ]\n",
       "    }\n",
       "    plots <- lapply(X = group.by, FUN = function(x) {\n",
       "        plot <- SingleDimPlot(data = data[, c(dims, x, split.by, \n",
       "            shape.by)], dims = dims, col.by = x, cols = cols, \n",
       "            pt.size = pt.size, shape.by = shape.by, order = order, \n",
       "            label = FALSE, cells.highlight = cells.highlight, \n",
       "            cols.highlight = cols.highlight, sizes.highlight = sizes.highlight, \n",
       "            na.value = na.value, raster = raster, raster.dpi = raster.dpi)\n",
       "        if (label) {\n",
       "            plot <- LabelClusters(plot = plot, id = x, repel = repel, \n",
       "                size = label.size, split.by = split.by, box = label.box, \n",
       "                color = label.color)\n",
       "        }\n",
       "        if (!is.null(x = split.by)) {\n",
       "            plot <- plot + FacetTheme() + facet_wrap(facets = vars(!!sym(x = split.by)), \n",
       "                ncol = if (length(x = group.by) > 1 || is.null(x = ncol)) {\n",
       "                  length(x = unique(x = data[, split.by]))\n",
       "                }\n",
       "                else {\n",
       "                  ncol\n",
       "                })\n",
       "        }\n",
       "        plot <- if (is.null(x = orig.groups)) {\n",
       "            plot + labs(title = NULL)\n",
       "        }\n",
       "        else {\n",
       "            plot + CenterTitle()\n",
       "        }\n",
       "    })\n",
       "    if (!is.null(x = split.by)) {\n",
       "        ncol <- 1\n",
       "    }\n",
       "    if (combine) {\n",
       "        plots <- wrap_plots(plots, ncol = orig.groups %iff% ncol)\n",
       "    }\n",
       "    return(plots)\n",
       "}\n",
       "```"
      ],
      "text/plain": [
       "function (object, dims = c(1, 2), cells = NULL, cols = NULL, \n",
       "    pt.size = NULL, reduction = NULL, group.by = NULL, split.by = NULL, \n",
       "    shape.by = NULL, order = NULL, shuffle = FALSE, seed = 1, \n",
       "    label = FALSE, label.size = 4, label.color = \"black\", label.box = FALSE, \n",
       "    repel = FALSE, cells.highlight = NULL, cols.highlight = \"#DE2D26\", \n",
       "    sizes.highlight = 1, na.value = \"grey50\", ncol = NULL, combine = TRUE, \n",
       "    raster = NULL, raster.dpi = c(512, 512)) \n",
       "{\n",
       "    if (length(x = dims) != 2) {\n",
       "        stop(\"'dims' must be a two-length vector\")\n",
       "    }\n",
       "    reduction <- reduction %||% DefaultDimReduc(object = object)\n",
       "    cells <- cells %||% colnames(x = object)\n",
       "    data <- Embeddings(object = object[[reduction]])[cells, dims]\n",
       "    data <- as.data.frame(x = data)\n",
       "    dims <- paste0(Key(object = object[[reduction]]), dims)\n",
       "    object[[\"ident\"]] <- Idents(object = object)\n",
       "    orig.groups <- group.by\n",
       "    group.by <- group.by %||% \"ident\"\n",
       "    data <- cbind(data, object[[group.by]][cells, , drop = FALSE])\n",
       "    group.by <- colnames(x = data)[3:ncol(x = data)]\n",
       "    for (group in group.by) {\n",
       "        if (!is.factor(x = data[, group])) {\n",
       "            data[, group] <- factor(x = data[, group])\n",
       "        }\n",
       "    }\n",
       "    if (!is.null(x = shape.by)) {\n",
       "        data[, shape.by] <- object[[shape.by, drop = TRUE]]\n",
       "    }\n",
       "    if (!is.null(x = split.by)) {\n",
       "        data[, split.by] <- object[[split.by, drop = TRUE]]\n",
       "    }\n",
       "    if (isTRUE(x = shuffle)) {\n",
       "        set.seed(seed = seed)\n",
       "        data <- data[sample(x = 1:nrow(x = data)), ]\n",
       "    }\n",
       "    plots <- lapply(X = group.by, FUN = function(x) {\n",
       "        plot <- SingleDimPlot(data = data[, c(dims, x, split.by, \n",
       "            shape.by)], dims = dims, col.by = x, cols = cols, \n",
       "            pt.size = pt.size, shape.by = shape.by, order = order, \n",
       "            label = FALSE, cells.highlight = cells.highlight, \n",
       "            cols.highlight = cols.highlight, sizes.highlight = sizes.highlight, \n",
       "            na.value = na.value, raster = raster, raster.dpi = raster.dpi)\n",
       "        if (label) {\n",
       "            plot <- LabelClusters(plot = plot, id = x, repel = repel, \n",
       "                size = label.size, split.by = split.by, box = label.box, \n",
       "                color = label.color)\n",
       "        }\n",
       "        if (!is.null(x = split.by)) {\n",
       "            plot <- plot + FacetTheme() + facet_wrap(facets = vars(!!sym(x = split.by)), \n",
       "                ncol = if (length(x = group.by) > 1 || is.null(x = ncol)) {\n",
       "                  length(x = unique(x = data[, split.by]))\n",
       "                }\n",
       "                else {\n",
       "                  ncol\n",
       "                })\n",
       "        }\n",
       "        plot <- if (is.null(x = orig.groups)) {\n",
       "            plot + labs(title = NULL)\n",
       "        }\n",
       "        else {\n",
       "            plot + CenterTitle()\n",
       "        }\n",
       "    })\n",
       "    if (!is.null(x = split.by)) {\n",
       "        ncol <- 1\n",
       "    }\n",
       "    if (combine) {\n",
       "        plots <- wrap_plots(plots, ncol = orig.groups %iff% ncol)\n",
       "    }\n",
       "    return(plots)\n",
       "}\n",
       "<bytecode: 0x55ddcab632c8>\n",
       "<environment: namespace:Seurat>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "DimPlot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-01-19T17:58:40.706009Z",
     "iopub.status.busy": "2023-01-19T17:58:40.704375Z",
     "iopub.status.idle": "2023-01-19T17:58:40.739117Z",
     "shell.execute_reply": "2023-01-19T17:58:40.737411Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".\n",
       "                                    fetal_ASDC \n",
       "                                            90 \n",
       "                              fetal_Adipocytes \n",
       "                                          3203 \n",
       "                                fetal_Arterial \n",
       "                                           172 \n",
       "                                  fetal_B cell \n",
       "                                          1982 \n",
       "                                   fetal_Basal \n",
       "                                           815 \n",
       "                              fetal_CD4 T cell \n",
       "                                          4675 \n",
       "                              fetal_CD8 T cell \n",
       "                                          1519 \n",
       "                 fetal_Capillary (venular tip) \n",
       "                                           487 \n",
       "          fetal_Capillary/postcapillary venule \n",
       "                                           609 \n",
       "                         fetal_Companion layer \n",
       "                                            27 \n",
       "                          fetal_Cuticle/cortex \n",
       "                                            21 \n",
       "                                     fetal_DC1 \n",
       "                                           748 \n",
       "                                     fetal_DC2 \n",
       "                                          2529 \n",
       "                       fetal_Dermal condensate \n",
       "                                          5013 \n",
       "                         fetal_Dermal papillia \n",
       "                                          5637 \n",
       "                                fetal_Early LE \n",
       "                                            65 \n",
       "                  fetal_Early endothelial cell \n",
       "                                           141 \n",
       "                         fetal_Early erythroid \n",
       "                                           427 \n",
       "             fetal_Early erythroid (embryonic) \n",
       "                                           420 \n",
       "                          fetal_Early myocytes \n",
       "                                           585 \n",
       "            fetal_Eo/baso/mast cell progenitor \n",
       "                                           138 \n",
       "                   fetal_Erythroid (embryonic) \n",
       "                                           682 \n",
       "                       fetal_Erythroid (fetal) \n",
       "                                          1939 \n",
       "                  fetal_FRZB+ early fibroblast \n",
       "                                         10877 \n",
       "                 fetal_HOXC5+ early fibroblast \n",
       "                                         15275 \n",
       "                                     fetal_HSC \n",
       "                                           112 \n",
       "                                    fetal_ILC2 \n",
       "                                           405 \n",
       "                                    fetal_ILC3 \n",
       "                                          1434 \n",
       "                          fetal_Immature basal \n",
       "                                            65 \n",
       "                     fetal_Immature suprabasal \n",
       "                                            15 \n",
       "                         fetal_Inflammatory DC \n",
       "                                            78 \n",
       "                          fetal_Innate T type1 \n",
       "                                           672 \n",
       "                          fetal_Innate T type3 \n",
       "                                           225 \n",
       "                       fetal_Inner root sheath \n",
       "                                            18 \n",
       "               fetal_Iron-recycling macrophage \n",
       "                                           299 \n",
       "                                      fetal_LC \n",
       "                                           379 \n",
       "                                      fetal_LE \n",
       "                                           517 \n",
       "                       fetal_LMCD1+ mural cell \n",
       "                                          1199 \n",
       "                                     fetal_LTi \n",
       "                                           691 \n",
       "                      fetal_LYVE1++ macrophage \n",
       "                                         20280 \n",
       "                     fetal_Lymphoid progenitor \n",
       "                                           121 \n",
       "                  fetal_MEMP - Early erythroid \n",
       "                                            89 \n",
       "                            fetal_MEMP - Megak \n",
       "                                           243 \n",
       "                       fetal_MHCII+ macrophage \n",
       "                                          2196 \n",
       "                    fetal_Mast cell (earliest) \n",
       "                                           189 \n",
       "                      fetal_Mast cell (medium) \n",
       "                                           762 \n",
       "                 fetal_Mast cell (most mature) \n",
       "                                           935 \n",
       "                          fetal_Matrix/placode \n",
       "                                           254 \n",
       "                           fetal_Megakaryocyte \n",
       "                                           173 \n",
       "                             fetal_Melanoblast \n",
       "                                           190 \n",
       "                              fetal_Melanocyte \n",
       "                                           604 \n",
       "                                fetal_Monocyte \n",
       "                                          3044 \n",
       "    fetal_Monocyte (activated/differentiating) \n",
       "                                          1895 \n",
       "                      fetal_Monocyte precursor \n",
       "                                          1734 \n",
       "               fetal_Myelinating Schwann cells \n",
       "                                          1211 \n",
       "                               fetal_Myoblasts \n",
       "                                          5175 \n",
       "                                fetal_Myocytes \n",
       "                                            51 \n",
       "                          fetal_Myofibroblasts \n",
       "                                          3154 \n",
       "                                 fetal_NK cell \n",
       "                                          6822 \n",
       "                          fetal_Neuroendocrine \n",
       "                                           281 \n",
       "                      fetal_Neuron progenitors \n",
       "                                           649 \n",
       "                             fetal_Neutrophil1 \n",
       "                                            88 \n",
       "                             fetal_Neutrophil2 \n",
       "                                           174 \n",
       "                       fetal_Outer root sheath \n",
       "                                            25 \n",
       "                       fetal_PEAR1+ fibroblast \n",
       "                                          3787 \n",
       "                    fetal_PID1+ schwann cellls \n",
       "                                           158 \n",
       "                         fetal_PLN+ mural cell \n",
       "                                          1928 \n",
       "                            fetal_POSTN+ basal \n",
       "                                            46 \n",
       "                               fetal_Pericytes \n",
       "                                          6250 \n",
       "                                fetal_Periderm \n",
       "                                           153 \n",
       "                    fetal_Postcapillary venule \n",
       "                                            48 \n",
       "                              fetal_Pre B cell \n",
       "                                           541 \n",
       "                          fetal_Pre pro B cell \n",
       "                                           176 \n",
       "                   fetal_Pre-dermal condensate \n",
       "                                         22724 \n",
       "                              fetal_Pro B cell \n",
       "                                           388 \n",
       "   fetal_SPP1+ proliferating neuron proneitors \n",
       "                                           774 \n",
       "              fetal_Schwann/Schwann precursors \n",
       "                                          2570 \n",
       "                          fetal_Suprabasal IFE \n",
       "                                            30 \n",
       "                       fetal_TREM2+ macrophage \n",
       "                                          1617 \n",
       "                     fetal_Tip cell (arterial) \n",
       "                                           134 \n",
       "                                    fetal_Treg \n",
       "                                           987 \n",
       "                        fetal_WNT2+ fibroblast \n",
       "                                         29109 \n",
       "                                     fetal_pDC \n",
       "                                           589 \n",
       "                           organoid_Adipocytes \n",
       "                                             0 \n",
       "                                organoid_Basal \n",
       "                                             0 \n",
       "                      organoid_Companion layer \n",
       "                                             0 \n",
       "                       organoid_Cuticle/cortex \n",
       "                                             0 \n",
       "                    organoid_Dermal condensate \n",
       "                                             0 \n",
       "                      organoid_Dermal papillia \n",
       "                                             0 \n",
       "                       organoid_Early myocytes \n",
       "                                             0 \n",
       "               organoid_FRZB+ early fibroblast \n",
       "                                             0 \n",
       "              organoid_HOXC5+ early fibroblast \n",
       "                                             0 \n",
       "                       organoid_Immature basal \n",
       "                                             0 \n",
       "                  organoid_Immature suprabasal \n",
       "                                             0 \n",
       "                    organoid_Inner root sheath \n",
       "                                             0 \n",
       "                       organoid_Matrix/placode \n",
       "                                             0 \n",
       "                          organoid_Melanoblast \n",
       "                                             0 \n",
       "                           organoid_Melanocyte \n",
       "                                             0 \n",
       "                         organoid_Merkel cells \n",
       "                                             0 \n",
       "                           organoid_Mural cell \n",
       "                                             0 \n",
       "            organoid_Myelinating Schwann cells \n",
       "                                             0 \n",
       "                            organoid_Myoblasts \n",
       "                                             0 \n",
       "                             organoid_Myocytes \n",
       "                                             0 \n",
       "                organoid_Neuroepithelial cells \n",
       "                                             0 \n",
       "                   organoid_Neuron progenitors \n",
       "                                             0 \n",
       "                    organoid_Outer root sheath \n",
       "                                             0 \n",
       "                         organoid_POSTN+ basal \n",
       "                                             0 \n",
       "                    organoid_POSTN+ fibroblast \n",
       "                                             0 \n",
       "                             organoid_Periderm \n",
       "                                             0 \n",
       "                organoid_Pre-dermal condensate \n",
       "                                             0 \n",
       "           organoid_Proliferating melanoblasts \n",
       "                                             0 \n",
       "organoid_SPP1+ proliferating neuron proneitors \n",
       "                                             0 \n",
       "           organoid_Schwann/Schwann precursors \n",
       "                                             0 \n",
       "                     organoid_Secreting neuron \n",
       "                                             0 \n",
       "                       organoid_Suprabasal IFE \n",
       "                                             0 \n",
       "                  organoid_Tip cell (arterial) \n",
       "                                             0 \n",
       "                            organoid_Uncertain \n",
       "                                             0 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "ERROR",
     "evalue": "Error: Cannot find 'tsne' in this Seurat object\n",
     "output_type": "error",
     "traceback": [
      "Error: Cannot find 'tsne' in this Seurat object\nTraceback:\n",
      "1. DimPlot(seuratObj, reduction = \"tsne\")",
      "2. Embeddings(object = object[[reduction]])",
      "3. object[[reduction]]",
      "4. `[[.Seurat`(object, reduction)",
      "5. stop(\"Cannot find '\", i, \"' in this Seurat object\", call. = FALSE)"
     ]
    }
   ],
   "source": [
    "seuratObj@meta.data$pooled_annot %>% table() # note that the number of cells of some cell types is very low and should preferably be higher for a real application \n",
    "## . \n",
    "##     B CD4 T CD8 T    DC  Mono    NK  Treg  \n",
    "##   382  2562  1645    18    90   131   199 \n",
    "DimPlot(seuratObj, reduction = \"tsne\") \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-23-1.png)<!-- --> \n",
    " \n",
    "Visualize the data to see to which condition cells belong. The metadata \n",
    "dataframe column that denotes the condition (steady-state or after LCMV \n",
    "infection) is here called ‘aggregate.’ \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seuratObj@meta.data$aggregate %>% table() \n",
    "## . \n",
    "## LCMV   SS  \n",
    "## 3886 1141 \n",
    "DimPlot(seuratObj, reduction = \"tsne\", group.by = \"aggregate\") \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-24-1.png)<!-- --> \n",
    " \n",
    "### Read in NicheNet’s ligand-target prior model, ligand-receptor network and weighted integrated networks: \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_target_matrix = readRDS(url(\"https://zenodo.org/record/3260758/files/ligand_target_matrix.rds\")) \n",
    "ligand_target_matrix[1:5,1:5] # target genes in rows, ligands in columns \n",
    "##                 CXCL1        CXCL2        CXCL3        CXCL5         PPBP \n",
    "## A1BG     3.534343e-04 4.041324e-04 3.729920e-04 3.080640e-04 2.628388e-04 \n",
    "## A1BG-AS1 1.650894e-04 1.509213e-04 1.583594e-04 1.317253e-04 1.231819e-04 \n",
    "## A1CF     5.787175e-04 4.596295e-04 3.895907e-04 3.293275e-04 3.211944e-04 \n",
    "## A2M      6.027058e-04 5.996617e-04 5.164365e-04 4.517236e-04 4.590521e-04 \n",
    "## A2M-AS1  8.898724e-05 8.243341e-05 7.484018e-05 4.912514e-05 5.120439e-05 \n",
    " \n",
    "lr_network = readRDS(url(\"https://zenodo.org/record/3260758/files/lr_network.rds\")) \n",
    "head(lr_network) \n",
    "## # A tibble: 6 x 4 \n",
    "##   from  to    source         database \n",
    "##   <chr> <chr> <chr>          <chr>    \n",
    "## 1 CXCL1 CXCR2 kegg_cytokines kegg     \n",
    "## 2 CXCL2 CXCR2 kegg_cytokines kegg     \n",
    "## 3 CXCL3 CXCR2 kegg_cytokines kegg     \n",
    "## 4 CXCL5 CXCR2 kegg_cytokines kegg     \n",
    "## 5 PPBP  CXCR2 kegg_cytokines kegg     \n",
    "## 6 CXCL6 CXCR2 kegg_cytokines kegg \n",
    " \n",
    "weighted_networks = readRDS(url(\"https://zenodo.org/record/3260758/files/weighted_networks.rds\")) \n",
    "weighted_networks_lr = weighted_networks$lr_sig %>% inner_join(lr_network %>% distinct(from,to), by = c(\"from\",\"to\")) \n",
    " \n",
    "head(weighted_networks$lr_sig) # interactions and their weights in the ligand-receptor + signaling network \n",
    "## # A tibble: 6 x 3 \n",
    "##   from  to     weight \n",
    "##   <chr> <chr>   <dbl> \n",
    "## 1 A1BG  ABCC6  0.422  \n",
    "## 2 A1BG  ACE2   0.101  \n",
    "## 3 A1BG  ADAM10 0.0970 \n",
    "## 4 A1BG  AGO1   0.0525 \n",
    "## 5 A1BG  AKT1   0.0855 \n",
    "## 6 A1BG  ANXA7  0.457 \n",
    "head(weighted_networks$gr) # interactions and their weights in the gene regulatory network \n",
    "## # A tibble: 6 x 3 \n",
    "##   from  to     weight \n",
    "##   <chr> <chr>   <dbl> \n",
    "## 1 A1BG  A2M    0.0294 \n",
    "## 2 AAAS  GFAP   0.0290 \n",
    "## 3 AADAC CYP3A4 0.0422 \n",
    "## 4 AADAC IRF8   0.0275 \n",
    "## 5 AATF  ATM    0.0330 \n",
    "## 6 AATF  ATR    0.0355 \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "Because the expression data is of mouse origin, we will convert the \n",
    "NicheNet network gene symbols from human to mouse based on one-to-one \n",
    "orthology: \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lr_network = lr_network %>% mutate(from = convert_human_to_mouse_symbols(from), to = convert_human_to_mouse_symbols(to)) %>% drop_na() \n",
    "colnames(ligand_target_matrix) = ligand_target_matrix %>% colnames() %>% convert_human_to_mouse_symbols() \n",
    "rownames(ligand_target_matrix) = ligand_target_matrix %>% rownames() %>% convert_human_to_mouse_symbols() \n",
    " \n",
    "ligand_target_matrix = ligand_target_matrix %>% .[!is.na(rownames(ligand_target_matrix)), !is.na(colnames(ligand_target_matrix))] \n",
    " \n",
    "weighted_networks_lr = weighted_networks_lr %>% mutate(from = convert_human_to_mouse_symbols(from), to = convert_human_to_mouse_symbols(to)) %>% drop_na() \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "# Perform the NicheNet analysis \n",
    " \n",
    "In this case study, we want to apply NicheNet to predict which ligands \n",
    "expressed by all immune cells in the T cell area of the lymph node are \n",
    "most likely to have induced the differential expression in CD8 T cells \n",
    "after LCMV infection. \n",
    " \n",
    "As described in the main vignette, the pipeline of a basic NicheNet \n",
    "analysis consist of the following steps: \n",
    " \n",
    "## 1. Define a “sender/niche” cell population and a “receiver/target” cell population present in your expression data and determine which genes are expressed in both populations \n",
    " \n",
    "In this case study, the receiver cell population is the ‘CD8 T’ cell \n",
    "population, whereas the sender cell populations are ‘CD4 T,’ ‘Treg,’ \n",
    "‘Mono,’ ‘NK,’ ‘B’ and ‘DC.’ We will consider a gene to be expressed when \n",
    "it is expressed in at least 10% of cells in one cluster. \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## receiver \n",
    "receiver = \"CD8 T\" \n",
    "expressed_genes_receiver = get_expressed_genes(receiver, seuratObj, pct = 0.10) \n",
    " \n",
    "background_expressed_genes = expressed_genes_receiver %>% .[. %in% rownames(ligand_target_matrix)] \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## sender \n",
    "sender_celltypes = c(\"CD4 T\",\"Treg\", \"Mono\", \"NK\", \"B\", \"DC\") \n",
    " \n",
    "list_expressed_genes_sender = sender_celltypes %>% unique() %>% lapply(get_expressed_genes, seuratObj, 0.10) # lapply to get the expressed genes of every sender cell type separately here \n",
    "expressed_genes_sender = list_expressed_genes_sender %>% unlist() %>% unique() \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "## 2. Define a gene set of interest: these are the genes in the “receiver/target” cell population that are potentially affected by ligands expressed by interacting cells (e.g. genes differentially expressed upon cell-cell interaction) \n",
    " \n",
    "Here, the gene set of interest are the genes differentially expressed in \n",
    "CD8 T cells after LCMV infection. The condition of interest is thus \n",
    "‘LCMV,’ whereas the reference/steady-state condition is ‘SS.’ The notion \n",
    "of conditions can be extracted from the metadata column ‘aggregate.’ The \n",
    "method to calculate the differential expression is here the standard \n",
    "Seurat Wilcoxon test, but this can be changed if necessary. \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seurat_obj_receiver= subset(seuratObj, idents = receiver) \n",
    "seurat_obj_receiver = SetIdent(seurat_obj_receiver, value = seurat_obj_receiver[[\"aggregate\"]]) \n",
    " \n",
    "condition_oi = \"LCMV\" \n",
    "condition_reference = \"SS\"  \n",
    "   \n",
    "DE_table_receiver = FindMarkers(object = seurat_obj_receiver, ident.1 = condition_oi, ident.2 = condition_reference, min.pct = 0.10) %>% rownames_to_column(\"gene\") \n",
    " \n",
    "geneset_oi = DE_table_receiver %>% filter(p_val_adj <= 0.05 & abs(avg_log2FC) >= 0.25) %>% pull(gene) \n",
    "geneset_oi = geneset_oi %>% .[. %in% rownames(ligand_target_matrix)] \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "## 3. Define a set of potential ligands: these are ligands that are expressed by the “sender/niche” cell population and bind a (putative) receptor expressed by the “receiver/target” population \n",
    " \n",
    "Because we combined the expressed genes of each sender cell type, in \n",
    "this example, we will perform one NicheNet analysis by pooling all \n",
    "ligands from all cell types together. Later on during the interpretation \n",
    "of the output, we will check which sender cell type expresses which \n",
    "ligand. \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ligands = lr_network %>% pull(from) %>% unique() \n",
    "receptors = lr_network %>% pull(to) %>% unique() \n",
    " \n",
    "expressed_ligands = intersect(ligands,expressed_genes_sender) \n",
    "expressed_receptors = intersect(receptors,expressed_genes_receiver) \n",
    " \n",
    "potential_ligands = lr_network %>% filter(from %in% expressed_ligands & to %in% expressed_receptors) %>% pull(from) %>% unique() \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "## 4) Perform NicheNet ligand activity analysis: rank the potential ligands based on the presence of their target genes in the gene set of interest (compared to the background set of genes) \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ligand_activities = predict_ligand_activities(geneset = geneset_oi, background_expressed_genes = background_expressed_genes, ligand_target_matrix = ligand_target_matrix, potential_ligands = potential_ligands) \n",
    " \n",
    "ligand_activities = ligand_activities %>% arrange(-pearson) %>% mutate(rank = rank(desc(pearson))) \n",
    "ligand_activities \n",
    "## # A tibble: 44 x 5 \n",
    "##    test_ligand auroc  aupr pearson  rank \n",
    "##    <chr>       <dbl> <dbl>   <dbl> <dbl> \n",
    "##  1 Ebi3        0.638 0.234  0.197      1 \n",
    "##  2 Il15        0.582 0.163  0.0961     2 \n",
    "##  3 Crlf2       0.549 0.163  0.0758     3 \n",
    "##  4 App         0.499 0.141  0.0655     4 \n",
    "##  5 Tgfb1       0.494 0.140  0.0558     5 \n",
    "##  6 Ptprc       0.536 0.149  0.0554     6 \n",
    "##  7 H2-M3       0.525 0.157  0.0528     7 \n",
    "##  8 Icam1       0.543 0.142  0.0486     8 \n",
    "##  9 Cxcl10      0.531 0.141  0.0408     9 \n",
    "## 10 Adam17      0.517 0.137  0.0359    10 \n",
    "## # ... with 34 more rows \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "The different ligand activity measures (auroc, aupr, pearson correlation \n",
    "coefficient) are a measure for how well a ligand can predict the \n",
    "observed differentially expressed genes compared to the background of \n",
    "expressed genes. In our validation study, we showed that the pearson \n",
    "correlation coefficient between a ligand’s target predictions and the \n",
    "observed transcriptional response was the most informative measure to \n",
    "define ligand activity. Therefore, NicheNet ranks the ligands based on \n",
    "their pearson correlation coefficient. This allows us to prioritize \n",
    "ligands inducing the antiviral response in CD8 T cells. \n",
    " \n",
    "The number of top-ranked ligands that are further used to predict active \n",
    "target genes and construct an active ligand-receptor network is here 20. \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "best_upstream_ligands = ligand_activities %>% top_n(20, pearson) %>% arrange(-pearson) %>% pull(test_ligand) %>% unique() \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "These ligands are expressed by one or more of the input sender cells. To \n",
    "see which cell population expresses which of these top-ranked ligands, \n",
    "you can run the following: \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DotPlot(seuratObj, features = best_upstream_ligands %>% rev(), cols = \"RdYlBu\") + RotatedAxis() \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-33-1.png)<!-- --> \n",
    " \n",
    "As you can see, most op the top-ranked ligands seem to be mainly \n",
    "expressed by dendritic cells and monocytes. \n",
    " \n",
    "## 5) Infer receptors and top-predicted target genes of ligands that are top-ranked in the ligand activity analysis \n",
    " \n",
    "### Active target gene inference \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "active_ligand_target_links_df = best_upstream_ligands %>% lapply(get_weighted_ligand_target_links,geneset = geneset_oi, ligand_target_matrix = ligand_target_matrix, n = 200) %>% bind_rows() %>% drop_na() \n",
    " \n",
    "active_ligand_target_links = prepare_ligand_target_visualization(ligand_target_df = active_ligand_target_links_df, ligand_target_matrix = ligand_target_matrix, cutoff = 0.33) \n",
    " \n",
    "order_ligands = intersect(best_upstream_ligands, colnames(active_ligand_target_links)) %>% rev() %>% make.names() \n",
    "order_targets = active_ligand_target_links_df$target %>% unique() %>% intersect(rownames(active_ligand_target_links)) %>% make.names() \n",
    "rownames(active_ligand_target_links) = rownames(active_ligand_target_links) %>% make.names() # make.names() for heatmap visualization of genes like H2-T23 \n",
    "colnames(active_ligand_target_links) = colnames(active_ligand_target_links) %>% make.names() # make.names() for heatmap visualization of genes like H2-T23 \n",
    " \n",
    "vis_ligand_target = active_ligand_target_links[order_targets,order_ligands] %>% t() \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p_ligand_target_network = vis_ligand_target %>% make_heatmap_ggplot(\"Prioritized ligands\",\"Predicted target genes\", color = \"purple\",legend_position = \"top\", x_axis_position = \"top\",legend_title = \"Regulatory potential\")  + theme(axis.text.x = element_text(face = \"italic\")) + scale_fill_gradient2(low = \"whitesmoke\",  high = \"purple\", breaks = c(0,0.0045,0.0090)) \n",
    "p_ligand_target_network \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-35-1.png)<!-- --> \n",
    " \n",
    "Note that not all ligands from the top 20 are present in this \n",
    "ligand-target heatmap. The left-out ligands are ligands that don’t have \n",
    "target genes with high enough regulatory potential scores. Therefore, \n",
    "they did not survive the used cutoffs. To include them, you can be less \n",
    "stringent in the used cutoffs. \n",
    " \n",
    "### Receptors of top-ranked ligands \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lr_network_top = lr_network %>% filter(from %in% best_upstream_ligands & to %in% expressed_receptors) %>% distinct(from,to) \n",
    "best_upstream_receptors = lr_network_top %>% pull(to) %>% unique() \n",
    " \n",
    "lr_network_top_df_large = weighted_networks_lr %>% filter(from %in% best_upstream_ligands & to %in% best_upstream_receptors) \n",
    " \n",
    "lr_network_top_df = lr_network_top_df_large %>% spread(\"from\",\"weight\",fill = 0) \n",
    "lr_network_top_matrix = lr_network_top_df %>% select(-to) %>% as.matrix() %>% magrittr::set_rownames(lr_network_top_df$to) \n",
    " \n",
    "dist_receptors = dist(lr_network_top_matrix, method = \"binary\") \n",
    "hclust_receptors = hclust(dist_receptors, method = \"ward.D2\") \n",
    "order_receptors = hclust_receptors$labels[hclust_receptors$order] \n",
    "     \n",
    "dist_ligands = dist(lr_network_top_matrix %>% t(), method = \"binary\") \n",
    "hclust_ligands = hclust(dist_ligands, method = \"ward.D2\") \n",
    "order_ligands_receptor = hclust_ligands$labels[hclust_ligands$order] \n",
    " \n",
    "order_receptors = order_receptors %>% intersect(rownames(lr_network_top_matrix)) \n",
    "order_ligands_receptor = order_ligands_receptor %>% intersect(colnames(lr_network_top_matrix)) \n",
    " \n",
    "vis_ligand_receptor_network = lr_network_top_matrix[order_receptors, order_ligands_receptor] \n",
    "rownames(vis_ligand_receptor_network) = order_receptors %>% make.names() \n",
    "colnames(vis_ligand_receptor_network) = order_ligands_receptor %>% make.names() \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p_ligand_receptor_network = vis_ligand_receptor_network %>% t() %>% make_heatmap_ggplot(\"Ligands\",\"Receptors\", color = \"mediumvioletred\", x_axis_position = \"top\",legend_title = \"Prior interaction potential\") \n",
    "p_ligand_receptor_network \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-37-1.png)<!-- --> \n",
    " \n",
    "### Receptors of top-ranked ligands, but after considering only bona fide ligand-receptor interactions documented in literature and publicly available databases \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lr_network_strict = lr_network %>% filter(database != \"ppi_prediction_go\" & database != \"ppi_prediction\") \n",
    "ligands_bona_fide = lr_network_strict %>% pull(from) %>% unique() \n",
    "receptors_bona_fide = lr_network_strict %>% pull(to) %>% unique() \n",
    " \n",
    "lr_network_top_df_large_strict = lr_network_top_df_large %>% distinct(from,to) %>% inner_join(lr_network_strict, by = c(\"from\",\"to\")) %>% distinct(from,to) \n",
    "lr_network_top_df_large_strict = lr_network_top_df_large_strict %>% inner_join(lr_network_top_df_large, by = c(\"from\",\"to\")) \n",
    " \n",
    "lr_network_top_df_strict = lr_network_top_df_large_strict %>% spread(\"from\",\"weight\",fill = 0) \n",
    "lr_network_top_matrix_strict = lr_network_top_df_strict %>% select(-to) %>% as.matrix() %>% magrittr::set_rownames(lr_network_top_df_strict$to) \n",
    " \n",
    "dist_receptors = dist(lr_network_top_matrix_strict, method = \"binary\") \n",
    "hclust_receptors = hclust(dist_receptors, method = \"ward.D2\") \n",
    "order_receptors = hclust_receptors$labels[hclust_receptors$order] \n",
    " \n",
    "dist_ligands = dist(lr_network_top_matrix_strict %>% t(), method = \"binary\") \n",
    "hclust_ligands = hclust(dist_ligands, method = \"ward.D2\") \n",
    "order_ligands_receptor = hclust_ligands$labels[hclust_ligands$order] \n",
    " \n",
    "order_receptors = order_receptors %>% intersect(rownames(lr_network_top_matrix_strict)) \n",
    "order_ligands_receptor = order_ligands_receptor %>% intersect(colnames(lr_network_top_matrix_strict)) \n",
    " \n",
    "vis_ligand_receptor_network_strict = lr_network_top_matrix_strict[order_receptors, order_ligands_receptor] \n",
    "rownames(vis_ligand_receptor_network_strict) = order_receptors %>% make.names() \n",
    "colnames(vis_ligand_receptor_network_strict) = order_ligands_receptor %>% make.names() \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "p_ligand_receptor_network_strict = vis_ligand_receptor_network_strict %>% t() %>% make_heatmap_ggplot(\"Ligands\",\"Receptors\", color = \"mediumvioletred\", x_axis_position = \"top\",legend_title = \"Prior interaction potential\\n(bona fide)\") \n",
    "p_ligand_receptor_network_strict \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-39-1.png)<!-- --> \n",
    " \n",
    "## 6) Add log fold change information of ligands from sender cells \n",
    " \n",
    "In some cases, it might be possible to also check upregulation of \n",
    "ligands in sender cells. This can add a useful extra layer of \n",
    "information next to the ligand activities defined by NicheNet, because \n",
    "you can assume that some of the ligands inducing DE in receiver cells, \n",
    "will be DE themselves in the sender cells. \n",
    " \n",
    "Here this is possible: we will define the log fold change between LCMV \n",
    "and steady-state in all sender cell types and visualize this as extra \n",
    "information. \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DE analysis for each sender cell type \n",
    "# this uses a new nichenetr function - reinstall nichenetr if necessary! \n",
    "DE_table_all = Idents(seuratObj) %>% levels() %>% intersect(sender_celltypes) %>% lapply(get_lfc_celltype, seurat_obj = seuratObj, condition_colname = \"aggregate\", condition_oi = condition_oi, condition_reference = condition_reference, expression_pct = 0.10, celltype_col = NULL) %>% reduce(full_join) # use this if cell type labels are the identities of your Seurat object -- if not: indicate the celltype_col properly \n",
    "DE_table_all[is.na(DE_table_all)] = 0 \n",
    " \n",
    "# Combine ligand activities with DE information \n",
    "ligand_activities_de = ligand_activities %>% select(test_ligand, pearson) %>% rename(ligand = test_ligand) %>% left_join(DE_table_all %>% rename(ligand = gene)) \n",
    "ligand_activities_de[is.na(ligand_activities_de)] = 0 \n",
    " \n",
    "# make LFC heatmap \n",
    "lfc_matrix = ligand_activities_de  %>% select(-ligand, -pearson) %>% as.matrix() %>% magrittr::set_rownames(ligand_activities_de$ligand) \n",
    "rownames(lfc_matrix) = rownames(lfc_matrix) %>% make.names() \n",
    " \n",
    "order_ligands = order_ligands[order_ligands %in% rownames(lfc_matrix)] \n",
    "vis_ligand_lfc = lfc_matrix[order_ligands,] \n",
    " \n",
    "colnames(vis_ligand_lfc) = vis_ligand_lfc %>% colnames() %>% make.names() \n",
    " \n",
    "p_ligand_lfc = vis_ligand_lfc %>% make_threecolor_heatmap_ggplot(\"Prioritized ligands\",\"LFC in Sender\", low_color = \"midnightblue\",mid_color = \"white\", mid = median(vis_ligand_lfc), high_color = \"red\",legend_position = \"top\", x_axis_position = \"top\", legend_title = \"LFC\") + theme(axis.text.y = element_text(face = \"italic\")) \n",
    "p_ligand_lfc \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-40-1.png)<!-- --> \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# change colors a bit to make them more stand out \n",
    "p_ligand_lfc = p_ligand_lfc + scale_fill_gradientn(colors = c(\"midnightblue\",\"blue\", \"grey95\", \"grey99\",\"firebrick1\",\"red\"),values = c(0,0.1,0.2,0.25, 0.40, 0.7,1), limits = c(vis_ligand_lfc %>% min() - 0.1, vis_ligand_lfc %>% max() + 0.1)) \n",
    "p_ligand_lfc \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-40-2.png)<!-- --> \n",
    " \n",
    "## 7) Summary visualizations of the NicheNet analysis \n",
    " \n",
    "For example, you can make a combined heatmap of ligand activities, \n",
    "ligand expression, ligand log fold change and the target genes of the \n",
    "top-ranked ligands. The plots for the log fold change and target genes \n",
    "were already made. Let’s now make the heatmap for ligand activities and \n",
    "for expression. \n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ligand activity heatmap \n",
    "ligand_pearson_matrix = ligand_activities %>% select(pearson) %>% as.matrix() %>% magrittr::set_rownames(ligand_activities$test_ligand) \n",
    " \n",
    "rownames(ligand_pearson_matrix) = rownames(ligand_pearson_matrix) %>% make.names() \n",
    "colnames(ligand_pearson_matrix) = colnames(ligand_pearson_matrix) %>% make.names() \n",
    " \n",
    "vis_ligand_pearson = ligand_pearson_matrix[order_ligands, ] %>% as.matrix(ncol = 1) %>% magrittr::set_colnames(\"Pearson\") \n",
    "p_ligand_pearson = vis_ligand_pearson %>% make_heatmap_ggplot(\"Prioritized ligands\",\"Ligand activity\", color = \"darkorange\",legend_position = \"top\", x_axis_position = \"top\", legend_title = \"Pearson correlation coefficient\\ntarget gene prediction ability)\") + theme(legend.text = element_text(size = 9)) \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ligand expression Seurat dotplot \n",
    "order_ligands_adapted = order_ligands \n",
    "order_ligands_adapted[order_ligands_adapted == \"H2.M3\"] = \"H2-M3\" # cf required use of make.names for heatmap visualization | this is not necessary if these ligands are not in the list of prioritized ligands! \n",
    "order_ligands_adapted[order_ligands_adapted == \"H2.T23\"] = \"H2-T23\" # cf required use of make.names for heatmap visualization | this is not necessary if these ligands are not in the list of prioritized ligands! \n",
    "rotated_dotplot = DotPlot(seuratObj %>% subset(celltype %in% sender_celltypes), features = order_ligands_adapted, cols = \"RdYlBu\") + coord_flip() + theme(legend.text = element_text(size = 10), legend.title = element_text(size = 12)) # flip of coordinates necessary because we want to show ligands in the rows when combining all plots \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "figures_without_legend = cowplot::plot_grid( \n",
    "  p_ligand_pearson + theme(legend.position = \"none\", axis.ticks = element_blank()) + theme(axis.title.x = element_text()), \n",
    "  rotated_dotplot + theme(legend.position = \"none\", axis.ticks = element_blank(), axis.title.x = element_text(size = 12), axis.text.y = element_text(face = \"italic\", size = 9), axis.text.x = element_text(size = 9,  angle = 90,hjust = 0)) + ylab(\"Expression in Sender\") + xlab(\"\") + scale_y_discrete(position = \"right\"), \n",
    "  p_ligand_lfc + theme(legend.position = \"none\", axis.ticks = element_blank()) + theme(axis.title.x = element_text()) + ylab(\"\"), \n",
    "  p_ligand_target_network + theme(legend.position = \"none\", axis.ticks = element_blank()) + ylab(\"\"), \n",
    "  align = \"hv\", \n",
    "  nrow = 1, \n",
    "  rel_widths = c(ncol(vis_ligand_pearson)+6, ncol(vis_ligand_lfc) + 7, ncol(vis_ligand_lfc) + 8, ncol(vis_ligand_target))) \n",
    " \n",
    "legends = cowplot::plot_grid( \n",
    "    ggpubr::as_ggplot(ggpubr::get_legend(p_ligand_pearson)), \n",
    "    ggpubr::as_ggplot(ggpubr::get_legend(rotated_dotplot)), \n",
    "    ggpubr::as_ggplot(ggpubr::get_legend(p_ligand_lfc)), \n",
    "    ggpubr::as_ggplot(ggpubr::get_legend(p_ligand_target_network)), \n",
    "    nrow = 1, \n",
    "    align = \"h\", rel_widths = c(1.5, 1, 1, 1)) \n",
    " \n",
    "combined_plot = cowplot::plot_grid(figures_without_legend, legends, rel_heights = c(10,5), nrow = 2, align = \"hv\") \n",
    "combined_plot \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " \n",
    "![](seurat_steps_files/figure-gfm/unnamed-chunk-43-1.png)<!-- --> \n",
    " \n",
    "# Remarks \n",
    " \n",
    "1.  Top-ranked ligands and target genes shown here differ from the \n",
    "    predictions shown in the respective case study in the NicheNet paper \n",
    "    because a different definition of expressed genes was used. \n",
    " \n",
    "# References \n",
    " \n",
    "<div id=\"refs\" class=\"references csl-bib-body hanging-indent\"> \n",
    " \n",
    "<div id=\"ref-medaglia_spatial_2017\" class=\"csl-entry\"> \n",
    " \n",
    "Medaglia, Chiara, Amir Giladi, Liat Stoler-Barak, Marco De Giovanni, \n",
    "Tomer Meir Salame, Adi Biram, Eyal David, et al. 2017. “Spatial \n",
    "Reconstruction of Immune Niches by Combining Photoactivatable Reporters \n",
    "and <span class=\"nocase\">scRNA</span>-Seq.” *Science*, December, \n",
    "eaao4277. <https://doi.org/10.1126/science.aao4277>. \n",
    " \n",
    "</div> \n",
    " \n",
    "</div> \n"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": "",
  "kernelspec": {
   "display_name": "nichenet",
   "language": "R",
   "name": "nichenet"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.5"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
