{
    "cells": [
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                "Differential NicheNet analysis between niches of interest \n",
                "================ \n",
                "Robin Browaeys \n",
                "20212-01-12 \n",
                " \n",
                "<!-- github markdown built using  \n",
                "rmarkdown::render(\"vignettes/differential_nichenet.Rmd\", output_format = \"github_document\") \n",
                "--> \n",
                " \n",
                "This vignette guides you in detail through all the steps of a \n",
                "Differential NicheNet analysis. As example expression data of \n",
                "interacting cells, we will here use subset of the liver scRNAseq data \n",
                "generated in the paper from Guilliams et al: [Spatial proteogenomics \n",
                "reveals distinct and evolutionarily conserved hepatic macrophage \n",
                "niches](https://www.sciencedirect.com/science/article/pii/S0092867421014811). \n",
                "We took a subset of the data (deposited on \n",
                "<https://zenodo.org/deposit/5840787>) for demonstration purposes because \n",
                "of the large size of the entire dataset. For exploration and downloading \n",
                "of all the data from the paper, we refer to: [Liver Atlas Data \n",
                "Portal](https://www.livercellatlas.org/). For the code used for all the \n",
                "Differential NicheNet analyses on the entire liver cell atlas dataset, \n",
                "see <https://github.com/saeyslab/NicheNet_LiverCellAtlas>. \n",
                " \n",
                "The goal of Differential NicheNet is to predict ligand-receptors pairs \n",
                "that are both differentially expressed and active between different \n",
                "niches of interest. \n",
                " \n",
                "In this vignette, we will look at cell-cell communication differences \n",
                "between the Kupffer cell niche, the bile duct macrophage niche, and the \n",
                "capsule macrophage niche, with the macrophages in each niche as receiver \n",
                "cell of interest. This means that we are interested in identifying the \n",
                "niche-specific ligands important for the identity of each of these \n",
                "macrophage subtypes. \n",
                " \n",
                "# 0. Read in the expression data of interest, and the NicheNet ligand-receptor network and ligand-target matrix \n",
                " \n",
                "## Load in packages \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "library(nichenetr) \n",
                "library(RColorBrewer) \n",
                "library(tidyverse) \n",
                "library(Seurat)  \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "## Read in the expression data \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "seurat_obj = readRDS(url(\"https://zenodo.org/record/5840787/files/seurat_obj_subset_integrated_zonation.rds\")) \n",
                "DimPlot(seurat_obj, group.by = \"celltype\", label = TRUE)  \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-2-1.png)<!-- --> \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "seurat_obj = SetIdent(seurat_obj, value = \"celltype\") \n",
                "\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "As you can see, the LSECs, hepatocytes and Stellate cells are each \n",
                "divided in two groups, based on their spatial location (periportal and \n",
                "pericentral). \n",
                " \n",
                "## Read in the NicheNet ligand-receptor network and ligand-target matrix \n",
                " \n",
                "The used ligand-receptor network and ligand-target matrix can be \n",
                "downloaded from Zenodo \n",
                "[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.3260758.svg)](https://doi.org/10.5281/zenodo.3260758). \n",
                "The Seurat object containing expression data of interacting cells in \n",
                "HNSCC can also be downloaded from Zenodo \n",
                "[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4675430.svg)](https://doi.org/10.5281/zenodo.4675430). \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "ligand_target_matrix = readRDS(url(\"https://zenodo.org/record/3260758/files/ligand_target_matrix.rds\")) \n",
                "ligand_target_matrix[1:5,1:5] # target genes in rows, ligands in columns \n",
                "##                 CXCL1        CXCL2        CXCL3        CXCL5         PPBP \n",
                "## A1BG     3.534343e-04 4.041324e-04 3.729920e-04 3.080640e-04 2.628388e-04 \n",
                "## A1BG-AS1 1.650894e-04 1.509213e-04 1.583594e-04 1.317253e-04 1.231819e-04 \n",
                "## A1CF     5.787175e-04 4.596295e-04 3.895907e-04 3.293275e-04 3.211944e-04 \n",
                "## A2M      6.027058e-04 5.996617e-04 5.164365e-04 4.517236e-04 4.590521e-04 \n",
                "## A2M-AS1  8.898724e-05 8.243341e-05 7.484018e-05 4.912514e-05 5.120439e-05 \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "lr_network = readRDS(url(\"https://zenodo.org/record/3260758/files/lr_network.rds\")) \n",
                "lr_network = lr_network %>% mutate(bonafide = ! database %in% c(\"ppi_prediction\",\"ppi_prediction_go\")) \n",
                "lr_network = lr_network %>% dplyr::rename(ligand = from, receptor = to) %>% distinct(ligand, receptor, bonafide) \n",
                " \n",
                "head(lr_network) \n",
                "## # A tibble: 6 x 3 \n",
                "##   ligand receptor bonafide \n",
                "##   <chr>  <chr>    <lgl>    \n",
                "## 1 CXCL1  CXCR2    TRUE     \n",
                "## 2 CXCL2  CXCR2    TRUE     \n",
                "## 3 CXCL3  CXCR2    TRUE     \n",
                "## 4 CXCL5  CXCR2    TRUE     \n",
                "## 5 PPBP   CXCR2    TRUE     \n",
                "## 6 CXCL6  CXCR2    TRUE \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "Note: because the data is of mouse origin: we need to convert human gene \n",
                "symbols to their murine one-to-one orthologs \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "organism = \"mouse\"  \n",
                "\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "if(organism == \"mouse\"){ \n",
                "  lr_network = lr_network %>% mutate(ligand = convert_human_to_mouse_symbols(ligand), receptor = convert_human_to_mouse_symbols(receptor)) %>% drop_na() \n",
                " \n",
                "  colnames(ligand_target_matrix) = ligand_target_matrix %>% colnames() %>% convert_human_to_mouse_symbols() \n",
                "  rownames(ligand_target_matrix) = ligand_target_matrix %>% rownames() %>% convert_human_to_mouse_symbols() \n",
                "  ligand_target_matrix = ligand_target_matrix %>% .[!is.na(rownames(ligand_target_matrix)), !is.na(colnames(ligand_target_matrix))] \n",
                "} \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 1. Define the niches/microenvironments of interest \n",
                " \n",
                "Each niche should have at least one “sender/niche” cell population and \n",
                "one “receiver/target” cell population (present in your expression data) \n",
                " \n",
                "In this case study, we are interested to find differences in cell-cell \n",
                "interactions to hepatic macrophages in three different niches: 1) the \n",
                "Kupffer cell niche, 2) the bile-duct or lipid-associated macrophage \n",
                "niche, and 3) the capsule macrophage niche. \n",
                " \n",
                "Based on imaging and spatial transcriptomics, the composition of each \n",
                "niche was defined as follows: \n",
                " \n",
                "The receiver cell population in the Kupffer cell niche is the “KCs” cell \n",
                "type, the sender cell types are: “LSECs_portal”,“Hepatocytes_portal”, \n",
                "and “Stellate cells_portal”. The receiver cell population in the \n",
                "lipid-associated macrophage (MoMac2) niche is the “MoMac2” cell type, \n",
                "the sender cell types are: “Cholangiocytes”, and “Fibroblast 2”. The \n",
                "receiver cell population in the capsule macrophage (MoMac1) niche is the \n",
                "“MoMac1” cell type, the sender cell types are: “Capsule fibroblasts”, \n",
                "and “Mesothelial cells”. \n",
                " \n",
                "! Important: your receiver cell type should consist of 1 cluster! \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "niches = list( \n",
                "    \"KC_niche\" = list( \n",
                "      \"sender\" = c(\"LSECs_portal\",\"Hepatocytes_portal\",\"Stellate cells_portal\"), \n",
                "      \"receiver\" = c(\"KCs\")), \n",
                "    \"MoMac2_niche\" = list( \n",
                "      \"sender\" = c(\"Cholangiocytes\",\"Fibroblast 2\"), \n",
                "      \"receiver\" = c(\"MoMac2\")), \n",
                "    \"MoMac1_niche\" = list( \n",
                "      \"sender\" = c(\"Capsule fibroblasts\",\"Mesothelial cells\"), \n",
                "      \"receiver\" = c(\"MoMac1\")) \n",
                "  ) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 2. Calculate differential expression between the niches \n",
                " \n",
                "In this step, we will determine DE between the different niches for both \n",
                "senders and receivers to define the DE of L-R pairs. \n",
                " \n",
                "### Calculate DE \n",
                " \n",
                "The method to calculate the differential expression is here the standard \n",
                "Seurat Wilcoxon test, but this can be replaced if wanted by the user \n",
                "(only requirement: output tables `DE_sender_processed` and \n",
                "`DE_receiver_processed` should be in the same format as shown here). \n",
                " \n",
                "DE will be calculated for each pairwise sender (or receiver) cell type \n",
                "comparision between the niches (so across niches, not within niche). In \n",
                "our case study, this means e.g. that DE of LSECs_portal ligands will be \n",
                "calculated by DE analysis of LSECs_portal vs Cholangiocytes; \n",
                "LSECs_portal vs Fibroblast 2; LSECs_portal vs Capsule fibroblasts; and \n",
                "LSECs_portal vs Mesothelial cells. We split the cells per cell type \n",
                "instead of merging all cells from the other niche to avoid that the DE \n",
                "analysis will be driven by the most abundant cell types. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "assay_oi = \"SCT\" # other possibilities: RNA,... \n",
                "seurat_obj = PrepSCTFindMarkers(seurat_obj, assay = \"SCT\", verbose = FALSE) \n",
                " \n",
                "DE_sender = calculate_niche_de(seurat_obj = seurat_obj %>% subset(features = lr_network$ligand %>% intersect(rownames(seurat_obj))), niches = niches, type = \"sender\", assay_oi = assay_oi) # only ligands important for sender cell types \n",
                "## [1] \"Calculate Sender DE between: LSECs_portal and Cholangiocytes\"      \n",
                "## [2] \"Calculate Sender DE between: LSECs_portal and Fibroblast 2\"        \n",
                "## [3] \"Calculate Sender DE between: LSECs_portal and Capsule fibroblasts\" \n",
                "## [4] \"Calculate Sender DE between: LSECs_portal and Mesothelial cells\"   \n",
                "## [1] \"Calculate Sender DE between: Hepatocytes_portal and Cholangiocytes\"      \n",
                "## [2] \"Calculate Sender DE between: Hepatocytes_portal and Fibroblast 2\"        \n",
                "## [3] \"Calculate Sender DE between: Hepatocytes_portal and Capsule fibroblasts\" \n",
                "## [4] \"Calculate Sender DE between: Hepatocytes_portal and Mesothelial cells\"   \n",
                "## [1] \"Calculate Sender DE between: Stellate cells_portal and Cholangiocytes\"      \n",
                "## [2] \"Calculate Sender DE between: Stellate cells_portal and Fibroblast 2\"        \n",
                "## [3] \"Calculate Sender DE between: Stellate cells_portal and Capsule fibroblasts\" \n",
                "## [4] \"Calculate Sender DE between: Stellate cells_portal and Mesothelial cells\"   \n",
                "## [1] \"Calculate Sender DE between: Cholangiocytes and LSECs_portal\"          \n",
                "## [2] \"Calculate Sender DE between: Cholangiocytes and Hepatocytes_portal\"    \n",
                "## [3] \"Calculate Sender DE between: Cholangiocytes and Stellate cells_portal\" \n",
                "## [4] \"Calculate Sender DE between: Cholangiocytes and Capsule fibroblasts\"   \n",
                "## [5] \"Calculate Sender DE between: Cholangiocytes and Mesothelial cells\"     \n",
                "## [1] \"Calculate Sender DE between: Fibroblast 2 and LSECs_portal\"          \n",
                "## [2] \"Calculate Sender DE between: Fibroblast 2 and Hepatocytes_portal\"    \n",
                "## [3] \"Calculate Sender DE between: Fibroblast 2 and Stellate cells_portal\" \n",
                "## [4] \"Calculate Sender DE between: Fibroblast 2 and Capsule fibroblasts\"   \n",
                "## [5] \"Calculate Sender DE between: Fibroblast 2 and Mesothelial cells\"     \n",
                "## [1] \"Calculate Sender DE between: Capsule fibroblasts and LSECs_portal\"          \n",
                "## [2] \"Calculate Sender DE between: Capsule fibroblasts and Hepatocytes_portal\"    \n",
                "## [3] \"Calculate Sender DE between: Capsule fibroblasts and Stellate cells_portal\" \n",
                "## [4] \"Calculate Sender DE between: Capsule fibroblasts and Cholangiocytes\"        \n",
                "## [5] \"Calculate Sender DE between: Capsule fibroblasts and Fibroblast 2\"          \n",
                "## [1] \"Calculate Sender DE between: Mesothelial cells and LSECs_portal\"          \n",
                "## [2] \"Calculate Sender DE between: Mesothelial cells and Hepatocytes_portal\"    \n",
                "## [3] \"Calculate Sender DE between: Mesothelial cells and Stellate cells_portal\" \n",
                "## [4] \"Calculate Sender DE between: Mesothelial cells and Cholangiocytes\"        \n",
                "## [5] \"Calculate Sender DE between: Mesothelial cells and Fibroblast 2\" \n",
                "DE_receiver = calculate_niche_de(seurat_obj = seurat_obj %>% subset(features = lr_network$receptor %>% unique()), niches = niches, type = \"receiver\", assay_oi = assay_oi) # only receptors now, later on: DE analysis to find targets \n",
                "## # A tibble: 3 x 2 \n",
                "##   receiver receiver_other_niche \n",
                "##   <chr>    <chr>                \n",
                "## 1 KCs      MoMac2               \n",
                "## 2 KCs      MoMac1               \n",
                "## 3 MoMac2   MoMac1               \n",
                "## [1] \"Calculate receiver DE between: KCs and MoMac2\" \"Calculate receiver DE between: KCs and MoMac1\" \n",
                "## [1] \"Calculate receiver DE between: MoMac2 and KCs\"    \"Calculate receiver DE between: MoMac2 and MoMac1\" \n",
                "## [1] \"Calculate receiver DE between: MoMac1 and KCs\"    \"Calculate receiver DE between: MoMac1 and MoMac2\" \n",
                " \n",
                "DE_sender = DE_sender %>% mutate(avg_log2FC = ifelse(avg_log2FC == Inf, max(avg_log2FC[is.finite(avg_log2FC)]), ifelse(avg_log2FC == -Inf, min(avg_log2FC[is.finite(avg_log2FC)]), avg_log2FC))) \n",
                "DE_receiver = DE_receiver %>% mutate(avg_log2FC = ifelse(avg_log2FC == Inf, max(avg_log2FC[is.finite(avg_log2FC)]), ifelse(avg_log2FC == -Inf, min(avg_log2FC[is.finite(avg_log2FC)]), avg_log2FC))) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "### Process DE results: \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "expression_pct = 0.10 \n",
                "DE_sender_processed = process_niche_de(DE_table = DE_sender, niches = niches, expression_pct = expression_pct, type = \"sender\") \n",
                "DE_receiver_processed = process_niche_de(DE_table = DE_receiver, niches = niches, expression_pct = expression_pct, type = \"receiver\") \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "### Combine sender-receiver DE based on L-R pairs: \n",
                " \n",
                "As mentioned above, DE of ligands from one sender cell type is \n",
                "determined be calculating DE between that cell type, and all the sender \n",
                "cell types of the other niche. To summarize the DE of ligands of that \n",
                "cell type we have several options: we could take the average LFC, but \n",
                "also the minimum LFC compared to the other niche. We recommend using the \n",
                "minimum LFC, because this is the strongest specificity measure of ligand \n",
                "expression, because a high min LFC means that a ligand is more strongly \n",
                "expressed in the cell type of niche 1 compared to all cell types of \n",
                "niche 2 (in contrast to a high average LFC, which does not exclude that \n",
                "one or more cell types in niche 2 also strongly express that ligand). \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "specificity_score_LR_pairs = \"min_lfc\" \n",
                "DE_sender_receiver = combine_sender_receiver_de(DE_sender_processed, DE_receiver_processed, lr_network, specificity_score = specificity_score_LR_pairs) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 3. Optional: Calculate differential expression between the different spatial regions \n",
                " \n",
                "To improve the cell-cell interaction predictions, you can consider \n",
                "spatial information if possible and applicable. Spatial information can \n",
                "come from microscopy data, or from spatial transcriptomics data such as \n",
                "Visium. \n",
                " \n",
                "There are several ways to incorporate spatial information in the \n",
                "Differential NicheNet pipeline. First, you can only consider cell types \n",
                "as belonging to the same niche if they are in the same spatial location. \n",
                "Another way is including spatial differential expression of \n",
                "ligand-receptor pairs within one cell type in the prioritization \n",
                "framework. \n",
                " \n",
                "For example: We have a cell type X, located in regions A and B, and we \n",
                "want to study cell-cell communication in region A. We first add only \n",
                "celltypeX of regionA in the niche definition, and then calculate DE \n",
                "between celltypeX-regionA and celltypeX-regionB to give higher \n",
                "prioritization weight to regionA-specific ligands. \n",
                " \n",
                "In this case study, our region of interest is the periportal region of \n",
                "the liver, because KCs in mouse are predominantly located in the \n",
                "periportal region. Therefore we will give higher weight to ligands that \n",
                "are in the niche cells of KCs higher expressed in the periportal \n",
                "compared to the pericentral region. \n",
                " \n",
                "We do this as follows, by first defining a ‘spatial info’ dataframe. If \n",
                "there is no spatial information in your data: set the following two \n",
                "parameters to FALSE, and make a mock ‘spatial_info’ data frame. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "include_spatial_info_sender = TRUE # if not spatial info to include: put this to false  \n",
                "include_spatial_info_receiver = FALSE # if spatial info to include: put this to true  \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "spatial_info = tibble(celltype_region_oi = c(\"LSECs_portal\",\"Hepatocytes_portal\",\"Stellate cells_portal\"),  \n",
                "                      celltype_other_region = c(\"LSECs_central\",\"Hepatocytes_central\",\"Stellate cells_central\") \n",
                "                      ) %>%  \n",
                "  mutate(niche =  \"KC_niche\", celltype_type = \"sender\") \n",
                "specificity_score_spatial = \"lfc\" \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "# this is how this should be defined if you don't have spatial info \n",
                "# mock spatial info \n",
                "if(include_spatial_info_sender == FALSE & include_spatial_info_receiver == FALSE){ \n",
                "    spatial_info = tibble(celltype_region_oi = NA, celltype_other_region = NA) %>% mutate(niche =  niches %>% names() %>% head(1), celltype_type = \"sender\") \n",
                "}  \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "if(include_spatial_info_sender == TRUE){ \n",
                "  sender_spatial_DE = calculate_spatial_DE(seurat_obj = seurat_obj %>% subset(features = lr_network$ligand %>% unique()), spatial_info = spatial_info %>% filter(celltype_type == \"sender\"), assay_oi = assay_oi) \n",
                "  sender_spatial_DE_processed = process_spatial_de(DE_table = sender_spatial_DE, type = \"sender\", lr_network = lr_network, expression_pct = expression_pct, specificity_score = specificity_score_spatial) \n",
                " \n",
                "  # add a neutral spatial score for sender celltypes in which the spatial is not known / not of importance \n",
                "  sender_spatial_DE_others = get_non_spatial_de(niches = niches, spatial_info = spatial_info, type = \"sender\", lr_network = lr_network) \n",
                "  sender_spatial_DE_processed = sender_spatial_DE_processed %>% bind_rows(sender_spatial_DE_others) \n",
                " \n",
                "  sender_spatial_DE_processed = sender_spatial_DE_processed %>% mutate(scaled_ligand_score_spatial = scale_quantile_adapted(ligand_score_spatial)) \n",
                " \n",
                "} else { \n",
                "  # # add a neutral spatial score for all sender celltypes (for none of them, spatial is relevant in this case) \n",
                "  sender_spatial_DE_processed = get_non_spatial_de(niches = niches, spatial_info = spatial_info, type = \"sender\", lr_network = lr_network) \n",
                "  sender_spatial_DE_processed = sender_spatial_DE_processed %>% mutate(scaled_ligand_score_spatial = scale_quantile_adapted(ligand_score_spatial))   \n",
                " \n",
                "} \n",
                "## [1] \"Calculate Spatial DE between: LSECs_portal and LSECs_central\" \n",
                "## [1] \"Calculate Spatial DE between: Hepatocytes_portal and Hepatocytes_central\" \n",
                "## [1] \"Calculate Spatial DE between: Stellate cells_portal and Stellate cells_central\" \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "if(include_spatial_info_receiver == TRUE){ \n",
                "  receiver_spatial_DE = calculate_spatial_DE(seurat_obj = seurat_obj %>% subset(features = lr_network$receptor %>% unique()), spatial_info = spatial_info %>% filter(celltype_type == \"receiver\"), assay_oi = assay_oi) \n",
                "  receiver_spatial_DE_processed = process_spatial_de(DE_table = receiver_spatial_DE, type = \"receiver\", lr_network = lr_network, expression_pct = expression_pct, specificity_score = specificity_score_spatial) \n",
                " \n",
                "  # add a neutral spatial score for receiver celltypes in which the spatial is not known / not of importance \n",
                "  receiver_spatial_DE_others = get_non_spatial_de(niches = niches, spatial_info = spatial_info, type = \"receiver\", lr_network = lr_network) \n",
                "  receiver_spatial_DE_processed = receiver_spatial_DE_processed %>% bind_rows(receiver_spatial_DE_others) \n",
                " \n",
                "  receiver_spatial_DE_processed = receiver_spatial_DE_processed %>% mutate(scaled_receptor_score_spatial = scale_quantile_adapted(receptor_score_spatial)) \n",
                " \n",
                "} else { \n",
                "    # # add a neutral spatial score for all receiver celltypes (for none of them, spatial is relevant in this case) \n",
                "  receiver_spatial_DE_processed = get_non_spatial_de(niches = niches, spatial_info = spatial_info, type = \"receiver\", lr_network = lr_network) \n",
                "  receiver_spatial_DE_processed = receiver_spatial_DE_processed %>% mutate(scaled_receptor_score_spatial = scale_quantile_adapted(receptor_score_spatial)) \n",
                "} \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 4. Calculate ligand activities and infer active ligand-target links \n",
                " \n",
                "In this step, we will predict ligand activities of each ligand for each \n",
                "of the receiver cell types across the different niches. This is similar \n",
                "to the ligand activity analysis done in the normal NicheNet pipeline. \n",
                " \n",
                "To calculate ligand activities, we first need to define a geneset of \n",
                "interest for each niche. In this case study, the geneset of interest for \n",
                "the Kupffer cell niche are the genes upregulated in Kupffer cells \n",
                "compared to the capsule and bile duct macrophages. The geneset of \n",
                "interest for the bile duct macrophage niche are the genes upregulated in \n",
                "bile duct macrophages compared to the capsule macrophages and Kupffer \n",
                "cells. And similarly for the capsule macrophage geneset of interest. \n",
                " \n",
                "Note that you can also define these geneset of interest in a different \n",
                "way! (eg pathway-based geneset etc) \n",
                " \n",
                "Ligand-target links are inferred in the same way as described in the \n",
                "basic NicheNet vignettes. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "lfc_cutoff = 0.15 # recommended for 10x as min_lfc cutoff.  \n",
                "specificity_score_targets = \"min_lfc\" \n",
                " \n",
                "DE_receiver_targets = calculate_niche_de_targets(seurat_obj = seurat_obj, niches = niches, lfc_cutoff = lfc_cutoff, expression_pct = expression_pct, assay_oi = assay_oi)  \n",
                "## [1] \"Calculate receiver DE between: KCs and MoMac2\" \"Calculate receiver DE between: KCs and MoMac1\" \n",
                "## [1] \"Calculate receiver DE between: MoMac2 and KCs\"    \"Calculate receiver DE between: MoMac2 and MoMac1\" \n",
                "## [1] \"Calculate receiver DE between: MoMac1 and KCs\"    \"Calculate receiver DE between: MoMac1 and MoMac2\" \n",
                "DE_receiver_processed_targets = process_receiver_target_de(DE_receiver_targets = DE_receiver_targets, niches = niches, expression_pct = expression_pct, specificity_score = specificity_score_targets) \n",
                "   \n",
                "background = DE_receiver_processed_targets  %>% pull(target) %>% unique() \n",
                "geneset_KC = DE_receiver_processed_targets %>% filter(receiver == niches$KC_niche$receiver & target_score >= lfc_cutoff & target_significant == 1 & target_present == 1) %>% pull(target) %>% unique() \n",
                "geneset_MoMac2 = DE_receiver_processed_targets %>% filter(receiver == niches$MoMac2_niche$receiver & target_score >= lfc_cutoff & target_significant == 1 & target_present == 1) %>% pull(target) %>% unique() \n",
                "geneset_MoMac1 = DE_receiver_processed_targets %>% filter(receiver == niches$MoMac1_niche$receiver & target_score >= lfc_cutoff & target_significant == 1 & target_present == 1) %>% pull(target) %>% unique() \n",
                "# Good idea to check which genes will be left out of the ligand activity analysis (=when not present in the rownames of the ligand-target matrix). \n",
                "# If many genes are left out, this might point to some issue in the gene naming (eg gene aliases and old gene symbols, bad human-mouse mapping) \n",
                "geneset_KC %>% setdiff(rownames(ligand_target_matrix)) \n",
                "##  [1] \"Fcna\"          \"Wfdc17\"        \"AW112010\"      \"mt-Co1\"        \"mt-Nd2\"        \"C4b\"           \"Adgre4\"        \"mt-Co3\"        \n",
                "##  [9] \"Pira2\"         \"mt-Co2\"        \"mt-Nd4\"        \"mt-Atp6\"       \"mt-Nd1\"        \"mt-Nd3\"        \"Ear2\"          \"2900097C17Rik\" \n",
                "## [17] \"Iigp1\"         \"Trim30a\"       \"B430306N03Rik\" \"mt-Cytb\"       \"Pilrb2\"        \"Anapc15\"       \"Arf2\"          \"Gbp8\"          \n",
                "## [25] \"AC149090.1\"    \"Cd209f\"        \"Xlr\"           \"Ifitm6\" \n",
                "geneset_MoMac2 %>% setdiff(rownames(ligand_target_matrix)) \n",
                "##  [1] \"Chil3\"         \"Lyz1\"          \"Ccl9\"          \"Tmsb10\"        \"Ly6c2\"         \"Gm21188\"       \"Gm10076\"       \"Ms4a6c\"        \n",
                "##  [9] \"Calm3\"         \"Atp5e\"         \"Ftl1-ps1\"      \"S100a11\"       \"Clec4a3\"       \"Snrpe\"         \"Cox6c\"         \"Ly6i\"          \n",
                "## [17] \"1810058I24Rik\" \"Rpl34\"         \"Aph1c\"         \"Atp5o.1\" \n",
                "geneset_MoMac1 %>% setdiff(rownames(ligand_target_matrix)) \n",
                "##  [1] \"H2-Ab1\"   \"Malat1\"   \"H2-Aa\"    \"Hspa1b\"   \"Gm26522\"  \"Ly6a\"     \"H2-D1\"    \"Klra2\"    \"Bcl2a1d\"  \"Kcnq1ot1\" \n",
                " \n",
                "length(geneset_KC) \n",
                "## [1] 443 \n",
                "length(geneset_MoMac2) \n",
                "## [1] 339 \n",
                "length(geneset_MoMac1) \n",
                "## [1] 84 \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "It is always useful to check the number of genes in the geneset before \n",
                "doing the ligand activity analysis. We recommend having between 20 and \n",
                "1000 genes in the geneset of interest, and a background of at least 5000 \n",
                "genes for a proper ligand activity analysis. If you retrieve too many DE \n",
                "genes, it is recommended to use a higher `lfc_cutoff` threshold. We \n",
                "recommend using a cutoff of 0.15 if you have \\> 2 receiver cells/niches \n",
                "to compare and use the min_lfc as specificity score. If you have only 2 \n",
                "receivers/niche, we recommend using a higher threshold (such as using \n",
                "0.25). If you have single-cell data like Smart-seq2 with high sequencing \n",
                "depth, we recommend to also use higher threshold. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "top_n_target = 250 \n",
                " \n",
                "niche_geneset_list = list( \n",
                "    \"KC_niche\" = list( \n",
                "      \"receiver\" = \"KCs\", \n",
                "      \"geneset\" = geneset_KC, \n",
                "      \"background\" = background), \n",
                "    \"MoMac1_niche\" = list( \n",
                "      \"receiver\" = \"MoMac1\", \n",
                "      \"geneset\" = geneset_MoMac1 , \n",
                "      \"background\" = background), \n",
                "    \"MoMac2_niche\" = list( \n",
                "      \"receiver\" = \"MoMac2\", \n",
                "      \"geneset\" = geneset_MoMac2 , \n",
                "      \"background\" = background)   \n",
                "  ) \n",
                "   \n",
                "ligand_activities_targets = get_ligand_activities_targets(niche_geneset_list = niche_geneset_list, ligand_target_matrix = ligand_target_matrix, top_n_target = top_n_target) \n",
                "## [1] \"Calculate Ligand activities for: KCs\" \n",
                "## [1] \"Calculate Ligand activities for: MoMac1\" \n",
                "## [1] \"Calculate Ligand activities for: MoMac2\" \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 5. Calculate (scaled) expression of ligands, receptors and targets across cell types of interest (log expression values and expression fractions) \n",
                " \n",
                "In this step, we will calculate average (scaled) expression, and \n",
                "fraction of expression, of ligands, receptors, and target genes across \n",
                "all cell types of interest. Now this is here demonstrated via the \n",
                "DotPlot function of Seurat, but this can also be done via other ways of \n",
                "course. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "features_oi = union(lr_network$ligand, lr_network$receptor) %>% union(ligand_activities_targets$target) %>% setdiff(NA) \n",
                "   \n",
                "dotplot = suppressWarnings(Seurat::DotPlot(seurat_obj %>% subset(idents = niches %>% unlist() %>% unique()), features = features_oi, assay = assay_oi)) \n",
                "exprs_tbl = dotplot$data %>% as_tibble() \n",
                "exprs_tbl = exprs_tbl %>% rename(celltype = id, gene = features.plot, expression = avg.exp, expression_scaled = avg.exp.scaled, fraction = pct.exp) %>% \n",
                "    mutate(fraction = fraction/100) %>% as_tibble() %>% select(celltype, gene, expression, expression_scaled, fraction) %>% distinct() %>% arrange(gene) %>% mutate(gene = as.character(gene)) \n",
                "   \n",
                "exprs_tbl_ligand = exprs_tbl %>% filter(gene %in% lr_network$ligand) %>% rename(sender = celltype, ligand = gene, ligand_expression = expression, ligand_expression_scaled = expression_scaled, ligand_fraction = fraction)  \n",
                "exprs_tbl_receptor = exprs_tbl %>% filter(gene %in% lr_network$receptor) %>% rename(receiver = celltype, receptor = gene, receptor_expression = expression, receptor_expression_scaled = expression_scaled, receptor_fraction = fraction) \n",
                "exprs_tbl_target = exprs_tbl %>% filter(gene %in% ligand_activities_targets$target) %>% rename(receiver = celltype, target = gene, target_expression = expression, target_expression_scaled = expression_scaled, target_fraction = fraction) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "exprs_tbl_ligand = exprs_tbl_ligand %>%  mutate(scaled_ligand_expression_scaled = scale_quantile_adapted(ligand_expression_scaled)) %>% mutate(ligand_fraction_adapted = ligand_fraction) %>% mutate_cond(ligand_fraction >= expression_pct, ligand_fraction_adapted = expression_pct)  %>% mutate(scaled_ligand_fraction_adapted = scale_quantile_adapted(ligand_fraction_adapted)) \n",
                " \n",
                "exprs_tbl_receptor = exprs_tbl_receptor %>% mutate(scaled_receptor_expression_scaled = scale_quantile_adapted(receptor_expression_scaled))  %>% mutate(receptor_fraction_adapted = receptor_fraction) %>% mutate_cond(receptor_fraction >= expression_pct, receptor_fraction_adapted = expression_pct)  %>% mutate(scaled_receptor_fraction_adapted = scale_quantile_adapted(receptor_fraction_adapted)) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 6. Expression fraction and receptor \n",
                " \n",
                "In this step, we will score ligand-receptor interactions based on \n",
                "expression strength of the receptor, in such a way that we give higher \n",
                "scores to the most strongly expressed receptor of a certain ligand, in a \n",
                "certain celltype. This will not effect the rank of individual ligands \n",
                "later on, but will help in prioritizing the most important receptors per \n",
                "ligand (next to other factors regarding the receptor - see later). \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "exprs_sender_receiver = lr_network %>%  \n",
                "  inner_join(exprs_tbl_ligand, by = c(\"ligand\")) %>%  \n",
                "  inner_join(exprs_tbl_receptor, by = c(\"receptor\")) %>% inner_join(DE_sender_receiver %>% distinct(niche, sender, receiver)) \n",
                "   \n",
                "ligand_scaled_receptor_expression_fraction_df = exprs_sender_receiver %>% group_by(ligand, receiver) %>% mutate(rank_receptor_expression = dense_rank(receptor_expression), rank_receptor_fraction  = dense_rank(receptor_fraction)) %>% mutate(ligand_scaled_receptor_expression_fraction = 0.5*( (rank_receptor_fraction / max(rank_receptor_fraction)) + ((rank_receptor_expression / max(rank_receptor_expression))) ) )  %>% distinct(ligand, receptor, receiver, ligand_scaled_receptor_expression_fraction, bonafide) %>% distinct() %>% ungroup()  \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 7. Prioritization of ligand-receptor and ligand-target links \n",
                " \n",
                "In this step, we will combine all the above calculated information to \n",
                "prioritize ligand-receptor-target links. We scale every property of \n",
                "interest between 0 and 1, and the final prioritization score is a \n",
                "weighted sum of the scaled scores of all the properties of interest. \n",
                " \n",
                "We provide the user the option to consider the following properties for \n",
                "prioritization (of which the weights are defined in \n",
                "`prioritizing_weights`) : \n",
                " \n",
                "-   Ligand DE score: niche-specific expression of the ligand: by \n",
                "    default, this the minimum logFC between the sender of interest and \n",
                "    all the senders of the other niche(s). The higher the min logFC, the \n",
                "    higher the niche-specificity of the ligand. Therefore we recommend \n",
                "    to give this factor a very high weight. `prioritizing_weights` \n",
                "    argument: `\"scaled_ligand_score\"`. Recommended weight: 5 (at least \n",
                "    1, max 5). \n",
                " \n",
                "-   Scaled ligand expression: scaled expression of a ligand in one \n",
                "    sender compared to the other cell types in the dataset. This might \n",
                "    be useful to rescue potentially interesting ligands that have a high \n",
                "    scaled expression value, but a relatively small min logFC compared \n",
                "    to the other niche. One reason why this logFC might be small occurs \n",
                "    when (some) genes are not picked up efficiently by the used \n",
                "    sequencing technology (or other reasons for low RNA expression of \n",
                "    ligands). For example, we have observed that many ligands from the \n",
                "    Tgf-beta/BMP family are not picked up efficiently with single-nuclei \n",
                "    RNA sequencing compared to single-cell sequencing. \n",
                "    `prioritizing_weights` argument: \n",
                "    `\"scaled_ligand_expression_scaled\"`. Recommended weight: 1 (unless \n",
                "    technical reason for lower gene detection such as while using \n",
                "    Nuc-seq: then recommended to use a higher weight: 2). \n",
                " \n",
                "-   Ligand expression fraction: Ligands that are expressed in a smaller \n",
                "    fraction of cells of a cell type than defined by \n",
                "    `exprs_cutoff`(default: 0.10) will get a lower ranking, proportional \n",
                "    to their fraction (eg ligand expressed in 9% of cells will be ranked \n",
                "    higher than ligand expressed in 0.5% of cells). We opted for this \n",
                "    weighting based on fraction, instead of removing ligands that are \n",
                "    not expressed in more cells than this cutoff, because some \n",
                "    interesting ligands could be removed that way. Fraction of \n",
                "    expression is not taken into account for the prioritization if it is \n",
                "    already higher than the cutoff. `prioritizing_weights` argument: \n",
                "    `\"ligand_fraction\"`. Recommended weight: 1. \n",
                " \n",
                "-   Ligand spatial DE score: spatial expression specificity of the \n",
                "    ligand. If the niche of interest is at a specific tissue location, \n",
                "    but some of the sender cell types of that niche are also present in \n",
                "    other locations, it can be very informative to further prioritize \n",
                "    ligands of that sender by looking how they are DE between the \n",
                "    spatial location of interest compared to the other locations. \n",
                "    `prioritizing_weights` argument: `\"scaled_ligand_score_spatial\"`. \n",
                "    Recommended weight: 2 (or 0 if not applicable). \n",
                " \n",
                "-   Receptor DE score: niche-specific expression of the receptor: by \n",
                "    default, this the minimum logFC between the receiver of interest and \n",
                "    all the receiver of the other niche(s). The higher the min logFC, \n",
                "    the higher the niche-specificity of the receptor. Based on our \n",
                "    experience, we don’t suggest to give this as high importance as the \n",
                "    ligand DE, but this might depend on the specific case study. \n",
                "    `prioritizing_weights` argument: `\"scaled_receptor_score\"`. \n",
                "    Recommended weight: 0.5 (at least 0.5, and lower than \n",
                "    `\"scaled_ligand_score\"`). \n",
                " \n",
                "-   Scaled receptor expression: scaled expression of a receptor in one \n",
                "    receiver compared to the other cell types in the dataset. This might \n",
                "    be useful to rescue potentially interesting receptors that have a \n",
                "    high scaled expression value, but a relatively small min logFC \n",
                "    compared to the other niche. One reason why this logFC might be \n",
                "    small occurs when (some) genes are not picked up efficiently by the \n",
                "    used sequencing technology. `prioritizing_weights` argument: \n",
                "    `\"scaled_receptor_expression_scaled\"`. Recommended weight: 0.5. \n",
                " \n",
                "-   Receptor expression fraction: Receptors that are expressed in a \n",
                "    smaller fraction of cells of a cell type than defined by \n",
                "    `exprs_cutoff`(default: 0.10) will get a lower ranking, proportional \n",
                "    to their fraction (eg receptor expressed in 9% of cells will be \n",
                "    ranked higher than receptor expressed in 0.5% of cells). We opted \n",
                "    for this weighting based on fraction, instead of removing receptors \n",
                "    that are not expressed in more cells than this cutoff, because some \n",
                "    interesting receptors could be removed that way. Fraction of \n",
                "    expression is not taken into account for the prioritization if it is \n",
                "    already higher than the cutoff. `prioritizing_weights` argument: \n",
                "    `\"receptor_fraction\"`. Recommended weight: 1. \n",
                " \n",
                "-   Receptor expression strength: this factor let us give higher weights \n",
                "    to the most highly expressed receptor of a ligand in the receiver. \n",
                "    This let us rank higher one member of a receptor family if it higher \n",
                "    expressed than the other members. `prioritizing_weights` argument: \n",
                "    `\"ligand_scaled_receptor_expression_fraction\"`. Recommended value: 1 \n",
                "    (minimum: 0.5). \n",
                " \n",
                "-   Receptor spatial DE score: spatial expression specificity of the \n",
                "    receptor. If the niche of interest is at a specific tissue location, \n",
                "    but the receiver cell type of that niche is also present in other \n",
                "    locations, it can be very informative to further prioritize \n",
                "    receptors of that receiver by looking how they are DE between the \n",
                "    spatial location of interest compared to the other locations. \n",
                "    `prioritizing_weights` argument: `\"scaled_receptor_score_spatial\"`. \n",
                "    Recommended weight: 1 (or 0 if not applicable). \n",
                " \n",
                "-   Absolute ligand activity: to further prioritize ligand-receptor \n",
                "    pairs based on their predicted effect of the ligand-receptor \n",
                "    interaction on the gene expression in the receiver cell type - \n",
                "    absolute ligand activity accords to ‘absolute’ enrichment of target \n",
                "    genes of a ligand within the affected receiver genes. \n",
                "    `prioritizing_weights` argument: `\"scaled_activity\"`. Recommended \n",
                "    weight: 0, unless absolute enrichment of target genes is of specific \n",
                "    interest. \n",
                " \n",
                "-   Normalized ligand activity: to further prioritize ligand-receptor \n",
                "    pairs based on their predicted effect of the ligand-receptor \n",
                "    interaction on the gene expression in the receiver cell type - \n",
                "    normalization of activity is done because we found that some \n",
                "    datasets/conditions/niches have higher baseline activity values than \n",
                "    others - normalized ligand activity accords to ‘relative’ enrichment \n",
                "    of target genes of a ligand within the affected receiver genes. \n",
                "    `prioritizing_weights` argument: `\"scaled_activity_normalized\"`. \n",
                "    Recommended weight: at least 1. \n",
                " \n",
                "-   Prior knowledge quality of the L-R interaction: the NicheNet LR \n",
                "    network consists of two types of interactions: L-R pairs documented \n",
                "    in curated databases, and L-R pairs predicted based on gene \n",
                "    annotation and PPIs. The former are categorized as ‘bona fide’ \n",
                "    interactions. To rank bona fide interactions higher, but not exlude \n",
                "    potentially interesting non-bona-fide ones, we give bona fide \n",
                "    interactions a score of 1, and non-bona-fide interactions a score \n",
                "    fof 0.5. `prioritizing_weights` argument: `\"bona_fide\"` Recommend \n",
                "    weight: at least 1. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "prioritizing_weights = c(\"scaled_ligand_score\" = 5, \n",
                "                         \"scaled_ligand_expression_scaled\" = 1, \n",
                "                         \"ligand_fraction\" = 1, \n",
                "                         \"scaled_ligand_score_spatial\" = 2,  \n",
                "                         \"scaled_receptor_score\" = 0.5, \n",
                "                         \"scaled_receptor_expression_scaled\" = 0.5, \n",
                "                          \"receptor_fraction\" = 1,  \n",
                "                         \"ligand_scaled_receptor_expression_fraction\" = 1, \n",
                "                         \"scaled_receptor_score_spatial\" = 0, \n",
                "                         \"scaled_activity\" = 0, \n",
                "                         \"scaled_activity_normalized\" = 1, \n",
                "                         \"bona_fide\" = 1) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": " \n"
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "output = list(DE_sender_receiver = DE_sender_receiver, ligand_scaled_receptor_expression_fraction_df = ligand_scaled_receptor_expression_fraction_df, sender_spatial_DE_processed = sender_spatial_DE_processed, receiver_spatial_DE_processed = receiver_spatial_DE_processed, \n",
                "         ligand_activities_targets = ligand_activities_targets, DE_receiver_processed_targets = DE_receiver_processed_targets, exprs_tbl_ligand = exprs_tbl_ligand,  exprs_tbl_receptor = exprs_tbl_receptor, exprs_tbl_target = exprs_tbl_target) \n",
                "prioritization_tables = get_prioritization_tables(output, prioritizing_weights) \n",
                " \n",
                "prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(receiver == niches[[1]]$receiver) %>% head(10) \n",
                "## # A tibble: 10 x 37 \n",
                "##    niche    receiver sender       ligand_receptor ligand   receptor bonafide ligand_score ligand_signific~ ligand_present ligand_expressi~ \n",
                "##    <chr>    <chr>    <chr>        <chr>           <chr>    <chr>    <lgl>           <dbl>            <dbl>          <dbl>            <dbl> \n",
                "##  1 KC_niche KCs      Hepatocytes~ Apoa1--Lrp1     Apoa1    Lrp1     FALSE            3.18                1              1            14.7  \n",
                "##  2 KC_niche KCs      Hepatocytes~ Apoa1--Msr1     Apoa1    Msr1     FALSE            3.18                1              1            14.7  \n",
                "##  3 KC_niche KCs      Hepatocytes~ Apoa1--Abca1    Apoa1    Abca1    FALSE            3.18                1              1            14.7  \n",
                "##  4 KC_niche KCs      Hepatocytes~ Apoa1--Scarb1   Apoa1    Scarb1   FALSE            3.18                1              1            14.7  \n",
                "##  5 KC_niche KCs      Hepatocytes~ Apoa1--Derl1    Apoa1    Derl1    FALSE            3.18                1              1            14.7  \n",
                "##  6 KC_niche KCs      Hepatocytes~ Serpina1a--Lrp1 Serpina~ Lrp1     TRUE             2.64                1              1             6.97 \n",
                "##  7 KC_niche KCs      Hepatocytes~ Apoa1--Atp5b    Apoa1    Atp5b    FALSE            3.18                1              1            14.7  \n",
                "##  8 KC_niche KCs      Hepatocytes~ Trf--Tfrc       Trf      Tfrc     TRUE             1.61                1              1             6.19 \n",
                "##  9 KC_niche KCs      Hepatocytes~ Apoa1--Cd36     Apoa1    Cd36     FALSE            3.18                1              1            14.7  \n",
                "## 10 KC_niche KCs      LSECs_portal Cxcl10--Fpr1    Cxcl10   Fpr1     FALSE            1.66                1              1             2.35 \n",
                "## # ... with 26 more variables: ligand_expression_scaled <dbl>, ligand_fraction <dbl>, ligand_score_spatial <dbl>, receptor_score <dbl>, \n",
                "## #   receptor_significant <dbl>, receptor_present <dbl>, receptor_expression <dbl>, receptor_expression_scaled <dbl>, \n",
                "## #   receptor_fraction <dbl>, receptor_score_spatial <dbl>, ligand_scaled_receptor_expression_fraction <dbl>, \n",
                "## #   avg_score_ligand_receptor <dbl>, activity <dbl>, activity_normalized <dbl>, scaled_ligand_score <dbl>, \n",
                "## #   scaled_ligand_expression_scaled <dbl>, scaled_receptor_score <dbl>, scaled_receptor_expression_scaled <dbl>, \n",
                "## #   scaled_avg_score_ligand_receptor <dbl>, scaled_ligand_score_spatial <dbl>, scaled_receptor_score_spatial <dbl>, \n",
                "## #   scaled_ligand_fraction_adapted <dbl>, scaled_receptor_fraction_adapted <dbl>, scaled_activity <dbl>, ... \n",
                "prioritization_tables$prioritization_tbl_ligand_target %>% filter(receiver == niches[[1]]$receiver) %>% head(10) \n",
                "## # A tibble: 10 x 20 \n",
                "##    niche    receiver sender  ligand_receptor ligand receptor bonafide target target_score target_signific~ target_present target_expressi~ \n",
                "##    <chr>    <chr>    <chr>   <chr>           <chr>  <chr>    <lgl>    <chr>         <dbl>            <dbl>          <dbl>            <dbl> \n",
                "##  1 KC_niche KCs      Hepato~ Apoa1--Lrp1     Apoa1  Lrp1     FALSE    Abca1         0.197                1              1            0.979 \n",
                "##  2 KC_niche KCs      Hepato~ Apoa1--Lrp1     Apoa1  Lrp1     FALSE    Actb          0.279                1              1           21.6   \n",
                "##  3 KC_niche KCs      Hepato~ Apoa1--Lrp1     Apoa1  Lrp1     FALSE    Ehd1          0.272                1              1            0.402 \n",
                "##  4 KC_niche KCs      Hepato~ Apoa1--Lrp1     Apoa1  Lrp1     FALSE    Hmox1         1.16                 1              1            5.23  \n",
                "##  5 KC_niche KCs      Hepato~ Apoa1--Lrp1     Apoa1  Lrp1     FALSE    Sgk1          0.265                1              1            0.629 \n",
                "##  6 KC_niche KCs      Hepato~ Apoa1--Lrp1     Apoa1  Lrp1     FALSE    Tcf7l2        0.811                1              1            1.32  \n",
                "##  7 KC_niche KCs      Hepato~ Apoa1--Lrp1     Apoa1  Lrp1     FALSE    Tsc22~        0.263                1              1            0.635 \n",
                "##  8 KC_niche KCs      Hepato~ Apoa1--Msr1     Apoa1  Msr1     FALSE    Abca1         0.197                1              1            0.979 \n",
                "##  9 KC_niche KCs      Hepato~ Apoa1--Msr1     Apoa1  Msr1     FALSE    Actb          0.279                1              1           21.6   \n",
                "## 10 KC_niche KCs      Hepato~ Apoa1--Msr1     Apoa1  Msr1     FALSE    Ehd1          0.272                1              1            0.402 \n",
                "## # ... with 8 more variables: target_expression_scaled <dbl>, target_fraction <dbl>, ligand_target_weight <dbl>, activity <dbl>, \n",
                "## #   activity_normalized <dbl>, scaled_activity <dbl>, scaled_activity_normalized <dbl>, prioritization_score <dbl> \n",
                " \n",
                "prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(receiver == niches[[2]]$receiver) %>% head(10) \n",
                "## # A tibble: 10 x 37 \n",
                "##    niche        receiver sender    ligand_receptor ligand receptor bonafide ligand_score ligand_significa~ ligand_present ligand_expressi~ \n",
                "##    <chr>        <chr>    <chr>     <chr>           <chr>  <chr>    <lgl>           <dbl>             <dbl>          <dbl>            <dbl> \n",
                "##  1 MoMac2_niche MoMac2   Cholangi~ Spp1--Itga4     Spp1   Itga4    TRUE            6.09                  1              1            72.4  \n",
                "##  2 MoMac2_niche MoMac2   Cholangi~ Spp1--Cd44      Spp1   Cd44     TRUE            6.09                  1              1            72.4  \n",
                "##  3 MoMac2_niche MoMac2   Cholangi~ Spp1--Itgb5     Spp1   Itgb5    TRUE            6.09                  1              1            72.4  \n",
                "##  4 MoMac2_niche MoMac2   Cholangi~ Spp1--Itgav     Spp1   Itgav    TRUE            6.09                  1              1            72.4  \n",
                "##  5 MoMac2_niche MoMac2   Cholangi~ Spp1--Itgb1     Spp1   Itgb1    TRUE            6.09                  1              1            72.4  \n",
                "##  6 MoMac2_niche MoMac2   Cholangi~ Spp1--Itga9     Spp1   Itga9    TRUE            6.09                  1              1            72.4  \n",
                "##  7 MoMac2_niche MoMac2   Cholangi~ Spp1--Ncstn     Spp1   Ncstn    FALSE           6.09                  1              1            72.4  \n",
                "##  8 MoMac2_niche MoMac2   Cholangi~ Spp1--Itga5     Spp1   Itga5    FALSE           6.09                  1              1            72.4  \n",
                "##  9 MoMac2_niche MoMac2   Fibrobla~ Lama2--Rpsa     Lama2  Rpsa     TRUE            1.51                  1              1             3.19 \n",
                "## 10 MoMac2_niche MoMac2   Cholangi~ Cyr61--Itgb2    Cyr61  Itgb2    TRUE            0.812                 1              1             3.11 \n",
                "## # ... with 26 more variables: ligand_expression_scaled <dbl>, ligand_fraction <dbl>, ligand_score_spatial <dbl>, receptor_score <dbl>, \n",
                "## #   receptor_significant <dbl>, receptor_present <dbl>, receptor_expression <dbl>, receptor_expression_scaled <dbl>, \n",
                "## #   receptor_fraction <dbl>, receptor_score_spatial <dbl>, ligand_scaled_receptor_expression_fraction <dbl>, \n",
                "## #   avg_score_ligand_receptor <dbl>, activity <dbl>, activity_normalized <dbl>, scaled_ligand_score <dbl>, \n",
                "## #   scaled_ligand_expression_scaled <dbl>, scaled_receptor_score <dbl>, scaled_receptor_expression_scaled <dbl>, \n",
                "## #   scaled_avg_score_ligand_receptor <dbl>, scaled_ligand_score_spatial <dbl>, scaled_receptor_score_spatial <dbl>, \n",
                "## #   scaled_ligand_fraction_adapted <dbl>, scaled_receptor_fraction_adapted <dbl>, scaled_activity <dbl>, ... \n",
                "prioritization_tables$prioritization_tbl_ligand_target %>% filter(receiver == niches[[2]]$receiver) %>% head(10) \n",
                "## # A tibble: 10 x 20 \n",
                "##    niche   receiver sender   ligand_receptor ligand receptor bonafide target target_score target_signific~ target_present target_expressi~ \n",
                "##    <chr>   <chr>    <chr>    <chr>           <chr>  <chr>    <lgl>    <chr>         <dbl>            <dbl>          <dbl>            <dbl> \n",
                "##  1 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Ahnak         1.05                 1              1            1.36  \n",
                "##  2 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Cdkn1a        0.609                1              1            0.801 \n",
                "##  3 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Cxcr4         0.374                1              1            0.717 \n",
                "##  4 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Dhrs3         0.371                1              1            0.743 \n",
                "##  5 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Fn1           0.360                1              1            0.456 \n",
                "##  6 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Gadd4~        0.180                1              1            0.474 \n",
                "##  7 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Gapdh         0.656                1              1            3.27  \n",
                "##  8 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Gdf15         0.479                1              1            0.521 \n",
                "##  9 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Gsn           0.221                1              1            0.647 \n",
                "## 10 MoMac2~ MoMac2   Cholang~ Spp1--Itga4     Spp1   Itga4    TRUE     Plec          0.154                1              1            0.164 \n",
                "## # ... with 8 more variables: target_expression_scaled <dbl>, target_fraction <dbl>, ligand_target_weight <dbl>, activity <dbl>, \n",
                "## #   activity_normalized <dbl>, scaled_activity <dbl>, scaled_activity_normalized <dbl>, prioritization_score <dbl> \n",
                " \n",
                "prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(receiver == niches[[3]]$receiver) %>% head(10) \n",
                "## # A tibble: 10 x 37 \n",
                "##    niche        receiver sender     ligand_receptor ligand receptor bonafide ligand_score ligand_signific~ ligand_present ligand_expressi~ \n",
                "##    <chr>        <chr>    <chr>      <chr>           <chr>  <chr>    <lgl>           <dbl>            <dbl>          <dbl>            <dbl> \n",
                "##  1 MoMac1_niche MoMac1   Mesotheli~ C3--C3ar1       C3     C3ar1    TRUE             3.52                1              1             22.6 \n",
                "##  2 MoMac1_niche MoMac1   Capsule f~ C3--C3ar1       C3     C3ar1    TRUE             3.42                1              1             20.9 \n",
                "##  3 MoMac1_niche MoMac1   Mesotheli~ C3--Itgb2       C3     Itgb2    TRUE             3.52                1              1             22.6 \n",
                "##  4 MoMac1_niche MoMac1   Mesotheli~ C3--Itgax       C3     Itgax    TRUE             3.52                1              1             22.6 \n",
                "##  5 MoMac1_niche MoMac1   Mesotheli~ C3--Lrp1        C3     Lrp1     TRUE             3.52                1              1             22.6 \n",
                "##  6 MoMac1_niche MoMac1   Capsule f~ C3--Itgb2       C3     Itgb2    TRUE             3.42                1              1             20.9 \n",
                "##  7 MoMac1_niche MoMac1   Capsule f~ C3--Itgax       C3     Itgax    TRUE             3.42                1              1             20.9 \n",
                "##  8 MoMac1_niche MoMac1   Capsule f~ C3--Lrp1        C3     Lrp1     TRUE             3.42                1              1             20.9 \n",
                "##  9 MoMac1_niche MoMac1   Capsule f~ Rarres2--Cmklr1 Rarre~ Cmklr1   TRUE             2.50                1              1             15.8 \n",
                "## 10 MoMac1_niche MoMac1   Mesotheli~ C3--Ccr5        C3     Ccr5     FALSE            3.52                1              1             22.6 \n",
                "## # ... with 26 more variables: ligand_expression_scaled <dbl>, ligand_fraction <dbl>, ligand_score_spatial <dbl>, receptor_score <dbl>, \n",
                "## #   receptor_significant <dbl>, receptor_present <dbl>, receptor_expression <dbl>, receptor_expression_scaled <dbl>, \n",
                "## #   receptor_fraction <dbl>, receptor_score_spatial <dbl>, ligand_scaled_receptor_expression_fraction <dbl>, \n",
                "## #   avg_score_ligand_receptor <dbl>, activity <dbl>, activity_normalized <dbl>, scaled_ligand_score <dbl>, \n",
                "## #   scaled_ligand_expression_scaled <dbl>, scaled_receptor_score <dbl>, scaled_receptor_expression_scaled <dbl>, \n",
                "## #   scaled_avg_score_ligand_receptor <dbl>, scaled_ligand_score_spatial <dbl>, scaled_receptor_score_spatial <dbl>, \n",
                "## #   scaled_ligand_fraction_adapted <dbl>, scaled_receptor_fraction_adapted <dbl>, scaled_activity <dbl>, ... \n",
                "prioritization_tables$prioritization_tbl_ligand_target %>% filter(receiver == niches[[3]]$receiver) %>% head(10) \n",
                "## # A tibble: 10 x 20 \n",
                "##    niche   receiver sender   ligand_receptor ligand receptor bonafide target target_score target_signific~ target_present target_expressi~ \n",
                "##    <chr>   <chr>    <chr>    <chr>           <chr>  <chr>    <lgl>    <chr>         <dbl>            <dbl>          <dbl>            <dbl> \n",
                "##  1 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Btg2          0.615                1              1            1.51  \n",
                "##  2 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Ccnd2         0.505                1              1            0.490 \n",
                "##  3 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Cdk6          0.221                1              1            0.320 \n",
                "##  4 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Ier5          0.396                1              1            1.16  \n",
                "##  5 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Il1b          0.956                1              1            3.74  \n",
                "##  6 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Jun           0.765                1              1            1.93  \n",
                "##  7 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Pdgfb         0.243                1              1            0.510 \n",
                "##  8 MoMac1~ MoMac1   Mesothe~ C3--C3ar1       C3     C3ar1    TRUE     Ubc           0.306                1              1            2.16  \n",
                "##  9 MoMac1~ MoMac1   Capsule~ C3--C3ar1       C3     C3ar1    TRUE     Btg2          0.615                1              1            1.51  \n",
                "## 10 MoMac1~ MoMac1   Capsule~ C3--C3ar1       C3     C3ar1    TRUE     Ccnd2         0.505                1              1            0.490 \n",
                "## # ... with 8 more variables: target_expression_scaled <dbl>, target_fraction <dbl>, ligand_target_weight <dbl>, activity <dbl>, \n",
                "## #   activity_normalized <dbl>, scaled_activity <dbl>, scaled_activity_normalized <dbl>, prioritization_score <dbl> \n",
                " \n",
                "prioritization_tables$prioritization_tbl_ligand_receptor = prioritization_tables$prioritization_tbl_ligand_receptor %>% mutate(receiver = factor(receiver, levels = c(\"KCs\",\"MoMac1\",\"MoMac2\")), niche = factor(niche, levels = c(\"KC_niche\",\"MoMac1_niche\",\"MoMac2_niche\")))  \n",
                "prioritization_tables$prioritization_tbl_ligand_target = prioritization_tables$prioritization_tbl_ligand_target %>% mutate(receiver = factor(receiver, levels = c(\"KCs\",\"MoMac1\",\"MoMac2\")), niche = factor(niche, levels = c(\"KC_niche\",\"MoMac1_niche\",\"MoMac2_niche\")))  \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "# 8. Visualization of the Differential NicheNet output \n",
                " \n",
                "## Differential expression of ligand and expression \n",
                " \n",
                "Before visualization, we need to define the most important \n",
                "ligand-receptor pairs per niche. We will do this by first determining \n",
                "for which niche the highest score is found for each \n",
                "ligand/ligand-receptor pair. And then getting the top 50 ligands per \n",
                "niche. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "top_ligand_niche_df = prioritization_tables$prioritization_tbl_ligand_receptor %>% select(niche, sender, receiver, ligand, receptor, prioritization_score) %>% group_by(ligand) %>% top_n(1, prioritization_score) %>% ungroup() %>% select(ligand, receptor, niche) %>% rename(top_niche = niche) \n",
                "top_ligand_receptor_niche_df = prioritization_tables$prioritization_tbl_ligand_receptor %>% select(niche, sender, receiver, ligand, receptor, prioritization_score) %>% group_by(ligand, receptor) %>% top_n(1, prioritization_score) %>% ungroup() %>% select(ligand, receptor, niche) %>% rename(top_niche = niche) \n",
                " \n",
                "ligand_prioritized_tbl_oi = prioritization_tables$prioritization_tbl_ligand_receptor %>% select(niche, sender, receiver, ligand, prioritization_score) %>% group_by(ligand, niche) %>% top_n(1, prioritization_score) %>% ungroup() %>% distinct() %>% inner_join(top_ligand_niche_df) %>% filter(niche == top_niche) %>% group_by(niche) %>% top_n(50, prioritization_score) %>% ungroup() # get the top50 ligands per niche \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "Now we will look first at the top ligand-receptor pairs for KCs (here, \n",
                "we will take the top 2 scoring receptors per prioritized ligand) \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "receiver_oi = \"KCs\"  \n",
                " \n",
                "filtered_ligands = ligand_prioritized_tbl_oi %>% filter(receiver == receiver_oi) %>% pull(ligand) %>% unique() \n",
                " \n",
                "prioritized_tbl_oi = prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(ligand %in% filtered_ligands) %>% select(niche, sender, receiver, ligand,  receptor, ligand_receptor, prioritization_score) %>% distinct() %>% inner_join(top_ligand_receptor_niche_df) %>% group_by(ligand) %>% filter(receiver == receiver_oi) %>% top_n(2, prioritization_score) %>% ungroup()  \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "Visualization: minimum LFC compared to other niches \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "lfc_plot = make_ligand_receptor_lfc_plot(receiver_oi, prioritized_tbl_oi, prioritization_tables$prioritization_tbl_ligand_receptor, plot_legend = FALSE, heights = NULL, widths = NULL) \n",
                "lfc_plot \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-25-1.png)<!-- --> \n",
                " \n",
                "Show the spatialDE as additional information \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "lfc_plot_spatial = make_ligand_receptor_lfc_spatial_plot(receiver_oi, prioritized_tbl_oi, prioritization_tables$prioritization_tbl_ligand_receptor, ligand_spatial = include_spatial_info_sender, receptor_spatial = include_spatial_info_receiver, plot_legend = FALSE, heights = NULL, widths = NULL) \n",
                "lfc_plot_spatial \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-26-1.png)<!-- --> \n",
                " \n",
                "From this plot, you can see that some KC-niche ligands like Dll4 (by \n",
                "LSEC) and Il34 (by Stellate cells) are higher expressed in the \n",
                "periportal LSEC/stellate cells vs the pericentral ones. This can be \n",
                "interesting information knowing that KCs are mainly located \n",
                "periportally. However, other ligands like Gdf2 (by Stellate cells) are \n",
                "not preferentially expressed by periportal stellate cells, but this does \n",
                "not mean they cannot be interesting. As you can see in the following \n",
                "figure, this ligand has one of the highest ligand activities, meaning \n",
                "that there is a strong enrichment of its target genes among the \n",
                "KC-specific genes. \n",
                " \n",
                "## Ligand expression, activity and target genes \n",
                " \n",
                "Active target gene inference - cf Default NicheNet \n",
                " \n",
                "Now: visualization of ligand activity and ligand-target links \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "exprs_activity_target_plot = make_ligand_activity_target_exprs_plot(receiver_oi, prioritized_tbl_oi,  prioritization_tables$prioritization_tbl_ligand_receptor,  prioritization_tables$prioritization_tbl_ligand_target, output$exprs_tbl_ligand,  output$exprs_tbl_target, lfc_cutoff, ligand_target_matrix, plot_legend = FALSE, heights = NULL, widths = NULL) \n",
                "exprs_activity_target_plot$combined_plot \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-27-1.png)<!-- --> \n",
                " \n",
                "On this plot, we can see that some strongly DE ligand-receptor pairs in \n",
                "the KC niche, have also high scaled ligand activity on KCs - making them \n",
                "strong predictions for further validation. An example of this is Gdf2 \n",
                "and Bmp10, who bind the receptor Acvrl1 (ALK1). The role of \n",
                "Gdf2/Bmp10-Acvrl1 in KC development was experimentally validated in the \n",
                "Guilliams et al paper. \n",
                " \n",
                "**important: ligand-receptor pairs with both high differential \n",
                "expression and ligand activity (=target gene enrichment) are very \n",
                "interesting predictions as key regulators of your intercellular \n",
                "communication process of interest !** \n",
                " \n",
                "If this plot contains too much information because we look at many hits \n",
                "(top 50 ligands), you can make this plot of course for less ligands as \n",
                "well, eg for the top20. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "filtered_ligands = ligand_prioritized_tbl_oi %>% filter(receiver == receiver_oi) %>% top_n(20, prioritization_score) %>% pull(ligand) %>% unique() \n",
                " \n",
                "prioritized_tbl_oi = prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(ligand %in% filtered_ligands) %>% select(niche, sender, receiver, ligand,  receptor, ligand_receptor, prioritization_score) %>% distinct() %>% inner_join(top_ligand_receptor_niche_df) %>% group_by(ligand) %>% filter(receiver == receiver_oi) %>% top_n(2, prioritization_score) %>% ungroup()  \n",
                " \n",
                "exprs_activity_target_plot = make_ligand_activity_target_exprs_plot(receiver_oi, prioritized_tbl_oi,  prioritization_tables$prioritization_tbl_ligand_receptor,  prioritization_tables$prioritization_tbl_ligand_target, output$exprs_tbl_ligand,  output$exprs_tbl_target, lfc_cutoff, ligand_target_matrix, plot_legend = FALSE, heights = NULL, widths = NULL) \n",
                "exprs_activity_target_plot$combined_plot \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-28-1.png)<!-- --> \n",
                " \n",
                "## Circos plot of prioritized ligand-receptor pairs \n",
                " \n",
                "Because a top50 is too much to visualize in a circos plot, we will only \n",
                "visualize the top 15. \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "filtered_ligands = ligand_prioritized_tbl_oi %>% filter(receiver == receiver_oi) %>% top_n(15, prioritization_score) %>% pull(ligand) %>% unique() \n",
                " \n",
                "prioritized_tbl_oi = prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(ligand %in% filtered_ligands) %>% select(niche, sender, receiver, ligand,  receptor, ligand_receptor, prioritization_score) %>% distinct() %>% inner_join(top_ligand_receptor_niche_df) %>% group_by(ligand) %>% filter(receiver == receiver_oi) %>% top_n(2, prioritization_score) %>% ungroup()  \n",
                " \n",
                "colors_sender = brewer.pal(n = prioritized_tbl_oi$sender %>% unique() %>% sort() %>% length(), name = 'Spectral') %>% magrittr::set_names(prioritized_tbl_oi$sender %>% unique() %>% sort()) \n",
                "colors_receiver = c(\"lavender\")  %>% magrittr::set_names(prioritized_tbl_oi$receiver %>% unique() %>% sort()) \n",
                " \n",
                "circos_output = make_circos_lr(prioritized_tbl_oi, colors_sender, colors_receiver) \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-29-1.png)<!-- -->![](differential_nichenet_files/figure-gfm/unnamed-chunk-29-2.png)<!-- --> \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "# circos_output$p_circos \n",
                "\n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "## Visualization for the other liver macrophages: central vein \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "receiver_oi = \"MoMac1\"   \n",
                "filtered_ligands = ligand_prioritized_tbl_oi %>% filter(receiver == receiver_oi) %>% top_n(50, prioritization_score) %>% pull(ligand) %>% unique() \n",
                " \n",
                "prioritized_tbl_oi = prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(ligand %in% filtered_ligands) %>% select(niche, sender, receiver, ligand,  receptor, ligand_receptor, prioritization_score) %>% distinct() %>% inner_join(top_ligand_receptor_niche_df) %>% group_by(ligand) %>% filter(receiver == receiver_oi) %>% top_n(2, prioritization_score) %>% ungroup()  \n",
                " \n",
                "lfc_plot = make_ligand_receptor_lfc_plot(receiver_oi, prioritized_tbl_oi, prioritization_tables$prioritization_tbl_ligand_receptor, plot_legend = FALSE, heights = NULL, widths = NULL) \n",
                "lfc_plot \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-30-1.png)<!-- --> \n",
                " \n",
                "## Visualization for the other liver macrophages: bile duct \n",
                " \n"
            ]
        },
        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {},
            "outputs": [],
            "source": [
                "receiver_oi = \"MoMac2\"   \n",
                "filtered_ligands = ligand_prioritized_tbl_oi %>% filter(receiver == receiver_oi) %>% top_n(50, prioritization_score) %>% pull(ligand) %>% unique() \n",
                " \n",
                "prioritized_tbl_oi = prioritization_tables$prioritization_tbl_ligand_receptor %>% filter(ligand %in% filtered_ligands) %>% select(niche, sender, receiver, ligand,  receptor, ligand_receptor, prioritization_score) %>% distinct() %>% inner_join(top_ligand_receptor_niche_df) %>% group_by(ligand) %>% filter(receiver == receiver_oi) %>% top_n(2, prioritization_score) %>% ungroup()  \n",
                " \n",
                "lfc_plot = make_ligand_receptor_lfc_plot(receiver_oi, prioritized_tbl_oi, prioritization_tables$prioritization_tbl_ligand_receptor, plot_legend = FALSE, heights = NULL, widths = NULL) \n",
                "lfc_plot \n"
            ]
        },
        {
            "cell_type": "markdown",
            "metadata": {},
            "source": [
                " \n",
                "![](differential_nichenet_files/figure-gfm/unnamed-chunk-31-1.png)<!-- --> \n",
                " \n",
                "# Notes, limitations, and comparison to default NicheNet. \n",
                " \n",
                "In the default NicheNet pipeline, expressed ligand-receptor pairs are \n",
                "prioritized based on their ligand activity alone. Here, in the \n",
                "Differential NicheNet pipeline, we also draw information based on \n",
                "differential expression of the L-R pairs compared to other niches (and \n",
                "if applicable: other spatial locations.) \n",
                " \n",
                "Because we here focus on differential expression of ligand-receptor \n",
                "pairs, and by using the default prioritizations weights more on DE than \n",
                "activity, we tend to find many different hits than with the default \n",
                "NicheNet pipeline. With Differential NicheNet, we tend to find more \n",
                "high-DE, low-activity hits, whereas with default NicheNet we find more \n",
                "low-DE, high-activity hits. \n",
                " \n",
                "It should be noted that some of the high-DE, low-activity hits might be \n",
                "really important because they just have low NicheNet activities due to \n",
                "limitations in the NicheNet activity prediction (eg improper/incomplete \n",
                "prior knowledge within NicheNet for that ligand), but some of them might \n",
                "also be high in DE but not in activity because they don’t have strong \n",
                "signaling effects (eg ligands involved in cell adhesion only). \n",
                " \n",
                "For the opposite pairs with low-DE and high-activity that are not \n",
                "strongly prioritized by Differential NicheNet, the following should be \n",
                "considered: 1) some ligands are regulated post-transcriptionally, and \n",
                "that the high predicted activities might still reflect true signaling; \n",
                "2) high predicted activity values might be due to limitations of \n",
                "NicheNet (inaccurate prior knowledge) and these lowDE ligands are not \n",
                "important in the biological process of interest (although a highDE \n",
                "family member of this ligand may! since signaling between family members \n",
                "tends to be very similar); 3) high activity in one condition might be \n",
                "due to downregulation in the other condition, leading to high activity \n",
                "but low DE. Currently, ligand activities are automatically calculated on \n",
                "upregulated genes per condition, but downregulated genes could also be a \n",
                "sign of ligand activity. \n",
                " \n",
                "When Ligand-Receptor pairs have both high DE and high activity, we can \n",
                "consider them to be very good candidates in regulating the process of \n",
                "interest, and we recommend testing these candidates for further \n",
                "experimental validation. \n",
                " \n",
                "# References \n",
                " \n",
                "Browaeys, R., Saelens, W. & Saeys, Y. NicheNet: modeling intercellular \n",
                "communication by linking ligands to target genes. Nat Methods (2019) \n",
                "<doi:10.1038/s41592-019-0667-5> \n",
                " \n",
                "Guilliams et al. Spatial proteogenomics reveals distinct and \n",
                "evolutionarily conserved hepatic macrophage niches. Cell (2022) \n",
                "<doi:10.1016/j.cell.2021.12.018> \n"
            ]
        }
    ],
    "metadata": {
        "anaconda-cloud": "",
        "kernelspec": {
            "display_name": "R",
            "langauge": "R",
            "name": "ir"
        },
        "language_info": {
            "codemirror_mode": "r",
            "file_extension": ".r",
            "mimetype": "text/x-r-source",
            "name": "R",
            "pygments_lexer": "r",
            "version": "3.4.1"
        }
    },
    "nbformat": 4,
    "nbformat_minor": 1
}
